{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Porting genome scale metabolic models for metabolomics\n",
    "- from CarveMe\n",
    "\n",
    "**source**\n",
    "- https://github.com/cdanielmachado/carveme\n",
    "\n",
    "**Use cobra to parse SBML models whereas applicable**\n",
    "\n",
    "Not all models comply with the formats in cobra. Models from USCD and Thiele labs should comply.\n",
    "\n",
    "**Base our code on metDataModel**\n",
    "\n",
    "Each model needs a list of Reactions, list of Pathways, and a list of Compounds.\n",
    "It's important to include with Compounds with all linked identifiers to other DBs (HMDB, PubChem, etc), and with formulae (usually charged form in these models) when available.\n",
    "We can alwasy update the data later. E.g. the neural formulae can be inferred from charged formula or retrieved from public metabolite database (e.g., HMDB) if linked.\n",
    "Save in Python pickle and in JSON.\n",
    "\n",
    "**No compartmentalization**\n",
    "- After decompartmentalization,\n",
    "  - transport reactions can be removed - they are identified by reactants and products being the same.\n",
    "  - redundant reactions can be merge - same reactions in diff compartments become one.\n",
    "\n",
    "Minghao Gong, 2022-04-26"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# !pip install cobra --user --ignore-installed ruamel.yaml\n",
    "# !pip install --upgrade metDataModel # https://github.com/shuzhao-li/metDataModel/ \n",
    "# !pip install --upgrade numpy pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra # https://cobrapy.readthedocs.io/en/latest/io.html#SBML\n",
    "from metDataModel.core import Compound, Reaction, Pathway, MetabolicModel\n",
    "import requests\n",
    "import sys\n",
    "import re\n",
    "\n",
    "sys.path.append(\"/Users/gongm/Documents/projects/mass2chem/\")\n",
    "sys.path.append(\"/Users/gongm/Documents/projects/JMS/JMS/JMS\")\n",
    "from mass2chem.formula import *\n",
    "from jms.formula import *\n",
    "from jms.utils.gems import *\n",
    "from jms.utils.git_download import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# download the most updated Rat-GEM.xml\n",
    "model_name = 'Clostridium_sporogenes-Carve'\n",
    "file_name = 'Clostridium_sporogenes_carveme.xml'\n",
    "local_path = output_fdr = f'../testdata/draftGenome-GEM-JMS/CarveMe/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaling...\n",
      " A: min|aij| =  1.000e+00  max|aij| =  1.000e+00  ratio =  1.000e+00\n",
      "Problem data seem to be well scaled\n"
     ]
    }
   ],
   "source": [
    "# Read the model via cobra\n",
    "model = cobra.io.read_sbml_model(os.path.join(local_path,file_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>Clostridium_sporogenes_ATCC_15579</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fc463a378e0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1219</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1715</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*Growth - 1.0*Growth_reverse_699ae</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, periplasm, extracellular space</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>nadh_c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Nicotinamide adenine dinucleotide - reduced</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fc463a6d700</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td>C21H27N7O14P2</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td>C_c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 128 reaction(s)</strong></td><td>\n",
       "                    ILEDHr, M1PD, IMPD, EAR160x, EAR161x, PPND, NHFRBO, EAR180x, 4MBZDH, ICDHx, PGCD, AKGDH, ALCD2x, NADH10, NADH7, PERD, SBTPD, FMNRx, NADFADOR, OIVD3, EAR60x, FRNDPR2r_1, NADH8, PXMO, OIVD1r, SHCHD2,...</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Metabolite nadh_c at 0x7fc463a6d700>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# metabolite entries, readily convert to list of metabolites\n",
    "model.metabolites[33]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['C20H21N7O7',\n",
       " 'C10H12N5O10P2',\n",
       " 'C10H12N5O13P3',\n",
       " 'C5H8NO4',\n",
       " 'HPO4;HO4P',\n",
       " 'C27H52O5',\n",
       " 'C27H52O5',\n",
       " 'C31H60O5',\n",
       " 'C31H60O5',\n",
       " 'C31H56O5',\n",
       " 'C31H56O5',\n",
       " 'C35H68O5',\n",
       " 'C35H68O5',\n",
       " 'C35H64O5',\n",
       " 'C35H64O5',\n",
       " 'C39H76O5',\n",
       " 'C39H76O5',\n",
       " 'C39H72O5',\n",
       " 'C39H72O5',\n",
       " 'H2O',\n",
       " 'C3H6O',\n",
       " 'C3H8O2',\n",
       " 'C3H8O2',\n",
       " 'C6H12O6',\n",
       " 'C6H12O6',\n",
       " 'H2O',\n",
       " 'H2O2',\n",
       " 'H4N;NH4',\n",
       " 'O2',\n",
       " 'H',\n",
       " 'C9H11N2O12P2',\n",
       " 'C15H22N2O17P2',\n",
       " 'C21H26N7O14P2',\n",
       " 'C21H27N7O14P2',\n",
       " 'C21H32N7O16P3S',\n",
       " 'C21H25N7O17P3',\n",
       " 'C21H26N7O17P3',\n",
       " 'H',\n",
       " 'C5H9O4',\n",
       " 'C6H11O4',\n",
       " 'C6H9O6PS',\n",
       " 'H2O',\n",
       " 'C7H14N2O4',\n",
       " 'C7H5NO4',\n",
       " 'C19H36O7P1',\n",
       " 'C19H36O7P1',\n",
       " 'C17H36NO7P1',\n",
       " 'C17H36NO7P1',\n",
       " 'C19H40NO7P1',\n",
       " 'C19H40NO7P1',\n",
       " 'C19H38NO7P1',\n",
       " 'C19H38NO7P1',\n",
       " 'C21H44NO7P1',\n",
       " 'C21H44NO7P1',\n",
       " 'C21H42NO7P1',\n",
       " 'C21H42NO7P1',\n",
       " 'C23H48NO7P1',\n",
       " 'C23H48NO7P1',\n",
       " 'C23H46NO7P1',\n",
       " 'C23H46NO7P1',\n",
       " 'C10H12N5O7P',\n",
       " 'C12H23O2',\n",
       " 'C29H58N1O8P1',\n",
       " 'HO7P2;P2HO7',\n",
       " 'C33H66N1O8P1',\n",
       " 'C14H27O2',\n",
       " 'C33H62N1O8P1',\n",
       " 'C14H25O2',\n",
       " 'C16H31O2',\n",
       " 'C37H74N1O8P1',\n",
       " 'C16H29O2',\n",
       " 'C37H70N1O8P1',\n",
       " 'C18H35O2',\n",
       " 'C41H82N1O8P1',\n",
       " 'C18H33O2',\n",
       " 'C41H78N1O8P1',\n",
       " 'C18H36O9P1',\n",
       " 'C18H36O9P1',\n",
       " 'C20H40O9P1',\n",
       " 'C20H40O9P1',\n",
       " 'C20H38O9P1',\n",
       " 'C20H38O9P1',\n",
       " 'C22H44O9P1',\n",
       " 'C22H44O9P1',\n",
       " 'C22H42O9P1',\n",
       " 'C22H42O9P1',\n",
       " 'C24H48O9P1',\n",
       " 'C24H48O9P1',\n",
       " 'C24H46O9P1',\n",
       " 'C24H46O9P1',\n",
       " 'C30H58O10P1',\n",
       " 'C34H66O10P1',\n",
       " 'C34H62O10P1',\n",
       " 'C38H74O10P1',\n",
       " 'C38H70O10P1',\n",
       " 'C42H82O10P1',\n",
       " 'C42H78O10P1',\n",
       " 'C3H3O3',\n",
       " 'C10H11N5O10P2',\n",
       " 'C10H11N5O13P2S',\n",
       " 'C3H6NO5S',\n",
       " 'C5H4O5',\n",
       " 'C4H7NO3',\n",
       " 'C7H10O10P',\n",
       " 'C6H11O7',\n",
       " 'C6H11O7',\n",
       " 'C6H9O4',\n",
       " 'C5H7O3',\n",
       " 'CH2O',\n",
       " 'C5H9O7P',\n",
       " 'C5H10O4',\n",
       " 'C4H5O3',\n",
       " 'H',\n",
       " 'CHO3;HCO3',\n",
       " 'Na',\n",
       " 'Na',\n",
       " 'CHO3;HCO3',\n",
       " 'C7H8O2',\n",
       " 'C7H8O2',\n",
       " 'C6H10O6',\n",
       " 'C6H8N3O7P2',\n",
       " 'C6H8N3O4P',\n",
       " 'C26H38N7O17P3S',\n",
       " 'C3H4O7P',\n",
       " 'C2H2O6P',\n",
       " 'C2H2O6P',\n",
       " 'HPO4;HO4P',\n",
       " 'C10H12N5O11P2',\n",
       " 'C10H12N5O14P3',\n",
       " 'C3H5O3',\n",
       " 'CO2',\n",
       " 'C14H20N6O5S',\n",
       " 'C15H23N6O5S',\n",
       " 'C6H13NO2',\n",
       " 'C6H13NO2',\n",
       " 'C9H7O4',\n",
       " 'C394H621O144N96P1S3;C21H39N2O9PRS',\n",
       " 'HSR;HX;C384H603N96O142P1S3;C11H21N2O7PRS',\n",
       " 'C396H625O144N96P1S3;C23H43N2O9PRS',\n",
       " 'C23H41N2O9PRS;C396H623O144N96P1S3',\n",
       " 'C25H47N2O9PRS;C398H629O144N96P1S3',\n",
       " 'C398H627O144N96P1S3;C25H45N2O9PRS',\n",
       " 'C17H31N2O9PRS;C390H613O144N96P1S3',\n",
       " 'C392H617O144N96P1S3;C19H35N2O9PRS',\n",
       " 'C21H37N2O8PRS;C394H619O143N96P1S3',\n",
       " 'C23H41N2O8PRS;C396H623O143N96P1S3',\n",
       " 'C396H621O143N96P1S3;C23H39N2O8PRS',\n",
       " 'C398H627O143N96P1S3;C25H45N2O8PRS',\n",
       " 'C25H43N2O8PRS;C398H625O143N96P1S3',\n",
       " 'C400H633O144N96P1S3;C27H51N2O9PRS',\n",
       " 'C400H631O143N96P1S3;C27H49N2O8PRS',\n",
       " 'C400H631O144N96P1S3;C27H49N2O9PRS',\n",
       " 'C27H47N2O8PRS;C400H629O143N96P1S3',\n",
       " 'C402H637O144N96P1S3;C29H55N2O9PRS',\n",
       " 'C402H635O143N96P1S3;C29H53N2O8PRS',\n",
       " 'C29H53N2O9PRS;C402H635O144N96P1S3',\n",
       " 'C29H51N2O8PRS;C402H633O143N96P1S3',\n",
       " 'C388H609O144N96P1S3;C15H27N2O9PRS',\n",
       " 'C15H25N2O8PRS;C388H607O143N96P1S3',\n",
       " 'C17H29N2O8PRS;C390H611O143N96P1S3',\n",
       " 'C19H33N2O8PRS;C392H615O143N96P1S3',\n",
       " 'C8H15NOS2',\n",
       " 'C6H9O3',\n",
       " 'C5H9O2',\n",
       " 'Na',\n",
       " 'C5H9O2',\n",
       " 'H2O2',\n",
       " 'H4N;NH4',\n",
       " 'O2',\n",
       " 'C10H13N5O4',\n",
       " 'C9H12N2O6',\n",
       " 'C9H13N3O5',\n",
       " 'C10H13N5O5;C10H12N5O5',\n",
       " 'C39H66N7O18P3S',\n",
       " 'C39H64N7O18P3S',\n",
       " 'C24H34N7O19P3S;C24H33N7O19P3S',\n",
       " 'C37H62N7O17P3S',\n",
       " 'C27H37N7O20P3S',\n",
       " 'C4H4O4',\n",
       " 'C25H35N7O19P3S',\n",
       " 'C21H37N2O9PRS;C394H619O144N96P1S3',\n",
       " 'C23H41N2O9PRS;C396H623O144N96P1S3',\n",
       " 'C396H621O144N96P1S3;C23H39N2O9PRS',\n",
       " 'C398H627O144N96P1S3;C25H45N2O9PRS',\n",
       " 'C25H43N2O9PRS;C398H625O144N96P1S3',\n",
       " 'C400H631O144N96P1S3;C27H49N2O9PRS',\n",
       " 'C400H629O144N96P1S3;C27H47N2O9PRS',\n",
       " 'C29H53N2O9PRS;C402H635O144N96P1S3',\n",
       " 'C29H51N2O9PRS;C402H633O144N96P1S3',\n",
       " 'C15H25N2O9PRS;C388H607O144N96P1S3',\n",
       " 'C390H611O144N96P1S3;C17H29N2O9PRS',\n",
       " 'C19H33N2O9PRS;C392H615O144N96P1S3',\n",
       " 'C387H604O145N96P1S3;C14H22N2O10PRS',\n",
       " 'C19H35N2O8PRS;C392H617O143N96P1S3',\n",
       " 'C394H621O143N96P1S3;C21H39N2O8PRS',\n",
       " 'C21H37N2O8PRS;C394H619O143N96P1S3',\n",
       " 'C23H43N2O8PRS;C396H625O143N96P1S3',\n",
       " 'C23H41N2O8PRS;C396H623O143N96P1S3',\n",
       " 'C398H629O143N96P1S3;C25H47N2O8PRS',\n",
       " 'C398H627O143N96P1S3;C25H45N2O8PRS',\n",
       " 'C400H633O143N96P1S3;C27H51N2O8PRS',\n",
       " 'C400H631O143N96P1S3;C27H49N2O8PRS',\n",
       " 'C388H609O143N96P1S3;C15H27N2O8PRS',\n",
       " 'C17H31N2O8PRS;C390H613O143N96P1S3',\n",
       " 'C23H34N7O17P3S',\n",
       " 'C15H25N4O8',\n",
       " 'C15H25N4O8',\n",
       " 'C3H4O7P',\n",
       " 'H2',\n",
       " 'C29H38N7O17P3S',\n",
       " 'C4H6NO4',\n",
       " 'O3S;HSO3',\n",
       " 'C4H9NO2',\n",
       " 'C4H10NO',\n",
       " 'C7H6NO2',\n",
       " 'C4H5O3',\n",
       " 'C8H7O3',\n",
       " 'C7H6O2',\n",
       " 'C7H6O2',\n",
       " 'C7H6O2',\n",
       " 'C7H8O2',\n",
       " 'C7H8O2',\n",
       " 'C7H5O3',\n",
       " 'C28H40N7O17P3S',\n",
       " 'Fe8S8X',\n",
       " 'Fe8S8X',\n",
       " 'C7H5O3',\n",
       " 'C7H5O3',\n",
       " 'C27H31N9O15P2',\n",
       " 'C27H33N9O15P2',\n",
       " 'C8H7O3',\n",
       " 'C3H7NO2',\n",
       " 'C3H7NO2',\n",
       " 'C4H9NO4',\n",
       " 'C4H8NO7P',\n",
       " 'C6H6N2O3;C6H7N2O3',\n",
       " 'C6H9O3',\n",
       " 'C8H8O',\n",
       " 'C8H10O',\n",
       " 'C8H7O2',\n",
       " 'C7H7O4',\n",
       " 'C7H8O2',\n",
       " 'CHO2;CH1O2',\n",
       " 'C5H5O3',\n",
       " 'C6H7O3',\n",
       " 'Cl',\n",
       " 'Cl',\n",
       " 'C18H30N5O9',\n",
       " 'C3H7NO2',\n",
       " 'C18H30N5O9',\n",
       " 'C3H7NO2',\n",
       " 'C29H38N7O18P3S;C29H42N7O18P3S',\n",
       " 'C5H9NO3',\n",
       " 'C6H9O7',\n",
       " 'C6H9O7',\n",
       " 'C5H10O4',\n",
       " 'C5H5N5',\n",
       " 'C10H13N5O3',\n",
       " 'C5H8O14P3',\n",
       " 'C20H21N7O7',\n",
       " 'C17H19N4O6;C17H20N4O6',\n",
       " 'C8H13N3O7P;C8H12N3O7P',\n",
       " 'C11H15N5O3S',\n",
       " 'C20H23N7O6;C20H24N7O6',\n",
       " 'C6H12O4S',\n",
       " 'C2H4O',\n",
       " 'C9H12N5O13P3;C9H11N5O13P3',\n",
       " 'HO10P3',\n",
       " 'C6H11O9P',\n",
       " 'C13H19O10P;C13H17O10P',\n",
       " 'C5H10N2O3',\n",
       " 'C3H7NO2S',\n",
       " 'C3H7NO3',\n",
       " 'C4H9NO3',\n",
       " 'C15H22N2O17P2',\n",
       " 'C5H9O8P',\n",
       " 'C5H9O8P',\n",
       " 'C6H9N3O2',\n",
       " 'C2H3O2',\n",
       " 'C25H36N7O18P3S',\n",
       " 'C402H635O143N96P1S3;C29H53N2O8PRS',\n",
       " 'C402H637O143N96P1S3;C29H55N2O8PRS',\n",
       " 'C10H19O2',\n",
       " 'C8H15O2',\n",
       " 'C3H4O2',\n",
       " 'C10H13N2O11P2',\n",
       " 'C5H10O5',\n",
       " 'C12H15O10P',\n",
       " 'C6H6O2',\n",
       " 'C5H10O5',\n",
       " 'C5H10O5',\n",
       " 'C4H9NO2',\n",
       " 'C4H9NO2',\n",
       " 'C27H40N7O18P3S',\n",
       " 'C25H38N7O17P3S',\n",
       " 'C29H44N7O18P3S',\n",
       " 'C27H42N7O17P3S',\n",
       " 'C31H48N7O18P3S',\n",
       " 'C29H46N7O17P3S',\n",
       " 'C33H52N7O18P3S',\n",
       " 'C31H50N7O17P3S',\n",
       " 'C31H50N7O17P3S',\n",
       " 'C35H56N7O18P3S',\n",
       " 'C33H54N7O17P3S',\n",
       " 'C33H54N7O17P3S',\n",
       " 'C37H60N7O18P3S',\n",
       " 'C35H58N7O17P3S',\n",
       " 'C4H8O2',\n",
       " 'C2H4O',\n",
       " 'C3H5O2',\n",
       " 'C24H36N7O17P3S',\n",
       " 'C5H7O3S',\n",
       " 'CO',\n",
       " 'C6H11O9P',\n",
       " 'C10H11N4O11P2',\n",
       " 'C10H11N4O14P3',\n",
       " 'C3H2O6P',\n",
       " 'C8H15NO6',\n",
       " 'C8H15NO6',\n",
       " 'C8H15NO6',\n",
       " 'C8H14NO9P',\n",
       " 'C6H13NO8P',\n",
       " 'C8H14NO9P',\n",
       " 'C17H25N3O17P2',\n",
       " 'C55H89O4P',\n",
       " 'C9H11N2O9P',\n",
       " 'C8H15NO6',\n",
       " 'C7H9NO5',\n",
       " 'C6H9O4',\n",
       " 'C2H3O2',\n",
       " 'C5H14NO',\n",
       " 'C2H6O',\n",
       " 'C6H11NO4',\n",
       " 'C6H5O7',\n",
       " 'C4H2O5',\n",
       " 'C2H3O5P',\n",
       " 'C5H7O4',\n",
       " 'C12H16N4O7P2S',\n",
       " 'C11H17NO11P',\n",
       " 'C3H5O3',\n",
       " 'C8H14NO9P',\n",
       " 'C8H15NO6',\n",
       " 'C6H11O9P',\n",
       " 'C11H18NO9',\n",
       " 'C9H12N3O14P3',\n",
       " 'C25H36N7O17P3S',\n",
       " 'C27H40N7O17P3S',\n",
       " 'C26H40N7O17P3S',\n",
       " 'C29H44N7O17P3S',\n",
       " 'C31H48N7O17P3S',\n",
       " 'C33H52N7O17P3S',\n",
       " 'C33H52N7O17P3S',\n",
       " 'C35H56N7O17P3S',\n",
       " 'C37H60N7O17P3S',\n",
       " 'C39H64N7O17P3S',\n",
       " 'C39H66N7O17P3S',\n",
       " 'C25H38N7O17P3S',\n",
       " 'C386H605O143N96P1S3;C13H23N2O8PRS',\n",
       " 'C7H14N2O3',\n",
       " 'C5H13N2O2',\n",
       " 'C5H13N2O2',\n",
       " 'C6H3O6',\n",
       " 'C6H5O7',\n",
       " 'HPO4;HO4P',\n",
       " 'C17H19N4O6;C17H20N4O6',\n",
       " 'C17H18N4O9P;C17H19N4O9P',\n",
       " 'C12H24O5P1',\n",
       " 'C14H28O5P',\n",
       " 'C14H26O5P',\n",
       " 'C16H32O5P',\n",
       " 'C16H30O5P',\n",
       " 'C18H36O5P',\n",
       " 'C18H34O5P',\n",
       " 'C11H20N2O7PS;C11H21N2O7PS',\n",
       " 'C5H9NO4',\n",
       " 'HS2O3;O3S2',\n",
       " 'C3H6NO5S2',\n",
       " 'C2H3O2',\n",
       " 'C4H8O2',\n",
       " 'C3H4O10P2',\n",
       " 'C10H12N4O5',\n",
       " 'C9H12N3O7P',\n",
       " 'C9H13N3O4',\n",
       " 'H4N;NH4',\n",
       " 'C10H10NO5',\n",
       " 'C58H84CoN16O11',\n",
       " 'C10H8O6',\n",
       " 'C5H4N4O',\n",
       " 'C5H5N5',\n",
       " 'C10H12N5O9P2',\n",
       " 'C10H12N5O6P',\n",
       " 'C10H12N5O12P3',\n",
       " 'C14H24N6O3S',\n",
       " 'C10H12N5O10P2',\n",
       " 'C10H12N5O13P3',\n",
       " 'C5H10O5',\n",
       " 'C72H100CoN18O17P',\n",
       " 'C10H12N5O8P',\n",
       " 'C7H14N2O4',\n",
       " 'C6H11O9P',\n",
       " 'C6H11O9P',\n",
       " 'C5H9O8P',\n",
       " 'C9H13N4O8P',\n",
       " 'C7H15NO3',\n",
       " 'C10H12N5O10PS',\n",
       " 'C14H14N5O11P',\n",
       " 'C4H2O4',\n",
       " 'C13H15N4O12P',\n",
       " 'C10H11N4O8P',\n",
       " 'C37H60N7O17P3S',\n",
       " 'C39H64N7O17P3S',\n",
       " 'C9H12N3O8P',\n",
       " 'C19H29N2O12',\n",
       " 'C34H52N6O19',\n",
       " 'C19H29N2O12',\n",
       " 'C34H52N6O19',\n",
       " 'C26H39N5O14',\n",
       " 'C37H57N7O20',\n",
       " 'C37H57N7O20',\n",
       " 'C29H44N6O15',\n",
       " 'C5H16N4',\n",
       " 'C11H16NO7',\n",
       " 'C6H12O6',\n",
       " 'C4H14N2',\n",
       " 'C5H16N4',\n",
       " 'C4H14N2',\n",
       " 'CH4N2O;CH5N2O',\n",
       " 'C15H29O7P1',\n",
       " 'C27H51O8P1',\n",
       " 'C17H33O7P1',\n",
       " 'C31H59O8P1',\n",
       " 'C17H31O7P1',\n",
       " 'C31H55O8P1',\n",
       " 'C19H37O7P1;C19H37O7P',\n",
       " 'C35H67O8P1',\n",
       " 'C19H35O7P1',\n",
       " 'C35H63O8P1',\n",
       " 'C21H41O7P;C21H41O7P1',\n",
       " 'C39H75O8P1',\n",
       " 'C21H39O7P1;C21H39O7P',\n",
       " 'C39H71O8P1',\n",
       " 'C2HO3;C2H1O3',\n",
       " 'C2H5NO2',\n",
       " 'C6H11O9P',\n",
       " 'C4H9NO2S',\n",
       " 'C9H17NO6S',\n",
       " 'C6H9N3O',\n",
       " 'C2H4O2',\n",
       " 'C5H11NO2S',\n",
       " 'CH4S',\n",
       " 'HS;H2S',\n",
       " 'X;C6H7NO2S2R2',\n",
       " 'XH2;C6H9NO2S2R2',\n",
       " 'C10H13N4O9P',\n",
       " 'C19H21N7O6',\n",
       " 'C9H11N3O9P',\n",
       " 'C9H11N3O9P',\n",
       " 'C12H20NO4S2',\n",
       " 'C8H17NOS2',\n",
       " 'C4H4O5',\n",
       " 'C4H4O5',\n",
       " 'C9H13N5O4',\n",
       " 'C3H5O4',\n",
       " 'C3H7NO2',\n",
       " 'C6H12N2O3',\n",
       " 'C6H12N2O3',\n",
       " 'C6H12N2O3',\n",
       " 'C4H8N2O3',\n",
       " 'C3H7NO2',\n",
       " 'C3H7NO2',\n",
       " 'C5H10N2O3',\n",
       " 'C2H5NO2',\n",
       " 'C8H13N2O5',\n",
       " 'C3H7NO3',\n",
       " 'C5H8NO4',\n",
       " 'C8H12N2O5P',\n",
       " 'C8H8NO6P;C8H7NO6P',\n",
       " 'C4H9NO3',\n",
       " 'C6H15N4O2',\n",
       " 'C6H15N4O2',\n",
       " 'CH4O1;CH4O',\n",
       " 'C3H6O3',\n",
       " 'C3H8O3',\n",
       " 'C3H6O2',\n",
       " 'C8H8O',\n",
       " 'C10H9NO',\n",
       " 'C4H8O',\n",
       " 'C4H10O',\n",
       " 'C8H7O2',\n",
       " 'C10H8NO2',\n",
       " 'C2H6NO',\n",
       " 'C4H7O2',\n",
       " 'C3H6O3',\n",
       " 'C3H5O6P',\n",
       " 'C4H7N4O4',\n",
       " 'C3H5N2O4',\n",
       " 'C4H5N4O3;C4H6N4O3',\n",
       " 'C4H5N4O3;C4H6N4O3',\n",
       " 'C37H74N1O8P1',\n",
       " 'C38H74O10P1',\n",
       " 'C6H9O6',\n",
       " 'C12H22O11',\n",
       " 'C6H12O6',\n",
       " 'C50H72O2',\n",
       " 'C51H74O2',\n",
       " 'C50H70O2',\n",
       " 'C51H72O2',\n",
       " 'C5H10N2O3S',\n",
       " 'C5H10N2O3S',\n",
       " 'C5H9NO2',\n",
       " 'C7H12N2O3',\n",
       " 'C12H22O11',\n",
       " 'C19H29N2O12',\n",
       " 'C11H14N2O8P;C11H13N2O8P',\n",
       " 'C6H11N3O4P',\n",
       " 'C6H7N2O5P;C6H8N2O5P',\n",
       " 'C20H24N10O19P4',\n",
       " 'C7H7NO4',\n",
       " 'C12H20N2O9PS;C12H19N2O9PS',\n",
       " 'C5H16N2',\n",
       " 'C3H7O6P',\n",
       " 'C6H15N2O2',\n",
       " 'C21H33N7O13P2S',\n",
       " 'C9H15N4O9P',\n",
       " 'C9H13N4O9P',\n",
       " 'C5H10O5',\n",
       " 'C5H10O5',\n",
       " 'C5H10O5',\n",
       " 'Fe',\n",
       " 'Fe',\n",
       " 'C17H21N4O9P',\n",
       " 'C12H16O7',\n",
       " 'C6H15N4O2',\n",
       " 'CO2',\n",
       " 'C6H13N3O3;C6H14N3O3',\n",
       " 'C6H15N2O2',\n",
       " 'C5H13N2O2',\n",
       " 'C5H13N2O2',\n",
       " 'C10H17N4O6',\n",
       " 'C9H11NO2',\n",
       " 'O4S;SO4',\n",
       " 'C4H6NO7P',\n",
       " 'C4H7NO3',\n",
       " 'C4H6NO4',\n",
       " 'C4H6NO4',\n",
       " 'AsO3',\n",
       " 'AsO3',\n",
       " 'AsO3',\n",
       " 'AsO4',\n",
       " 'AsO4',\n",
       " 'AsO4',\n",
       " 'C5H7N2O5;C5H6N2O5',\n",
       " 'CH2NO5P',\n",
       " 'K',\n",
       " 'K',\n",
       " 'C4H3NO4',\n",
       " 'C49H74O4',\n",
       " 'C49H76O4',\n",
       " 'C5H9NO2',\n",
       " 'C20H30N6O12S2;C20H28N6O12S2',\n",
       " 'C10H15N3O6S;C10H16N3O6S',\n",
       " 'C4H9NO3',\n",
       " 'C15H19N5O20P4',\n",
       " 'C10H15N2O3S',\n",
       " 'C7H5O2',\n",
       " 'CHO3;HCO3',\n",
       " 'C6H13NO2',\n",
       " 'C16H23N5O16P2',\n",
       " 'C5H11NO2',\n",
       " 'C12H21O14P',\n",
       " 'C12H22O11',\n",
       " 'C12H22O11',\n",
       " 'C12H21O14P',\n",
       " 'CH4O1;CH4O',\n",
       " 'C7H14O6',\n",
       " 'C7H14O6',\n",
       " 'C15H19N2O18P2',\n",
       " 'C4H7O7P',\n",
       " 'Cu',\n",
       " 'C10H10N5O8PS2Mo',\n",
       " 'C20H22N10O12P2S2Cu',\n",
       " 'C32H39N7O20P3S',\n",
       " 'C4H10O2',\n",
       " 'C4H10O',\n",
       " 'C4H10O',\n",
       " 'C10H17N2O3',\n",
       " 'S',\n",
       " 'SFe2',\n",
       " 'S2Fe2',\n",
       " 'C4H7N2O3;C4H8N2O3',\n",
       " 'C4H7O5P',\n",
       " 'C4H7O2',\n",
       " 'C4H7O2',\n",
       " 'C7H6O',\n",
       " 'C7H6O',\n",
       " 'C7H6O',\n",
       " 'C6H6O2',\n",
       " 'C7H5O2',\n",
       " 'C31H50N7O18P3S',\n",
       " 'C33H54N7O18P3S',\n",
       " 'C35H58N7O18P3S',\n",
       " 'C26H40N7O17P3S',\n",
       " 'Ca',\n",
       " 'Ca',\n",
       " 'Ca',\n",
       " 'XH2;Fe8S8XH2;Fe2S2X',\n",
       " 'C5H16N2',\n",
       " 'C6H15N2O2',\n",
       " 'C6H6O4',\n",
       " 'C6H6O2',\n",
       " 'C6H6O2',\n",
       " 'C48H72CoN11O8',\n",
       " 'C62CoH88N13O14P;C62H88CoN13O14P',\n",
       " 'Cd',\n",
       " 'Cd',\n",
       " 'Cd',\n",
       " 'K',\n",
       " 'C36H63N3O15P2',\n",
       " 'C40H71N3O15P2',\n",
       " 'C40H67N3O15P2',\n",
       " 'C44H79N3O15P2',\n",
       " 'C44H75N3O15P2',\n",
       " 'C48H87N3O15P2',\n",
       " 'C48H83N3O15P2',\n",
       " 'XH',\n",
       " 'XH',\n",
       " 'C20H22N10O14P2',\n",
       " 'C20H23N10O15P2',\n",
       " 'C14H22N3O17P3',\n",
       " 'C14H23N3O14P2',\n",
       " 'C9H12N3O11P2',\n",
       " 'C3H3O3',\n",
       " 'C5H10N2O3S',\n",
       " 'C5H14NO',\n",
       " 'C5H13NO4S',\n",
       " 'C5H13NO4S',\n",
       " 'C10H8O6',\n",
       " 'C10H9O10P',\n",
       " 'C6H5O7',\n",
       " 'Mg',\n",
       " 'Mg',\n",
       " 'Mn',\n",
       " 'Mn',\n",
       " 'Ni',\n",
       " 'Ni',\n",
       " 'Co',\n",
       " 'Co',\n",
       " 'Zn',\n",
       " 'Zn',\n",
       " 'C4H4O4',\n",
       " 'C4H4O4',\n",
       " 'CHO2;CH1O2',\n",
       " 'C114H202N2O39P2',\n",
       " 'C114H202N2O39P2',\n",
       " 'C2O4',\n",
       " 'C2O4',\n",
       " 'C27H51O8P1',\n",
       " 'C30H58O10P1',\n",
       " 'C31H59O8P1',\n",
       " 'C34H66O10P1',\n",
       " 'C31H55O8P1',\n",
       " 'C34H62O10P1',\n",
       " 'C35H67O8P1',\n",
       " 'C35H63O8P1',\n",
       " 'C38H70O10P1',\n",
       " 'C39H75O8P1',\n",
       " 'C42H82O10P1',\n",
       " 'C39H71O8P1',\n",
       " 'C42H78O10P1',\n",
       " 'C3H8O3',\n",
       " 'Cl',\n",
       " 'Fe',\n",
       " 'C4H5N3O',\n",
       " 'CO2',\n",
       " 'Co',\n",
       " 'Fe8S8X',\n",
       " 'Fe8S8X',\n",
       " 'C19CoN4R21',\n",
       " 'C20H3CoN4R21',\n",
       " 'CO',\n",
       " 'C10H13N5O8P',\n",
       " 'C36H40N4O8',\n",
       " 'C34H38N4O4',\n",
       " 'C7H8O',\n",
       " 'C7H15NO3',\n",
       " 'C7H15NO3',\n",
       " 'C7H15NO2',\n",
       " 'C7H15NO3',\n",
       " 'C7H15NO3',\n",
       " 'C7H15NO2',\n",
       " 'C7H15NO2',\n",
       " 'CrO4',\n",
       " 'CrO4',\n",
       " 'CrO4',\n",
       " 'C4H4N2O2',\n",
       " 'C7H22N3',\n",
       " 'C35H56N7O17P3S',\n",
       " 'C9H11N2O15P3',\n",
       " 'Cu',\n",
       " 'Cu',\n",
       " 'Cu',\n",
       " 'Cu',\n",
       " 'CHN',\n",
       " 'O3S;HSO3',\n",
       " 'CNS',\n",
       " 'HS2O3;O3S2',\n",
       " 'CHN',\n",
       " 'C14H18O4',\n",
       " 'C14H20O4',\n",
       " 'C10H17N5O4R',\n",
       " 'R',\n",
       " 'C3H7NO2S',\n",
       " 'O2',\n",
       " 'C6H12N2O4S2;C7H14N2O4S',\n",
       " 'C5H8NO4',\n",
       " 'C9H13N3O5',\n",
       " 'C9H12N3O10P2',\n",
       " 'C9H12N3O13P3',\n",
       " 'C5H7N3O',\n",
       " 'C10H13N5O3',\n",
       " 'C10H12N4O4',\n",
       " 'C7H14N2O4',\n",
       " 'C5H10O5',\n",
       " 'C10H11N4O13P3',\n",
       " 'C4H7O6P',\n",
       " 'C11H12NO9P',\n",
       " 'C10H19O2',\n",
       " 'C10H19O2',\n",
       " 'C9H11N2O8P',\n",
       " 'C15H25O7P2',\n",
       " 'C5H9O7P2',\n",
       " 'C9H11N2O14P3',\n",
       " 'C9H12N2O5',\n",
       " 'C9H13N3O4',\n",
       " 'C12H23O2',\n",
       " 'C6H9O6',\n",
       " 'C6H8O9P',\n",
       " 'C3H5O6P',\n",
       " 'Fe2S2X',\n",
       " 'C6H12O6',\n",
       " 'C10H12N5O7P',\n",
       " 'C10H13N5O4',\n",
       " 'C8H9O4',\n",
       " 'C27H39N7O20P3S',\n",
       " 'C6H9O3',\n",
       " 'C19H19N7O6',\n",
       " 'C14H13N6O3',\n",
       " 'C11H7O4',\n",
       " 'C40H65O7P2',\n",
       " 'C7H9N5O2',\n",
       " 'C5H5N2O4',\n",
       " 'C5H3N2O4',\n",
       " 'C29H40N7O19P3S',\n",
       " 'C9H12N5O7P',\n",
       " 'C4H6N2O2',\n",
       " 'C9H14N5O8P',\n",
       " 'C9H14N5O8P',\n",
       " 'C20H21N7O6',\n",
       " 'C5H8O4',\n",
       " 'C5H8O4',\n",
       " 'C7H9O6',\n",
       " 'C7H7O5',\n",
       " 'C10H11N4O7P;C10H12N4O7P',\n",
       " 'C6H10O12P2',\n",
       " 'C3H5O3',\n",
       " 'C5H9O7P2',\n",
       " 'C10H17O7P2',\n",
       " 'C9H16N4O6',\n",
       " 'C5H9O8P2',\n",
       " 'C21H24N6O15P2',\n",
       " 'C9H11N2O11P2',\n",
       " 'C10H13N2O14P3',\n",
       " 'C6H10O10P2',\n",
       " 'C6H11O4',\n",
       " 'C5H9O7P',\n",
       " 'C5H10O4',\n",
       " 'C10H15N3O6S;C10H16N3O6S',\n",
       " 'C42H40N4O16;C42H41N4O16',\n",
       " 'C3H5NO3',\n",
       " 'C3H7NO3',\n",
       " 'C3H7NO3',\n",
       " 'C10H13N2O8P',\n",
       " 'C5H6N2O2',\n",
       " 'C5H11O7P',\n",
       " 'C5H9O7P',\n",
       " 'C4H6O8P',\n",
       " 'C55H89O7P2',\n",
       " 'C25H38N7O18P3S',\n",
       " 'C25H38N7O18P3S',\n",
       " 'C27H42N7O18P3S',\n",
       " 'C29H46N7O18P3S',\n",
       " 'C37H62N7O18P3S',\n",
       " 'C26H40N7O18P3S',\n",
       " 'C84H148N2O37P2',\n",
       " 'C96H170N2O38P2',\n",
       " 'C110H196N2O39P2',\n",
       " 'C100H176N2O38P2',\n",
       " 'C114H202N2O39P2',\n",
       " 'C112H202N3O42P3',\n",
       " 'C112H202N3O42P3',\n",
       " 'C30H27N3O15',\n",
       " 'C30H27FeN3O15',\n",
       " 'C42H47N3O25',\n",
       " 'C30H27N3O15',\n",
       " 'C6H9O7',\n",
       " 'C11H12N2O2',\n",
       " 'C5H11NO2',\n",
       " 'C9H11NO3',\n",
       " 'C2H8NO',\n",
       " 'C2H8NO',\n",
       " 'C2H6O',\n",
       " 'C2H6O',\n",
       " 'H2',\n",
       " 'C6H12O6',\n",
       " 'C14H27O2',\n",
       " 'C14H25O2',\n",
       " 'C16H31O2',\n",
       " 'C16H29O2',\n",
       " 'C18H35O2',\n",
       " 'C18H33O2',\n",
       " 'C4H7O2',\n",
       " 'C6H11O2',\n",
       " 'C8H15O2',\n",
       " 'C6H11O9P',\n",
       " 'C7H12O13P2',\n",
       " 'C34H32FeN4O4;C34FeH30N4O4;C34H30FeN4O4',\n",
       " 'C34H32N4O4',\n",
       " 'C22H36N7O17P3S;C22H32N7O17P3S',\n",
       " 'CHO2;CH1O2',\n",
       " 'CH3O3S',\n",
       " 'C2H7NO3S;C2H6NO3S',\n",
       " 'Fe',\n",
       " 'Fe',\n",
       " 'Fe',\n",
       " 'C30H27FeN3O15',\n",
       " 'C30H27N3O15',\n",
       " 'C30H27FeN3O15',\n",
       " 'S4Fe3',\n",
       " 'S4Fe4',\n",
       " 'N2O',\n",
       " 'NO',\n",
       " 'C12H21O14P',\n",
       " 'C6H9N2O4;C6H8N2O4',\n",
       " 'X',\n",
       " 'XH2',\n",
       " 'C7H9N5O8P2;C7H8N5O8P2',\n",
       " 'C20H20N7O6;C20H21N7O6',\n",
       " 'C23H31N7O19P3S',\n",
       " 'C6H12O6',\n",
       " 'C6H12O6',\n",
       " 'C20H22N8O6',\n",
       " 'C4H2O4',\n",
       " 'C4H2O4',\n",
       " 'C6H13NO8P',\n",
       " 'C6H12O6',\n",
       " 'C5H9NO3',\n",
       " 'C5H14NO6P',\n",
       " 'C5H14NO6P',\n",
       " 'C5H14NO6P',\n",
       " 'C6H14O8P',\n",
       " 'C6H14O8P',\n",
       " 'C6H14O8P',\n",
       " 'C5H6NO2',\n",
       " 'C5H9NO3',\n",
       " 'C6H11O9P',\n",
       " 'C6H11O7',\n",
       " 'C6H12O6',\n",
       " 'C6H12O6',\n",
       " 'C6H11O9P',\n",
       " 'C6H13O9P',\n",
       " 'C6H14O6',\n",
       " 'C6H9O7',\n",
       " 'C6H9O7',\n",
       " 'C6H14NO5',\n",
       " 'C8H12N2O9P;C8H13N2O9P',\n",
       " 'C7H14N2O8P;C7H13N2O8P',\n",
       " 'C2H3O3',\n",
       " 'CH6NS2X',\n",
       " 'S2X',\n",
       " 'H2S2X',\n",
       " 'C10H12N5O20P5;C10H11N5O20P5',\n",
       " 'C6H9O7',\n",
       " 'C17H20N2O8',\n",
       " 'C6H11O7',\n",
       " 'C3H3O4',\n",
       " 'C6H9O7',\n",
       " 'C6H9O7',\n",
       " 'C5H10N2O3',\n",
       " 'C95H152N8O28P2',\n",
       " 'C5H8NO4',\n",
       " 'C5H11NO7P',\n",
       " 'C5H7NO3R;C5H8NO4X',\n",
       " 'R;HOX;C15H21N5O10PR',\n",
       " 'C3H7O6P',\n",
       " 'C2H5NO2',\n",
       " 'C5H11NO2',\n",
       " 'C3H3O4',\n",
       " 'C3H8O3',\n",
       " 'C13H19N3O8S;C13H20N3O8S',\n",
       " 'C10H11N4O9P',\n",
       " 'C5H4N5O;C5H5N5O',\n",
       " 'C3H7NO3',\n",
       " 'C6H12O6',\n",
       " 'X',\n",
       " 'XH2',\n",
       " 'C17H36N6O5S',\n",
       " 'H2O2',\n",
       " 'C10H11N4O15P3',\n",
       " 'C5H4N4O2',\n",
       " 'C5H4N5O;C5H5N5O',\n",
       " 'C5H4N5O;C5H5N5O',\n",
       " 'HS;H2S',\n",
       " 'H2',\n",
       " 'C26H38N7O18P3S',\n",
       " 'H3NO',\n",
       " 'C47H69O3',\n",
       " 'C6H13N3O;C6H12N3O',\n",
       " 'XH2',\n",
       " 'X',\n",
       " 'C34H32FeN4O4;C34FeH30N4O4;C34H30FeN4O4',\n",
       " 'C6H9NOS',\n",
       " 'C6H8NO4PS',\n",
       " 'Hg',\n",
       " 'Hg',\n",
       " 'Hg',\n",
       " 'C6H9N3O2',\n",
       " 'C6H9N3O2',\n",
       " 'C6H5N2O2',\n",
       " 'C40H38N4O17',\n",
       " 'C10H13N2O4',\n",
       " 'C27H40N7O20P3S;C27H39N7O20P3S',\n",
       " 'C6H11O4',\n",
       " 'C4H9NO3',\n",
       " 'C5H7O4',\n",
       " 'C6H9O4',\n",
       " 'C8H7O3',\n",
       " 'C6H6O2',\n",
       " 'C4H8NO6P',\n",
       " 'C8H12NO6',\n",
       " 'C5H4N4O',\n",
       " 'C5H4N4O',\n",
       " 'C6H11O2',\n",
       " 'C8H13N2O2',\n",
       " 'C2H7NO3S;C2H6NO3S',\n",
       " 'HSR',\n",
       " 'HS2R',\n",
       " 'H8O2S6R',\n",
       " 'H4O2S8Fe2R',\n",
       " 'O2S10Fe4R',\n",
       " 'H4O2S10Fe4R',\n",
       " 'C4H7O2',\n",
       " 'C6H5O7',\n",
       " 'C10H9NO',\n",
       " 'C6H11O7',\n",
       " 'C6H11O7',\n",
       " 'C6H9N2O6P',\n",
       " 'C15H22N5O15P2;C15H21N5O15P2',\n",
       " 'C6H13NO2',\n",
       " 'C6H13NO2',\n",
       " 'C11H8NO3',\n",
       " 'C6H12O6',\n",
       " 'C31H39N8O17P3S',\n",
       " 'C9H7O3',\n",
       " 'C84H148N2O37P2',\n",
       " 'C84H148N2O37P2',\n",
       " 'C17H24N3O15P',\n",
       " 'C8H13O8',\n",
       " 'C8H12O11P',\n",
       " 'C12H22O11',\n",
       " 'C12H22O11',\n",
       " 'C12H22O11',\n",
       " 'C8H13N2O5',\n",
       " 'C8H13N2O5',\n",
       " 'C3H6NO5S',\n",
       " 'C3H6NO5S',\n",
       " 'C6H13NO2',\n",
       " 'C110H196N2O39P2',\n",
       " 'C110H196N2O39P2',\n",
       " 'C8H13OS2',\n",
       " 'C8H15O',\n",
       " 'C68H127N2O20P',\n",
       " 'C34H64NO12P',\n",
       " 'C43H75N3O20P2',\n",
       " 'C17H36NO7P1',\n",
       " 'C19H40NO7P1',\n",
       " 'C23H46NO7P1',\n",
       " 'C18H36O9P1',\n",
       " 'C20H40O9P1',\n",
       " 'C22H44O9P1',\n",
       " 'C24H48O9P1',\n",
       " 'C24H46O9P1',\n",
       " 'C6H13O9P',\n",
       " 'C12H21O14P',\n",
       " 'C4H4O5',\n",
       " 'C12H22O11',\n",
       " 'C12H22O11',\n",
       " 'C18H32O16',\n",
       " 'C6H12O6',\n",
       " 'C12H22O11',\n",
       " 'C18H32O16',\n",
       " ...]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# CarveMe has some particularly strange \n",
    "[x.formula for x in model.metabolites]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': 'nadh_c',\n",
       " 'name': 'Nicotinamide adenine dinucleotide - reduced',\n",
       " 'notes': {'FORMULA': 'C21H27N7O14P2',\n",
       "  'BioCyc': 'META:NADH',\n",
       "  'SEED Compound': 'cpd00004',\n",
       "  'UniPathway Compound': 'UPC00004',\n",
       "  'KEGG Compound': 'C00004',\n",
       "  'BioPath Molecule': 'NADH',\n",
       "  'MetaNetX (MNX) Chemical': 'MNXM10',\n",
       "  'Reactome': '192305;194697;29362;73473',\n",
       "  'Human Metabolome Database': 'HMDB01487'},\n",
       " '_annotation': {},\n",
       " '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       " '_reaction': {<Reaction 4MBZALDH at 0x7fc449961b50>,\n",
       "  <Reaction 4MBZDH at 0x7fc44993c790>,\n",
       "  <Reaction ABUTD at 0x7fc4499838e0>,\n",
       "  <Reaction ACALD at 0x7fc4499830d0>,\n",
       "  <Reaction ACOAD1 at 0x7fc44999dee0>,\n",
       "  <Reaction ACOAD2 at 0x7fc4499a7ee0>,\n",
       "  <Reaction ACOAD4_1 at 0x7fc4499a7f70>,\n",
       "  <Reaction ACOAD5_1 at 0x7fc44997b040>,\n",
       "  <Reaction ACTD2 at 0x7fc4499ccf70>,\n",
       "  <Reaction AKGDH at 0x7fc449a12850>,\n",
       "  <Reaction ALAD_L at 0x7fc449a12df0>,\n",
       "  <Reaction ALCD19 at 0x7fc449a2eb80>,\n",
       "  <Reaction ALCD1 at 0x7fc4499f8d60>,\n",
       "  <Reaction ALCD2x at 0x7fc449a2e850>,\n",
       "  <Reaction ALCD4 at 0x7fc449a23d00>,\n",
       "  <Reaction ALDD19xr at 0x7fc449a2ee50>,\n",
       "  <Reaction ALDD20x at 0x7fc449a37a60>,\n",
       "  <Reaction ALDD2x at 0x7fc449a2edc0>,\n",
       "  <Reaction ALDD31 at 0x7fc449a2ed60>,\n",
       "  <Reaction AMPMS2 at 0x7fc449a51d90>,\n",
       "  <Reaction BTDD_RR at 0x7fc449aa7f70>,\n",
       "  <Reaction BZDH at 0x7fc449ab9e80>,\n",
       "  <Reaction DH4MCHCDH at 0x7fc449b91ca0>,\n",
       "  <Reaction DMPPS at 0x7fc449bb96a0>,\n",
       "  <Reaction DURADx at 0x7fc449bcef40>,\n",
       "  <Reaction E4PD at 0x7fc449bcef10>,\n",
       "  <Reaction EAR100x at 0x7fc449bcefa0>,\n",
       "  <Reaction EAR120x at 0x7fc449be2f70>,\n",
       "  <Reaction EAR121x at 0x7fc449be2e20>,\n",
       "  <Reaction EAR140x at 0x7fc449be2c10>,\n",
       "  <Reaction EAR141x at 0x7fc449be2cd0>,\n",
       "  <Reaction EAR160x at 0x7fc449be2580>,\n",
       "  <Reaction EAR161x at 0x7fc449be25e0>,\n",
       "  <Reaction EAR180x at 0x7fc449be2730>,\n",
       "  <Reaction EAR181x at 0x7fc449bcee20>,\n",
       "  <Reaction EAR40x at 0x7fc449be2d90>,\n",
       "  <Reaction EAR60x at 0x7fc449be2a00>,\n",
       "  <Reaction EAR80x at 0x7fc449be2e50>,\n",
       "  <Reaction FAO3 at 0x7fc449c34c10>,\n",
       "  <Reaction FLNDPR2r at 0x7fc449c6eeb0>,\n",
       "  <Reaction FMNRx at 0x7fc449c6e9d0>,\n",
       "  <Reaction FRNDPR2r_1 at 0x7fc449c74a30>,\n",
       "  <Reaction G3PD1 at 0x7fc449c94c40>,\n",
       "  <Reaction GAPD at 0x7fc449ca7940>,\n",
       "  <Reaction GCALDD at 0x7fc449cbdf70>,\n",
       "  <Reaction GCCc at 0x7fc449cbdc70>,\n",
       "  <Reaction GLTPD at 0x7fc449cd1820>,\n",
       "  <Reaction GLUDxi at 0x7fc449cf6fa0>,\n",
       "  <Reaction GLYALDDr at 0x7fc449d07ca0>,\n",
       "  <Reaction GLYCDx at 0x7fc449d07a90>,\n",
       "  <Reaction GLYCL at 0x7fc449d07e80>,\n",
       "  <Reaction HACD1 at 0x7fc449d57be0>,\n",
       "  <Reaction HACD1_2 at 0x7fc449d57dc0>,\n",
       "  <Reaction HACD1i at 0x7fc449d43b20>,\n",
       "  <Reaction HACD2 at 0x7fc449d57e80>,\n",
       "  <Reaction HACD2i at 0x7fc449d43fa0>,\n",
       "  <Reaction HACD3 at 0x7fc449d57670>,\n",
       "  <Reaction HACD3i at 0x7fc449d57d90>,\n",
       "  <Reaction HACD4 at 0x7fc449d57af0>,\n",
       "  <Reaction HACD4i at 0x7fc449d43dc0>,\n",
       "  <Reaction HACD5 at 0x7fc449d57bb0>,\n",
       "  <Reaction HACD5i at 0x7fc449d57940>,\n",
       "  <Reaction HACD6 at 0x7fc449d43f70>,\n",
       "  <Reaction HACD6i at 0x7fc449d57ac0>,\n",
       "  <Reaction HACD7 at 0x7fc449d69d60>,\n",
       "  <Reaction HACD7i at 0x7fc449d69970>,\n",
       "  <Reaction HACD8 at 0x7fc449d69b20>,\n",
       "  <Reaction HACD8i at 0x7fc449d69eb0>,\n",
       "  <Reaction HACD9 at 0x7fc449d69ca0>,\n",
       "  <Reaction HADPCOADH3 at 0x7fc449d43cd0>,\n",
       "  <Reaction HAMR at 0x7fc449d57d00>,\n",
       "  <Reaction HISTD at 0x7fc449d7eb80>,\n",
       "  <Reaction HXAND at 0x7fc449dab7c0>,\n",
       "  <Reaction HYDFDN at 0x7fc449dabb20>,\n",
       "  <Reaction ICDHx at 0x7fc449dbe7c0>,\n",
       "  <Reaction ILEDHr at 0x7fc449cf6040>,\n",
       "  <Reaction IMPD at 0x7fc45d380580>,\n",
       "  <Reaction INS2D at 0x7fc449ce3c70>,\n",
       "  <Reaction IPDPS at 0x7fc449dd2fa0>,\n",
       "  <Reaction LCARS at 0x7fc449e334c0>,\n",
       "  <Reaction LDH_D at 0x7fc449e33d00>,\n",
       "  <Reaction LDH_L at 0x7fc449e33af0>,\n",
       "  <Reaction LIPOS at 0x7fc449e48eb0>,\n",
       "  <Reaction LPD5 at 0x7fc449e48f40>,\n",
       "  <Reaction M1PD at 0x7fc449e82460>,\n",
       "  <Reaction MDH at 0x7fc449ea77c0>,\n",
       "  <Reaction ME1 at 0x7fc449ea7e20>,\n",
       "  <Reaction MECDPDH2 at 0x7fc449ea7940>,\n",
       "  <Reaction MMSAD2 at 0x7fc449ee0df0>,\n",
       "  <Reaction MOADSUx at 0x7fc449ef36a0>,\n",
       "  <Reaction MTHFR2 at 0x7fc449f12d30>,\n",
       "  <Reaction MTHFR2_1 at 0x7fc449f12eb0>,\n",
       "  <Reaction N2OR at 0x7fc449f12fa0>,\n",
       "  <Reaction NADFADOR at 0x7fc449f2c9d0>,\n",
       "  <Reaction NADH10 at 0x7fc449f2c940>,\n",
       "  <Reaction NADH5 at 0x7fc449f2cee0>,\n",
       "  <Reaction NADH6 at 0x7fc449f2cf70>,\n",
       "  <Reaction NADH7 at 0x7fc449f12970>,\n",
       "  <Reaction NADH8 at 0x7fc449f2ca60>,\n",
       "  <Reaction NADH9 at 0x7fc449f2cf40>,\n",
       "  <Reaction NADHDH at 0x7fc449f12cd0>,\n",
       "  <Reaction NADHNQR at 0x7fc449f2cb50>,\n",
       "  <Reaction NHFRBO at 0x7fc449f51f70>,\n",
       "  <Reaction NOX at 0x7fc449f65b50>,\n",
       "  <Reaction OBDHc at 0x7fc449f8eca0>,\n",
       "  <Reaction OIVD1r at 0x7fc449fb8a90>,\n",
       "  <Reaction OIVD2 at 0x7fc449fb8be0>,\n",
       "  <Reaction OIVD3 at 0x7fc449fb89d0>,\n",
       "  <Reaction P5CRx at 0x7fc449fccca0>,\n",
       "  <Reaction PDH at 0x7fc449fdfe80>,\n",
       "  <Reaction PDHcr at 0x7fc44a007af0>,\n",
       "  <Reaction PDX5PO2 at 0x7fc449fdfaf0>,\n",
       "  <Reaction PDX5PS at 0x7fc449fccf10>,\n",
       "  <Reaction PERD at 0x7fc4498d0970>,\n",
       "  <Reaction PGCD at 0x7fc4498d07f0>,\n",
       "  <Reaction PPND at 0x7fc449a1c5e0>,\n",
       "  <Reaction PTOLDOX at 0x7fc44a0e7ac0>,\n",
       "  <Reaction PXMO at 0x7fc44a10ca60>,\n",
       "  <Reaction RDXR at 0x7fc44a130e20>,\n",
       "  <Reaction RNF at 0x7fc44a14bf10>,\n",
       "  <Reaction SBTD_D2 at 0x7fc44a179fd0>,\n",
       "  <Reaction SBTPD at 0x7fc44a18c970>,\n",
       "  <Reaction SHCHD2 at 0x7fc44a18cac0>,\n",
       "  <Reaction SSALx at 0x7fc44a1b5b20>,\n",
       "  <Reaction THRD at 0x7fc44a1ebca0>,\n",
       "  <Reaction UDPGD at 0x7fc44a23aca0>,\n",
       "  <Reaction VNDH_2 at 0x7fc44a263f40>,\n",
       "  <Reaction XAND at 0x7fc44a277d30>},\n",
       " 'formula': 'C21H27N7O14P2',\n",
       " 'compartment': 'C_c',\n",
       " 'charge': 0,\n",
       " '_bound': 0.0}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.metabolites[33].__dict__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>2AGPGAT160</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>2-acyl-glycerophospho-glycerol acyltransferase (n-C16:0)</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fc4498fceb0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>2agpg160_c + atp_c + hdca_c --> amp_c + pg160_c + ppi_c</p>\n",
       "                    <p style='text-align:right'>2-Acyl-sn-glycero-3-phosphoglycerol (n-C16:0) + ATP + Hexadecanoate (n-C16:0) --> AMP + Phosphatidylglycerol (dihexadecanoyl, n-C16:0) + Diphosphate</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td>EDU36256_1</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction 2AGPGAT160 at 0x7fc4498fceb0>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# reaction entries, Readily convert to list of reactions\n",
    "model.reactions[33]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': '2AGPGAT160',\n",
       " 'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C16:0)',\n",
       " 'notes': {'MetaNetX (MNX) Equation': 'MNXR68143'},\n",
       " '_annotation': {},\n",
       " '_gene_reaction_rule': 'EDU36256_1',\n",
       " 'subsystem': '',\n",
       " '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       " '_metabolites': {<Metabolite 2agpg160_c at 0x7fc463a6dfd0>: -1.0,\n",
       "  <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "  <Metabolite hdca_c at 0x7fc463a6dd60>: -1.0,\n",
       "  <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "  <Metabolite pg160_c at 0x7fc463adc310>: 1.0,\n",
       "  <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       " '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       " '_lower_bound': 0.0,\n",
       " '_upper_bound': 1000.0}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions[33].__dict__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# NO group or pathway information\n",
    "# model.groups[33]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': 'nadh_c',\n",
       " 'name': 'Nicotinamide adenine dinucleotide - reduced',\n",
       " 'notes': {'FORMULA': 'C21H27N7O14P2',\n",
       "  'BioCyc': 'META:NADH',\n",
       "  'SEED Compound': 'cpd00004',\n",
       "  'UniPathway Compound': 'UPC00004',\n",
       "  'KEGG Compound': 'C00004',\n",
       "  'BioPath Molecule': 'NADH',\n",
       "  'MetaNetX (MNX) Chemical': 'MNXM10',\n",
       "  'Reactome': '192305;194697;29362;73473',\n",
       "  'Human Metabolome Database': 'HMDB01487'},\n",
       " '_annotation': {},\n",
       " '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       " '_reaction': {<Reaction 4MBZALDH at 0x7fc449961b50>,\n",
       "  <Reaction 4MBZDH at 0x7fc44993c790>,\n",
       "  <Reaction ABUTD at 0x7fc4499838e0>,\n",
       "  <Reaction ACALD at 0x7fc4499830d0>,\n",
       "  <Reaction ACOAD1 at 0x7fc44999dee0>,\n",
       "  <Reaction ACOAD2 at 0x7fc4499a7ee0>,\n",
       "  <Reaction ACOAD4_1 at 0x7fc4499a7f70>,\n",
       "  <Reaction ACOAD5_1 at 0x7fc44997b040>,\n",
       "  <Reaction ACTD2 at 0x7fc4499ccf70>,\n",
       "  <Reaction AKGDH at 0x7fc449a12850>,\n",
       "  <Reaction ALAD_L at 0x7fc449a12df0>,\n",
       "  <Reaction ALCD19 at 0x7fc449a2eb80>,\n",
       "  <Reaction ALCD1 at 0x7fc4499f8d60>,\n",
       "  <Reaction ALCD2x at 0x7fc449a2e850>,\n",
       "  <Reaction ALCD4 at 0x7fc449a23d00>,\n",
       "  <Reaction ALDD19xr at 0x7fc449a2ee50>,\n",
       "  <Reaction ALDD20x at 0x7fc449a37a60>,\n",
       "  <Reaction ALDD2x at 0x7fc449a2edc0>,\n",
       "  <Reaction ALDD31 at 0x7fc449a2ed60>,\n",
       "  <Reaction AMPMS2 at 0x7fc449a51d90>,\n",
       "  <Reaction BTDD_RR at 0x7fc449aa7f70>,\n",
       "  <Reaction BZDH at 0x7fc449ab9e80>,\n",
       "  <Reaction DH4MCHCDH at 0x7fc449b91ca0>,\n",
       "  <Reaction DMPPS at 0x7fc449bb96a0>,\n",
       "  <Reaction DURADx at 0x7fc449bcef40>,\n",
       "  <Reaction E4PD at 0x7fc449bcef10>,\n",
       "  <Reaction EAR100x at 0x7fc449bcefa0>,\n",
       "  <Reaction EAR120x at 0x7fc449be2f70>,\n",
       "  <Reaction EAR121x at 0x7fc449be2e20>,\n",
       "  <Reaction EAR140x at 0x7fc449be2c10>,\n",
       "  <Reaction EAR141x at 0x7fc449be2cd0>,\n",
       "  <Reaction EAR160x at 0x7fc449be2580>,\n",
       "  <Reaction EAR161x at 0x7fc449be25e0>,\n",
       "  <Reaction EAR180x at 0x7fc449be2730>,\n",
       "  <Reaction EAR181x at 0x7fc449bcee20>,\n",
       "  <Reaction EAR40x at 0x7fc449be2d90>,\n",
       "  <Reaction EAR60x at 0x7fc449be2a00>,\n",
       "  <Reaction EAR80x at 0x7fc449be2e50>,\n",
       "  <Reaction FAO3 at 0x7fc449c34c10>,\n",
       "  <Reaction FLNDPR2r at 0x7fc449c6eeb0>,\n",
       "  <Reaction FMNRx at 0x7fc449c6e9d0>,\n",
       "  <Reaction FRNDPR2r_1 at 0x7fc449c74a30>,\n",
       "  <Reaction G3PD1 at 0x7fc449c94c40>,\n",
       "  <Reaction GAPD at 0x7fc449ca7940>,\n",
       "  <Reaction GCALDD at 0x7fc449cbdf70>,\n",
       "  <Reaction GCCc at 0x7fc449cbdc70>,\n",
       "  <Reaction GLTPD at 0x7fc449cd1820>,\n",
       "  <Reaction GLUDxi at 0x7fc449cf6fa0>,\n",
       "  <Reaction GLYALDDr at 0x7fc449d07ca0>,\n",
       "  <Reaction GLYCDx at 0x7fc449d07a90>,\n",
       "  <Reaction GLYCL at 0x7fc449d07e80>,\n",
       "  <Reaction HACD1 at 0x7fc449d57be0>,\n",
       "  <Reaction HACD1_2 at 0x7fc449d57dc0>,\n",
       "  <Reaction HACD1i at 0x7fc449d43b20>,\n",
       "  <Reaction HACD2 at 0x7fc449d57e80>,\n",
       "  <Reaction HACD2i at 0x7fc449d43fa0>,\n",
       "  <Reaction HACD3 at 0x7fc449d57670>,\n",
       "  <Reaction HACD3i at 0x7fc449d57d90>,\n",
       "  <Reaction HACD4 at 0x7fc449d57af0>,\n",
       "  <Reaction HACD4i at 0x7fc449d43dc0>,\n",
       "  <Reaction HACD5 at 0x7fc449d57bb0>,\n",
       "  <Reaction HACD5i at 0x7fc449d57940>,\n",
       "  <Reaction HACD6 at 0x7fc449d43f70>,\n",
       "  <Reaction HACD6i at 0x7fc449d57ac0>,\n",
       "  <Reaction HACD7 at 0x7fc449d69d60>,\n",
       "  <Reaction HACD7i at 0x7fc449d69970>,\n",
       "  <Reaction HACD8 at 0x7fc449d69b20>,\n",
       "  <Reaction HACD8i at 0x7fc449d69eb0>,\n",
       "  <Reaction HACD9 at 0x7fc449d69ca0>,\n",
       "  <Reaction HADPCOADH3 at 0x7fc449d43cd0>,\n",
       "  <Reaction HAMR at 0x7fc449d57d00>,\n",
       "  <Reaction HISTD at 0x7fc449d7eb80>,\n",
       "  <Reaction HXAND at 0x7fc449dab7c0>,\n",
       "  <Reaction HYDFDN at 0x7fc449dabb20>,\n",
       "  <Reaction ICDHx at 0x7fc449dbe7c0>,\n",
       "  <Reaction ILEDHr at 0x7fc449cf6040>,\n",
       "  <Reaction IMPD at 0x7fc45d380580>,\n",
       "  <Reaction INS2D at 0x7fc449ce3c70>,\n",
       "  <Reaction IPDPS at 0x7fc449dd2fa0>,\n",
       "  <Reaction LCARS at 0x7fc449e334c0>,\n",
       "  <Reaction LDH_D at 0x7fc449e33d00>,\n",
       "  <Reaction LDH_L at 0x7fc449e33af0>,\n",
       "  <Reaction LIPOS at 0x7fc449e48eb0>,\n",
       "  <Reaction LPD5 at 0x7fc449e48f40>,\n",
       "  <Reaction M1PD at 0x7fc449e82460>,\n",
       "  <Reaction MDH at 0x7fc449ea77c0>,\n",
       "  <Reaction ME1 at 0x7fc449ea7e20>,\n",
       "  <Reaction MECDPDH2 at 0x7fc449ea7940>,\n",
       "  <Reaction MMSAD2 at 0x7fc449ee0df0>,\n",
       "  <Reaction MOADSUx at 0x7fc449ef36a0>,\n",
       "  <Reaction MTHFR2 at 0x7fc449f12d30>,\n",
       "  <Reaction MTHFR2_1 at 0x7fc449f12eb0>,\n",
       "  <Reaction N2OR at 0x7fc449f12fa0>,\n",
       "  <Reaction NADFADOR at 0x7fc449f2c9d0>,\n",
       "  <Reaction NADH10 at 0x7fc449f2c940>,\n",
       "  <Reaction NADH5 at 0x7fc449f2cee0>,\n",
       "  <Reaction NADH6 at 0x7fc449f2cf70>,\n",
       "  <Reaction NADH7 at 0x7fc449f12970>,\n",
       "  <Reaction NADH8 at 0x7fc449f2ca60>,\n",
       "  <Reaction NADH9 at 0x7fc449f2cf40>,\n",
       "  <Reaction NADHDH at 0x7fc449f12cd0>,\n",
       "  <Reaction NADHNQR at 0x7fc449f2cb50>,\n",
       "  <Reaction NHFRBO at 0x7fc449f51f70>,\n",
       "  <Reaction NOX at 0x7fc449f65b50>,\n",
       "  <Reaction OBDHc at 0x7fc449f8eca0>,\n",
       "  <Reaction OIVD1r at 0x7fc449fb8a90>,\n",
       "  <Reaction OIVD2 at 0x7fc449fb8be0>,\n",
       "  <Reaction OIVD3 at 0x7fc449fb89d0>,\n",
       "  <Reaction P5CRx at 0x7fc449fccca0>,\n",
       "  <Reaction PDH at 0x7fc449fdfe80>,\n",
       "  <Reaction PDHcr at 0x7fc44a007af0>,\n",
       "  <Reaction PDX5PO2 at 0x7fc449fdfaf0>,\n",
       "  <Reaction PDX5PS at 0x7fc449fccf10>,\n",
       "  <Reaction PERD at 0x7fc4498d0970>,\n",
       "  <Reaction PGCD at 0x7fc4498d07f0>,\n",
       "  <Reaction PPND at 0x7fc449a1c5e0>,\n",
       "  <Reaction PTOLDOX at 0x7fc44a0e7ac0>,\n",
       "  <Reaction PXMO at 0x7fc44a10ca60>,\n",
       "  <Reaction RDXR at 0x7fc44a130e20>,\n",
       "  <Reaction RNF at 0x7fc44a14bf10>,\n",
       "  <Reaction SBTD_D2 at 0x7fc44a179fd0>,\n",
       "  <Reaction SBTPD at 0x7fc44a18c970>,\n",
       "  <Reaction SHCHD2 at 0x7fc44a18cac0>,\n",
       "  <Reaction SSALx at 0x7fc44a1b5b20>,\n",
       "  <Reaction THRD at 0x7fc44a1ebca0>,\n",
       "  <Reaction UDPGD at 0x7fc44a23aca0>,\n",
       "  <Reaction VNDH_2 at 0x7fc44a263f40>,\n",
       "  <Reaction XAND at 0x7fc44a277d30>},\n",
       " 'formula': 'C21H27N7O14P2',\n",
       " 'compartment': 'C_c',\n",
       " 'charge': 0,\n",
       " '_bound': 0.0}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.metabolites[33].__dict__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Port metabolite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def port_metabolite(M):\n",
    "    # convert cobra Metabolite to metDataModel Compound\n",
    "    Cpd = Compound()\n",
    "    Cpd.src_id = remove_compartment_by_split(M.id,'_')\n",
    "    Cpd.id = remove_compartment_by_split(M.id,'_')              # temporarily the same with the source id\n",
    "    Cpd.name = M.name\n",
    "    Cpd.charged_formula = M.formula\n",
    "    Cpd.db_ids = [[model_name,Cpd.src_id]] # using src_id to also reference Rat-GEM ID in db_ids field\n",
    "    for k,v in M.notes.items():\n",
    "        if isinstance(v,list):\n",
    "            Cpd.db_ids.append([[k,x] for x in v])\n",
    "        else: \n",
    "            if \":\" in v:\n",
    "                Cpd.db_ids.append([k,v.split(\":\")[1]])\n",
    "            else:\n",
    "                Cpd.db_ids.append([k,v])\n",
    "    \n",
    "    inchi_list = [x[1].split('=')[1] for x in Cpd.db_ids if x[0] == 'inchi']\n",
    "    if len(inchi_list) ==1:\n",
    "        Cpd.inchi = inchi_list[0]\n",
    "    elif len(inchi_list) >1:\n",
    "        Cpd.inchi = inchi_list\n",
    "        \n",
    "    return Cpd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "myCpds = []\n",
    "for i in range(len(model.metabolites)):\n",
    "    myCpds.append(port_metabolite(model.metabolites[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1219"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(myCpds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove duplicated compounds\n",
    "myCpds = remove_duplicate_cpd(myCpds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "854"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(myCpds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fetch_CarveMe_GEM_charge_formula(compound_list,Bigg_json_path,overwrite = True):\n",
    "    with open(Bigg_json_path,'r') as f:\n",
    "        list_Bigg_cpd = json.load(f)\n",
    "    Bigg_dict = {}\n",
    "    for Bigg_cpd in list_Bigg_cpd:\n",
    "        Bigg_dict.update({Bigg_cpd['id']:Bigg_cpd})\n",
    "    new_cpd_list = []\n",
    "    for myCpd in compound_list:\n",
    "        for k,v in Bigg_dict.items():\n",
    "            if myCpd.id == k:\n",
    "                if overwrite == True:\n",
    "                    myCpd.charge = v['charge']\n",
    "                    myCpd.charged_formula = v['charged_formula'] # the Bigg json is using `identifiers` rather than `db_ids`\n",
    "                break\n",
    "        new_cpd_list.append(myCpd)\n",
    "    return new_cpd_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('../jms/data/staged/Bigg_FM_CG_updated0427.json','r') as f:\n",
    "    list_Bigg_cpd = json.load(f)\n",
    "Bigg_dict = {}\n",
    "for Bigg_cpd in list_Bigg_cpd:\n",
    "    Bigg_dict.update({Bigg_cpd['id']:Bigg_cpd})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'idon__L',\n",
       " 'name': 'L-Idonate',\n",
       " 'identifiers': [['BIGG', 'idon_DASH_L_p'],\n",
       "  ['BIGG', 'idon_L_p'],\n",
       "  ['BIGG', 'idon__L'],\n",
       "  ['BIGG', 'idon__L_p'],\n",
       "  ['KEGG', 'C00770'],\n",
       "  ['CHEBI', '13126'],\n",
       "  ['CHEBI', '17796'],\n",
       "  ['CHEBI', '21335'],\n",
       "  ['CHEBI', '21336'],\n",
       "  ['CHEBI', '57659'],\n",
       "  ['CHEBI', '58494'],\n",
       "  ['CHEBI', '6250'],\n",
       "  ['BioCyc', 'L-IDONATE'],\n",
       "  ['MetaNetX', 'MNXM1565'],\n",
       "  ['inchikey', 'RGHNJXZEOKUKBD-SKNVOMKLSA-M'],\n",
       "  ['SEED', 'cpd00573']],\n",
       " 'neutral_formula': '',\n",
       " 'charge': -1,\n",
       " 'charged_formula': 'C6H11O7',\n",
       " 'neutral_mono_mass': 0.0,\n",
       " 'SMILES': '',\n",
       " 'inchi': ''}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Bigg_dict['idon__L']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "myCpds = fetch_CarveMe_GEM_charge_formula(myCpds,'../jms/data/staged/Bigg_FM_CG_updated0427.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'internal_id': '',\n",
       " 'id': 'udp',\n",
       " 'name': 'UDP',\n",
       " 'db_ids': [['Clostridium_sporogenes-Carve', 'udp'],\n",
       "  ['FORMULA', 'C9H11N2O12P2'],\n",
       "  ['BioCyc', 'UDP;META'],\n",
       "  ['SEED Compound', 'cpd00014'],\n",
       "  ['UniPathway Compound', 'UPC00015'],\n",
       "  ['KEGG Compound', 'C00015;G10619'],\n",
       "  ['BioPath Molecule', 'Uridine-5-prime-diphosphate'],\n",
       "  ['MetaNetX (MNX) Chemical', 'MNXM17'],\n",
       "  ['Reactome', '110096;111814;158602;205687'],\n",
       "  ['Human Metabolome Database', 'HMDB00295']],\n",
       " 'neutral_formula': '',\n",
       " 'neutral_mono_mass': 0.0,\n",
       " 'charge': -3,\n",
       " 'charged_formula': 'C9H11N2O12P2',\n",
       " 'SMILES': '',\n",
       " 'inchi': '',\n",
       " 'src_id': 'udp'}"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myCpds[20].__dict__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "def update_neutral_formula_mass(compound_list):\n",
    "    for myCpd in compound_list:\n",
    "        try:\n",
    "            myCpd.neutral_formula = adjust_charge_in_formula(myCpd.charged_formula,myCpd.charge) \n",
    "            myCpd.neutral_mono_mass = neutral_formula2mass(myCpd.neutral_formula)\n",
    "        except:\n",
    "            warnings.warn(f'{myCpd.id} do not have legal charge or charge formula to convert')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "update_neutral_formula_mass(myCpds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'internal_id': '',\n",
       "  'id': 'idon__L',\n",
       "  'name': 'L-Idonate',\n",
       "  'db_ids': [['Clostridium_sporogenes-Carve', 'idon__L'],\n",
       "   ['FORMULA', 'C6H11O7'],\n",
       "   ['BioCyc', 'L-IDONATE'],\n",
       "   ['SEED Compound', 'cpd00573'],\n",
       "   ['UniPathway Compound', 'UPC00770'],\n",
       "   ['KEGG Compound', 'C00770'],\n",
       "   ['MetaNetX (MNX) Chemical', 'MNXM1565']],\n",
       "  'neutral_formula': 'C6H12O7',\n",
       "  'neutral_mono_mass': 196.058302,\n",
       "  'charge': -1,\n",
       "  'charged_formula': 'C6H11O7',\n",
       "  'SMILES': '',\n",
       "  'inchi': '',\n",
       "  'src_id': 'idon__L'}]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check those that doesn't have the charge and charged formula\n",
    "[x.__dict__ for x in myCpds if x.id == 'idon__L']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Port reactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'_id': '12DGR120tipp',\n",
       "  'name': '1,2 diacylglycerol transport via flipping (periplasm to cytoplasm, n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR7'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr120_p at 0x7fc463a6d1c0>: -1.0,\n",
       "   <Metabolite 12dgr120_c at 0x7fc463a6d220>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '12DGR140tipp',\n",
       "  'name': '1,2 diacylglycerol transport via flipping (periplasm to cytoplasm, n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81306'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr140_p at 0x7fc463a6d130>: -1.0,\n",
       "   <Metabolite 12dgr140_c at 0x7fc463a6d1f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '12DGR141tipp',\n",
       "  'name': '1,2 diacylglycerol transport via flipping (periplasm to cytoplasm, n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr141_p at 0x7fc463a6d280>: -1.0,\n",
       "   <Metabolite 12dgr141_c at 0x7fc463a6d0d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '12DGR160tipp',\n",
       "  'name': '1,2 diacylglycerol transport via flipping (periplasm to cytoplasm, n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR10'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr160_p at 0x7fc463a6d2e0>: -1.0,\n",
       "   <Metabolite 12dgr160_c at 0x7fc463a6d2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '12DGR161tipp',\n",
       "  'name': '1,2 diacylglycerol transport via flipping (periplasm to cytoplasm, n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR11'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr161_p at 0x7fc463a6d340>: -1.0,\n",
       "   <Metabolite 12dgr161_c at 0x7fc463a6d310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '12DGR180tipp',\n",
       "  'name': '1,2 diacylglycerol transport via flipping (periplasm to cytoplasm, n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR12'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr180_p at 0x7fc463a6d3a0>: -1.0,\n",
       "   <Metabolite 12dgr180_c at 0x7fc463a6d370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '12DGR181tipp',\n",
       "  'name': '1,2 diacylglycerol transport via flipping (periplasm to cytoplasm, n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR13'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr181_p at 0x7fc463a6d400>: -1.0,\n",
       "   <Metabolite 12dgr181_c at 0x7fc463a6d3d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '12PPDt',\n",
       "  'name': 'S-Propane-1,2-diol facilitated transport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR15'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12ppd__S_e at 0x7fc463a6d490>: -1.0,\n",
       "   <Metabolite 12ppd__S_c at 0x7fc463a6d4c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPA161tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphatidate (n-C16:1) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR52'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2hdec9eg3p_p at 0x7fc463a6d7c0>: -1.0,\n",
       "   <Metabolite 2hdec9eg3p_c at 0x7fc463a6d760>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPE120tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoethanolamine (n-C12:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68130'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpe120_p at 0x7fc463a6d940>: -1.0,\n",
       "   <Metabolite 2agpe120_c at 0x7fc463a6d910>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPE140tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoethanolamine (n-C14:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68131'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpe140_p at 0x7fc463a6d9a0>: -1.0,\n",
       "   <Metabolite 2agpe140_c at 0x7fc463a6d970>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPE141tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoethanolamine (n-C14:1) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR57'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpe141_p at 0x7fc463a6da00>: -1.0,\n",
       "   <Metabolite 2agpe141_c at 0x7fc463a6d9d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPE160tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoethanolamine (n-C16:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68132'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpe160_p at 0x7fc463a6da60>: -1.0,\n",
       "   <Metabolite 2agpe160_c at 0x7fc463a6da30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPE161tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoethanolamine (n-C16:1) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpe161_p at 0x7fc463a6d5b0>: -1.0,\n",
       "   <Metabolite 2agpe161_c at 0x7fc463a6da90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPE180tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoethanolamine (n-C18:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68133'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpe180_p at 0x7fc463a6daf0>: -1.0,\n",
       "   <Metabolite 2agpe180_c at 0x7fc463a6dac0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPE181tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoethanolamine (n-C18:1) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR61'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpe181_p at 0x7fc463a6db50>: -1.0,\n",
       "   <Metabolite 2agpe181_c at 0x7fc463a6db20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPEAT120',\n",
       "  'name': '2-acyl-glycerophospho-ethanolamine acyltransferase (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81336'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpe120_c at 0x7fc463a6d910>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ddca_c at 0x7fc463a6dbb0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pe120_c at 0x7fc463a6dbe0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPEAT140',\n",
       "  'name': '2-acyl-glycerophospho-ethanolamine acyltransferase (n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81337'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpe140_c at 0x7fc463a6d970>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ttdca_c at 0x7fc463a6dc70>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pe140_c at 0x7fc463a6dc40>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPEAT141',\n",
       "  'name': '2-acyl-glycerophospho-ethanolamine acyltransferase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR50741'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpe141_c at 0x7fc463a6d9d0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ttdcea_c at 0x7fc463a6dd00>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pe141_c at 0x7fc463a6dca0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPEAT160',\n",
       "  'name': '2-acyl-glycerophospho-ethanolamine acyltransferase (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68136', 'EC Number': '2.3.1.40'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpe160_c at 0x7fc463a6da30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pe160_c at 0x7fc463a6dd90>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPEAT161',\n",
       "  'name': '2-acyl-glycerophospho-ethanolamine acyltransferase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81338'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpe161_c at 0x7fc463a6da90>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pe161_c at 0x7fc463a6ddf0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPEAT180',\n",
       "  'name': '2-acyl-glycerophospho-ethanolamine acyltransferase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68137', 'EC Number': '2.3.1.40'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpe180_c at 0x7fc463a6dac0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ocdca_c at 0x7fc463a6de20>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pe180_c at 0x7fc463a6de50>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPEAT181',\n",
       "  'name': '2-acyl-glycerophospho-ethanolamine acyltransferase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpe181_c at 0x7fc463a6db20>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ocdcea_c at 0x7fc463a6de80>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pe181_c at 0x7fc463a6deb0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPG120tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoglycerol (n-C12:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg120_p at 0x7fc463a6df10>: -1.0,\n",
       "   <Metabolite 2agpg120_c at 0x7fc463a6dee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPG140tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoglycerol (n-C14:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68139'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg140_p at 0x7fc463a6dd30>: -1.0,\n",
       "   <Metabolite 2agpg140_c at 0x7fc463a6df40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPG141tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoglycerol (n-C14:1) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR71'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg141_p at 0x7fc463a6df70>: -1.0,\n",
       "   <Metabolite 2agpg141_c at 0x7fc463a6dcd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPG160tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoglycerol (n-C16:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68140'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg160_p at 0x7fc463a6dfa0>: -1.0,\n",
       "   <Metabolite 2agpg160_c at 0x7fc463a6dfd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPG161tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoglycerol (n-C16:1) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR73'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg161_p at 0x7fc463adc160>: -1.0,\n",
       "   <Metabolite 2agpg161_c at 0x7fc463adc130>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPG180tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoglycerol (n-C18:0) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68141'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg180_p at 0x7fc463adc0a0>: -1.0,\n",
       "   <Metabolite 2agpg180_c at 0x7fc463adc190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPG181tipp',\n",
       "  'name': '2-Acyl-sn-glycero-3-phosphoglycerol (n-C18:1) transporter via facilitated diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR75'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg181_p at 0x7fc463adc070>: -1.0,\n",
       "   <Metabolite 2agpg181_c at 0x7fc463adc040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPGAT120',\n",
       "  'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR76'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpg120_c at 0x7fc463a6dee0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ddca_c at 0x7fc463a6dbb0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pg120_c at 0x7fc4604a8d90>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPGAT140',\n",
       "  'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68142'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpg140_c at 0x7fc463a6df40>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ttdca_c at 0x7fc463a6dc70>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pg140_c at 0x7fc463adc100>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPGAT141',\n",
       "  'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR50742'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpg141_c at 0x7fc463a6dcd0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ttdcea_c at 0x7fc463a6dd00>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pg141_c at 0x7fc463adc1c0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPGAT160',\n",
       "  'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68143'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpg160_c at 0x7fc463a6dfd0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pg160_c at 0x7fc463adc310>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPGAT161',\n",
       "  'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68144'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpg161_c at 0x7fc463adc130>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pg161_c at 0x7fc463adc220>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPGAT180',\n",
       "  'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68145'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpg180_c at 0x7fc463adc190>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ocdca_c at 0x7fc463a6de20>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pg180_c at 0x7fc463adc250>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2AGPGAT181',\n",
       "  'name': '2-acyl-glycerophospho-glycerol acyltransferase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite 2agpg181_c at 0x7fc463adc040>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ocdcea_c at 0x7fc463a6de80>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pg181_c at 0x7fc463adc280>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2HXMPt6',\n",
       "  'name': '2  hydroxymethyl phenol transport inout via proton symport',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite 2hxmp_e at 0x7fc463adc6a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 2hxmp_c at 0x7fc463adc670>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2MAHMP',\n",
       "  'name': '2-Methyl-4-amino-5-hydroxymethylpyrimidine diphosphatase',\n",
       "  'notes': {'RHEA': '27914;27915;27916;27917',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68153',\n",
       "   'BioCyc': 'META:RXN0-3543'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39211_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39211_1 at 0x7fc449820310>},\n",
       "  '_metabolites': {<Metabolite 2mahmp_c at 0x7fc463adc700>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 4ampm_c at 0x7fc463adc730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '2PGLYCt6',\n",
       "  'name': '2 Phosphoglycolate transport inout via proton symport',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite 2pglyc_e at 0x7fc463adc7f0>: -1.0,\n",
       "   <Metabolite 2pglyc_c at 0x7fc463adc7c0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD100',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C10:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68167',\n",
       "   'KEGG Reaction': 'R04535',\n",
       "   'EC Number': '4.2.1.59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hdecACP_c at 0x7fc463adca00>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite tdec2eACP_c at 0x7fc463adcb80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD120',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68168',\n",
       "   'KEGG Reaction': 'R04965',\n",
       "   'EC Number': '4.2.1.59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hddecACP_c at 0x7fc463adca60>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite tddec2eACP_c at 0x7fc463adcbb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD121',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C12:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68169', 'EC Number': '4.2.1.61'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hcddec5eACP_c at 0x7fc463adca90>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite t3c5ddeceACP_c at 0x7fc463adcbe0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD140',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C14:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9537',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68170',\n",
       "   'KEGG Reaction': 'R04568',\n",
       "   'EC Number': '4.2.1.59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hmrsACP_c at 0x7fc463adcac0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite tmrs2eACP_c at 0x7fc463adcc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD141',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68171', 'EC Number': '4.2.1.61'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hcmrs7eACP_c at 0x7fc463adcaf0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite t3c7mrseACP_c at 0x7fc463adcc40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD160',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68172',\n",
       "   'KEGG Reaction': 'R04544',\n",
       "   'EC Number': '4.2.1.59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hpalmACP_c at 0x7fc463adcc70>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite tpalm2eACP_c at 0x7fc463adcca0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD161',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68173', 'EC Number': '4.2.1.61'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hcpalm9eACP_c at 0x7fc463adccd0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite t3c9palmeACP_c at 0x7fc463adcd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD180',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68174', 'EC Number': '4.2.1.61'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hoctaACP_c at 0x7fc463adcd30>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite toctd2eACP_c at 0x7fc463adcd60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD181',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81364', 'EC Number': '4.2.1.61'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hcvac11eACP_c at 0x7fc463adcd90>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite t3c11vaceACP_c at 0x7fc463adcdc0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD40',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C4:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68176', 'EC Number': '4.2.1.58'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3haACP_c at 0x7fc463adcdf0>: -1.0,\n",
       "   <Metabolite but2eACP_c at 0x7fc463adce20>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD60',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C6:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68177',\n",
       "   'KEGG Reaction': 'R04954',\n",
       "   'EC Number': '4.2.1.59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hhexACP_c at 0x7fc463adcb20>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite thex2eACP_c at 0x7fc463adce50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3HAD80',\n",
       "  'name': '3-hydroxyacyl-[acyl-carrier-protein] dehydratase (n-C8:0)',\n",
       "  'notes': {'BioCyc': 'META:4.2.1.59-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68178',\n",
       "   'KEGG Reaction': 'R04537',\n",
       "   'EC Number': '4.2.1.59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38148_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38148_1 at 0x7fc449820280>},\n",
       "  '_metabolites': {<Metabolite 3hoctACP_c at 0x7fc463adcb50>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite toct2eACP_c at 0x7fc463adce80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3MBtex',\n",
       "  'name': '3-methylbutanoic acid transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81374'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite 3mb_e at 0x7fc463adcf10>: -1.0,\n",
       "   <Metabolite 3mb_p at 0x7fc463adcfa0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR100',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C10:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9528',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68183',\n",
       "   'KEGG Reaction': 'R04534',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3odecACP_c at 0x7fc463a82280>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hdecACP_c at 0x7fc463adca00>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR120',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68184',\n",
       "   'KEGG Reaction': 'R04964',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3oddecACP_c at 0x7fc463a822b0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hddecACP_c at 0x7fc463adca60>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR121',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C12:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR187', 'EC Number': '1.1.1.100'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3ocddec5eACP_c at 0x7fc463a822e0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hcddec5eACP_c at 0x7fc463adca90>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR140',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C14:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9536',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68185',\n",
       "   'KEGG Reaction': 'R04566',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3omrsACP_c at 0x7fc463a82310>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hmrsACP_c at 0x7fc463adcac0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR141',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR189', 'EC Number': '1.1.1.100'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3ocmrs7eACP_c at 0x7fc463a82340>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hcmrs7eACP_c at 0x7fc463adcaf0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR160',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68186',\n",
       "   'KEGG Reaction': 'R04543',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3opalmACP_c at 0x7fc463a82370>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hpalmACP_c at 0x7fc463adcc70>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR161',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR191', 'EC Number': '1.1.1.100'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3ocpalm9eACP_c at 0x7fc463a823a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hcpalm9eACP_c at 0x7fc463adccd0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR180',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR192', 'EC Number': '1.1.1.100'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3ooctdACP_c at 0x7fc463a823d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hoctaACP_c at 0x7fc463adcd30>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR181',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR23892',\n",
       "   'EC Number': '1.1.1.100',\n",
       "   'BioCyc': 'META:RXN-9556'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3ocvac11eACP_c at 0x7fc463a82400>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hcvac11eACP_c at 0x7fc463adcd90>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR40',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C4:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68189', 'EC Number': '1.1.1.100'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite actACP_c at 0x7fc463a82430>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3haACP_c at 0x7fc463adcdf0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR60',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C6:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68190',\n",
       "   'KEGG Reaction': 'R04953',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3ohexACP_c at 0x7fc463a82460>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hhexACP_c at 0x7fc463adcb20>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAR80',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] reductase (n-C8:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9524',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68191',\n",
       "   'KEGG Reaction': 'R04536',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38145_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38145_1 at 0x7fc449820190>},\n",
       "  '_metabolites': {<Metabolite 3ooctACP_c at 0x7fc463a82490>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hoctACP_c at 0x7fc463adcb50>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS100',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C10:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9527',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81380',\n",
       "   'KEGG Reaction': 'R04960',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite ocACP_c at 0x7fc463a824f0>: -1.0,\n",
       "   <Metabolite 3odecACP_c at 0x7fc463a82280>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS120',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C12:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9531',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81381',\n",
       "   'KEGG Reaction': 'R04963',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite dcaACP_c at 0x7fc463a82520>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite 3oddecACP_c at 0x7fc463a822b0>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS121',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C12:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR47319', 'EC Number': '2.3.1.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite cdec3eACP_c at 0x7fc463a82550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite 3ocddec5eACP_c at 0x7fc463a822e0>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS140',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C14:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9535',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81382',\n",
       "   'KEGG Reaction': 'R04726',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite ddcaACP_c at 0x7fc463a82580>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite 3omrsACP_c at 0x7fc463a82310>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS141',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR47320', 'EC Number': '2.3.1.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite cddec5eACP_c at 0x7fc463a825b0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite 3ocmrs7eACP_c at 0x7fc463a82340>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS160',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C16:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9539',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81383',\n",
       "   'KEGG Reaction': 'R04968',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: -1.0,\n",
       "   <Metabolite 3opalmACP_c at 0x7fc463a82370>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS161',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR47321', 'EC Number': '2.3.1.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite tdeACP_c at 0x7fc463a82610>: -1.0,\n",
       "   <Metabolite 3ocpalm9eACP_c at 0x7fc463a823a0>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS180',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR48165', 'EC Number': '2.3.1.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite palmACP_c at 0x7fc463a82640>: -1.0,\n",
       "   <Metabolite 3ooctdACP_c at 0x7fc463a823d0>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS181',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81384',\n",
       "   'EC Number': '2.3.1.179',\n",
       "   'BioCyc': 'META:2.3.1.179-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite hdeACP_c at 0x7fc463a82670>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite 3ocvac11eACP_c at 0x7fc463a82400>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS60',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C6:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9516',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81385',\n",
       "   'KEGG Reaction': 'R04952',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite butACP_c at 0x7fc463a826a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite 3ohexACP_c at 0x7fc463a82460>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OAS80',\n",
       "  'name': '3-oxoacyl-[acyl-carrier-protein] synthase (n-C8:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9523',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81386',\n",
       "   'KEGG Reaction': 'R04957',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite hexACP_c at 0x7fc463a826d0>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite 3ooctACP_c at 0x7fc463a82490>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3OXCOAT',\n",
       "  'name': '3-oxoadipyl-CoA thiolase',\n",
       "  'notes': {'EC Number': '2.3.1.174',\n",
       "   'BioCyc': 'META:RXN-3641',\n",
       "   'BioPath Reaction': 'RXN00946',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR209',\n",
       "   'KEGG Reaction': 'R00829',\n",
       "   'RHEA': '19481;19482;19483;19484',\n",
       "   'UniPathway Reaction': 'UCR00829;UER00263'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39341_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39341_1 at 0x7fc449820370>},\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite oxadpcoa_c at 0x7fc463a821f0>: -1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: 1.0,\n",
       "   <Metabolite succoa_c at 0x7fc463a82250>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '3PEPTabcpp',\n",
       "  'name': 'Tripeptide (LalaDgluMdap) transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68200'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37419_1 or EDU39393_1 or (EDU39392_1 and EDU39393_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37419_1 at 0x7fc4498203d0>,\n",
       "   <Gene EDU39392_1 at 0x7fc4498203a0>,\n",
       "   <Gene EDU39393_1 at 0x7fc449820400>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite LalaDgluMdap_p at 0x7fc463a82760>: -1.0,\n",
       "   <Metabolite LalaDgluMdap_c at 0x7fc463a82730>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4HBALDt2pp',\n",
       "  'name': '4-hydroxybenzoaldehyde transport in via proton symport  (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81396'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite 4hbald_p at 0x7fc463a829a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 4hbald_c at 0x7fc463a82970>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4HBALDtex',\n",
       "  'name': '4-hydroxybenzaldehyde transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81397'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite 4hbald_e at 0x7fc463a829d0>: -1.0,\n",
       "   <Metabolite 4hbald_p at 0x7fc463a829a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4HBAtex',\n",
       "  'name': '4-hydroxy-benzyl alcohol transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68734',\n",
       "   'BioCyc': 'META:TRANS-RXN0-274'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite 4hba_e at 0x7fc463a82a00>: -1.0,\n",
       "   <Metabolite 4hba_p at 0x7fc463a82a30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4HBZtex',\n",
       "  'name': '4HBZtex',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR233'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite 4hbz_e at 0x7fc463a82b20>: -1.0,\n",
       "   <Metabolite 4hbz_p at 0x7fc463a82b50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4HOXPACt2pp',\n",
       "  'name': '4 hydroxyphenylacetate transport in via proton symport periplasm ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69250'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36008_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36008_1 at 0x7fc449820430>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite 4hphac_p at 0x7fc463a82be0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 4hphac_c at 0x7fc463a82940>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4HTHRK',\n",
       "  'name': '4-Hydroxy-L-threonine kinase',\n",
       "  'notes': {'RHEA': '28787;28788;28789;28790',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR24335',\n",
       "   'EC Number': '2.7.1',\n",
       "   'BioCyc': 'META:RXN0-6564'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 4hthr_c at 0x7fc463a82c70>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite phthr_c at 0x7fc463a82ca0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4HTHRS',\n",
       "  'name': '4-Hydroxy-L-threonine synthase',\n",
       "  'notes': {'RHEA': '30659;30660;30661;30662',\n",
       "   'BioCyc': 'META:RXN-14125',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR239',\n",
       "   'KEGG Reaction': 'R05086',\n",
       "   'EC Number': '4.2.3.1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37120_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37120_1 at 0x7fc449820460>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite phthr_c at 0x7fc463a82ca0>: -1.0,\n",
       "   <Metabolite 4hthr_c at 0x7fc463a82c70>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4MBZALDH',\n",
       "  'name': '4MBZALDH',\n",
       "  'notes': {'BioCyc': 'META:RXN-8581',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR242',\n",
       "   'KEGG Reaction': 'R05282',\n",
       "   'EC Number': '1.1.1.90'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 4mbzalc_c at 0x7fc463a82d60>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite 4mbzald_c at 0x7fc463a82d30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4MBZDH',\n",
       "  'name': '4MBZDH',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR243', 'BioCyc': 'META:RXN-8582'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 4mbzald_c at 0x7fc463a82d30>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite p_tol_c at 0x7fc463a82d90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4MCAT23DOX',\n",
       "  'name': '4MCAT23DOX',\n",
       "  'notes': {'BioCyc': 'META:RXN-11219',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR54920',\n",
       "   'KEGG Reaction': 'R05295',\n",
       "   'EC Number': '1.13.11.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite o2_c at 0x7fc463a6d610>: -1.0,\n",
       "   <Metabolite 4mcat_c at 0x7fc463a82df0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hmccms_c at 0x7fc463a82dc0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4PCP',\n",
       "  'name': 'Tetrapeptide L,D-carboxypeptidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68217'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38089_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38089_1 at 0x7fc449820490>},\n",
       "  '_metabolites': {<Metabolite LalaDgluMdapDala_c at 0x7fc463a82fa0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite LalaDgluMdap_c at 0x7fc463a82730>: 1.0,\n",
       "   <Metabolite ala__D_c at 0x7fc463a82fd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '4PEPTabcpp',\n",
       "  'name': 'Tetrapeptide (LalaDgluMdapDala) transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68218'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37419_1 or EDU39393_1 or (EDU39392_1 and EDU39393_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37419_1 at 0x7fc4498203d0>,\n",
       "   <Gene EDU39392_1 at 0x7fc4498203a0>,\n",
       "   <Gene EDU39393_1 at 0x7fc449820400>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite LalaDgluMdapDala_p at 0x7fc463a82f10>: -1.0,\n",
       "   <Metabolite LalaDgluMdapDala_c at 0x7fc463a82fa0>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '5DGLCNt2rpp',\n",
       "  'name': '5-Dehydro-D-gluconate transport via proton symport, reversible (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR54924',\n",
       "   'BioCyc': 'META:TRANS-RXN0-228'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36576_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36576_1 at 0x7fc4498204c0>},\n",
       "  '_metabolites': {<Metabolite 5dglcn_p at 0x7fc463aae190>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite 5dglcn_c at 0x7fc463aae160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': '5DOAN',\n",
       "  'name': \"5'-deoxyadenosine nuclosidase\",\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68220'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite dad_5_c at 0x7fc463aae070>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 5drib_c at 0x7fc463aae0a0>: 1.0,\n",
       "   <Metabolite ade_c at 0x7fc463aae040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'A5PISO',\n",
       "  'name': 'Arabinose-5-phosphate isomerase',\n",
       "  'notes': {'BioCyc': 'META:DARAB5PISOM-RXN',\n",
       "   'EC Number': '5.3.1.13',\n",
       "   'KEGG Reaction': 'R01530',\n",
       "   'UniPathway Reaction': 'UCR01530;UER00473',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR300'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ru5p__D_c at 0x7fc463aae4c0>: -1.0,\n",
       "   <Metabolite ara5p_c at 0x7fc463aae490>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS1',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase (n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR50749',\n",
       "   'EC Number': '6.2.1.20',\n",
       "   'BioCyc': 'META:RXN-13266'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ttdca_c at 0x7fc463a6dc70>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS2',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68229', 'EC Number': '6.2.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ttdcea_c at 0x7fc463a6dd00>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite tdeACP_c at 0x7fc463a82610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS3',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68230', 'EC Number': '6.2.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite palmACP_c at 0x7fc463a82640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS4',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81444'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite hdeACP_c at 0x7fc463a82670>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS5',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81445'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ocdcea_c at 0x7fc463a6de80>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite octeACP_c at 0x7fc463aae580>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS6',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81446', 'EC Number': '6.2.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ocdca_c at 0x7fc463a6de20>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite ocdcaACP_c at 0x7fc463aae5b0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS7',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR54932',\n",
       "   'EC Number': '6.2.1.l',\n",
       "   'BioCyc': 'META:RXN-13264'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ddca_c at 0x7fc463a6dbb0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite ddcaACP_c at 0x7fc463a82580>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS8',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C10:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68232', 'EC Number': '6.2.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dca_c at 0x7fc463aae5e0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite dcaACP_c at 0x7fc463a82520>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AACPS9',\n",
       "  'name': 'Acyl-[acyl-carrier-protein] synthetase  (n-C8:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68233',\n",
       "   'EC Number': '6.2.1.l',\n",
       "   'BioCyc': 'META:RXN-5741'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite octa_c at 0x7fc463aae610>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite ocACP_c at 0x7fc463a824f0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AB6PGH',\n",
       "  'name': 'Arbutin 6-phosphate glucohydrolase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR80352', 'EC Number': '3.2.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35994_1 or EDU39174_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35994_1 at 0x7fc4498204f0>,\n",
       "   <Gene EDU39174_1 at 0x7fc449820520>},\n",
       "  '_metabolites': {<Metabolite arbt6p_c at 0x7fc463aae6d0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0,\n",
       "   <Metabolite hqn_c at 0x7fc463aae700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ABUTD',\n",
       "  'name': 'Aminobutyraldehyde dehydrogenase',\n",
       "  'notes': {'EC Number': '1.2.1.3',\n",
       "   'BioCyc': 'META:AMINOBUTDEHYDROG-RXN',\n",
       "   'BioPath Reaction': 'RXN00431',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR337',\n",
       "   'KEGG Reaction': 'R02549',\n",
       "   'RHEA': '19105;19106;19107;19108',\n",
       "   'UniPathway Reaction': 'UCR02549;UER00292'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38973_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38973_1 at 0x7fc449820550>},\n",
       "  '_metabolites': {<Metabolite 4abutn_c at 0x7fc463a828b0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 4abut_c at 0x7fc463a82880>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACACT1r',\n",
       "  'name': 'Acetyl-CoA C-acetyltransferase',\n",
       "  'notes': {'EC Number': '2.3.1.9',\n",
       "   'BioCyc': 'META:ACETYL-COA-ACETYLTRANSFER-RXN',\n",
       "   'BioPath Reaction': 'RXN00225;RXN01163',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR341',\n",
       "   'KEGG Reaction': 'R00238',\n",
       "   'RHEA': '21036;21037;21038;21039',\n",
       "   'UniPathway Reaction': 'UCR00238;UER00101'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39341_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39341_1 at 0x7fc449820370>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -2.0,\n",
       "   <Metabolite aacoa_c at 0x7fc463aae550>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACACT2r',\n",
       "  'name': 'Acetyl-CoA C-acyltransferase (butanoyl-CoA) (r)',\n",
       "  'notes': {'EC Number': '2.3.1.9',\n",
       "   'BioCyc': 'META:RXN-12565',\n",
       "   'BioPath Reaction': 'RXN00254',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR343',\n",
       "   'RHEA': '31111;31112;31113;31114',\n",
       "   'KEGG Reaction': 'R01177'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39341_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39341_1 at 0x7fc449820370>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite btcoa_c at 0x7fc463aae820>: -1.0,\n",
       "   <Metabolite 3ohcoa_c at 0x7fc463aae7f0>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACACT4r',\n",
       "  'name': 'Acetyl-CoA C-acyltransferase (octanoyl-CoA) (r)',\n",
       "  'notes': {'EC Number': '2.3.1.9',\n",
       "   'BioCyc': 'META:RXN-13617',\n",
       "   'BioPath Reaction': 'RXN01338',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR345',\n",
       "   'RHEA': '31087;31088;31089;31090',\n",
       "   'KEGG Reaction': 'R03778'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite occoa_c at 0x7fc463aae8e0>: -1.0,\n",
       "   <Metabolite 3odcoa_c at 0x7fc463aae8b0>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACACT5r_1',\n",
       "  'name': 'Acetyl CoA C acyltransferase  decanoyl CoA ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite dccoa_c at 0x7fc463aae970>: -1.0,\n",
       "   <Metabolite 3oddcoa_c at 0x7fc463aae910>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACALD',\n",
       "  'name': 'Acetaldehyde dehydrogenase (acetylating)',\n",
       "  'notes': {'EC Number': '1.2.1.10',\n",
       "   'BioCyc': 'META:ACETALD-DEHYDROG-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR353',\n",
       "   'KEGG Reaction': 'R00228',\n",
       "   'RHEA': '23288;23289;23290;23291',\n",
       "   'UniPathway Reaction': 'UCR00228;UER00798'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36705_1 or EDU37339_1 or EDU37784_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36705_1 at 0x7fc4498205b0>,\n",
       "   <Gene EDU37339_1 at 0x7fc449820580>,\n",
       "   <Gene EDU37784_1 at 0x7fc4498205e0>},\n",
       "  '_metabolites': {<Metabolite acald_c at 0x7fc463aae2b0>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACALDt',\n",
       "  'name': 'Acetaldehyde reversible transport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR354'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite acald_e at 0x7fc463aaeac0>: -1.0,\n",
       "   <Metabolite acald_c at 0x7fc463aae2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACCOAC',\n",
       "  'name': 'Acetyl-CoA carboxylase',\n",
       "  'notes': {'EC Number': '6.4.1.2',\n",
       "   'BioCyc': 'META:ACETYL-COA-CARBOXYLTRANSFER-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR358',\n",
       "   'KEGG Reaction': 'R00742',\n",
       "   'RHEA': '11308;11309;11310;11311',\n",
       "   'UniPathway Reaction': 'UCR00742;UER00711'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38147_1 and EDU38149_1 and EDU38150_1 and EDU38151_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38147_1 at 0x7fc449820670>,\n",
       "   <Gene EDU38149_1 at 0x7fc4498206a0>,\n",
       "   <Gene EDU38150_1 at 0x7fc449820640>,\n",
       "   <Gene EDU38151_1 at 0x7fc4498206d0>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hco3_c at 0x7fc463adc5b0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite malcoa_c at 0x7fc463a82100>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACCOAL',\n",
       "  'name': 'Acetate-CoA ligase (ADP-forming)',\n",
       "  'notes': {'RHEA': '29843;29844;29845;29846',\n",
       "   'BioCyc': 'META:PROPANOATECOA-LIGASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR359',\n",
       "   'KEGG Reaction': 'R00920',\n",
       "   'EC Number': '6.2.1.13'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite ppa_c at 0x7fc463aaeaf0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite ppcoa_c at 0x7fc463aaeb20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACGAMK',\n",
       "  'name': 'N-acetylglucosamine kinase',\n",
       "  'notes': {'RHEA': '17417;17418;17419;17420',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81465',\n",
       "   'KEGG Reaction': 'R01201',\n",
       "   'EC Number': '2.7.1.59'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite acgam_c at 0x7fc463aaeca0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite acgam6p_c at 0x7fc463aaed00>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACGApts',\n",
       "  'name': 'N-Acetyl-D-glucosamine transport via PEP:Pyr PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81468'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU36005_1 and EDU36437_1) or (EDU36005_1 and EDU36437_1 and EDU39079_1) or (EDU36006_1 and EDU36437_1 and EDU37160_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36005_1 at 0x7fc4498207c0>,\n",
       "   <Gene EDU36006_1 at 0x7fc449820790>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU37160_1 at 0x7fc449820700>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite acgam_e at 0x7fc463aaee20>: -1.0,\n",
       "   <Metabolite acgam6p_c at 0x7fc463aaed00>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACGAptspp',\n",
       "  'name': 'N-Acetyl-D-glucosamine transport via PEP:Pyr PTS  (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81468'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU36005_1 and EDU36437_1) or (EDU36006_1 and EDU36437_1) or (EDU36437_1 and EDU37843_1) or (EDU36006_1 and EDU36437_1 and EDU37160_1) or (EDU36437_1 and EDU37160_1 and EDU37843_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36005_1 at 0x7fc4498207c0>,\n",
       "   <Gene EDU36006_1 at 0x7fc449820790>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU37160_1 at 0x7fc449820700>,\n",
       "   <Gene EDU37843_1 at 0x7fc4498207f0>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite acgam_p at 0x7fc463aaec70>: -1.0,\n",
       "   <Metabolite acgam6p_c at 0x7fc463aaed00>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACGAtex',\n",
       "  'name': 'N-Acetyl-D-glucosamine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR379'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite acgam_e at 0x7fc463aaee20>: -1.0,\n",
       "   <Metabolite acgam_p at 0x7fc463aaec70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACHBS',\n",
       "  'name': '2-aceto-2-hydroxybutanoate synthase',\n",
       "  'notes': {'RHEA': '27654;27655;27656;27657',\n",
       "   'BioCyc': 'META:ACETOOHBUTSYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR76182',\n",
       "   'KEGG Reaction': 'R08648',\n",
       "   'EC Number': '2.2.1.6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2obut_c at 0x7fc463adc550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: -1.0,\n",
       "   <Metabolite 2ahbut_c at 0x7fc463aaee80>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACKr',\n",
       "  'name': 'Acetate kinase',\n",
       "  'notes': {'EC Number': '2.7.2.15',\n",
       "   'BioCyc': 'META:ACETATEKIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00036',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR390',\n",
       "   'KEGG Reaction': 'R00315',\n",
       "   'RHEA': '11352;11353;11354;11355',\n",
       "   'UniPathway Reaction': 'UCR00315;UER00458'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36374_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36374_1 at 0x7fc449820820>},\n",
       "  '_metabolites': {<Metabolite ac_c at 0x7fc463aae520>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite actp_c at 0x7fc463aaef40>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACLS',\n",
       "  'name': 'Acetolactate synthase',\n",
       "  'notes': {'EC Number': '2.2.1.6',\n",
       "   'BioCyc': 'META:ACETOLACTSYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68259',\n",
       "   'KEGG Reaction': 'R00226',\n",
       "   'RHEA': '25249;25250;25251;25252',\n",
       "   'UniPathway Reaction': 'UCR00226;UER00677'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: -2.0,\n",
       "   <Metabolite alac__S_c at 0x7fc448e1c130>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACM6PH',\n",
       "  'name': 'N-acetylmuramate 6-phosphate hydrolase',\n",
       "  'notes': {'RHEA': '26410;26411;26412;26413',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81471',\n",
       "   'KEGG Reaction': 'R08555',\n",
       "   'EC Number': '4.2.1.126'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37441_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37441_1 at 0x7fc449820850>},\n",
       "  '_metabolites': {<Metabolite acmum6p_c at 0x7fc448e1c190>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite acgam6p_c at 0x7fc463aaed00>: 1.0,\n",
       "   <Metabolite lac__D_c at 0x7fc448e1c0a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACMANApts',\n",
       "  'name': 'N-acetyl-D-mannosamine transport via PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81472'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite acmana_e at 0x7fc448e1c070>: -1.0,\n",
       "   <Metabolite acmanap_c at 0x7fc448e1c040>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACNML',\n",
       "  'name': 'N-Acetylneuraminate lyase',\n",
       "  'notes': {'EC Number': '4.1.3.3',\n",
       "   'BioCyc': 'META:ACNEULY-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26498',\n",
       "   'KEGG Reaction': 'R01811',\n",
       "   'RHEA': '23296;23297;23298;23299',\n",
       "   'UniPathway Reaction': 'UCR01811;UER00680'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite acnam_c at 0x7fc448e1c100>: -1.0,\n",
       "   <Metabolite acmana_c at 0x7fc463aaecd0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACNPLYS',\n",
       "  'name': 'N-Acetylneuraminate pyruvate-lyase',\n",
       "  'notes': {'EC Number': '2.5.1.56',\n",
       "   'BioCyc': 'META:N-ACETYLNEURAMINATE-SYNTHASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00982',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26433',\n",
       "   'RHEA': '19273;19274;19275;19276',\n",
       "   'KEGG Reaction': 'R01804'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36131_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36131_1 at 0x7fc449820880>},\n",
       "  '_metabolites': {<Metabolite acmana_c at 0x7fc463aaecd0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite acnam_c at 0x7fc448e1c100>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD1',\n",
       "  'name': 'Acyl-CoA dehydrogenase (butanoyl-CoA)',\n",
       "  'notes': {'BioCyc': 'META:RXN-12558',\n",
       "   'BioPath Reaction': 'RXN00780',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR410',\n",
       "   'KEGG Reaction': 'R01171',\n",
       "   'EC Number': '1.3.99.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35996_1 or EDU36591_1 or EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35996_1 at 0x7fc4498208b0>,\n",
       "   <Gene EDU36591_1 at 0x7fc4498208e0>,\n",
       "   <Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite b2coa_c at 0x7fc448e1c1f0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite btcoa_c at 0x7fc463aae820>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD1f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (butanoyl-CoA)',\n",
       "  'notes': {'RHEA': '24004;24005;24006;24007;30731;30732;30733;30734',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81478',\n",
       "   'KEGG Reaction': 'R01175',\n",
       "   'EC Number': '1.3.99.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite btcoa_c at 0x7fc463aae820>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite b2coa_c at 0x7fc448e1c1f0>: 1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD2',\n",
       "  'name': 'Acyl-CoA dehydrogenase (hexanoyl-CoA)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR413',\n",
       "   'EC Number': '1.3.1.44',\n",
       "   'BioCyc': 'META:RXN-12559'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35996_1 or EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35996_1 at 0x7fc4498208b0>,\n",
       "   <Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite hx2coa_c at 0x7fc448e1c220>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite hxcoa_c at 0x7fc463aae880>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD20',\n",
       "  'name': 'Acyl CoA dehydrogenase  2 methylbutnoyl CoA ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36591_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36591_1 at 0x7fc4498208e0>},\n",
       "  '_metabolites': {<Metabolite o2_c at 0x7fc463a6d610>: -1.0,\n",
       "   <Metabolite 2mbcoa_c at 0x7fc448e1c250>: -2.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 2.0,\n",
       "   <Metabolite 2mb2coa_c at 0x7fc463adc760>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD2f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (hexanoyl-CoA)',\n",
       "  'notes': {'RHEA': '31207;31208;31209;31210;43464;43465;43466;43467',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81479',\n",
       "   'KEGG Reaction': 'R04751',\n",
       "   'EC Number': '1.3.99.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite hxcoa_c at 0x7fc463aae880>: -1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0,\n",
       "   <Metabolite hx2coa_c at 0x7fc448e1c220>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD3f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (octanoyl-CoA)',\n",
       "  'notes': {'RHEA': '30943;30944;30945;30946',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81480',\n",
       "   'KEGG Reaction': 'R03777',\n",
       "   'EC Number': '1.3.99.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite occoa_c at 0x7fc463aae8e0>: -1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0,\n",
       "   <Metabolite oc2coa_c at 0x7fc448e1c280>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD4_1',\n",
       "  'name': 'Acyl CoA dehydrogenase  decanoyl CoA ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35996_1 or EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35996_1 at 0x7fc4498208b0>,\n",
       "   <Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite dccoa_c at 0x7fc463aae970>: -1.0,\n",
       "   <Metabolite dc2coa_c at 0x7fc448e1c2b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD4f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (decanoyl-CoA)',\n",
       "  'notes': {'RHEA': '28354;28355;28356;28357',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81474',\n",
       "   'KEGG Reaction': 'R04754',\n",
       "   'EC Number': '1.3.99.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite dcacoa_c at 0x7fc463aae940>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite dc2coa_c at 0x7fc448e1c2b0>: 1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD5_1',\n",
       "  'name': 'Acyl CoA dehydrogenase  dodecanoyl CoA ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35996_1 or EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35996_1 at 0x7fc4498208b0>,\n",
       "   <Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite ddcoa_c at 0x7fc463aaea00>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite trans_dd2coa_c at 0x7fc448e1c310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD5f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (dodecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '31039;31040;31041;31042',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81475',\n",
       "   'KEGG Reaction': 'R03857',\n",
       "   'EC Number': '1.3.99.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite ddcacoa_c at 0x7fc463aae9d0>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite dd2coa_c at 0x7fc448e1c2e0>: 1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD6f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (tetradecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '28358;28359;28360;28361',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81476',\n",
       "   'KEGG Reaction': 'R03990',\n",
       "   'EC Number': '1.3.99.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite tdcoa_c at 0x7fc463aaea60>: -1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0,\n",
       "   <Metabolite td2coa_c at 0x7fc45d333c10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD7f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (hexadecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '30747;30748;30749;30750;43448;43449;43450;43451',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81477',\n",
       "   'KEGG Reaction': 'R01279',\n",
       "   'EC Number': '1.3.99.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite pmtcoa_c at 0x7fc463a821c0>: -1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0,\n",
       "   <Metabolite hdd2coa_c at 0x7fc463a40f10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOAD8f',\n",
       "  'name': 'Acyl-CoA dehydrogenase (octadecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '40843;40844;40845;40846',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81483'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39342_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39342_1 at 0x7fc449820910>},\n",
       "  '_metabolites': {<Metabolite fad_c at 0x7fc463a82b80>: -1.0,\n",
       "   <Metabolite stcoa_c at 0x7fc448e1c370>: -1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 1.0,\n",
       "   <Metabolite od2coa_c at 0x7fc448e1c3a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACOATA',\n",
       "  'name': 'Acetyl-CoA ACP transacylase',\n",
       "  'notes': {'BioCyc': 'META:ACP-S-ACETYLTRANSFER-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR35617',\n",
       "   'KEGG Reaction': 'R01624',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38560_1 or (EDU36378_1 and EDU38141_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU36378_1 at 0x7fc4498209a0>,\n",
       "   <Gene EDU38141_1 at 0x7fc449820970>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite acACP_c at 0x7fc448e1c430>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACONT',\n",
       "  'name': 'Aconitase',\n",
       "  'notes': {'EC Number': '4.2.1.3',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR30608',\n",
       "   'KEGG Reaction': 'R01324',\n",
       "   'RHEA': '10336;10337;10338;10339',\n",
       "   'UniPathway Reaction': 'UER00718'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38240_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38240_1 at 0x7fc4498209d0>},\n",
       "  '_metabolites': {<Metabolite cit_c at 0x7fc463aaef10>: -1.0,\n",
       "   <Metabolite icit_c at 0x7fc448e1c4c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACONTa',\n",
       "  'name': 'Aconitase (half-reaction A, Citrate hydro-lyase)',\n",
       "  'notes': {'EC Number': '4.2.1.3',\n",
       "   'BioCyc': 'META:ACONITATEDEHYDR-RXN',\n",
       "   'BioPath Reaction': 'RXN00006',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR443',\n",
       "   'KEGG Reaction': 'R01325',\n",
       "   'RHEA': '10228;10229;10230;10231',\n",
       "   'UniPathway Reaction': 'UCR01325'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38240_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38240_1 at 0x7fc4498209d0>},\n",
       "  '_metabolites': {<Metabolite cit_c at 0x7fc463aaef10>: -1.0,\n",
       "   <Metabolite acon_C_c at 0x7fc448e1c460>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACONTb',\n",
       "  'name': 'Aconitase (half-reaction B, Isocitrate hydro-lyase)',\n",
       "  'notes': {'RHEA': '22144;22145;22146;22147',\n",
       "   'EC Number': '4.2.1.3',\n",
       "   'KEGG Reaction': 'R01900',\n",
       "   'UniPathway Reaction': 'UCR01900',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR30609'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38240_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38240_1 at 0x7fc4498209d0>},\n",
       "  '_metabolites': {<Metabolite acon_C_c at 0x7fc448e1c460>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite icit_c at 0x7fc448e1c4c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPPAT120',\n",
       "  'name': 'Acyl-(acyl carrier protein):phosphate acyltransferase (C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81535'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ddcaACP_c at 0x7fc463a82580>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite ddcap_c at 0x7fc448e1c580>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPPAT140',\n",
       "  'name': 'Acyl-(acyl carrier protein):phosphate acyltransferase (C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81536'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite ttdcap_c at 0x7fc448e1c5b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPPAT141',\n",
       "  'name': 'Acyl-(acyl carrier protein):phosphate acyltransferase (C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81537'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite tdeACP_c at 0x7fc463a82610>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite ttdceap_c at 0x7fc448e1c5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPPAT160',\n",
       "  'name': 'Acyl-(acyl carrier protein):phosphate acyltransferase (C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81538'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite palmACP_c at 0x7fc463a82640>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite hdcap_c at 0x7fc448e1c610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPPAT161',\n",
       "  'name': 'Acyl-(acyl carrier protein):phosphate acyltransferase (C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81539'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite hdeACP_c at 0x7fc463a82670>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite hdceap_c at 0x7fc448e1c640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPPAT180',\n",
       "  'name': 'Acyl-(acyl carrier protein):phosphate acyltransferase (C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81540'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite ocdcaACP_c at 0x7fc463aae5b0>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite ocdcap_c at 0x7fc448e1c670>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPPAT181',\n",
       "  'name': 'Acyl-(acyl carrier protein):phosphate acyltransferase (C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81541'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite octeACP_c at 0x7fc463aae580>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite ocdceap_c at 0x7fc448e1c6a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACPS',\n",
       "  'name': 'ACPS',\n",
       "  'notes': {'BioCyc': 'META:RXN-10811',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR449',\n",
       "   'KEGG Reaction': 'R10747',\n",
       "   'EC Number': '3.6.1.-'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37368_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37368_1 at 0x7fc449820a00>},\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite pan4p_c at 0x7fc448e1c6d0>: 1.0,\n",
       "   <Metabolite pap_c at 0x7fc463adc1f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACTD2',\n",
       "  'name': 'Acetoin dehydrogenase',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36343_1 or (EDU37132_1 and EDU37133_1 and EDU37135_1 and EDU37136_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36343_1 at 0x7fc449820ac0>,\n",
       "   <Gene EDU37132_1 at 0x7fc449820a30>,\n",
       "   <Gene EDU37133_1 at 0x7fc449820af0>,\n",
       "   <Gene EDU37135_1 at 0x7fc449820a60>,\n",
       "   <Gene EDU37136_1 at 0x7fc449820a90>},\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite actn__R_c at 0x7fc463aaea90>: -1.0,\n",
       "   <Metabolite acald_c at 0x7fc463aae2b0>: 1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACtex',\n",
       "  'name': 'Acetate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR465',\n",
       "   'BioCyc': 'META:RXN0-1981;META:TRANS-RXN0-567'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite ac_e at 0x7fc463aaeeb0>: -1.0,\n",
       "   <Metabolite ac_p at 0x7fc448e1c790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACTNabc',\n",
       "  'name': 'Acetoin transport via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36062_1 and EDU37582_1 and EDU37584_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36062_1 at 0x7fc449820b50>,\n",
       "   <Gene EDU37582_1 at 0x7fc449820b20>,\n",
       "   <Gene EDU37584_1 at 0x7fc449820b80>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite actn__R_e at 0x7fc448e1c7c0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite actn__R_c at 0x7fc463aaea90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ACTNabc1',\n",
       "  'name': 'Acetoin efflux  via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36062_1 and EDU37582_1 and EDU37584_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36062_1 at 0x7fc449820b50>,\n",
       "   <Gene EDU37582_1 at 0x7fc449820b20>,\n",
       "   <Gene EDU37584_1 at 0x7fc449820b80>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite actn__R_c at 0x7fc463aaea90>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite actn__R_e at 0x7fc448e1c7c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADA',\n",
       "  'name': 'Adenosine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81551', 'EC Number': '3.5.4.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38839_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38839_1 at 0x7fc449820bb0>},\n",
       "  '_metabolites': {<Metabolite adn_c at 0x7fc463a82160>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ins_c at 0x7fc448e1c820>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADCL',\n",
       "  'name': '4-aminobenzoate synthase',\n",
       "  'notes': {'EC Number': '4.1.3.38',\n",
       "   'BioCyc': 'META:ADCLY-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR467',\n",
       "   'KEGG Reaction': 'R05553',\n",
       "   'RHEA': '16201;16202;16203;16204',\n",
       "   'UniPathway Reaction': 'UCR05553;UER00150'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36284_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36284_1 at 0x7fc449820be0>},\n",
       "  '_metabolites': {<Metabolite 4adcho_c at 0x7fc448e1c8e0>: -1.0,\n",
       "   <Metabolite 4abz_c at 0x7fc463a828e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADCS',\n",
       "  'name': '4-amino-4-deoxychorismate synthase',\n",
       "  'notes': {'RHEA': '11672;11673;11674;11675',\n",
       "   'BioCyc': 'META:PABASYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR30623',\n",
       "   'KEGG Reaction': 'R01716',\n",
       "   'EC Number': '2.6.1.85'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite chor_c at 0x7fc448e1c940>: -1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite 4adcho_c at 0x7fc448e1c8e0>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADD',\n",
       "  'name': 'Adenine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81555', 'EC Number': '3.5.4.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37825_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37825_1 at 0x7fc449820c10>},\n",
       "  '_metabolites': {<Metabolite ade_c at 0x7fc463aae040>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hxan_c at 0x7fc448e1c970>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADEt2rpp',\n",
       "  'name': 'Adenine transport via proton symport (reversible) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR475',\n",
       "   'BioCyc': 'META:TRANS-RXN0-447;META:TRANS-RXN-198'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35943_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35943_1 at 0x7fc449820c40>},\n",
       "  '_metabolites': {<Metabolite ade_p at 0x7fc448e1c9a0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ade_c at 0x7fc463aae040>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADK1',\n",
       "  'name': 'Adenylate kinase',\n",
       "  'notes': {'EC Number': '2.7.4.3',\n",
       "   'BioCyc': 'META:ADENYL-KIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00680',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR479',\n",
       "   'KEGG Reaction': 'R00127',\n",
       "   'RHEA': '12973;12974;12975;12976',\n",
       "   'UniPathway Reaction': 'UCR00127;UER00649'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39058_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39058_1 at 0x7fc449820c70>},\n",
       "  '_metabolites': {<Metabolite amp_c at 0x7fc463a6db80>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADK2',\n",
       "  'name': 'ADK2',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR480', 'EC Number': '2.7.4.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39058_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39058_1 at 0x7fc449820c70>},\n",
       "  '_metabolites': {<Metabolite amp_c at 0x7fc463a6db80>: -1.0,\n",
       "   <Metabolite pppi_c at 0x7fc463aae310>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADK4',\n",
       "  'name': 'Adentylate kinase (ITP)',\n",
       "  'notes': {'RHEA': '29867;29868;29869;29870',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68284'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite amp_c at 0x7fc463a6db80>: -1.0,\n",
       "   <Metabolite itp_c at 0x7fc463aaec10>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite idp_c at 0x7fc463aaebe0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADMDC',\n",
       "  'name': 'Adenosylmethionine decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.50',\n",
       "   'BioCyc': 'META:SAMDECARB-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR484',\n",
       "   'KEGG Reaction': 'R00178',\n",
       "   'RHEA': '15981;15982;15983;15984',\n",
       "   'UniPathway Reaction': 'UCR00178;UER00451'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37027_1 or EDU39172_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37027_1 at 0x7fc449820ca0>,\n",
       "   <Gene EDU39172_1 at 0x7fc449820cd0>},\n",
       "  '_metabolites': {<Metabolite amet_c at 0x7fc463adc940>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite ametam_c at 0x7fc448e1ca60>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADPT',\n",
       "  'name': 'Adenine phosphoribosyltransferase',\n",
       "  'notes': {'EC Number': '2.4.2.8',\n",
       "   'BioCyc': 'META:ADENPRIBOSYLTRAN-RXN',\n",
       "   'BioPath Reaction': 'RXN01461',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68290',\n",
       "   'KEGG Reaction': 'R00190',\n",
       "   'RHEA': '16609;16610;16611;16612',\n",
       "   'UniPathway Reaction': 'UCR00190;UER00646'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35786_1 or EDU36804_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35786_1 at 0x7fc449820d30>,\n",
       "   <Gene EDU36804_1 at 0x7fc449820d00>},\n",
       "  '_metabolites': {<Metabolite ade_c at 0x7fc463aae040>: -1.0,\n",
       "   <Metabolite prpp_c at 0x7fc463aae0d0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADSK',\n",
       "  'name': 'Adenylyl-sulfate kinase',\n",
       "  'notes': {'EC Number': '2.7.1.25',\n",
       "   'BioCyc': 'META:ADENYLYLSULFKIN-RXN',\n",
       "   'BioPath Reaction': 'RXN01225',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR513',\n",
       "   'KEGG Reaction': 'R00509',\n",
       "   'RHEA': '24152;24153;24154;24155',\n",
       "   'UniPathway Reaction': 'UCR00509;UER00205'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36146_1 or EDU38050_1 or EDU39034_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36146_1 at 0x7fc449820d90>,\n",
       "   <Gene EDU38050_1 at 0x7fc449820d60>,\n",
       "   <Gene EDU39034_1 at 0x7fc449820dc0>},\n",
       "  '_metabolites': {<Metabolite aps_c at 0x7fc448e1cca0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite paps_c at 0x7fc463adc0d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADSL1r',\n",
       "  'name': 'Adenylsuccinate lyase',\n",
       "  'notes': {'EC Number': '4.3.2.2',\n",
       "   'BioCyc': 'META:AMPSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN00398',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR514',\n",
       "   'KEGG Reaction': 'R01083',\n",
       "   'RHEA': '16853;16854;16855;16856',\n",
       "   'UniPathway Reaction': 'UCR01083;UER00336'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36439_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36439_1 at 0x7fc449820df0>},\n",
       "  '_metabolites': {<Metabolite dcamp_c at 0x7fc448e1ccd0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite fum_c at 0x7fc448e1cd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADSL2r',\n",
       "  'name': 'Adenylosuccinate lyase',\n",
       "  'notes': {'EC Number': '4.3.2.2',\n",
       "   'BioCyc': 'META:AICARSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN01106',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR515',\n",
       "   'KEGG Reaction': 'R04559',\n",
       "   'RHEA': '23920;23921;23922;23923',\n",
       "   'UniPathway Reaction': 'UCR04559;UER00132'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36439_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36439_1 at 0x7fc449820df0>},\n",
       "  '_metabolites': {<Metabolite 25aics_c at 0x7fc448e1cd30>: -1.0,\n",
       "   <Metabolite aicar_c at 0x7fc448e1cc40>: 1.0,\n",
       "   <Metabolite fum_c at 0x7fc448e1cd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ADSS',\n",
       "  'name': 'Adenylosuccinate synthase',\n",
       "  'notes': {'EC Number': '6.3.4.4',\n",
       "   'BioCyc': 'META:ADENYLOSUCCINATE-SYNTHASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01479',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68295',\n",
       "   'KEGG Reaction': 'R01135',\n",
       "   'RHEA': '15753;15754;15755;15756',\n",
       "   'UniPathway Reaction': 'UCR01135;UER00335'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38130_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38130_1 at 0x7fc449820e20>},\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite imp_c at 0x7fc448e1cd60>: -1.0,\n",
       "   <Metabolite dcamp_c at 0x7fc448e1ccd0>: 1.0,\n",
       "   <Metabolite gdp_c at 0x7fc463adc850>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGDC',\n",
       "  'name': 'N-acetylglucosamine-6-phosphate deacetylase',\n",
       "  'notes': {'EC Number': '3.5.1.25',\n",
       "   'BioCyc': 'META:NAG6PDEACET-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81575',\n",
       "   'KEGG Reaction': 'R02059',\n",
       "   'RHEA': '22936;22937;22938;22939',\n",
       "   'UniPathway Reaction': 'UCR02059;UER00683'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36010_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36010_1 at 0x7fc449820e50>},\n",
       "  '_metabolites': {<Metabolite acgam6p_c at 0x7fc463aaed00>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ac_c at 0x7fc463aae520>: 1.0,\n",
       "   <Metabolite gam6p_c at 0x7fc463aaed30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGM3PA',\n",
       "  'name': 'N-Acetyl-D-glucosamine(anhydrous)N-Acetylmuramyl-tripeptide amidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68303'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37399_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37399_1 at 0x7fc449820e80>},\n",
       "  '_metabolites': {<Metabolite anhgm3p_c at 0x7fc448e1ce50>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite LalaDgluMdap_c at 0x7fc463a82730>: 1.0,\n",
       "   <Metabolite anhgm_c at 0x7fc448e1ce20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGM3PApp',\n",
       "  'name': 'N-Acetyl-D-glucosamine(anhydrous)N-Acetylmuramyl-tripeptide amidase (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68303'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37399_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37399_1 at 0x7fc449820e80>},\n",
       "  '_metabolites': {<Metabolite anhgm3p_p at 0x7fc448e1ceb0>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite LalaDgluMdap_p at 0x7fc463a82760>: 1.0,\n",
       "   <Metabolite anhgm_p at 0x7fc448e1ce80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGM4PA',\n",
       "  'name': 'N-Acetyl-D-glucosamine(anhydrous)N-Acetylmuramyl-tetrapeptide amidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68306'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37399_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37399_1 at 0x7fc449820e80>},\n",
       "  '_metabolites': {<Metabolite anhgm4p_c at 0x7fc448e1cfa0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite LalaDgluMdapDala_c at 0x7fc463a82fa0>: 1.0,\n",
       "   <Metabolite anhgm_c at 0x7fc448e1ce20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGM4PApp',\n",
       "  'name': 'N-Acetyl-D-glucosamine(anhydrous)N-Acetylmuramyl-tetrapeptide amidase (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68306'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37399_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37399_1 at 0x7fc449820e80>},\n",
       "  '_metabolites': {<Metabolite anhgm4p_p at 0x7fc448e1cfd0>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite LalaDgluMdapDala_p at 0x7fc463a82f10>: 1.0,\n",
       "   <Metabolite anhgm_p at 0x7fc448e1ce80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGM4PCP',\n",
       "  'name': 'N-Acetyl-D-glucosamine(anhydrous)N-Acetylmuramyl-tetrapeptide L,D-carboxypeptidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68307'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38089_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38089_1 at 0x7fc449820490>},\n",
       "  '_metabolites': {<Metabolite anhgm4p_c at 0x7fc448e1cfa0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ala__D_c at 0x7fc463a82fd0>: 1.0,\n",
       "   <Metabolite anhgm3p_c at 0x7fc448e1ce50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGM4PH',\n",
       "  'name': 'N-Acetyl-D-glucosamine(anhydrous)N-Acetylmuramyl-tetrapeptide beta -1,4-N-acetylglucosaminidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81580'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite anhgm4p_c at 0x7fc448e1cfa0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite acgam_c at 0x7fc463aaeca0>: 1.0,\n",
       "   <Metabolite anhm4p_c at 0x7fc448e1cf10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGM4Pt2pp',\n",
       "  'name': 'GlcNAc-anhMurNAc tetrapeptide transport in via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68309'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite anhgm4p_p at 0x7fc448e1cfd0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite anhgm4p_c at 0x7fc448e1cfa0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGMH',\n",
       "  'name': 'N-Acetyl-D-glucosamine(anhydrous)N-Acetylmuramyl beta -1,4-N-acetylglucosaminidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68310'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite anhgm_c at 0x7fc448e1ce20>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite acgam_c at 0x7fc463aaeca0>: 1.0,\n",
       "   <Metabolite anhm_c at 0x7fc448e1cf40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGMT',\n",
       "  'name': 'Agmatinase',\n",
       "  'notes': {'EC Number': '3.5.3.11',\n",
       "   'BioCyc': 'META:AGMATIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00171',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR558',\n",
       "   'KEGG Reaction': 'R01157',\n",
       "   'RHEA': '13929;13930;13931;13932',\n",
       "   'UniPathway Reaction': 'UCR01157;UER00287'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite agm_c at 0x7fc448e1cf70>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ptrc_c at 0x7fc448e49160>: 1.0,\n",
       "   <Metabolite urea_c at 0x7fc448e49040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGPAT120',\n",
       "  'name': '1-tetradecanoyl-sn-glycerol 3-phosphate O-acyltransferase (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68311', 'EC Number': '2.3.1.51'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37934_1 or EDU38238_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37934_1 at 0x7fc449820eb0>,\n",
       "   <Gene EDU38238_1 at 0x7fc449820ee0>},\n",
       "  '_metabolites': {<Metabolite 1ddecg3p_c at 0x7fc448e49070>: -1.0,\n",
       "   <Metabolite ddcaACP_c at 0x7fc463a82580>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite pa120_c at 0x7fc448e490d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGPAT140',\n",
       "  'name': '1-tetradecanoyl-sn-glycerol 3-phosphate O-acyltransferase (n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81582', 'EC Number': '2.3.1.51'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37934_1 or EDU38238_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37934_1 at 0x7fc449820eb0>,\n",
       "   <Gene EDU38238_1 at 0x7fc449820ee0>},\n",
       "  '_metabolites': {<Metabolite 1tdecg3p_c at 0x7fc448e49100>: -1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite pa140_c at 0x7fc448e491c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGPAT141',\n",
       "  'name': '1-tetradec-7-enoyl-sn-glycerol 3-phosphate O-acyltransferase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68313', 'EC Number': '2.3.1.51'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 1tdec7eg3p_c at 0x7fc448e491f0>: -1.0,\n",
       "   <Metabolite tdeACP_c at 0x7fc463a82610>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite pa141_c at 0x7fc448e49220>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGPAT160',\n",
       "  'name': '1-hexadecanoyl-sn-glycerol 3-phosphate O-acyltransferase (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81583', 'EC Number': '2.3.1.51'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37934_1 or EDU38238_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37934_1 at 0x7fc449820eb0>,\n",
       "   <Gene EDU38238_1 at 0x7fc449820ee0>},\n",
       "  '_metabolites': {<Metabolite 1hdecg3p_c at 0x7fc448e49250>: -1.0,\n",
       "   <Metabolite palmACP_c at 0x7fc463a82640>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite pa160_c at 0x7fc448e49280>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGPAT161',\n",
       "  'name': '1-hexadec-7-enoyl-sn-glycerol 3-phosphate O-acyltransferase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81585', 'EC Number': '2.3.1.51'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37934_1 or EDU38238_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37934_1 at 0x7fc449820eb0>,\n",
       "   <Gene EDU38238_1 at 0x7fc449820ee0>},\n",
       "  '_metabolites': {<Metabolite 1hdec9eg3p_c at 0x7fc448e492b0>: -1.0,\n",
       "   <Metabolite hdeACP_c at 0x7fc463a82670>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite pa161_c at 0x7fc448e492e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGPAT180',\n",
       "  'name': '1-octadecanoyl-sn-glycerol 3-phosphate O-acyltransferase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81586', 'EC Number': '2.3.1.51'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37934_1 or EDU38238_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37934_1 at 0x7fc449820eb0>,\n",
       "   <Gene EDU38238_1 at 0x7fc449820ee0>},\n",
       "  '_metabolites': {<Metabolite 1odecg3p_c at 0x7fc448e49310>: -1.0,\n",
       "   <Metabolite ocdcaACP_c at 0x7fc463aae5b0>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite pa180_c at 0x7fc448e49340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AGPAT181',\n",
       "  'name': '1-octadec-7-enoyl-sn-glycerol 3-phosphate O-acyltransferase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81588', 'EC Number': '2.3.1.51'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37934_1 or EDU38238_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37934_1 at 0x7fc449820eb0>,\n",
       "   <Gene EDU38238_1 at 0x7fc449820ee0>},\n",
       "  '_metabolites': {<Metabolite 1odec11eg3p_c at 0x7fc448e49370>: -1.0,\n",
       "   <Metabolite octeACP_c at 0x7fc463aae580>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite pa181_c at 0x7fc448e493a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AHCYSNS',\n",
       "  'name': 'S-adenosylhomocysteine nucleosidase',\n",
       "  'notes': {'RHEA': '17805;17806;17807;17808',\n",
       "   'BioCyc': 'META:ADENOSYLHOMOCYSTEINE-NUCLEOSIDASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR579',\n",
       "   'KEGG Reaction': 'R00194',\n",
       "   'EC Number': '3.2.2.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37304_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37304_1 at 0x7fc449820f10>},\n",
       "  '_metabolites': {<Metabolite ahcys_c at 0x7fc463adc910>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ade_c at 0x7fc463aae040>: 1.0,\n",
       "   <Metabolite rhcys_c at 0x7fc448e49490>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AHMMPS',\n",
       "  'name': '4 amino 5 hydroxymethyl 2 methylpyrimidine synthetase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68324'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35932_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35932_1 at 0x7fc449820f40>},\n",
       "  '_metabolites': {<Metabolite air_c at 0x7fc463aae1f0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite 4ahmmp_c at 0x7fc448e494c0>: 1.0,\n",
       "   <Metabolite gcald_c at 0x7fc448e494f0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AHSERL2',\n",
       "  'name': 'O acetylhomoserine  thiol  lyase',\n",
       "  'notes': {'EC Number': '2.5.1.49',\n",
       "   'BioCyc': 'META:ACETYLHOMOSER-CYS-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68325',\n",
       "   'KEGG Reaction': 'R01287',\n",
       "   'RHEA': '27822;27823;27824;27825',\n",
       "   'UniPathway Reaction': 'UCR01287;UER00079'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36946_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36946_1 at 0x7fc449820f70>},\n",
       "  '_metabolites': {<Metabolite h2s_c at 0x7fc448e49580>: -1.0,\n",
       "   <Metabolite achms_c at 0x7fc463aaefd0>: -1.0,\n",
       "   <Metabolite ac_c at 0x7fc463aae520>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hcys__L_c at 0x7fc448e49460>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AHSERL4',\n",
       "  'name': 'Cysteine synthase  Thiosulfate ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite acser_c at 0x7fc448e1c700>: -1.0,\n",
       "   <Metabolite trdrd_c at 0x7fc448e495e0>: -1.0,\n",
       "   <Metabolite tsul_c at 0x7fc448e1c730>: -1.0,\n",
       "   <Metabolite ac_c at 0x7fc463aae520>: 1.0,\n",
       "   <Metabolite cys__L_c at 0x7fc463aae3d0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite so3_c at 0x7fc463a82850>: 1.0,\n",
       "   <Metabolite trdox_c at 0x7fc448e495b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AICART',\n",
       "  'name': 'Phosphoribosylaminoimidazolecarboxamide formyltransferase',\n",
       "  'notes': {'EC Number': '2.1.2.3',\n",
       "   'BioPath Reaction': 'RXN00808',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR588',\n",
       "   'KEGG Reaction': 'R04560',\n",
       "   'RHEA': '22192;22193;22194;22195',\n",
       "   'UniPathway Reaction': 'UCR04560;UER00133'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35969_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35969_1 at 0x7fc449820fa0>},\n",
       "  '_metabolites': {<Metabolite 10fthf_c at 0x7fc463a377c0>: -1.0,\n",
       "   <Metabolite aicar_c at 0x7fc448e1cc40>: -1.0,\n",
       "   <Metabolite fprica_c at 0x7fc448e49610>: 1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AIRC2',\n",
       "  'name': 'Phosphoribosylaminoimidazole carboxylase',\n",
       "  'notes': {'EC Number': '6.3.4.18',\n",
       "   'BioCyc': 'META:RXN0-742',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR590',\n",
       "   'KEGG Reaction': 'R07404',\n",
       "   'RHEA': '19317;19318;19319;19320',\n",
       "   'UniPathway Reaction': 'UCR07404;UER00942'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38033_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38033_1 at 0x7fc449820fd0>},\n",
       "  '_metabolites': {<Metabolite air_c at 0x7fc463aae1f0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hco3_c at 0x7fc463adc5b0>: -1.0,\n",
       "   <Metabolite 5caiz_c at 0x7fc448e496a0>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AIRC3',\n",
       "  'name': 'Phosphoribosylaminoimidazole carboxylase (mutase rxn)',\n",
       "  'notes': {'EC Number': '5.4.99.18',\n",
       "   'BioCyc': 'META:RXN0-743',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81637',\n",
       "   'KEGG Reaction': 'R07405',\n",
       "   'RHEA': '13193;13194;13195;13196',\n",
       "   'UniPathway Reaction': 'UCR07405;UER00943'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38033_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38033_1 at 0x7fc449820fd0>},\n",
       "  '_metabolites': {<Metabolite 5aizc_c at 0x7fc448e49670>: -1.0,\n",
       "   <Metabolite 5caiz_c at 0x7fc448e496a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AIRCr',\n",
       "  'name': 'Phosphoribosylaminoimidazole carboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.21',\n",
       "   'BioCyc': 'META:AIRCARBOXY-RXN',\n",
       "   'BioPath Reaction': 'RXN00676',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81638',\n",
       "   'KEGG Reaction': 'R04209',\n",
       "   'RHEA': '10792;10793;10794;10795',\n",
       "   'UniPathway Reaction': 'UCR04209;UER00130'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35964_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35964_1 at 0x7fc449820100>},\n",
       "  '_metabolites': {<Metabolite air_c at 0x7fc463aae1f0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: -1.0,\n",
       "   <Metabolite 5aizc_c at 0x7fc448e49670>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AKGDa',\n",
       "  'name': 'Oxoglutarate dehydrogenase (lipoamide)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68326', 'EC Number': '1.2.4.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite akg_c at 0x7fc463adc370>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite lpam_c at 0x7fc463adceb0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite sdhlam_c at 0x7fc448e496d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AKGDb',\n",
       "  'name': 'Oxoglutarate dehydrogenase (dihydrolipoamide S-succinyltransferase)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68327', 'EC Number': '2.3.1.61'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite sdhlam_c at 0x7fc448e496d0>: -1.0,\n",
       "   <Metabolite succoa_c at 0x7fc463a82250>: 1.0,\n",
       "   <Metabolite dhlam_c at 0x7fc448e49700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AKGDH',\n",
       "  'name': '2-Oxogluterate dehydrogenase',\n",
       "  'notes': {'EC Number': '2.3.1.61',\n",
       "   'BioCyc': 'META:2OXOGLUTARATEDEH-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR595',\n",
       "   'KEGG Reaction': 'R08549',\n",
       "   'RHEA': '27786;27787;27788;27789',\n",
       "   'UniPathway Reaction': 'UCR08549;UER00997'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37132_1 or EDU37133_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37132_1 at 0x7fc449820a30>,\n",
       "   <Gene EDU37133_1 at 0x7fc449820af0>},\n",
       "  '_metabolites': {<Metabolite akg_c at 0x7fc463adc370>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite succoa_c at 0x7fc463a82250>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAabc',\n",
       "  'name': 'L-alanine transport via ABC system',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR635'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU38276_1 and EDU38278_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1 and EDU38279_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36982_1 at 0x7fc44983b280>,\n",
       "   <Gene EDU38275_1 at 0x7fc44983b220>,\n",
       "   <Gene EDU38276_1 at 0x7fc44983b250>,\n",
       "   <Gene EDU38278_1 at 0x7fc44983b2e0>,\n",
       "   <Gene EDU38279_1 at 0x7fc44983b2b0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ala__L_e at 0x7fc463a82c40>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAabcpp',\n",
       "  'name': 'L-alanine transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR635'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU36982_1 and EDU38275_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1) or (EDU36982_1 and EDU38275_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1 and EDU38279_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36982_1 at 0x7fc44983b280>,\n",
       "   <Gene EDU38275_1 at 0x7fc44983b220>,\n",
       "   <Gene EDU38276_1 at 0x7fc44983b250>,\n",
       "   <Gene EDU38279_1 at 0x7fc44983b2b0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ala__L_p at 0x7fc448e497f0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAALAabcpp',\n",
       "  'name': 'D-alanyl-D-alanine (DalaDala) transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR613'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37580_1 or (EDU37419_1 and EDU38656_1) or (EDU37419_1 and EDU37423_1 and EDU37580_1) or (EDU37419_1 and EDU37580_1 and EDU38656_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37419_1 at 0x7fc4498203d0>,\n",
       "   <Gene EDU37423_1 at 0x7fc4498201f0>,\n",
       "   <Gene EDU37580_1 at 0x7fc44983b040>,\n",
       "   <Gene EDU38656_1 at 0x7fc44983b0d0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite alaala_p at 0x7fc448e49850>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite alaala_c at 0x7fc448e49820>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAALAr',\n",
       "  'name': 'D-alanine-D-alanine ligase (reversible)',\n",
       "  'notes': {'RHEA': '11224;11225;11226;11227',\n",
       "   'BioCyc': 'META:DALADALALIG-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR614',\n",
       "   'KEGG Reaction': 'R01150',\n",
       "   'EC Number': '6.3.2.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38274_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38274_1 at 0x7fc44983b0a0>},\n",
       "  '_metabolites': {<Metabolite ala__D_c at 0x7fc463a82fd0>: -2.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite alaala_c at 0x7fc448e49820>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAD_L',\n",
       "  'name': 'L-alanine dehydrogenase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81648', 'EC Number': '1.4.1.1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38340_1 at 0x7fc44983b130>},\n",
       "  '_metabolites': {<Metabolite ala__L_c at 0x7fc463a82c10>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAR',\n",
       "  'name': 'Alanine racemase',\n",
       "  'notes': {'EC Number': '5.1.1.1',\n",
       "   'BioCyc': 'META:ALARACECAT-RXN',\n",
       "   'BioPath Reaction': 'RXN01498',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR626',\n",
       "   'KEGG Reaction': 'R00401',\n",
       "   'RHEA': '20249;20250;20251;20252',\n",
       "   'UniPathway Reaction': 'UCR00401;UER00497'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37097_1 or EDU37846_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37097_1 at 0x7fc44983b190>,\n",
       "   <Gene EDU37846_1 at 0x7fc44983b160>},\n",
       "  '_metabolites': {<Metabolite ala__L_c at 0x7fc463a82c10>: -1.0,\n",
       "   <Metabolite ala__D_c at 0x7fc463a82fd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAt2r',\n",
       "  'name': 'L alanine reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR636', 'BioCyc': 'META:RXN0-5202'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36264_1 or EDU37508_1 or EDU37559_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36264_1 at 0x7fc44983b340>,\n",
       "   <Gene EDU37508_1 at 0x7fc44983b310>,\n",
       "   <Gene EDU37559_1 at 0x7fc44983b370>},\n",
       "  '_metabolites': {<Metabolite ala__L_e at 0x7fc463a82c40>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAt4',\n",
       "  'name': 'ALAt4',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR637',\n",
       "   'BioCyc': 'META:TRANS-RXN-125'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36264_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36264_1 at 0x7fc44983b340>},\n",
       "  '_metabolites': {<Metabolite ala__L_e at 0x7fc463a82c40>: -1.0,\n",
       "   <Metabolite na1_e at 0x7fc463adc610>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAt4pp',\n",
       "  'name': 'L-alanine transport in via sodium symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR637',\n",
       "   'BioCyc': 'META:TRANS-RXN-125'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35942_1 or EDU36264_1 or EDU37508_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35942_1 at 0x7fc44983b3a0>,\n",
       "   <Gene EDU36264_1 at 0x7fc44983b340>,\n",
       "   <Gene EDU37508_1 at 0x7fc44983b310>},\n",
       "  '_metabolites': {<Metabolite ala__L_p at 0x7fc448e497f0>: -1.0,\n",
       "   <Metabolite na1_p at 0x7fc463adcfd0>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALATA_L',\n",
       "  'name': 'L-alanine transaminase',\n",
       "  'notes': {'EC Number': '2.6.1.2',\n",
       "   'BioCyc': 'META:ALANINE-AMINOTRANSFERASE-RXN;META:RXN-13698',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR32635',\n",
       "   'KEGG Reaction': 'R00258',\n",
       "   'RHEA': '19453;19454;19455;19456',\n",
       "   'UniPathway Reaction': 'UCR00258;UER00586'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37030_1 or EDU39385_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37030_1 at 0x7fc44983b1c0>,\n",
       "   <Gene EDU39385_1 at 0x7fc44983b1f0>},\n",
       "  '_metabolites': {<Metabolite akg_c at 0x7fc463adc370>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALAtex',\n",
       "  'name': 'L-alanine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR639',\n",
       "   'BioCyc': 'META:TRANS-RXN0-469'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ala__L_e at 0x7fc463a82c40>: -1.0,\n",
       "   <Metabolite ala__L_p at 0x7fc448e497f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALCD1',\n",
       "  'name': 'Alcohol dehydrogenase (methanol)',\n",
       "  'notes': {'EC Number': '1.1.1.244',\n",
       "   'BioCyc': 'META:METHANOL-DEHYDROGENASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR640',\n",
       "   'KEGG Reaction': 'R00605',\n",
       "   'RHEA': '19401;19402;19403;19404',\n",
       "   'UniPathway Reaction': 'UCR00605;UER00893'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite meoh_c at 0x7fc448e49b20>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite fald_c at 0x7fc463adc4c0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALCD19',\n",
       "  'name': 'Alcohol dehydrogenase (glycerol)',\n",
       "  'notes': {'RHEA': '30027;30028;30029;30030',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68334',\n",
       "   'KEGG Reaction': 'R01036',\n",
       "   'EC Number': '1.1.1.72'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37339_1 at 0x7fc449820580>},\n",
       "  '_metabolites': {<Metabolite glyald_c at 0x7fc448e49b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite glyc_c at 0x7fc448e49b80>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALCD19y',\n",
       "  'name': 'Alcohol dehydrogenase  glycerol  NADP ',\n",
       "  'notes': {'EC Number': '1.1.1.72',\n",
       "   'BioCyc': 'META:GLYCEROL-DEHYDROGENASE-NADP+-RXN',\n",
       "   'BioPath Reaction': 'RXN00332',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68335',\n",
       "   'RHEA': '23592;23593;23594;23595',\n",
       "   'KEGG Reaction': 'R01041'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39153_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39153_1 at 0x7fc44983b3d0>},\n",
       "  '_metabolites': {<Metabolite glyald_c at 0x7fc448e49b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite glyc_c at 0x7fc448e49b80>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALCD2x',\n",
       "  'name': 'Alcohol dehydrogenase (ethanol)',\n",
       "  'notes': {'EC Number': '1.1.1.71',\n",
       "   'BioCyc': 'META:ALCOHOL-DEHYDROG-RXN',\n",
       "   'BioPath Reaction': 'RXN01075',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR648',\n",
       "   'KEGG Reaction': 'R00754',\n",
       "   'RHEA': '25290;25291;25292;25293',\n",
       "   'UniPathway Reaction': 'UCR00754;UER00767'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36688_1 or EDU37784_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36688_1 at 0x7fc44983b400>,\n",
       "   <Gene EDU37784_1 at 0x7fc4498205e0>},\n",
       "  '_metabolites': {<Metabolite etoh_c at 0x7fc463aaefa0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite acald_c at 0x7fc463aae2b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALCD2yi',\n",
       "  'name': 'Alcohol dehydrogenase (ethanol, NADP)',\n",
       "  'notes': {'BioCyc': 'META:RXN-12484',\n",
       "   'BioPath Reaction': 'RXN00183',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR649',\n",
       "   'KEGG Reaction': 'R00746',\n",
       "   'EC Number': '1.1.1.71'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35940_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35940_1 at 0x7fc44983b430>},\n",
       "  '_metabolites': {<Metabolite acald_c at 0x7fc463aae2b0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite etoh_c at 0x7fc463aaefa0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALCD4',\n",
       "  'name': 'Alcohol dehydrogenase (butanol)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81653'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37784_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37784_1 at 0x7fc4498205e0>},\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite btoh_c at 0x7fc448e49c70>: -1.0,\n",
       "   <Metabolite btal_c at 0x7fc448e49c40>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALCD4y',\n",
       "  'name': 'Alcohol dehydrogenase (butanol) NADP depnedent',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81654'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37103_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37103_1 at 0x7fc44983b460>},\n",
       "  '_metabolites': {<Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite btoh_c at 0x7fc448e49c70>: -1.0,\n",
       "   <Metabolite btal_c at 0x7fc448e49c40>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALDD19xr',\n",
       "  'name': 'Aldehyde dehydrogenase (phenylacetaldehyde, NAD)',\n",
       "  'notes': {'EC Number': '1.2.1.5',\n",
       "   'BioCyc': 'META:PHENDEHYD-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR652',\n",
       "   'KEGG Reaction': 'R02536',\n",
       "   'RHEA': '21392;21393;21394;21395',\n",
       "   'UniPathway Reaction': 'UCR02536;UER00724'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite pacald_c at 0x7fc448e49be0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite pac_c at 0x7fc448e49ca0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALDD20x',\n",
       "  'name': 'Aldehyde dehydrogenase (indole-3-acetaldehyde, NAD)',\n",
       "  'notes': {'RHEA': '30867;30868;30869;30870',\n",
       "   'BioCyc': 'META:RXN-10715',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR654',\n",
       "   'KEGG Reaction': 'R02678',\n",
       "   'EC Number': '1.2.1.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38973_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38973_1 at 0x7fc449820550>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite id3acald_c at 0x7fc448e49c10>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite ind3ac_c at 0x7fc448e49cd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALDD2x',\n",
       "  'name': 'Aldehyde dehydrogenase (acetaldehyde, NAD)',\n",
       "  'notes': {'EC Number': '1.2.1.5',\n",
       "   'BioCyc': 'META:RXN66-3',\n",
       "   'BioPath Reaction': 'RXN01270',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR657',\n",
       "   'KEGG Reaction': 'R00710',\n",
       "   'RHEA': '25294;25295;25296;25297',\n",
       "   'UniPathway Reaction': 'UCR00710;UER00768'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37555_1 or EDU38973_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37555_1 at 0x7fc44983b490>,\n",
       "   <Gene EDU38973_1 at 0x7fc449820550>},\n",
       "  '_metabolites': {<Metabolite acald_c at 0x7fc463aae2b0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite ac_c at 0x7fc463aae520>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALDD31',\n",
       "  'name': 'Aminoacetaldehyde oxidation',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR27865'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite aacald_c at 0x7fc448e49d00>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite gly_c at 0x7fc448e49400>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALLTAHr',\n",
       "  'name': 'Allantoate amidinohydrolase  reversible',\n",
       "  'notes': {'EC Number': '3.5.3.4',\n",
       "   'BioCyc': 'META:ALLANTOICASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01269',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR665',\n",
       "   'KEGG Reaction': 'R02422',\n",
       "   'RHEA': '11016;11017;11018;11019',\n",
       "   'UniPathway Reaction': 'UCR02422;UER00654'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38456_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38456_1 at 0x7fc44983b4c0>},\n",
       "  '_metabolites': {<Metabolite alltt_c at 0x7fc448e49dc0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite urdglyc_c at 0x7fc448e49df0>: 1.0,\n",
       "   <Metabolite urea_c at 0x7fc448e49040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALLTAMH',\n",
       "  'name': 'Allantoate amidohydrolase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81658', 'EC Number': '3.5.3.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38456_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38456_1 at 0x7fc44983b4c0>},\n",
       "  '_metabolites': {<Metabolite alltt_c at 0x7fc448e49dc0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -2.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 2.0,\n",
       "   <Metabolite urdglyc_c at 0x7fc448e49df0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALLTN',\n",
       "  'name': 'Allantoinase',\n",
       "  'notes': {'RHEA': '30819;30820;30821;30822',\n",
       "   'BioPath Reaction': 'RXN00055',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81659',\n",
       "   'EC Number': '3.5.2.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39082_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39082_1 at 0x7fc44983b4f0>},\n",
       "  '_metabolites': {<Metabolite alltn_c at 0x7fc448e49e20>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite alltt_c at 0x7fc448e49dc0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ALLTNt2r',\n",
       "  'name': 'Allantoin transport in via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81660'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite alltn_e at 0x7fc448e49e50>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite alltn_c at 0x7fc448e49e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AM3PA',\n",
       "  'name': 'Anhydrous-N-Acetylmuramyl-tripeptide amidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68346'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37399_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37399_1 at 0x7fc449820e80>},\n",
       "  '_metabolites': {<Metabolite anhm3p_c at 0x7fc448e1cee0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite LalaDgluMdap_c at 0x7fc463a82730>: 1.0,\n",
       "   <Metabolite anhm_c at 0x7fc448e1cf40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AM4PA',\n",
       "  'name': 'Anhydrous-N-Acetylmuramyl-tetrapeptide amidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81668'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37399_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37399_1 at 0x7fc449820e80>},\n",
       "  '_metabolites': {<Metabolite anhm4p_c at 0x7fc448e1cf10>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite LalaDgluMdapDala_c at 0x7fc463a82fa0>: 1.0,\n",
       "   <Metabolite anhm_c at 0x7fc448e1cf40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AM4PCP',\n",
       "  'name': 'Anhydrous-N-Acetylmuramyl-tetrapeptide L,D-carboxypeptidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81669'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38089_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38089_1 at 0x7fc449820490>},\n",
       "  '_metabolites': {<Metabolite anhm4p_c at 0x7fc448e1cf10>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ala__D_c at 0x7fc463a82fd0>: 1.0,\n",
       "   <Metabolite anhm3p_c at 0x7fc448e1cee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AMANAPEr',\n",
       "  'name': 'N-acetylmannosamine 6-phosphate epimerase',\n",
       "  'notes': {'EC Number': '5.1.3.9',\n",
       "   'BioCyc': 'META:NANE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81674',\n",
       "   'KEGG Reaction': 'R02087',\n",
       "   'RHEA': '25257;25258;25259;25260',\n",
       "   'UniPathway Reaction': 'UCR02087;UER00682'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite acmanap_c at 0x7fc448e1c040>: -1.0,\n",
       "   <Metabolite acgam6p_c at 0x7fc463aaed00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AMANK',\n",
       "  'name': 'N-acetyl-D-mannosamine kinase',\n",
       "  'notes': {'EC Number': '2.7.1.60',\n",
       "   'BioCyc': 'META:NANK-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26557',\n",
       "   'KEGG Reaction': 'R02705',\n",
       "   'RHEA': '25253;25254;25255;25256',\n",
       "   'UniPathway Reaction': 'UCR02705;UER00681'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36232_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36232_1 at 0x7fc44983b520>},\n",
       "  '_metabolites': {<Metabolite acmana_c at 0x7fc463aaecd0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite acmanap_c at 0x7fc448e1c040>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AMMQLT8',\n",
       "  'name': 'S-adenosylmethione:2-demthylmenaquinole methyltransferase (menaquinone 8)',\n",
       "  'notes': {'RHEA': '30063;30064;30065;30066',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR710',\n",
       "   'EC Number': '2.1.1.163',\n",
       "   'BioCyc': 'META:ADOMET-DMK-METHYLTRANSFER-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37078_1 or EDU38267_1 or EDU38289_1 or EDU38600_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37078_1 at 0x7fc44983b550>,\n",
       "   <Gene EDU38267_1 at 0x7fc44983b5b0>,\n",
       "   <Gene EDU38289_1 at 0x7fc44983b5e0>,\n",
       "   <Gene EDU38600_1 at 0x7fc44983b580>},\n",
       "  '_metabolites': {<Metabolite 2dmmql8_c at 0x7fc448e49f10>: -1.0,\n",
       "   <Metabolite amet_c at 0x7fc463adc940>: -1.0,\n",
       "   <Metabolite ahcys_c at 0x7fc463adc910>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AMMQT8_2',\n",
       "  'name': 'S-adenosylmethione:2-demethylmenaquinone methyltransferase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68360'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38600_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38600_1 at 0x7fc44983b580>},\n",
       "  '_metabolites': {<Metabolite 2dmmq8_c at 0x7fc448e49f40>: -1.0,\n",
       "   <Metabolite amet_c at 0x7fc463adc940>: -1.0,\n",
       "   <Metabolite ahcys_c at 0x7fc463adc910>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AMPMS2',\n",
       "  'name': '4-amino-2-methyl-5-phosphomethylpyrimidine synthetase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68362'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35932_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35932_1 at 0x7fc449820f40>},\n",
       "  '_metabolites': {<Metabolite air_c at 0x7fc463aae1f0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 4ampm_c at 0x7fc463adc730>: 1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 3.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AMPTASECG',\n",
       "  'name': 'Alanyl aminopeptidase (cys-gly)',\n",
       "  'notes': {'EC Number': '3.4.13.18',\n",
       "   'BioCyc': 'META:RXN-6622',\n",
       "   'BioPath Reaction': 'RXN00091',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR718',\n",
       "   'RHEA': '28783;28784;28785;28786',\n",
       "   'KEGG Reaction': 'R00899'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35978_1 or EDU36273_1 or EDU36540_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35978_1 at 0x7fc44983b670>,\n",
       "   <Gene EDU36273_1 at 0x7fc44983b610>,\n",
       "   <Gene EDU36540_1 at 0x7fc44983b640>},\n",
       "  '_metabolites': {<Metabolite cgly_c at 0x7fc448e74160>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cys__L_c at 0x7fc463aae3d0>: 1.0,\n",
       "   <Metabolite gly_c at 0x7fc448e49400>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AMPTASEPG',\n",
       "  'name': 'Aminopeptidase (pro-gly)',\n",
       "  'notes': {'RHEA': '28799;28800;28801;28802',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26681'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35978_1 or EDU36540_1 or EDU37387_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35978_1 at 0x7fc44983b670>,\n",
       "   <Gene EDU36540_1 at 0x7fc44983b640>,\n",
       "   <Gene EDU37387_1 at 0x7fc44983b6a0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite progly_c at 0x7fc448e74040>: -1.0,\n",
       "   <Metabolite gly_c at 0x7fc448e49400>: 1.0,\n",
       "   <Metabolite pro__L_c at 0x7fc448e740a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ANHGMtex',\n",
       "  'name': 'GlcNAc-anhMurNAc transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR725'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite anhgm_e at 0x7fc448e740d0>: -1.0,\n",
       "   <Metabolite anhgm_p at 0x7fc448e1ce80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ANHMK',\n",
       "  'name': '1,6-anhydrous-N-Acetylmuramate kinase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68364'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite anhm_c at 0x7fc448e1cf40>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite acmum6p_c at 0x7fc448e1c190>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AOR_CL',\n",
       "  'name': 'Acetaldehyde:ferredoxin oxidoreductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81455'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36883_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36883_1 at 0x7fc44983b6d0>},\n",
       "  '_metabolites': {<Metabolite ac_c at 0x7fc463aae520>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -3.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: -1.0,\n",
       "   <Metabolite acald_c at 0x7fc463aae2b0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 0.0},\n",
       " {'_id': 'AP4AH',\n",
       "  'name': 'Ap4A hydrolase',\n",
       "  'notes': {'RHEA': '24252;24253;24254;24255',\n",
       "   'BioCyc': 'META:3.6.1.41-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR734',\n",
       "   'KEGG Reaction': 'R00125',\n",
       "   'EC Number': '3.6.1.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36908_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36908_1 at 0x7fc44983b700>},\n",
       "  '_metabolites': {<Metabolite ap4a_c at 0x7fc448e74220>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'AP4AS',\n",
       "  'name': 'Ap4A synthetase',\n",
       "  'notes': {'RHEA': '34935;34936;34937;34938',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR57293',\n",
       "   'EC Number': '2.7.7',\n",
       "   'BioCyc': 'META:RXN0-5208'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38229_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38229_1 at 0x7fc44983b730>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite ap4a_c at 0x7fc448e74220>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APG3PAT120',\n",
       "  'name': 'Acyl-phosphate:glycerol-3-phosphate acyltransferase (C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81691'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite ddcap_c at 0x7fc448e1c580>: -1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite 1ddecg3p_c at 0x7fc448e49070>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APG3PAT140',\n",
       "  'name': 'Acyl-phosphate:glycerol-3-phosphate acyltransferase (C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81692'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite ttdcap_c at 0x7fc448e1c5b0>: -1.0,\n",
       "   <Metabolite 1tdecg3p_c at 0x7fc448e49100>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APG3PAT141',\n",
       "  'name': 'Acyl-phosphate:glycerol-3-phosphate acyltransferase (C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81693'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite ttdceap_c at 0x7fc448e1c5e0>: -1.0,\n",
       "   <Metabolite 1tdec7eg3p_c at 0x7fc448e491f0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APG3PAT160',\n",
       "  'name': 'Acyl-phosphate:glycerol-3-phosphate acyltransferase (C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81694'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite hdcap_c at 0x7fc448e1c610>: -1.0,\n",
       "   <Metabolite 1hdecg3p_c at 0x7fc448e49250>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APG3PAT161',\n",
       "  'name': 'Acyl-phosphate:glycerol-3-phosphate acyltransferase (C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81695'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite hdceap_c at 0x7fc448e1c640>: -1.0,\n",
       "   <Metabolite 1hdec9eg3p_c at 0x7fc448e492b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APG3PAT180',\n",
       "  'name': 'Acyl-phosphate:glycerol-3-phosphate acyltransferase (C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81696'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite ocdcap_c at 0x7fc448e1c670>: -1.0,\n",
       "   <Metabolite 1odecg3p_c at 0x7fc448e49310>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APG3PAT181',\n",
       "  'name': 'Acyl-phosphate:glycerol-3-phosphate acyltransferase (C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81697'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite ocdceap_c at 0x7fc448e1c6a0>: -1.0,\n",
       "   <Metabolite 1odec11eg3p_c at 0x7fc448e49370>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'APRAUR',\n",
       "  'name': '5-amino-6-(5-phosphoribosylamino)uracil reductase',\n",
       "  'notes': {'EC Number': '1.1.1.193',\n",
       "   'BioCyc': 'META:RIBOFLAVINSYNREDUC-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR745',\n",
       "   'KEGG Reaction': 'R03458',\n",
       "   'RHEA': '17845;17846;17847;17848',\n",
       "   'UniPathway Reaction': 'UCR03458;UER00402'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35979_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35979_1 at 0x7fc44983b790>},\n",
       "  '_metabolites': {<Metabolite 5apru_c at 0x7fc448e743a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 5aprbu_c at 0x7fc448e74370>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARABDI',\n",
       "  'name': 'L-fucose isomerase',\n",
       "  'notes': {'RHEA': '13849;13850;13851;13852',\n",
       "   'BioCyc': 'META:DARABISOM-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81718',\n",
       "   'KEGG Reaction': 'R01577',\n",
       "   'EC Number': '5.3.1.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37034_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37034_1 at 0x7fc44983b7c0>},\n",
       "  '_metabolites': {<Metabolite arab__D_c at 0x7fc448e743d0>: -1.0,\n",
       "   <Metabolite rbl__D_c at 0x7fc448e74400>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARAI',\n",
       "  'name': 'L-arabinose isomerase',\n",
       "  'notes': {'EC Number': '5.3.1.4',\n",
       "   'BioCyc': 'META:ARABISOM-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR5440',\n",
       "   'KEGG Reaction': 'R01761',\n",
       "   'RHEA': '14821;14822;14823;14824',\n",
       "   'UniPathway Reaction': 'UCR01761;UER00565'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite arab__L_c at 0x7fc463aae6a0>: -1.0,\n",
       "   <Metabolite rbl__L_c at 0x7fc448e74430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARBabc',\n",
       "  'name': 'ARBabc',\n",
       "  'notes': {'RHEA': '30007;30008;30009;30010',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR774'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38031_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38031_1 at 0x7fc44983b820>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite arab__L_e at 0x7fc463aae730>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite arab__L_c at 0x7fc463aae6a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARBTpts',\n",
       "  'name': 'Arbutin transport via PEPPyr PTS',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35995_1 and EDU36437_1 and EDU39079_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35995_1 at 0x7fc44983b7f0>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite arbt_e at 0x7fc448e744f0>: -1.0,\n",
       "   <Metabolite arbt6p_c at 0x7fc463aae6d0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGabc',\n",
       "  'name': 'L-arginine transport via ABC system',\n",
       "  'notes': {'RHEA': '29879;29880;29881;29882',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR785',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-4-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36792_1 and EDU37373_1 and EDU38265_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36792_1 at 0x7fc44983ba00>,\n",
       "   <Gene EDU37373_1 at 0x7fc44983ba60>,\n",
       "   <Gene EDU38265_1 at 0x7fc44983ba30>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite arg__L_e at 0x7fc448e49af0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGabcpp',\n",
       "  'name': 'L-arginine transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29879;29880;29881;29882',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR785',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-4-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37427_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37427_1 at 0x7fc44983ba90>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite arg__L_p at 0x7fc448e74520>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGAGMt7pp',\n",
       "  'name': 'Arginine/agmatine antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR776', 'BioCyc': 'META:RXN0-2162'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38352_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38352_1 at 0x7fc44983b850>},\n",
       "  '_metabolites': {<Metabolite agm_c at 0x7fc448e1cf70>: -1.0,\n",
       "   <Metabolite arg__L_p at 0x7fc448e74520>: -1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite agm_p at 0x7fc448e49190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGDC',\n",
       "  'name': 'Arginine decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.19',\n",
       "   'BioCyc': 'META:ARGDECARBOX-RXN',\n",
       "   'BioPath Reaction': 'RXN00721',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR778',\n",
       "   'KEGG Reaction': 'R00566',\n",
       "   'RHEA': '17641;17642;17643;17644',\n",
       "   'UniPathway Reaction': 'UCR00566;UER00284'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38084_1 or EDU39429_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38084_1 at 0x7fc44983b8b0>,\n",
       "   <Gene EDU39429_1 at 0x7fc44983b880>},\n",
       "  '_metabolites': {<Metabolite arg__L_c at 0x7fc448e49ac0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite agm_c at 0x7fc448e1cf70>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGDr',\n",
       "  'name': 'Arginine deiminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81722', 'EC Number': '3.5.3.6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38072_1 or EDU39351_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38072_1 at 0x7fc44983b8e0>,\n",
       "   <Gene EDU39351_1 at 0x7fc44983b910>},\n",
       "  '_metabolites': {<Metabolite arg__L_c at 0x7fc448e49ac0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite citr__L_c at 0x7fc448e74580>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGORNt7',\n",
       "  'name': 'Arginine/ornithine antiporter',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR30738'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37176_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37176_1 at 0x7fc44983b940>},\n",
       "  '_metabolites': {<Metabolite orn_c at 0x7fc448e1c490>: -1.0,\n",
       "   <Metabolite arg__L_e at 0x7fc448e49af0>: -1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite orn_e at 0x7fc448e745e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGORNt7pp',\n",
       "  'name': 'Arginine/ornithine antiporter (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR30738'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37176_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37176_1 at 0x7fc44983b940>},\n",
       "  '_metabolites': {<Metabolite orn_c at 0x7fc448e1c490>: -1.0,\n",
       "   <Metabolite arg__L_p at 0x7fc448e74520>: -1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite orn_p at 0x7fc448e74610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGSL',\n",
       "  'name': 'Argininosuccinate lyase',\n",
       "  'notes': {'EC Number': '4.3.2.1',\n",
       "   'BioCyc': 'META:ARGSUCCINLYA-RXN',\n",
       "   'BioPath Reaction': 'RXN00832',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR782',\n",
       "   'KEGG Reaction': 'R01086',\n",
       "   'RHEA': '24020;24021;24022;24023',\n",
       "   'UniPathway Reaction': 'UCR01086;UER00114;UER00273'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36169_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36169_1 at 0x7fc44983b970>},\n",
       "  '_metabolites': {<Metabolite argsuc_c at 0x7fc448e74640>: -1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite fum_c at 0x7fc448e1cd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGSS',\n",
       "  'name': 'Argininosuccinate synthase',\n",
       "  'notes': {'EC Number': '6.3.4.5',\n",
       "   'BioCyc': 'META:ARGSUCCINSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN01485',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR783',\n",
       "   'KEGG Reaction': 'R01954',\n",
       "   'RHEA': '10932;10933;10934;10935',\n",
       "   'UniPathway Reaction': 'UCR01954;UER00113;UER00272'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35938_1 or EDU36168_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35938_1 at 0x7fc44983b9a0>,\n",
       "   <Gene EDU36168_1 at 0x7fc44983b9d0>},\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite citr__L_c at 0x7fc448e74580>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite argsuc_c at 0x7fc448e74640>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGt2r',\n",
       "  'name': 'L arganine reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR786'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37733_1 or EDU38353_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37733_1 at 0x7fc44983baf0>,\n",
       "   <Gene EDU38353_1 at 0x7fc44983bac0>},\n",
       "  '_metabolites': {<Metabolite arg__L_e at 0x7fc448e49af0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ARGtex',\n",
       "  'name': 'L-arginine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR788', 'BioCyc': 'META:RXN66-448'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite arg__L_e at 0x7fc448e49af0>: -1.0,\n",
       "   <Metabolite arg__L_p at 0x7fc448e74520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASAD',\n",
       "  'name': 'Aspartate-semialdehyde dehydrogenase',\n",
       "  'notes': {'EC Number': '1.2.1.11',\n",
       "   'BioCyc': 'META:ASPARTATE-SEMIALDEHYDE-DEHYDROGENASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR832',\n",
       "   'KEGG Reaction': 'R02291',\n",
       "   'RHEA': '24284;24285;24286;24287',\n",
       "   'UniPathway Reaction': 'UCR02291;UER00016;UER00463;UER00464'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39388_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39388_1 at 0x7fc44983bb20>},\n",
       "  '_metabolites': {<Metabolite aspsa_c at 0x7fc448e74700>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite 4pasp_c at 0x7fc448e746d0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASADi',\n",
       "  'name': 'Aspartate semialdehyde dehydrogenase  irreversible',\n",
       "  'notes': {'EC Number': '1.2.1.11',\n",
       "   'BioCyc': 'META:ASPARTATE-SEMIALDEHYDE-DEHYDROGENASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR832',\n",
       "   'KEGG Reaction': 'R02291',\n",
       "   'RHEA': '24284;24285;24286;24287',\n",
       "   'UniPathway Reaction': 'UCR02291;UER00016;UER00463;UER00464'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 4pasp_c at 0x7fc448e746d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite aspsa_c at 0x7fc448e74700>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASNN',\n",
       "  'name': 'L-asparaginase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81754', 'EC Number': '3.5.1.38'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36315_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36315_1 at 0x7fc44983bb50>},\n",
       "  '_metabolites': {<Metabolite asn__L_c at 0x7fc448e498b0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASNS2',\n",
       "  'name': 'Asparagine synthetase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81756', 'EC Number': '6.3.1.1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37274_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37274_1 at 0x7fc44983bb80>},\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite asn__L_c at 0x7fc448e498b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASO3t4pp',\n",
       "  'name': 'Arsenite transporter via sodium antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81760'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite aso3_c at 0x7fc448e74790>: -1.0,\n",
       "   <Metabolite na1_p at 0x7fc463adcfd0>: -1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0,\n",
       "   <Metabolite aso3_p at 0x7fc448e747c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASO3tex',\n",
       "  'name': 'Arsenite transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR854',\n",
       "   'BioCyc': 'META:TRANS-RXN0-551'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite aso3_e at 0x7fc448e747f0>: -1.0,\n",
       "   <Metabolite aso3_p at 0x7fc448e747c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASO4rt2rpp',\n",
       "  'name': 'Arsenate reversible transport via symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR67939',\n",
       "   'BioCyc': 'META:TRANS-RXN0-550'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite aso4_p at 0x7fc448e74850>: -1.0,\n",
       "   <Metabolite aso4_c at 0x7fc448e74820>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASO4tex',\n",
       "  'name': 'Arsenate transporter (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR66725'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite aso4_e at 0x7fc448e74880>: -1.0,\n",
       "   <Metabolite aso4_p at 0x7fc448e74850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASP1DC',\n",
       "  'name': 'Aspartate 1-decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.15',\n",
       "   'BioCyc': 'META:ASPDECARBOX-RXN',\n",
       "   'BioPath Reaction': 'RXN00901',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR855',\n",
       "   'KEGG Reaction': 'R00489',\n",
       "   'RHEA': '19497;19498;19499;19500',\n",
       "   'UniPathway Reaction': 'UCR00489;UER00002'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35915_1 or EDU38245_1 or EDU39421_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35915_1 at 0x7fc44983bc10>,\n",
       "   <Gene EDU38245_1 at 0x7fc44983bbb0>,\n",
       "   <Gene EDU39421_1 at 0x7fc44983bbe0>},\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite ala_B_c at 0x7fc448e498e0>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASP4DC',\n",
       "  'name': 'Aspartate 4-decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.12',\n",
       "   'BioCyc': 'META:ASPARTATE-4-DECARBOXYLASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00395',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR5617',\n",
       "   'RHEA': '12621;12622;12623;12624',\n",
       "   'KEGG Reaction': 'R00397'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37650_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37650_1 at 0x7fc44983bc40>},\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPabc',\n",
       "  'name': 'L-aspartate transport via ABC system',\n",
       "  'notes': {'RHEA': '29039;29040;29041;29042',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR872',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:TRANS-RXN0-222'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37242_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37242_1 at 0x7fc44983be50>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite asp__L_e at 0x7fc448e74730>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPabcpp',\n",
       "  'name': 'L-aspartate transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29039;29040;29041;29042',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR872',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:TRANS-RXN0-222'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37242_1 or EDU38743_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37242_1 at 0x7fc44983be50>,\n",
       "   <Gene EDU38743_1 at 0x7fc44983be80>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite asp__L_p at 0x7fc448e74760>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPCT',\n",
       "  'name': 'Aspartate carbamoyltransferase',\n",
       "  'notes': {'EC Number': '2.1.3.2',\n",
       "   'BioCyc': 'META:ASPCARBTRANS-RXN',\n",
       "   'BioPath Reaction': 'RXN00366',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR856',\n",
       "   'KEGG Reaction': 'R01397',\n",
       "   'RHEA': '20013;20014;20015;20016',\n",
       "   'UniPathway Reaction': 'UCR01397;UER00116'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39301_1 and EDU39302_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39301_1 at 0x7fc44983bca0>,\n",
       "   <Gene EDU39302_1 at 0x7fc44983bc70>},\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite cbp_c at 0x7fc448e748e0>: -1.0,\n",
       "   <Metabolite cbasp_c at 0x7fc448e748b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPK',\n",
       "  'name': 'Aspartate kinase',\n",
       "  'notes': {'EC Number': '2.7.2.4',\n",
       "   'BioCyc': 'META:ASPARTATEKIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00382',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR861',\n",
       "   'KEGG Reaction': 'R00480',\n",
       "   'RHEA': '23776;23777;23778;23779',\n",
       "   'UniPathway Reaction': 'UCR00480;UER00015;UER00461;UER00462'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36425_1 or EDU37118_1 or EDU39205_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36425_1 at 0x7fc44983bd00>,\n",
       "   <Gene EDU37118_1 at 0x7fc44983bd30>,\n",
       "   <Gene EDU39205_1 at 0x7fc44983bcd0>},\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite 4pasp_c at 0x7fc448e746d0>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPO4',\n",
       "  'name': 'L-aspartate oxidase',\n",
       "  'notes': {'RHEA': '29191;29192;29193;29194',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81765'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite iasp_c at 0x7fc448e74970>: 1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPt2_2pp',\n",
       "  'name': 'Aspartate transport via proton symport (2 H) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR874',\n",
       "   'BioCyc': 'META:TRANS-RXN-122A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39112_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39112_1 at 0x7fc44983beb0>},\n",
       "  '_metabolites': {<Metabolite asp__L_p at 0x7fc448e74760>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -2.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPTA',\n",
       "  'name': 'Aspartate transaminase',\n",
       "  'notes': {'EC Number': '2.6.1.1',\n",
       "   'BioCyc': 'META:ASPAMINOTRANS-RXN;META:RXN-13697',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR32641',\n",
       "   'KEGG Reaction': 'R00355',\n",
       "   'RHEA': '21824;21825;21826;21827',\n",
       "   'UniPathway Reaction': 'UCR00355;UER01024'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36436_1 or EDU36770_1 or EDU36788_1 or EDU36793_1 or EDU37030_1 or EDU38950_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36436_1 at 0x7fc44983bdc0>,\n",
       "   <Gene EDU36770_1 at 0x7fc44983bd90>,\n",
       "   <Gene EDU36788_1 at 0x7fc44983be20>,\n",
       "   <Gene EDU36793_1 at 0x7fc44983bdf0>,\n",
       "   <Gene EDU37030_1 at 0x7fc44983b1c0>,\n",
       "   <Gene EDU38950_1 at 0x7fc44983bd60>},\n",
       "  '_metabolites': {<Metabolite akg_c at 0x7fc463adc370>: -1.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite oaa_c at 0x7fc463aaef70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPTA6',\n",
       "  'name': 'E L-alanine-alpha-keto acid aminotransferase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR14595',\n",
       "   'KEGG Reaction': 'R00400',\n",
       "   'EC Number': '2.6.1.12'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36788_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36788_1 at 0x7fc44983be20>},\n",
       "  '_metabolites': {<Metabolite ala__L_c at 0x7fc463a82c10>: -1.0,\n",
       "   <Metabolite oaa_c at 0x7fc463aaef70>: -1.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASPtex',\n",
       "  'name': 'L-aspartate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR877',\n",
       "   'BioCyc': 'META:TRANS-RXN0-554'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite asp__L_e at 0x7fc448e74730>: -1.0,\n",
       "   <Metabolite asp__L_p at 0x7fc448e74760>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ASR2',\n",
       "  'name': 'Arsenate reductase (thioredoxin)',\n",
       "  'notes': {'BioCyc': 'META:RXN-10737',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81779',\n",
       "   'KEGG Reaction': 'R10927',\n",
       "   'EC Number': '1.20.4.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38592_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38592_1 at 0x7fc44983bee0>},\n",
       "  '_metabolites': {<Metabolite aso4_c at 0x7fc448e74820>: -1.0,\n",
       "   <Metabolite trdrd_c at 0x7fc448e495e0>: -1.0,\n",
       "   <Metabolite aso3_c at 0x7fc448e74790>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite trdox_c at 0x7fc448e495b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ATHRDHr',\n",
       "  'name': 'L-allo-threonine dehydrogenase',\n",
       "  'notes': {'RHEA': '30055;30056;30057;30058',\n",
       "   'BioCyc': 'META:RXN-16000',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68421',\n",
       "   'KEGG Reaction': 'R10851',\n",
       "   'EC Number': '1.1.1.381'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36536_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36536_1 at 0x7fc44983bf10>},\n",
       "  '_metabolites': {<Metabolite athr__L_c at 0x7fc448e74a90>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite 2aobut_c at 0x7fc463adc3a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ATPHs',\n",
       "  'name': 'ATP amine hydrolysis (spontaneous)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81781', 'EC Number': '3.5.4.18'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite itp_c at 0x7fc463aaec10>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ATPPRT',\n",
       "  'name': 'ATP phosphoribosyltransferase',\n",
       "  'notes': {'EC Number': '2.4.2.17',\n",
       "   'BioCyc': 'META:ATPPHOSPHORIBOSYLTRANS-RXN',\n",
       "   'BioPath Reaction': 'RXN00983',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68424',\n",
       "   'KEGG Reaction': 'R01071',\n",
       "   'RHEA': '18473;18474;18475;18476',\n",
       "   'UniPathway Reaction': 'UCR01071;UER00006'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37198_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37198_1 at 0x7fc44983bf40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite prpp_c at 0x7fc463aae0d0>: -1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite prbatp_c at 0x7fc448e74ac0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ATPS4rpp',\n",
       "  'name': 'ATP synthase (four protons for one ATP) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81785',\n",
       "   'EC Number': '3.6.3.14',\n",
       "   'BioCyc': 'META:ATPSYN-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38005_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38005_1 at 0x7fc44983bf70>},\n",
       "  '_metabolites': {<Metabolite adp_c at 0x7fc463a6d160>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -4.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 3.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BCT1_syn',\n",
       "  'name': 'Bicarbonate transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81806'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39295_1 and EDU39370_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39295_1 at 0x7fc44983bfa0>,\n",
       "   <Gene EDU39370_1 at 0x7fc44983bfd0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hco3_p at 0x7fc448e74b50>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hco3_c at 0x7fc463adc5b0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BG_CELLB',\n",
       "  'name': 'Beta glucosidase  cellobiose ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37619_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37619_1 at 0x7fc44983b100>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite cellb_e at 0x7fc448e74c40>: -1.0,\n",
       "   <Metabolite glc__D_e at 0x7fc463a6d520>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BGLA1',\n",
       "  'name': '6-phospho-beta-glucosidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81810'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35994_1 or EDU39174_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35994_1 at 0x7fc4498204f0>,\n",
       "   <Gene EDU39174_1 at 0x7fc449820520>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 6pgg_c at 0x7fc448e74ca0>: -1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0,\n",
       "   <Metabolite glc__D_c at 0x7fc463a6d4f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BG_MADG',\n",
       "  'name': 'Beta glucosidase  methyl alpha D glucoside ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37619_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37619_1 at 0x7fc44983b100>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite madg_e at 0x7fc448e74d00>: -1.0,\n",
       "   <Metabolite glc__D_e at 0x7fc463a6d520>: 1.0,\n",
       "   <Metabolite meoh_e at 0x7fc448e74cd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BG_MBDG',\n",
       "  'name': 'Beta glucosidase',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37619_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37619_1 at 0x7fc44983b100>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite mbdg_e at 0x7fc448e74d30>: -1.0,\n",
       "   <Metabolite glc__D_e at 0x7fc463a6d520>: 1.0,\n",
       "   <Metabolite meoh_e at 0x7fc448e74cd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BPNT',\n",
       "  'name': \"3',5'-bisphosphate nucleotidase\",\n",
       "  'notes': {'RHEA': '10040;10041;10042;10043',\n",
       "   'BioCyc': 'META:325-BISPHOSPHATE-NUCLEOTIDASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR965',\n",
       "   'KEGG Reaction': 'R00188',\n",
       "   'EC Number': '3.1.3.7'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite pap_c at 0x7fc463adc1f0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BSCT',\n",
       "  'name': 'Benzoylsuccinyl-CoA thiolase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81819'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -4.0,\n",
       "   <Metabolite sbzcoa_c at 0x7fc448e74e50>: -1.0,\n",
       "   <Metabolite succoa_c at 0x7fc463a82250>: 1.0,\n",
       "   <Metabolite benzcoa_c at 0x7fc463a82a90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BTDD_RR',\n",
       "  'name': ' R R  butanediol dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.4',\n",
       "   'BioCyc': 'META:RR-BUTANEDIOL-DEHYDROGENASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00480',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR5668',\n",
       "   'KEGG Reaction': 'R02946',\n",
       "   'RHEA': '24340;24341;24342;24343',\n",
       "   'UniPathway Reaction': 'UCR02946;UER00679'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36557_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36557_1 at 0x7fc44983b070>},\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite btd_RR_c at 0x7fc448e74e80>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite actn__R_c at 0x7fc463aaea90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BTOHtex',\n",
       "  'name': 'Butanol transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81821'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite btoh_e at 0x7fc448e74eb0>: -1.0,\n",
       "   <Metabolite btoh_p at 0x7fc448e74ee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BTS_1',\n",
       "  'name': 'Biotin synthase',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36506_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36506_1 at 0x7fc44985d040>},\n",
       "  '_metabolites': {<Metabolite dtbt_c at 0x7fc448e74fa0>: -1.0,\n",
       "   <Metabolite s_c at 0x7fc448e74fd0>: -2.0,\n",
       "   <Metabolite btn_c at 0x7fc448e74af0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite h2s_c at 0x7fc448e49580>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BTS3r',\n",
       "  'name': 'Biotin synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR30815', 'EC Number': '2.8.1.6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38163_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38163_1 at 0x7fc44985d0d0>},\n",
       "  '_metabolites': {<Metabolite dtbt_c at 0x7fc448e74fa0>: -1.0,\n",
       "   <Metabolite s_c at 0x7fc448e74fd0>: -2.0,\n",
       "   <Metabolite btn_c at 0x7fc448e74af0>: 1.0,\n",
       "   <Metabolite h2s_c at 0x7fc448e49580>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BTS4',\n",
       "  'name': 'Biotin synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81823', 'EC Number': '2.8.1.6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36506_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36506_1 at 0x7fc44985d040>},\n",
       "  '_metabolites': {<Metabolite amet_c at 0x7fc463adc940>: -1.0,\n",
       "   <Metabolite dtbt_c at 0x7fc448e74fa0>: -1.0,\n",
       "   <Metabolite s_c at 0x7fc448e74fd0>: -1.0,\n",
       "   <Metabolite btn_c at 0x7fc448e74af0>: 1.0,\n",
       "   <Metabolite dad_5_c at 0x7fc463aae070>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BTS5',\n",
       "  'name': 'Biotin synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81824'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36506_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36506_1 at 0x7fc44985d040>},\n",
       "  '_metabolites': {<Metabolite 2fe2s_c at 0x7fc448e74f70>: -1.0,\n",
       "   <Metabolite amet_c at 0x7fc463adc940>: -1.0,\n",
       "   <Metabolite dtbt_c at 0x7fc448e74fa0>: -1.0,\n",
       "   <Metabolite 2fe1s_c at 0x7fc448e74f10>: 1.0,\n",
       "   <Metabolite btn_c at 0x7fc448e74af0>: 1.0,\n",
       "   <Metabolite dad_5_c at 0x7fc463aae070>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BUTKr',\n",
       "  'name': 'Butyrate kinase',\n",
       "  'notes': {'EC Number': '2.7.2.7',\n",
       "   'BioCyc': 'META:BUTYRATE-KINASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01168',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR6040',\n",
       "   'RHEA': '13585;13586;13587;13588',\n",
       "   'KEGG Reaction': 'R01688'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39089_1 or EDU39091_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39089_1 at 0x7fc44985d130>,\n",
       "   <Gene EDU39091_1 at 0x7fc44985d0a0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite but_c at 0x7fc448e49d30>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite butpi_c at 0x7fc448e9f130>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BUTt2r',\n",
       "  'name': 'Butyrate transport via proton symport, reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR983'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite but_e at 0x7fc448e9f160>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite but_c at 0x7fc448e49d30>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BUTtex',\n",
       "  'name': 'Butyrate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR984'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite but_e at 0x7fc448e9f160>: -1.0,\n",
       "   <Metabolite but_p at 0x7fc448e9f190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BZALt2pp',\n",
       "  'name': 'Benzaldehyde transport in via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81833'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite bzal_p at 0x7fc448e9f040>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite bzal_c at 0x7fc448e9f0a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BZALtex',\n",
       "  'name': 'Benzaldehyde transport via diffusion (Extra-organism to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81834'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite bzal_p at 0x7fc448e9f040>: -1.0,\n",
       "   <Metabolite bzal_e at 0x7fc448e9f1c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BZDH',\n",
       "  'name': 'BZDH',\n",
       "  'notes': {'EC Number': '1.2.1.29',\n",
       "   'BioCyc': 'META:BENZALDEHYDE-DEHYDROGENASE-NAD+-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR988',\n",
       "   'KEGG Reaction': 'R01419;R01486',\n",
       "   'RHEA': '11840;11841;11842;11843',\n",
       "   'UniPathway Reaction': 'UCR01419;UER00988'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37555_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37555_1 at 0x7fc44983b490>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite bzal_c at 0x7fc448e9f0a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite bz_c at 0x7fc448e74b20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'BZt',\n",
       "  'name': 'Benzoate transport (diffusion)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR990'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite bz_e at 0x7fc448e9f100>: -1.0,\n",
       "   <Metabolite bz_c at 0x7fc448e74b20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CA2abc',\n",
       "  'name': 'CA2abc',\n",
       "  'notes': {'RHEA': '18105;18106;18107;18108',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1035',\n",
       "   'EC Number': '3.6.3.8',\n",
       "   'BioCyc': 'META:3.6.3.8-RXN;META:TRANS-RXN-193;META:TRANS-RXN-194'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38356_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38356_1 at 0x7fc44985d160>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ca2_e at 0x7fc448e9f2e0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite ca2_c at 0x7fc448e9f2b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CA2abc1',\n",
       "  'name': 'Calcium efflux  via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38356_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38356_1 at 0x7fc44985d160>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite ca2_c at 0x7fc448e9f2b0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite ca2_e at 0x7fc448e9f2e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CA2tex',\n",
       "  'name': 'Calcium (Ca+2) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'BioPath Reaction': 'RXN00877',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1037',\n",
       "   'BioCyc': 'META:TRANS-RXN0-497;META:TRANS-RXN-144'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite ca2_e at 0x7fc448e9f2e0>: -1.0,\n",
       "   <Metabolite ca2_p at 0x7fc448e9f310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CADVt',\n",
       "  'name': 'Lysine/Cadaverine antiporter',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1040'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 15dap_c at 0x7fc448e742b0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite lys__L_e at 0x7fc448e745b0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lys__L_c at 0x7fc448e74310>: 1.0,\n",
       "   <Metabolite 15dap_e at 0x7fc448e9f370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CAT23DOX',\n",
       "  'name': 'CAT23DOX',\n",
       "  'notes': {'EC Number': '1.13.11.2',\n",
       "   'BioCyc': 'META:CATECHOL-23-DIOXYGENASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00565',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68471',\n",
       "   'RHEA': '17337;17338;17339;17340',\n",
       "   'KEGG Reaction': 'R00816'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite o2_c at 0x7fc463a6d610>: -1.0,\n",
       "   <Metabolite catechol_c at 0x7fc448e9f0d0>: -1.0,\n",
       "   <Metabolite 2hmcnsad_c at 0x7fc448e9f3d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CATECHOLtex',\n",
       "  'name': 'CATECHOLtex',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1049'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite catechol_e at 0x7fc448e9f430>: -1.0,\n",
       "   <Metabolite catechol_p at 0x7fc448e9f400>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CATECHOLtpp',\n",
       "  'name': 'CATECHOLtpp',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1049'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite catechol_p at 0x7fc448e9f400>: -1.0,\n",
       "   <Metabolite catechol_c at 0x7fc448e9f0d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CBIAT',\n",
       "  'name': 'Cobinamide adenyltransferase',\n",
       "  'notes': {'RHEA': '14725;14726;14727;14728',\n",
       "   'BioCyc': 'META:BTUR2-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81868',\n",
       "   'KEGG Reaction': 'R07268',\n",
       "   'EC Number': '2.5.1.17'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38801_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38801_1 at 0x7fc44985d190>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cbi_c at 0x7fc448e9f460>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite adocbi_c at 0x7fc448e1c910>: 1.0,\n",
       "   <Metabolite pppi_c at 0x7fc463aae310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CBIAT_DELETE',\n",
       "  'name': 'Cobinamide adenyltransferase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81869'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cbi_c at 0x7fc448e9f460>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adocbi_c at 0x7fc448e1c910>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CBLAT',\n",
       "  'name': 'Cob(I)alamin adenosyltransferase',\n",
       "  'notes': {'RHEA': '28671;28672;28673;28674',\n",
       "   'BioCyc': 'META:COBALADENOSYLTRANS-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81874',\n",
       "   'KEGG Reaction': 'R01492',\n",
       "   'EC Number': '2.5.1.17'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38801_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38801_1 at 0x7fc44985d190>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cbl1_c at 0x7fc448e9f490>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite adocbl_c at 0x7fc448e1cb20>: 1.0,\n",
       "   <Metabolite pppi_c at 0x7fc463aae310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CBLAT_DELETE',\n",
       "  'name': 'Cob(I)alamin adenosyltransferase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81875', 'EC Number': '2.5.1.17'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cbl1_c at 0x7fc448e9f490>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adocbl_c at 0x7fc448e1cb20>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CBMKr',\n",
       "  'name': 'Carbamate kinase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81878', 'EC Number': '2.7.2.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36246_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36246_1 at 0x7fc44985d1c0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cbp_c at 0x7fc448e748e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CBPS',\n",
       "  'name': 'Carbamoyl-phosphate synthase (glutamine-hydrolysing)',\n",
       "  'notes': {'EC Number': '6.3.5.5',\n",
       "   'BioCyc': 'META:CARBPSYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR33037',\n",
       "   'KEGG Reaction': 'R00575',\n",
       "   'RHEA': '18633;18634;18635;18636',\n",
       "   'UniPathway Reaction': 'UCR00575;UER00115;UER00171'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36986_1 and EDU36987_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36986_1 at 0x7fc44985d1f0>,\n",
       "   <Gene EDU36987_1 at 0x7fc44985d220>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -2.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hco3_c at 0x7fc463adc5b0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 2.0,\n",
       "   <Metabolite cbp_c at 0x7fc448e748e0>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CD2abc1',\n",
       "  'name': 'CD2abc1',\n",
       "  'notes': {'RHEA': '12132;12133;12134;12135',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR19981',\n",
       "   'EC Number': '3.6.3.46',\n",
       "   'BioCyc': 'META:3.6.3.3-RXN;META:3.6.3.46-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37480_1 or EDU38602_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37480_1 at 0x7fc44985d250>,\n",
       "   <Gene EDU38602_1 at 0x7fc44985d280>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cd2_c at 0x7fc448e9f4c0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite cd2_e at 0x7fc448e9f4f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CD2t3pp',\n",
       "  'name': 'Cadmium (Cd+2) transport out via proton antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68486',\n",
       "   'BioCyc': 'META:TRANS-RXN0-244'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39203_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39203_1 at 0x7fc44985d2b0>},\n",
       "  '_metabolites': {<Metabolite cd2_c at 0x7fc448e9f4c0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite cd2_p at 0x7fc448e9f520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CD2t4pp',\n",
       "  'name': 'Cadmium (Cd+2) transport out via proton antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR79902'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37535_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37535_1 at 0x7fc44985d2e0>},\n",
       "  '_metabolites': {<Metabolite cd2_c at 0x7fc448e9f4c0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite k_p at 0x7fc448e9f550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite k_c at 0x7fc448e74910>: 1.0,\n",
       "   <Metabolite cd2_p at 0x7fc448e9f520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CD2tex',\n",
       "  'name': 'Cadmium (Cd+2) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68487', 'BioCyc': 'META:RXN0-10'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite cd2_e at 0x7fc448e9f4f0>: -1.0,\n",
       "   <Metabolite cd2_p at 0x7fc448e9f520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CD2tpp',\n",
       "  'name': 'Cadmium (+2) transport in via permease (no H+)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68487', 'BioCyc': 'META:RXN0-10'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37639_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37639_1 at 0x7fc44985d310>},\n",
       "  '_metabolites': {<Metabolite cd2_p at 0x7fc448e9f520>: -1.0,\n",
       "   <Metabolite cd2_c at 0x7fc448e9f4c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CDGUNPD',\n",
       "  'name': '3,5-cyclic-nucleotide phosphodiesterase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81889'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36303_1 or EDU37256_1 or EDU37724_1 or EDU37769_1 or EDU37796_1 or EDU38268_1 or EDU39184_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36303_1 at 0x7fc44985d3a0>,\n",
       "   <Gene EDU37256_1 at 0x7fc44985d340>,\n",
       "   <Gene EDU37724_1 at 0x7fc44985d370>,\n",
       "   <Gene EDU37769_1 at 0x7fc44985d460>,\n",
       "   <Gene EDU37796_1 at 0x7fc44985d430>,\n",
       "   <Gene EDU38268_1 at 0x7fc44985d400>,\n",
       "   <Gene EDU39184_1 at 0x7fc44985d3d0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cdigmp_c at 0x7fc448e9f730>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 5pg35pg_c at 0x7fc448e9f760>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CDPMEK',\n",
       "  'name': \"4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol kinase\",\n",
       "  'notes': {'EC Number': '2.7.1.148',\n",
       "   'BioCyc': 'META:2.7.1.148-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1102',\n",
       "   'KEGG Reaction': 'R05634',\n",
       "   'RHEA': '18437;18438;18439;18440',\n",
       "   'UniPathway Reaction': 'UCR05634;UER00094'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38018_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38018_1 at 0x7fc44985d490>},\n",
       "  '_metabolites': {<Metabolite 4c2me_c at 0x7fc448e9f7c0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite 2p4c2me_c at 0x7fc448e9f790>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CELBpts',\n",
       "  'name': 'Cellobiose transport via PEPPyr PTS',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU36437_1 and EDU39079_1 and EDU39175_1) or (EDU36437_1 and EDU39079_1 and EDU39175_1 and EDU39177_1 and EDU39178_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>,\n",
       "   <Gene EDU39175_1 at 0x7fc44985d4f0>,\n",
       "   <Gene EDU39177_1 at 0x7fc44985d520>,\n",
       "   <Gene EDU39178_1 at 0x7fc44985d4c0>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite cellb_e at 0x7fc448e74c40>: -1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0,\n",
       "   <Metabolite 6pgg_c at 0x7fc448e74ca0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CELLBabc',\n",
       "  'name': 'Cellobiose transport via ATP transporter',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81937'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37083_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37083_1 at 0x7fc44985d550>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cellb_e at 0x7fc448e74c40>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite cellb_c at 0x7fc448e74c70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CEPA',\n",
       "  'name': 'Cellobiose phosphorylase',\n",
       "  'notes': {'RHEA': '19493;19494;19495;19496',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR32712',\n",
       "   'KEGG Reaction': 'R00952;R06056',\n",
       "   'EC Number': '2.4.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39207_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39207_1 at 0x7fc44985d580>},\n",
       "  '_metabolites': {<Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite cellb_c at 0x7fc448e74c70>: -1.0,\n",
       "   <Metabolite g1p_c at 0x7fc448e1cbb0>: 1.0,\n",
       "   <Metabolite glc__D_c at 0x7fc463a6d4f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CGLYabcpp',\n",
       "  'name': 'L-Cysteinylglycine (Cys-Gly) transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1130'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU37419_1 and EDU38656_1) or (EDU37419_1 and EDU37423_1 and EDU37580_1) or (EDU37419_1 and EDU37580_1 and EDU38656_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37419_1 at 0x7fc4498203d0>,\n",
       "   <Gene EDU37423_1 at 0x7fc4498201f0>,\n",
       "   <Gene EDU37580_1 at 0x7fc44983b040>,\n",
       "   <Gene EDU38656_1 at 0x7fc44983b0d0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cgly_p at 0x7fc448e9f850>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cgly_c at 0x7fc448e74160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CGLYtex',\n",
       "  'name': 'L-Cysteinylglycine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1132'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite cgly_e at 0x7fc448e74190>: -1.0,\n",
       "   <Metabolite cgly_p at 0x7fc448e9f850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CHLabc',\n",
       "  'name': 'Choline transport via ABC system',\n",
       "  'notes': {'RHEA': '38715;38716;38717;38718',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1145'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37130_1 or (EDU36480_1 and EDU37129_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36480_1 at 0x7fc44985d5b0>,\n",
       "   <Gene EDU37129_1 at 0x7fc44985d610>,\n",
       "   <Gene EDU37130_1 at 0x7fc44985d5e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite chol_e at 0x7fc463aaeee0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite chol_c at 0x7fc448e9f880>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CHLt2',\n",
       "  'name': 'Choline transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1146',\n",
       "   'BioCyc': 'META:TRANS-RXN-99'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite chol_e at 0x7fc463aaeee0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite chol_c at 0x7fc448e9f880>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CHOLSabc_1',\n",
       "  'name': 'Choline sulfate transport in via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37130_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37130_1 at 0x7fc44985d5e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite chols_e at 0x7fc448e9f8e0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite chols_c at 0x7fc448e9f8b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CHOLSH',\n",
       "  'name': 'CHOLSH',\n",
       "  'notes': {'EC Number': '3.1.6.6',\n",
       "   'BioCyc': 'META:CHOLINE-SULFATASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1156',\n",
       "   'KEGG Reaction': 'R01028',\n",
       "   'RHEA': '20820;20821;20822;20823',\n",
       "   'UniPathway Reaction': 'UCR01028;UER00431'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite chols_c at 0x7fc448e9f8b0>: -1.0,\n",
       "   <Metabolite chol_c at 0x7fc448e9f880>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite so4_c at 0x7fc448e746a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CHORM',\n",
       "  'name': 'Chorismate mutase',\n",
       "  'notes': {'EC Number': '5.4.99.5',\n",
       "   'BioCyc': 'META:CHORISMATEMUT-RXN',\n",
       "   'BioPath Reaction': 'RXN00744',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1159',\n",
       "   'KEGG Reaction': 'R01715',\n",
       "   'RHEA': '13897;13898;13899;13900',\n",
       "   'UniPathway Reaction': 'UCR01715;UER00203'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35836_1 or EDU37024_1 or EDU37303_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35836_1 at 0x7fc44985d640>,\n",
       "   <Gene EDU37024_1 at 0x7fc44985d670>,\n",
       "   <Gene EDU37303_1 at 0x7fc44985d6a0>},\n",
       "  '_metabolites': {<Metabolite chor_c at 0x7fc448e1c940>: -1.0,\n",
       "   <Metabolite pphn_c at 0x7fc448e9f910>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CHORS',\n",
       "  'name': 'Chorismate synthase',\n",
       "  'notes': {'EC Number': '4.2.3.5',\n",
       "   'BioCyc': 'META:CHORISMATE-SYNTHASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01245',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1160',\n",
       "   'KEGG Reaction': 'R01714',\n",
       "   'RHEA': '21020;21021;21022;21023',\n",
       "   'UniPathway Reaction': 'UCR01714;UER00090'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35831_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35831_1 at 0x7fc44985d6d0>},\n",
       "  '_metabolites': {<Metabolite 3psme_c at 0x7fc448e9f940>: -1.0,\n",
       "   <Metabolite chor_c at 0x7fc448e1c940>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CLIPAabctex',\n",
       "  'name': 'Cold lipid A transport via ABC system (periplasm to extracellular)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69316'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37217_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37217_1 at 0x7fc44985d700>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite lipa_cold_p at 0x7fc448e9fc40>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite lipa_cold_e at 0x7fc448e9fc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CLt3_2pp',\n",
       "  'name': 'Chloride transport out via proton antiport (2:1) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1217', 'BioCyc': 'META:RXN0-2501'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite cl_p at 0x7fc448e9ff10>: -2.0,\n",
       "   <Metabolite cl_c at 0x7fc463a82eb0>: 2.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CLtex',\n",
       "  'name': 'Chloride (Cl-1) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1218',\n",
       "   'BioCyc': 'META:TRANS-RXN-139'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite cl_e at 0x7fc463a82ee0>: -1.0,\n",
       "   <Metabolite cl_p at 0x7fc448e9ff10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CMPDAi',\n",
       "  'name': 'CMP deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82012', 'EC Number': '3.5.4.12'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37993_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37993_1 at 0x7fc44985d730>},\n",
       "  '_metabolites': {<Metabolite cmp_c at 0x7fc448e1cdf0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite ump_c at 0x7fc463aaedf0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CO2t',\n",
       "  'name': 'CO2 transporter via diffusion',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1225',\n",
       "   'BioCyc': 'META:TRANS-RXN0-545'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite co2_e at 0x7fc448ec8130>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CO2tex',\n",
       "  'name': 'CO2 transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1225',\n",
       "   'BioCyc': 'META:TRANS-RXN0-545'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite co2_e at 0x7fc448ec8130>: -1.0,\n",
       "   <Metabolite co2_p at 0x7fc448e74550>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CO2tpp',\n",
       "  'name': 'CO2 transporter via diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1225',\n",
       "   'BioCyc': 'META:TRANS-RXN0-545'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite co2_p at 0x7fc448e74550>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'COabc',\n",
       "  'name': 'Cobalt transport via ABC system',\n",
       "  'notes': {'RHEA': '32779;32780;32781;32782',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR82019'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36840_1 or EDU38763_1 or (EDU36840_1 and EDU36841_1) or (EDU39067_1 and EDU39068_1 and EDU39069_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36840_1 at 0x7fc44985d8e0>,\n",
       "   <Gene EDU36841_1 at 0x7fc44985d850>,\n",
       "   <Gene EDU38763_1 at 0x7fc44985d940>,\n",
       "   <Gene EDU39067_1 at 0x7fc44985d8b0>,\n",
       "   <Gene EDU39068_1 at 0x7fc44985d910>,\n",
       "   <Gene EDU39069_1 at 0x7fc44985d880>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cobalt2_e at 0x7fc448e9faf0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cobalt2_c at 0x7fc448e9fac0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'Cobalt2abcppI',\n",
       "  'name': 'Cobalt II transport via ABC system periplasm I',\n",
       "  'notes': {'RHEA': '32779;32780;32781;32782',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR82019'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38762_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38762_1 at 0x7fc44986a0d0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cobalt2_p at 0x7fc448ec8160>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cobalt2_c at 0x7fc448e9fac0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'COBALT2t3pp',\n",
       "  'name': 'Cobalt (Co+2) transport out via proton antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82020'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39203_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39203_1 at 0x7fc44985d2b0>},\n",
       "  '_metabolites': {<Metabolite cobalt2_c at 0x7fc448e9fac0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite cobalt2_p at 0x7fc448ec8160>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'COBALT2tex',\n",
       "  'name': 'Cobalt (Co+2) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82021',\n",
       "   'BioCyc': 'META:TRANS-RXN-141A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite cobalt2_e at 0x7fc448e9faf0>: -1.0,\n",
       "   <Metabolite cobalt2_p at 0x7fc448ec8160>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'COBALT2tpp',\n",
       "  'name': 'Cobalt transport in via permease (no H+)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82021',\n",
       "   'BioCyc': 'META:TRANS-RXN-141A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37639_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37639_1 at 0x7fc44985d310>},\n",
       "  '_metabolites': {<Metabolite cobalt2_p at 0x7fc448ec8160>: -1.0,\n",
       "   <Metabolite cobalt2_c at 0x7fc448e9fac0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'COBALTt4pp',\n",
       "  'name': 'Cobalt transport out via proton antiporter (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82022'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37535_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37535_1 at 0x7fc44985d2e0>},\n",
       "  '_metabolites': {<Metabolite cobalt2_c at 0x7fc448e9fac0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite k_p at 0x7fc448e9f550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite k_c at 0x7fc448e74910>: 1.0,\n",
       "   <Metabolite cobalt2_p at 0x7fc448ec8160>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CODH2r',\n",
       "  'name': 'CODH2r',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82025'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36556_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36556_1 at 0x7fc44985d760>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fdox_c at 0x7fc448ec8190>: -2.0,\n",
       "   <Metabolite co_c at 0x7fc463aaeb80>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite fdred_c at 0x7fc448ec80a0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CODH4_1',\n",
       "  'name': 'Carbon monoxide dehydrogenase / acetyl-CoA synthase 2',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82026'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37458_1 and EDU37459_1 and EDU37460_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37458_1 at 0x7fc44985d7f0>,\n",
       "   <Gene EDU37459_1 at 0x7fc44985d790>,\n",
       "   <Gene EDU37460_1 at 0x7fc44985d7c0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite co_c at 0x7fc463aaeb80>: -1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CODH_ACS',\n",
       "  'name': 'CODH/Acetyl-CoA synthase (Wood-Ljundahl)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82027'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37285_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37285_1 at 0x7fc44985d820>},\n",
       "  '_metabolites': {<Metabolite co2_c at 0x7fc463adc8e0>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: -1.0,\n",
       "   <Metabolite mecfsp_c at 0x7fc448ec8070>: -1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: 1.0,\n",
       "   <Metabolite cfesp_c at 0x7fc448ec8040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'COt',\n",
       "  'name': 'COt',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1256',\n",
       "   'BioCyc': 'META:TRANS-RXN0-556'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite co_c at 0x7fc463aaeb80>: -1.0,\n",
       "   <Metabolite co_e at 0x7fc448ec80d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CPK1',\n",
       "  'name': 'Adenylate kinase',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite cmp_c at 0x7fc448e1cdf0>: -1.0,\n",
       "   <Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite cdp_c at 0x7fc448e9f7f0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CPMPS',\n",
       "  'name': 'Cyclic pyranopterin monophosphate synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR61132',\n",
       "   'KEGG Reaction': 'R09394',\n",
       "   'EC Number': '4.1.99.18'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36879_1 or EDU36880_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36879_1 at 0x7fc44985d970>,\n",
       "   <Gene EDU36880_1 at 0x7fc44985d9a0>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cpmp_c at 0x7fc448ec8100>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CPPPGO',\n",
       "  'name': 'Coproporphyrinogen oxidase (O2 required)',\n",
       "  'notes': {'EC Number': '1.3.3.3',\n",
       "   'BioCyc': 'META:RXN0-1461',\n",
       "   'BioPath Reaction': 'RXN00664',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1272',\n",
       "   'KEGG Reaction': 'R03220',\n",
       "   'RHEA': '18257;18258;18259;18260',\n",
       "   'UniPathway Reaction': 'UCR03220;UER00322'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35790_1 or EDU36380_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35790_1 at 0x7fc44985da00>,\n",
       "   <Gene EDU36380_1 at 0x7fc44985d9d0>},\n",
       "  '_metabolites': {<Metabolite cpppg3_c at 0x7fc448ec81c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite o2_c at 0x7fc463a6d610>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 2.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 2.0,\n",
       "   <Metabolite pppg9_c at 0x7fc448ec81f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CPPPGOAN2_1',\n",
       "  'name': 'Oxygen Independent coproporphyrinogen-III oxidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82048'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35884_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35884_1 at 0x7fc44985da30>},\n",
       "  '_metabolites': {<Metabolite amet_c at 0x7fc463adc940>: -2.0,\n",
       "   <Metabolite cpppg3_c at 0x7fc448ec81c0>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -2.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: -2.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 2.0,\n",
       "   <Metabolite dad_5_c at 0x7fc463aae070>: 2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 2.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 2.0,\n",
       "   <Metabolite pppg9_c at 0x7fc448ec81f0>: 1.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRNabc',\n",
       "  'name': 'L-carnitine transport via ABC system',\n",
       "  'notes': {'RHEA': '38719;38720;38721;38722',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68589'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37130_1 or (EDU36480_1 and EDU37129_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36480_1 at 0x7fc44985d5b0>,\n",
       "   <Gene EDU37129_1 at 0x7fc44985d610>,\n",
       "   <Gene EDU37130_1 at 0x7fc44985d5e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite crn_e at 0x7fc448ec8250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite crn_c at 0x7fc448e1cc70>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRNabcpp',\n",
       "  'name': 'L-carnitine transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '38719;38720;38721;38722',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68589'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36889_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36889_1 at 0x7fc44985da60>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite crn_p at 0x7fc448ec8280>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite crn_c at 0x7fc448e1cc70>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRNDabcpp',\n",
       "  'name': 'D-carnitine transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1285'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36889_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36889_1 at 0x7fc44985da60>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite crn__D_p at 0x7fc448ec8310>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite crn__D_c at 0x7fc448ec82e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRNDt2rpp',\n",
       "  'name': 'D-carnitine outward transport (H+ antiport)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1286'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite crn__D_p at 0x7fc448ec8310>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite crn__D_c at 0x7fc448ec82e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRNt2rpp',\n",
       "  'name': 'L-carnitine outward transport (H+ antiport)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68591'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite crn_p at 0x7fc448ec8280>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite crn_c at 0x7fc448e1cc70>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRNt7pp',\n",
       "  'name': 'Carnitine/butyrobetaine antiporter (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82054',\n",
       "   'BioCyc': 'META:TRANS-RXN-100'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite gbbtn_c at 0x7fc448ec82b0>: -1.0,\n",
       "   <Metabolite crn_p at 0x7fc448ec8280>: -1.0,\n",
       "   <Metabolite crn_c at 0x7fc448e1cc70>: 1.0,\n",
       "   <Metabolite gbbtn_p at 0x7fc448ec8370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRNtex',\n",
       "  'name': 'L-carnitine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68590'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite crn_e at 0x7fc448ec8250>: -1.0,\n",
       "   <Metabolite crn_p at 0x7fc448ec8280>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRO4abcpp',\n",
       "  'name': 'Chromate transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82055'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37652_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37652_1 at 0x7fc44985da90>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cro4_p at 0x7fc448ec83d0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite cro4_c at 0x7fc448ec83a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRO4t3',\n",
       "  'name': 'Chromate transport out via proton antiport',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38358_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38358_1 at 0x7fc44985dac0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite cro4_c at 0x7fc448ec83a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite cro4_e at 0x7fc448ec8400>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRO4t3pp',\n",
       "  'name': 'Chromate transport out via proton antiport (cytoplasm to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR79905'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38357_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38357_1 at 0x7fc44985daf0>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite cro4_c at 0x7fc448ec83a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite cro4_p at 0x7fc448ec83d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CRO4tex',\n",
       "  'name': 'Cromate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82056'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite cro4_e at 0x7fc448ec8400>: -1.0,\n",
       "   <Metabolite cro4_p at 0x7fc448ec83d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CS',\n",
       "  'name': 'Citrate synthase',\n",
       "  'notes': {'EC Number': '2.3.3.3',\n",
       "   'BioCyc': 'META:CITSYN-RXN;META:RXN-14905',\n",
       "   'BioPath Reaction': 'RXN00484',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1298',\n",
       "   'KEGG Reaction': 'R00351',\n",
       "   'RHEA': '16845;16846;16847;16848',\n",
       "   'UniPathway Reaction': 'UCR00351;UER00717'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38239_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38239_1 at 0x7fc44985db20>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite oaa_c at 0x7fc463aaef70>: -1.0,\n",
       "   <Metabolite cit_c at 0x7fc463aaef10>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CSND',\n",
       "  'name': 'Cytosine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82063', 'EC Number': '3.5.4.1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38989_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38989_1 at 0x7fc44985db50>},\n",
       "  '_metabolites': {<Metabolite csn_c at 0x7fc448e9ff40>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite ura_c at 0x7fc448ec8430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CTECOAI6',\n",
       "  'name': '3-cis-2-trans-enoyl-CoA isomerase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69942', 'EC Number': '5.3.3.8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1 or EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>,\n",
       "   <Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite tdecoa_c at 0x7fc448ec8490>: -1.0,\n",
       "   <Metabolite td2coa_c at 0x7fc45d333c10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CTECOAI7',\n",
       "  'name': '3-cis-2-trans-enoyl-CoA isomerase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69222', 'EC Number': '5.3.3.8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1 or EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>,\n",
       "   <Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite hdcoa_c at 0x7fc448e1cd90>: -1.0,\n",
       "   <Metabolite hdd2coa_c at 0x7fc463a40f10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CTECOAI8',\n",
       "  'name': '3-cis-2-trans-enoyl-CoA isomerase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR80399', 'EC Number': '5.3.3.8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1 or EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>,\n",
       "   <Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite odecoa_c at 0x7fc448e1cdc0>: -1.0,\n",
       "   <Metabolite od2coa_c at 0x7fc448e1c3a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CTPS1',\n",
       "  'name': 'CTP synthase  NH3 ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82065', 'EC Number': '6.3.4.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38007_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38007_1 at 0x7fc44985dbe0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: -1.0,\n",
       "   <Metabolite utp_c at 0x7fc448ec84c0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite ctp_c at 0x7fc448e1c1c0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CTPS2',\n",
       "  'name': 'CTP synthase (glutamine)',\n",
       "  'notes': {'EC Number': '6.3.4.2',\n",
       "   'BioCyc': 'META:CTPSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN00356',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR32700',\n",
       "   'RHEA': '26426;26427;26428;26429',\n",
       "   'KEGG Reaction': 'R00573'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38007_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38007_1 at 0x7fc44985dbe0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite utp_c at 0x7fc448ec84c0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite ctp_c at 0x7fc448e1c1c0>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CU1Opp',\n",
       "  'name': 'Cuprous Oxidase (Cu+1)',\n",
       "  'notes': {'RHEA': '30083;30084;30085;30086',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR24156',\n",
       "   'EC Number': '1.16.3',\n",
       "   'BioCyc': 'META:RXN0-2945'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38903_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38903_1 at 0x7fc44985dc10>},\n",
       "  '_metabolites': {<Metabolite cu_p at 0x7fc448ec84f0>: -4.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -4.0,\n",
       "   <Metabolite o2_p at 0x7fc463a82130>: -1.0,\n",
       "   <Metabolite cu2_p at 0x7fc448ec8520>: 4.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CU2tex',\n",
       "  'name': 'Copper (Cu+2) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR25169', 'BioCyc': 'META:RXN0-14'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite cu2_e at 0x7fc448ec8550>: -1.0,\n",
       "   <Metabolite cu2_p at 0x7fc448ec8520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CU2tpp',\n",
       "  'name': 'Copper transport in via permease (no H+)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR25169', 'BioCyc': 'META:RXN0-14'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37639_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37639_1 at 0x7fc44985d310>},\n",
       "  '_metabolites': {<Metabolite cu2_p at 0x7fc448ec8520>: -1.0,\n",
       "   <Metabolite cu2_c at 0x7fc448e74dc0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CUabcpp',\n",
       "  'name': 'Copper transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '10376;10377;10378;10379',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR19990',\n",
       "   'EC Number': '3.6.3.4',\n",
       "   'BioCyc': 'META:3.6.3.4-RXN;META:TRANS-RXN-178'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38941_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38941_1 at 0x7fc44985dc40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cu2_p at 0x7fc448ec8520>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cu2_c at 0x7fc448e74dc0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'Cut1',\n",
       "  'name': 'Copper export via ATPase',\n",
       "  'notes': {'RHEA': '10376;10377;10378;10379',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR19990',\n",
       "   'EC Number': '3.6.3.4',\n",
       "   'BioCyc': 'META:3.6.3.4-RXN;META:TRANS-RXN-178'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37498_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37498_1 at 0x7fc44986a040>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cu2_c at 0x7fc448e74dc0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite cu2_e at 0x7fc448ec8550>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CUtex',\n",
       "  'name': 'Copper (Cu+1) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68598'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite cu_e at 0x7fc448ec8580>: -1.0,\n",
       "   <Metabolite cu_p at 0x7fc448ec84f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYANSTpp',\n",
       "  'name': 'Cyanide sulfurtransferase (periplasmic)',\n",
       "  'notes': {'RHEA': '16881;16882;16883;16884',\n",
       "   'BioCyc': 'META:THIOSULFATE-SULFURTRANSFERASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68599',\n",
       "   'KEGG Reaction': 'R01931',\n",
       "   'EC Number': '2.8.1.1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37670_1 or EDU37676_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37670_1 at 0x7fc44985dca0>,\n",
       "   <Gene EDU37676_1 at 0x7fc44985dc70>},\n",
       "  '_metabolites': {<Metabolite cyan_p at 0x7fc448ec85b0>: -1.0,\n",
       "   <Metabolite tsul_p at 0x7fc448ec8640>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0,\n",
       "   <Metabolite so3_p at 0x7fc448ec85e0>: 1.0,\n",
       "   <Metabolite tcynt_p at 0x7fc448ec8610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYANtex',\n",
       "  'name': 'Cyanide transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1327'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite cyan_e at 0x7fc448ec8670>: -1.0,\n",
       "   <Metabolite cyan_p at 0x7fc448ec85b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYPHYC',\n",
       "  'name': 'Cyanophycinase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82080'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38020_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38020_1 at 0x7fc44985dcd0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -2.0,\n",
       "   <Metabolite cyanphy_c at 0x7fc448ec8700>: -1.0,\n",
       "   <Metabolite arg__L_c at 0x7fc448e49ac0>: 1.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: 1.0,\n",
       "   <Metabolite precyanphy_c at 0x7fc448ec8730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYPHYS',\n",
       "  'name': 'Cyanophycin sinthetase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82081'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38019_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38019_1 at 0x7fc44985dd00>},\n",
       "  '_metabolites': {<Metabolite arg__L_c at 0x7fc448e49ac0>: -1.0,\n",
       "   <Metabolite asp__L_c at 0x7fc463a82820>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -2.0,\n",
       "   <Metabolite precyanphy_c at 0x7fc448ec8730>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 2.0,\n",
       "   <Metabolite cyanphy_c at 0x7fc448ec8700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSabc2pp',\n",
       "  'name': 'L-cysteine export via ABC system (cytoplasm to periplasm)',\n",
       "  'notes': {'RHEA': '29783;29784;29785;29786',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1366',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:RXN0-3;META:TRANS-RXN-265'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37455_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37455_1 at 0x7fc44985de80>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cys__L_c at 0x7fc463aae3d0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite cys__L_p at 0x7fc448ec8760>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSabcpp',\n",
       "  'name': 'L-cysteine uptake via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29783;29784;29785;29786',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1366',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:RXN0-3;META:TRANS-RXN-265'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37428_1 and EDU37636_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37428_1 at 0x7fc44985deb0>,\n",
       "   <Gene EDU37636_1 at 0x7fc44985dee0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cys__L_p at 0x7fc448ec8760>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cys__L_c at 0x7fc463aae3d0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSDS',\n",
       "  'name': 'Cysteine Desulfhydrase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82085', 'EC Number': '4.4.1.8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38570_1 or EDU38988_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38570_1 at 0x7fc44985dd60>,\n",
       "   <Gene EDU38988_1 at 0x7fc44985dd30>},\n",
       "  '_metabolites': {<Metabolite cys__L_c at 0x7fc463aae3d0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite h2s_c at 0x7fc448e49580>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSDSF',\n",
       "  'name': 'Cysteine desulfurase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82086'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>},\n",
       "  '_metabolites': {<Metabolite cys__L_c at 0x7fc463aae3d0>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite s_c at 0x7fc448e74fd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSLY3',\n",
       "  'name': 'Cysteine lyase (nadph)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82090'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38112_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38112_1 at 0x7fc44985dd90>},\n",
       "  '_metabolites': {<Metabolite cys__L_c at 0x7fc463aae3d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite h2s_c at 0x7fc448e49580>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSS',\n",
       "  'name': 'Cysteine synthase',\n",
       "  'notes': {'EC Number': '2.5.1.65',\n",
       "   'BioCyc': 'META:ACSERLY-RXN',\n",
       "   'BioPath Reaction': 'RXN00245',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR30965',\n",
       "   'KEGG Reaction': 'R00897',\n",
       "   'RHEA': '14829;14830;14831;14832',\n",
       "   'UniPathway Reaction': 'UCR00897;UER00200'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37637_1 or EDU37937_1 or EDU38568_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37637_1 at 0x7fc44985ddf0>,\n",
       "   <Gene EDU37937_1 at 0x7fc44985de20>,\n",
       "   <Gene EDU38568_1 at 0x7fc44985ddc0>},\n",
       "  '_metabolites': {<Metabolite acser_c at 0x7fc448e1c700>: -1.0,\n",
       "   <Metabolite h2s_c at 0x7fc448e49580>: -1.0,\n",
       "   <Metabolite ac_c at 0x7fc463aae520>: 1.0,\n",
       "   <Metabolite cys__L_c at 0x7fc463aae3d0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSTGL',\n",
       "  'name': 'Cystathionine g lyase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82092'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38570_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38570_1 at 0x7fc44985dd60>},\n",
       "  '_metabolites': {<Metabolite cyst__L_c at 0x7fc448ec87c0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 2obut_c at 0x7fc463adc550>: 1.0,\n",
       "   <Metabolite cys__L_c at 0x7fc463aae3d0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSTL',\n",
       "  'name': 'Cystathionine b-lyase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR81870'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37374_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37374_1 at 0x7fc44985de50>},\n",
       "  '_metabolites': {<Metabolite cyst__L_c at 0x7fc448ec87c0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hcys__L_c at 0x7fc448e49460>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYStpp',\n",
       "  'name': 'L-cysteine export via facilitated transport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1369', 'BioCyc': 'META:RXN0-1924'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37649_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37649_1 at 0x7fc44985df10>},\n",
       "  '_metabolites': {<Metabolite cys__L_c at 0x7fc463aae3d0>: -1.0,\n",
       "   <Metabolite cys__L_p at 0x7fc448ec8760>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYSTS',\n",
       "  'name': 'Cystathionine beta synthase',\n",
       "  'notes': {'EC Number': '4.2.1.22',\n",
       "   'BioCyc': 'META:CYSTATHIONINE-BETA-SYNTHASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01128',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68615',\n",
       "   'KEGG Reaction': 'R01290',\n",
       "   'RHEA': '10112;10113;10114;10115',\n",
       "   'UniPathway Reaction': 'UCR01290;UER00201'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38568_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38568_1 at 0x7fc44985ddc0>},\n",
       "  '_metabolites': {<Metabolite hcys__L_c at 0x7fc448e49460>: -1.0,\n",
       "   <Metabolite ser__L_c at 0x7fc463aae400>: -1.0,\n",
       "   <Metabolite cyst__L_c at 0x7fc448ec87c0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTBD2pp',\n",
       "  'name': 'Cytochrome oxidase bd (menaquinol-8: 2 protons) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68616'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37455_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37455_1 at 0x7fc44985de80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: -1.0,\n",
       "   <Metabolite o2_c at 0x7fc463a6d610>: -0.5,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTD',\n",
       "  'name': 'Cytidine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82096', 'EC Number': '3.5.4.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35900_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35900_1 at 0x7fc44985df40>},\n",
       "  '_metabolites': {<Metabolite cytd_c at 0x7fc463a820a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite uri_c at 0x7fc463a82190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTDK1',\n",
       "  'name': 'CYTDK1',\n",
       "  'notes': {'EC Number': '2.7.1.48',\n",
       "   'BioCyc': 'META:CYTIKIN-RXN',\n",
       "   'BioPath Reaction': 'RXN01333',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1376',\n",
       "   'KEGG Reaction': 'R00513',\n",
       "   'RHEA': '24674;24675;24676;24677',\n",
       "   'UniPathway Reaction': 'UCR00513;UER00640'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36287_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36287_1 at 0x7fc44985df70>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cytd_c at 0x7fc463a820a0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cmp_c at 0x7fc448e1cdf0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTDK2',\n",
       "  'name': 'Cytidine kinase (GTP)',\n",
       "  'notes': {'RHEA': '28162;28163;28164;28165',\n",
       "   'BioCyc': 'META:CYTIDINEKIN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68618',\n",
       "   'KEGG Reaction': 'R00517',\n",
       "   'EC Number': '2.7.1.48'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36287_1 or EDU37946_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36287_1 at 0x7fc44985df70>,\n",
       "   <Gene EDU37946_1 at 0x7fc44985dfa0>},\n",
       "  '_metabolites': {<Metabolite cytd_c at 0x7fc463a820a0>: -1.0,\n",
       "   <Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite cmp_c at 0x7fc448e1cdf0>: 1.0,\n",
       "   <Metabolite gdp_c at 0x7fc463adc850>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTDt2',\n",
       "  'name': 'Cytidine transport in via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1379',\n",
       "   'BioCyc': 'META:TRANS-RXN-108B'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38420_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38420_1 at 0x7fc44985dfd0>},\n",
       "  '_metabolites': {<Metabolite cytd_e at 0x7fc448ec8820>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite cytd_c at 0x7fc463a820a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTK1',\n",
       "  'name': 'Cytidylate kinase (CMP)',\n",
       "  'notes': {'EC Number': '2.7.4.25',\n",
       "   'BioCyc': 'META:RXN-11832',\n",
       "   'BioPath Reaction': 'RXN00034',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68619',\n",
       "   'KEGG Reaction': 'R00512',\n",
       "   'RHEA': '11600;11601;11602;11603',\n",
       "   'UniPathway Reaction': 'UCR00512;UER00641'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37025_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37025_1 at 0x7fc44985d100>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite cmp_c at 0x7fc448e1cdf0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cdp_c at 0x7fc448e9f7f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTK2',\n",
       "  'name': 'Cytidylate kinase (dCMP)',\n",
       "  'notes': {'EC Number': '2.7.4.25',\n",
       "   'BioCyc': 'META:RXN-7913',\n",
       "   'BioPath Reaction': 'RXN00572',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68625',\n",
       "   'RHEA': '25094;25095;25096;25097',\n",
       "   'KEGG Reaction': 'R01665'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37025_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37025_1 at 0x7fc44985d100>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dcmp_c at 0x7fc448e1c850>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite dcdp_c at 0x7fc448ec8850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'CYTOM',\n",
       "  'name': 'Cytosine 5 methyltransferase',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35719_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35719_1 at 0x7fc44985d070>},\n",
       "  '_metabolites': {<Metabolite amet_c at 0x7fc463adc940>: -1.0,\n",
       "   <Metabolite csn_c at 0x7fc448e9ff40>: -1.0,\n",
       "   <Metabolite ahcys_c at 0x7fc463adc910>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 5mcsn_c at 0x7fc448ec88b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DADA',\n",
       "  'name': 'Deoxyadenosine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82107', 'EC Number': '3.5.4.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38839_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38839_1 at 0x7fc449820bb0>},\n",
       "  '_metabolites': {<Metabolite dad_2_c at 0x7fc448ec88e0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite din_c at 0x7fc448ec8910>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DADK',\n",
       "  'name': 'Deoxyadenylate kinase',\n",
       "  'notes': {'EC Number': '2.7.4.3',\n",
       "   'BioCyc': 'META:DEOXYADENYLATE-KINASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01182',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68634',\n",
       "   'RHEA': '23100;23101;23102;23103',\n",
       "   'KEGG Reaction': 'R01547'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36460_1 or EDU39058_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36460_1 at 0x7fc44986a0a0>,\n",
       "   <Gene EDU39058_1 at 0x7fc449820c70>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite damp_c at 0x7fc448e1ca00>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite dadp_c at 0x7fc448e1c9d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DADNK',\n",
       "  'name': 'DADNK',\n",
       "  'notes': {'EC Number': '2.7.1.76',\n",
       "   'BioCyc': 'META:DEOXYADENOSINE-KINASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00390',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1405',\n",
       "   'RHEA': '23452;23453;23454;23455',\n",
       "   'KEGG Reaction': 'R02089'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dad_2_c at 0x7fc448ec88e0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite damp_c at 0x7fc448e1ca00>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAGK120',\n",
       "  'name': 'Diacylglycerol kinase (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68636'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35899_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35899_1 at 0x7fc44986a130>},\n",
       "  '_metabolites': {<Metabolite 12dgr120_c at 0x7fc463a6d220>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pa120_c at 0x7fc448e490d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAGK140',\n",
       "  'name': 'Diacylglycerol kinase (n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82116'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35899_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35899_1 at 0x7fc44986a130>},\n",
       "  '_metabolites': {<Metabolite 12dgr140_c at 0x7fc463a6d1f0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pa140_c at 0x7fc448e491c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAGK141',\n",
       "  'name': 'Diacylglycerol kinase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68638'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 12dgr141_c at 0x7fc463a6d0d0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pa141_c at 0x7fc448e49220>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAGK160',\n",
       "  'name': 'Diacylglycerol kinase (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82117', 'EC Number': '2.7.1.107'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35899_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35899_1 at 0x7fc44986a130>},\n",
       "  '_metabolites': {<Metabolite 12dgr160_c at 0x7fc463a6d2b0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pa160_c at 0x7fc448e49280>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAGK161',\n",
       "  'name': 'Diacylglycerol kinase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68640'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35899_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35899_1 at 0x7fc44986a130>},\n",
       "  '_metabolites': {<Metabolite 12dgr161_c at 0x7fc463a6d310>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pa161_c at 0x7fc448e492e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAGK180',\n",
       "  'name': 'Diacylglycerol kinase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82120'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35899_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35899_1 at 0x7fc44986a130>},\n",
       "  '_metabolites': {<Metabolite 12dgr180_c at 0x7fc463a6d370>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pa180_c at 0x7fc448e49340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAGK181',\n",
       "  'name': 'Diacylglycerol kinase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68642'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35899_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35899_1 at 0x7fc44986a130>},\n",
       "  '_metabolites': {<Metabolite 12dgr181_c at 0x7fc463a6d3d0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pa181_c at 0x7fc448e493a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DALAt2r',\n",
       "  'name': 'D-alanine transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1424',\n",
       "   'BioCyc': 'META:TRANS-RXN-62A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36264_1 or EDU37508_1 or EDU37559_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36264_1 at 0x7fc44983b340>,\n",
       "   <Gene EDU37508_1 at 0x7fc44983b310>,\n",
       "   <Gene EDU37559_1 at 0x7fc44983b370>},\n",
       "  '_metabolites': {<Metabolite ala__D_e at 0x7fc448e49910>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite ala__D_c at 0x7fc463a82fd0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DALAtex',\n",
       "  'name': 'D-Alanine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1425'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ala__D_e at 0x7fc448e49910>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAPabc',\n",
       "  'name': 'M-diaminopimelic acid ABC transport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1433'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 26dap__M_e at 0x7fc463a6d820>: -1.0,\n",
       "   <Metabolite 26dap__M_c at 0x7fc448ec8940>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAPDC',\n",
       "  'name': 'Diaminopimelate decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.20',\n",
       "   'BioCyc': 'META:DIAMINOPIMDECARB-RXN',\n",
       "   'BioPath Reaction': 'RXN01425',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1430',\n",
       "   'KEGG Reaction': 'R00451',\n",
       "   'RHEA': '15101;15102;15103;15104',\n",
       "   'UniPathway Reaction': 'UCR00451;UER00027'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39206_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39206_1 at 0x7fc44986a160>},\n",
       "  '_metabolites': {<Metabolite 26dap__M_c at 0x7fc448ec8940>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite lys__L_c at 0x7fc448e74310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DAPE',\n",
       "  'name': 'Diaminopimelate epimerase',\n",
       "  'notes': {'EC Number': '5.1.1.7',\n",
       "   'BioCyc': 'META:DIAMINOPIMEPIM-RXN',\n",
       "   'BioPath Reaction': 'RXN00670',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1432',\n",
       "   'KEGG Reaction': 'R02735',\n",
       "   'RHEA': '15393;15394;15395;15396',\n",
       "   'UniPathway Reaction': 'UCR02735;UER00025'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36911_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36911_1 at 0x7fc44986a190>},\n",
       "  '_metabolites': {<Metabolite 26dap_LL_c at 0x7fc448e1cb80>: -1.0,\n",
       "   <Metabolite 26dap__M_c at 0x7fc448ec8940>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DARBt2rpp',\n",
       "  'name': 'D-arabinose transport via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR29522'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite arab__D_p at 0x7fc448ec8970>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite arab__D_c at 0x7fc448e743d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DARBt3ipp',\n",
       "  'name': 'D-arabinose transport via proton antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR29522'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37163_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37163_1 at 0x7fc44986a1c0>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite arab__D_c at 0x7fc448e743d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite arab__D_p at 0x7fc448ec8970>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DASYN120',\n",
       "  'name': 'CDP-diacylglycerol synthetase (n-C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68647', 'EC Number': '2.7.7.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36404_1 or EDU36418_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36404_1 at 0x7fc44986a1f0>,\n",
       "   <Gene EDU36418_1 at 0x7fc44986a220>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pa120_c at 0x7fc448e490d0>: -1.0,\n",
       "   <Metabolite cdpdddecg_c at 0x7fc448e9f580>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DASYN140',\n",
       "  'name': 'CDP-diacylglycerol synthetase (n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82152', 'EC Number': '2.7.7.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36404_1 or EDU36418_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36404_1 at 0x7fc44986a1f0>,\n",
       "   <Gene EDU36418_1 at 0x7fc44986a220>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pa140_c at 0x7fc448e491c0>: -1.0,\n",
       "   <Metabolite cdpdtdecg_c at 0x7fc448e9f5b0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DASYN141',\n",
       "  'name': 'CDP-diacylglycerol synthetase (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68649', 'EC Number': '2.7.7.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36404_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36404_1 at 0x7fc44986a1f0>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pa141_c at 0x7fc448e49220>: -1.0,\n",
       "   <Metabolite cdpdtdec7eg_c at 0x7fc448e9f5e0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DASYN160',\n",
       "  'name': 'CDP-diacylglycerol synthetase (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82153', 'EC Number': '2.7.7.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36404_1 or EDU36418_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36404_1 at 0x7fc44986a1f0>,\n",
       "   <Gene EDU36418_1 at 0x7fc44986a220>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pa160_c at 0x7fc448e49280>: -1.0,\n",
       "   <Metabolite cdpdhdecg_c at 0x7fc448e9f610>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DASYN161',\n",
       "  'name': 'CDP-diacylglycerol synthetase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68652', 'EC Number': '2.7.7.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36404_1 or EDU36418_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36404_1 at 0x7fc44986a1f0>,\n",
       "   <Gene EDU36418_1 at 0x7fc44986a220>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pa161_c at 0x7fc448e492e0>: -1.0,\n",
       "   <Metabolite cdpdhdec9eg_c at 0x7fc448e9f640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DASYN180',\n",
       "  'name': 'CDP-diacylglycerol synthetase (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82154', 'EC Number': '2.7.7.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36404_1 or EDU36418_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36404_1 at 0x7fc44986a1f0>,\n",
       "   <Gene EDU36418_1 at 0x7fc44986a220>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pa180_c at 0x7fc448e49340>: -1.0,\n",
       "   <Metabolite cdpdodecg_c at 0x7fc448e9f670>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DASYN181',\n",
       "  'name': 'CDP-diacylglycerol synthetase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68654', 'EC Number': '2.7.7.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36404_1 or EDU36418_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36404_1 at 0x7fc44986a1f0>,\n",
       "   <Gene EDU36418_1 at 0x7fc44986a220>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pa181_c at 0x7fc448e493a0>: -1.0,\n",
       "   <Metabolite cdpdodec11eg_c at 0x7fc448e9f6a0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DATPHs',\n",
       "  'name': 'DATP amine hydrolysis (spontaneous)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82161'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite datp_c at 0x7fc448e1ca30>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ditp_c at 0x7fc448ec89a0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DB4PS',\n",
       "  'name': '3,4-Dihydroxy-2-butanone-4-phosphate synthase',\n",
       "  'notes': {'EC Number': '4.1.99.12',\n",
       "   'BioCyc': 'META:DIOHBUTANONEPSYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26412',\n",
       "   'KEGG Reaction': 'R07281',\n",
       "   'RHEA': '18457;18458;18459;18460',\n",
       "   'UniPathway Reaction': 'UCR07281;UER00399'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35924_1 or EDU35981_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35924_1 at 0x7fc44986a280>,\n",
       "   <Gene EDU35981_1 at 0x7fc44986a250>},\n",
       "  '_metabolites': {<Metabolite ru5p__D_c at 0x7fc463aae4c0>: -1.0,\n",
       "   <Metabolite db4p_c at 0x7fc448ec89d0>: 1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DCAtex',\n",
       "  'name': 'Decanoate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1456'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite dca_e at 0x7fc448ec8a30>: -1.0,\n",
       "   <Metabolite dca_p at 0x7fc448ec8a60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DCMPDA',\n",
       "  'name': 'DCMP deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82164', 'EC Number': '3.5.4.12'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37993_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37993_1 at 0x7fc44985d730>},\n",
       "  '_metabolites': {<Metabolite dcmp_c at 0x7fc448e1c850>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite dump_c at 0x7fc448ec8a90>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DCTPD2',\n",
       "  'name': 'DCTPD2',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82167', 'EC Number': '3.5.4.13'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite utp_c at 0x7fc448ec84c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DCYTD',\n",
       "  'name': 'Deoxycytidine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82168', 'EC Number': '3.5.4.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35900_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35900_1 at 0x7fc44985df40>},\n",
       "  '_metabolites': {<Metabolite dcyt_c at 0x7fc448e1c880>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite duri_c at 0x7fc448ec8b50>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DCYTt2',\n",
       "  'name': 'Deoxycytidine transport in via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1480',\n",
       "   'BioCyc': 'META:TRANS-RXN-108D'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38420_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38420_1 at 0x7fc44985dfd0>},\n",
       "  '_metabolites': {<Metabolite dcyt_e at 0x7fc448ec8b80>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite dcyt_c at 0x7fc448e1c880>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DDCAt2pp',\n",
       "  'name': 'Dodecanoate transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR29418'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ddca_p at 0x7fc448ec8bb0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ddca_c at 0x7fc463a6dbb0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DDGLCNt2r',\n",
       "  'name': '2-dehydro-3-deoxy-D-gluconate transport via proton symport, reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1483',\n",
       "   'BioCyc': 'META:TRANS-RXN-113'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2ddglcn_e at 0x7fc448ec8be0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite 2ddglcn_c at 0x7fc448e49ee0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DDGLK',\n",
       "  'name': '2-dehydro-3-deoxygluconokinase',\n",
       "  'notes': {'EC Number': '2.7.1.45',\n",
       "   'BioCyc': 'META:DEOXYGLUCONOKIN-RXN',\n",
       "   'BioPath Reaction': 'RXN01439',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1485',\n",
       "   'KEGG Reaction': 'R01541',\n",
       "   'RHEA': '14797;14798;14799;14800',\n",
       "   'UniPathway Reaction': 'UCR01541;UER00828'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37826_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37826_1 at 0x7fc44986a2b0>},\n",
       "  '_metabolites': {<Metabolite 2ddglcn_c at 0x7fc448e49ee0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite 2ddg6p_c at 0x7fc448ec8c10>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DDPA',\n",
       "  'name': '3-deoxy-D-arabino-heptulosonate 7-phosphate synthetase',\n",
       "  'notes': {'EC Number': '2.5.1.54',\n",
       "   'BioCyc': 'META:DAHPSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN01379',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1486',\n",
       "   'KEGG Reaction': 'R01826',\n",
       "   'RHEA': '14717;14718;14719;14720',\n",
       "   'UniPathway Reaction': 'UCR01826;UER00084'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37303_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37303_1 at 0x7fc44985d6a0>},\n",
       "  '_metabolites': {<Metabolite e4p_c at 0x7fc448e74d90>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite 2dda7p_c at 0x7fc463adc3d0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DGK1',\n",
       "  'name': 'Deoxyguanylate kinase (dGMP:ATP)',\n",
       "  'notes': {'EC Number': '2.7.4.8',\n",
       "   'BioCyc': 'META:GMKALT-RXN',\n",
       "   'BioPath Reaction': 'RXN01294',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68689',\n",
       "   'RHEA': '12697;12698;12699;12700',\n",
       "   'KEGG Reaction': 'R02090'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36318_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36318_1 at 0x7fc44986a2e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dgmp_c at 0x7fc448ec8cd0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite dgdp_c at 0x7fc448e1ca90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DGUNC',\n",
       "  'name': 'Diguanylate cyclase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82264'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36802_1 or EDU37256_1 or EDU37769_1 or EDU37775_1 or EDU37796_1 or EDU38047_1 or EDU38616_1 or EDU38745_1 or EDU39185_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36802_1 at 0x7fc44986a3d0>,\n",
       "   <Gene EDU37256_1 at 0x7fc44985d340>,\n",
       "   <Gene EDU37769_1 at 0x7fc44985d460>,\n",
       "   <Gene EDU37775_1 at 0x7fc44986a3a0>,\n",
       "   <Gene EDU37796_1 at 0x7fc44985d430>,\n",
       "   <Gene EDU38047_1 at 0x7fc44986a400>,\n",
       "   <Gene EDU38616_1 at 0x7fc44986a340>,\n",
       "   <Gene EDU38745_1 at 0x7fc44986a370>,\n",
       "   <Gene EDU39185_1 at 0x7fc44986a310>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -2.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 2.0,\n",
       "   <Metabolite cdigmp_c at 0x7fc448e9f730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DH4MCHCDH',\n",
       "  'name': 'DH4MCHCDH',\n",
       "  'notes': {'RHEA': '24228;24229;24230;24231',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26518',\n",
       "   'KEGG Reaction': 'R05292',\n",
       "   'EC Number': '1.3.1.67'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite dh4mchc_c at 0x7fc448ec8d30>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite 4mcat_c at 0x7fc463a82df0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHAD1',\n",
       "  'name': 'Dihydroxy-acid dehydratase (2,3-dihydroxy-3-methylbutanoate)',\n",
       "  'notes': {'EC Number': '4.2.1.9',\n",
       "   'BioCyc': 'META:DIHYDROXYISOVALDEHYDRAT-RXN',\n",
       "   'BioPath Reaction': 'RXN00558',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68691',\n",
       "   'KEGG Reaction': 'R04441',\n",
       "   'RHEA': '24809;24810;24811;24812',\n",
       "   'UniPathway Reaction': 'UCR04441;UER00061'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37828_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37828_1 at 0x7fc44986a430>},\n",
       "  '_metabolites': {<Metabolite 23dhmb_c at 0x7fc463a6d7f0>: -1.0,\n",
       "   <Metabolite 3mob_c at 0x7fc463adc490>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHAD2',\n",
       "  'name': 'Dihydroxy-acid dehydratase (2,3-dihydroxy-3-methylpentanoate)',\n",
       "  'notes': {'EC Number': '4.2.1.9',\n",
       "   'BioCyc': 'META:DIHYDROXYMETVALDEHYDRAT-RXN',\n",
       "   'BioPath Reaction': 'RXN00766',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1536',\n",
       "   'KEGG Reaction': 'R05070',\n",
       "   'RHEA': '27694;27695;27696;27697',\n",
       "   'UniPathway Reaction': 'UCR05070;UER00057'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37828_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37828_1 at 0x7fc44986a430>},\n",
       "  '_metabolites': {<Metabolite 23dhmp_c at 0x7fc463a6d8b0>: -1.0,\n",
       "   <Metabolite 3mop_c at 0x7fc463adcee0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHAD3',\n",
       "  'name': 'Dihydroxy acid dehydratase  2 3 dihydroxy 3 methylpentanoate ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37828_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37828_1 at 0x7fc44986a430>},\n",
       "  '_metabolites': {<Metabolite 23dhmp_c at 0x7fc463a6d8b0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite 2o3mpt_c at 0x7fc448ec8d90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHAK',\n",
       "  'name': 'Dihydroxyacetone kinase',\n",
       "  'notes': {'EC Number': '2.7.1.29',\n",
       "   'BioCyc': 'META:GLYCERONE-KINASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1537',\n",
       "   'KEGG Reaction': 'R01011',\n",
       "   'RHEA': '15773;15774;15775;15776',\n",
       "   'UniPathway Reaction': 'UCR01011;UER00669'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38822_1 or EDU39098_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38822_1 at 0x7fc44986a460>,\n",
       "   <Gene EDU39098_1 at 0x7fc44986a490>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dha_c at 0x7fc448e49d60>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite dhap_c at 0x7fc448e49d90>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHAPT',\n",
       "  'name': 'Dihydroxyacetone phosphotransferase',\n",
       "  'notes': {'RHEA': '18381;18382;18383;18384',\n",
       "   'BioCyc': 'META:2.7.1.121-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1539',\n",
       "   'KEGG Reaction': 'R01012',\n",
       "   'EC Number': '2.7.1.121'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1 and EDU38822_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU38822_1 at 0x7fc44986a460>},\n",
       "  '_metabolites': {<Metabolite dha_c at 0x7fc448e49d60>: -1.0,\n",
       "   <Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite dhap_c at 0x7fc448e49d90>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHDPRy',\n",
       "  'name': 'Dihydrodipicolinate reductase (NADPH)',\n",
       "  'notes': {'RHEA': '24344;24345;24346;24347',\n",
       "   'BioPath Reaction': 'RXN01489',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1556',\n",
       "   'EC Number': '1.3.1.26'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38647_1 or EDU39386_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38647_1 at 0x7fc44986a4f0>,\n",
       "   <Gene EDU39386_1 at 0x7fc44986a4c0>},\n",
       "  '_metabolites': {<Metabolite 23dhdp_c at 0x7fc463a6d8e0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0,\n",
       "   <Metabolite thdp_c at 0x7fc448e74250>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHDPS',\n",
       "  'name': 'Dihydrodipicolinate synthase',\n",
       "  'notes': {'RHEA': '14845;14846;14847;14848',\n",
       "   'BioPath Reaction': 'RXN00081',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1557'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39387_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39387_1 at 0x7fc44986a520>},\n",
       "  '_metabolites': {<Metabolite aspsa_c at 0x7fc448e74700>: -1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: -1.0,\n",
       "   <Metabolite 23dhdp_c at 0x7fc463a6d8e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHFR',\n",
       "  'name': 'Dihydrofolate reductase',\n",
       "  'notes': {'EC Number': '1.5.1.3',\n",
       "   'BioPath Reaction': 'RXN00447',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1561',\n",
       "   'KEGG Reaction': 'R00939',\n",
       "   'RHEA': '15009;15010;15011;15012',\n",
       "   'UniPathway Reaction': 'UCR00939;UER00158'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37867_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37867_1 at 0x7fc44986a550>},\n",
       "  '_metabolites': {<Metabolite dhf_c at 0x7fc448ec8dc0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHFS',\n",
       "  'name': 'Dihydrofolate synthase',\n",
       "  'notes': {'EC Number': '6.3.2.17',\n",
       "   'BioCyc': 'META:DIHYDROFOLATESYNTH-RXN',\n",
       "   'BioPath Reaction': 'RXN00210',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR32670',\n",
       "   'KEGG Reaction': 'R02237',\n",
       "   'RHEA': '23584;23585;23586;23587',\n",
       "   'UniPathway Reaction': 'UCR02237;UER00157'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39377_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39377_1 at 0x7fc44986a580>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dhpt_c at 0x7fc448ec8df0>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite dhf_c at 0x7fc448ec8dc0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHNAOT',\n",
       "  'name': '1,4-dihydroxy-2-naphthoate octaprenyltransferase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68700'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite dhna_c at 0x7fc448ec8e20>: -1.0,\n",
       "   <Metabolite octdp_c at 0x7fc448ec8e50>: -1.0,\n",
       "   <Metabolite 2dmmq8_c at 0x7fc448e49f40>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHNPA_1',\n",
       "  'name': 'Dihydroneopterin aldolase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82283'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38671_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38671_1 at 0x7fc44986a5b0>},\n",
       "  '_metabolites': {<Metabolite dhnpt_c at 0x7fc448e49790>: -1.0,\n",
       "   <Metabolite gcald_c at 0x7fc448e494f0>: 1.0,\n",
       "   <Metabolite 2ahhmp_c at 0x7fc448ec8e80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHORD5',\n",
       "  'name': 'Dihydroorotic acid (menaquinone-8)',\n",
       "  'notes': {'RHEA': '29199;29200;29201;29202',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26702'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39305_1 and EDU39306_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39305_1 at 0x7fc44986a610>,\n",
       "   <Gene EDU39306_1 at 0x7fc44986a5e0>},\n",
       "  '_metabolites': {<Metabolite dhor__S_c at 0x7fc448ec8eb0>: -1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0,\n",
       "   <Metabolite orot_c at 0x7fc448ec8ee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHORDi',\n",
       "  'name': 'Dihydoorotic acid dehydrogenase  irreversible ',\n",
       "  'notes': {'RHEA': '15441;15442;15443;15444',\n",
       "   'BioPath Reaction': 'RXN00151',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1569',\n",
       "   'EC Number': '1.3.99.11'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38226_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38226_1 at 0x7fc44986a640>},\n",
       "  '_metabolites': {<Metabolite dhor__S_c at 0x7fc448ec8eb0>: -1.0,\n",
       "   <Metabolite o2_c at 0x7fc463a6d610>: -1.0,\n",
       "   <Metabolite h2o2_c at 0x7fc463a6d580>: 1.0,\n",
       "   <Metabolite orot_c at 0x7fc448ec8ee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHORTS',\n",
       "  'name': 'Dihydroorotase',\n",
       "  'notes': {'EC Number': '3.5.2.3',\n",
       "   'BioCyc': 'META:DIHYDROOROT-RXN',\n",
       "   'BioPath Reaction': 'RXN00773;RXN01007',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1576',\n",
       "   'KEGG Reaction': 'R01993',\n",
       "   'RHEA': '24296;24297;24298;24299',\n",
       "   'UniPathway Reaction': 'UCR01993;UER00117'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39082_1 or EDU39303_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39082_1 at 0x7fc44983b4f0>,\n",
       "   <Gene EDU39303_1 at 0x7fc44986a670>},\n",
       "  '_metabolites': {<Metabolite dhor__S_c at 0x7fc448ec8eb0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cbasp_c at 0x7fc448e748b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHPACCOAHIT',\n",
       "  'name': 'DHPACCOAHIT',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68709'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -2.0,\n",
       "   <Metabolite 2dhphaccoa_c at 0x7fc448ec8fa0>: -1.0,\n",
       "   <Metabolite 3hadpcoa_c at 0x7fc448ec8d60>: 1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite h2_c at 0x7fc463a827c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHPM1',\n",
       "  'name': 'DHPM1',\n",
       "  'notes': {'EC Number': '3.5.2.2',\n",
       "   'BioCyc': 'META:DIHYDROPYRIMIDINASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01531',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1580',\n",
       "   'RHEA': '16121;16122;16123;16124',\n",
       "   'KEGG Reaction': 'R02269'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 56dura_c at 0x7fc448ec8f10>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite cala_c at 0x7fc448e74f40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHPPDA',\n",
       "  'name': 'Diaminohydroxyphosphoribosylaminopyrimidine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82286'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35979_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35979_1 at 0x7fc44983b790>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 25dhpp_c at 0x7fc448ec8f70>: -1.0,\n",
       "   <Metabolite 5apru_c at 0x7fc448e743a0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHPPDA2',\n",
       "  'name': 'Diaminohydroxyphosphoribosylaminopryrimidine deaminase (25drapp)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82287', 'EC Number': '3.5.4.26'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 25drapp_c at 0x7fc448ec8f40>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 5apru_c at 0x7fc448e743a0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHPTDCs2',\n",
       "  'name': '4,5-dihydroxy-2,3-pentanedione cyclization (spontaneous)',\n",
       "  'notes': {'RHEA': '35155;35156;35157;35158',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR62124',\n",
       "   'BioCyc': 'META:RXN-10015'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite dhptd_c at 0x7fc448eef160>: -1.0,\n",
       "   <Metabolite mdhdhf_c at 0x7fc448eef190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHQS',\n",
       "  'name': '3-dehydroquinate synthase',\n",
       "  'notes': {'EC Number': '4.2.3.4',\n",
       "   'BioCyc': 'META:3-DEHYDROQUINATE-SYNTHASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01223',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1594',\n",
       "   'KEGG Reaction': 'R03083',\n",
       "   'RHEA': '21968;21969;21970;21971',\n",
       "   'UniPathway Reaction': 'UCR03083;UER00085'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36656_1 or EDU37305_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36656_1 at 0x7fc44986a6a0>,\n",
       "   <Gene EDU37305_1 at 0x7fc44986a6d0>},\n",
       "  '_metabolites': {<Metabolite 2dda7p_c at 0x7fc463adc3d0>: -1.0,\n",
       "   <Metabolite 3dhq_c at 0x7fc448eef0a0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DHQTi',\n",
       "  'name': '3-dehydroquinate dehydratase, irreversible',\n",
       "  'notes': {'EC Number': '4.2.1.10',\n",
       "   'BioCyc': 'META:3-DEHYDROQUINATE-DEHYDRATASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00107',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55233',\n",
       "   'KEGG Reaction': 'R03084',\n",
       "   'RHEA': '21096;21097;21098;21099',\n",
       "   'UniPathway Reaction': 'UCR03084;UER00086;UER00178'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36924_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36924_1 at 0x7fc44986a700>},\n",
       "  '_metabolites': {<Metabolite 3dhq_c at 0x7fc448eef0a0>: -1.0,\n",
       "   <Metabolite 3dhsk_c at 0x7fc448eef040>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DIPEPabc15',\n",
       "  'name': 'Dipeptide transport via ABC system  cgly ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39389_1 and EDU39390_1 and EDU39391_1 and EDU39392_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39389_1 at 0x7fc44986a730>,\n",
       "   <Gene EDU39390_1 at 0x7fc44986a790>,\n",
       "   <Gene EDU39391_1 at 0x7fc44986a760>,\n",
       "   <Gene EDU39392_1 at 0x7fc4498203a0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cgly_e at 0x7fc448e74190>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cgly_c at 0x7fc448e74160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DKMPPD2',\n",
       "  'name': '2 3 diketo 5 methylthio 1 phosphopentane degradation reaction',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82299'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -3.0,\n",
       "   <Metabolite dkmpp_c at 0x7fc463a6d850>: -1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 6.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite 2kmb_c at 0x7fc463aaeb50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'D_LACt2',\n",
       "  'name': 'D lactate transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1398',\n",
       "   'BioCyc': 'META:TRANS-RXN0-515'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38858_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38858_1 at 0x7fc44986ab50>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite lac__D_e at 0x7fc448eef100>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lac__D_c at 0x7fc448e1c0a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DMATT',\n",
       "  'name': 'Dimethylallyltranstransferase',\n",
       "  'notes': {'EC Number': '2.5.1.1',\n",
       "   'BioCyc': 'META:GPPSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN00099',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55238',\n",
       "   'KEGG Reaction': 'R01658',\n",
       "   'RHEA': '22408;22409;22410;22411',\n",
       "   'UniPathway Reaction': 'UCR01658;UER00368'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36939_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36939_1 at 0x7fc44986a7c0>},\n",
       "  '_metabolites': {<Metabolite dmpp_c at 0x7fc448eef1c0>: -1.0,\n",
       "   <Metabolite ipdp_c at 0x7fc448ec8af0>: -1.0,\n",
       "   <Metabolite grdp_c at 0x7fc448eef1f0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DMPPS',\n",
       "  'name': '1-hydroxy-2-methyl-2-(E)-butenyl 4-diphosphate reductase (dmpp)',\n",
       "  'notes': {'RHEA': '24821;24822;24823;24824',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68731',\n",
       "   'KEGG Reaction': 'R08210',\n",
       "   'EC Number': '1.17.1.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37026_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37026_1 at 0x7fc44986a7f0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2mb4p_c at 0x7fc448eef250>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite dmpp_c at 0x7fc448eef1c0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DNMPPA',\n",
       "  'name': 'Dihydroneopterin monophosphate dephosphorylase',\n",
       "  'notes': {'EC Number': '3.6.1.-',\n",
       "   'BioCyc': 'META:DIHYDRONEOPTERIN-MONO-P-DEPHOS-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1716',\n",
       "   'KEGG Reaction': 'R04621',\n",
       "   'RHEA': '25306;25307;25308;25309',\n",
       "   'UniPathway Reaction': 'UCR04621;UER00153'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38595_1 or EDU38845_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38595_1 at 0x7fc44986a820>,\n",
       "   <Gene EDU38845_1 at 0x7fc44986a850>},\n",
       "  '_metabolites': {<Metabolite dhpmp_c at 0x7fc448ec8fd0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite dhnpt_c at 0x7fc448e49790>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DNTPPA',\n",
       "  'name': 'Dihydroneopterin triphosphate pyrophosphatase',\n",
       "  'notes': {'EC Number': '3.6.1.67',\n",
       "   'BioCyc': 'META:H2NEOPTERINP3PYROPHOSPHOHYDRO-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1717',\n",
       "   'KEGG Reaction': 'R04638',\n",
       "   'RHEA': '25302;25303;25304;25305',\n",
       "   'UniPathway Reaction': 'UCR04638;UER00152'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38845_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38845_1 at 0x7fc44986a850>},\n",
       "  '_metabolites': {<Metabolite ahdt_c at 0x7fc463aae2e0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite dhpmp_c at 0x7fc448ec8fd0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DPCOAK',\n",
       "  'name': 'Dephospho-CoA kinase',\n",
       "  'notes': {'EC Number': '2.7.1.24',\n",
       "   'BioCyc': 'META:DEPHOSPHOCOAKIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00293',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1772',\n",
       "   'KEGG Reaction': 'R00130',\n",
       "   'RHEA': '18245;18246;18247;18248',\n",
       "   'UniPathway Reaction': 'UCR00130;UER00356'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35833_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35833_1 at 0x7fc44986a880>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dpcoa_c at 0x7fc448e74340>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DPMVD',\n",
       "  'name': 'Diphosphomevalonate decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.33',\n",
       "   'BioCyc': 'META:DIPHOSPHOMEVALONTE-DECARBOXYLASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00266',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68753',\n",
       "   'KEGG Reaction': 'R01121',\n",
       "   'RHEA': '23732;23733;23734;23735',\n",
       "   'UniPathway Reaction': 'UCR01121;UER00100'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite 5dpmev_c at 0x7fc448eef310>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite ipdp_c at 0x7fc448ec8af0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DPR',\n",
       "  'name': '2-dehydropantoate 2-reductase',\n",
       "  'notes': {'EC Number': '1.1.1.169',\n",
       "   'BioCyc': 'META:2-DEHYDROPANTOATE-REDUCT-RXN',\n",
       "   'BioPath Reaction': 'RXN00209',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1788',\n",
       "   'KEGG Reaction': 'R02472',\n",
       "   'RHEA': '16233;16234;16235;16236',\n",
       "   'UniPathway Reaction': 'UCR02472;UER00004'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37646_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37646_1 at 0x7fc44986a8b0>},\n",
       "  '_metabolites': {<Metabolite 2dhp_c at 0x7fc463adc460>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0,\n",
       "   <Metabolite pant__R_c at 0x7fc448eef340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DRBK',\n",
       "  'name': 'Deoxyribokinase',\n",
       "  'notes': {'RHEA': '30871;30872;30873;30874',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR82393',\n",
       "   'EC Number': '2.7.1.15',\n",
       "   'BioCyc': 'META:RXN-14223'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37200_1 or EDU37648_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37200_1 at 0x7fc44986a8e0>,\n",
       "   <Gene EDU37648_1 at 0x7fc44986a910>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite drib_c at 0x7fc463adc520>: -1.0,\n",
       "   <Metabolite 2dr5p_c at 0x7fc448eef370>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DRIBabc',\n",
       "  'name': 'Deoxyribose transport via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37922_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37922_1 at 0x7fc44986a940>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite drib_e at 0x7fc448eef3a0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite drib_c at 0x7fc463adc520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DRPA',\n",
       "  'name': 'Deoxyribose-phosphate aldolase',\n",
       "  'notes': {'EC Number': '4.1.2.4',\n",
       "   'BioCyc': 'META:DEOXYRIBOSE-P-ALD-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR50871',\n",
       "   'KEGG Reaction': 'R01066',\n",
       "   'RHEA': '12821;12822;12823;12824',\n",
       "   'UniPathway Reaction': 'UCR01066;UER00468'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37201_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37201_1 at 0x7fc44986a970>},\n",
       "  '_metabolites': {<Metabolite 2dr5p_c at 0x7fc448eef370>: -1.0,\n",
       "   <Metabolite acald_c at 0x7fc463aae2b0>: 1.0,\n",
       "   <Metabolite g3p_c at 0x7fc448ec8c40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DSERt2',\n",
       "  'name': 'DSERt2',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1803', 'BioCyc': 'META:RXN0-5130'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36264_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36264_1 at 0x7fc44983b340>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite ser__D_e at 0x7fc448eef490>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ser__D_c at 0x7fc448eef460>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DTMPK',\n",
       "  'name': 'DTMP kinase',\n",
       "  'notes': {'EC Number': '2.7.4.9',\n",
       "   'BioCyc': 'META:DTMPKI-RXN',\n",
       "   'BioPath Reaction': 'RXN00496',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68764',\n",
       "   'RHEA': '13517;13518;13519;13520',\n",
       "   'KEGG Reaction': 'R02094'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38083_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38083_1 at 0x7fc44986a9a0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite dtmp_c at 0x7fc448eef4c0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite dtdp_c at 0x7fc463aae670>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DURAD',\n",
       "  'name': 'DURAD',\n",
       "  'notes': {'EC Number': '1.3.1.2',\n",
       "   'BioCyc': 'META:1.3.1.2-RXN',\n",
       "   'BioPath Reaction': 'RXN00652',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1810',\n",
       "   'RHEA': '18093;18094;18095;18096',\n",
       "   'KEGG Reaction': 'R00978'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36025_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36025_1 at 0x7fc44986a9d0>},\n",
       "  '_metabolites': {<Metabolite 56dura_c at 0x7fc448ec8f10>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0,\n",
       "   <Metabolite ura_c at 0x7fc448ec8430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DURADx',\n",
       "  'name': 'Dihydrouracil dehydrogenase (NAD)',\n",
       "  'notes': {'EC Number': '1.3.1.1',\n",
       "   'BioCyc': 'META:DIHYDROURACIL-DEHYDROGENASE-NAD+-RXN',\n",
       "   'BioPath Reaction': 'RXN00751',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR5810',\n",
       "   'RHEA': '20189;20190;20191;20192',\n",
       "   'KEGG Reaction': 'R00977'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36024_1 or EDU36025_1 or (EDU36024_1 and EDU36025_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36024_1 at 0x7fc44986aa00>,\n",
       "   <Gene EDU36025_1 at 0x7fc44986a9d0>},\n",
       "  '_metabolites': {<Metabolite 56dura_c at 0x7fc448ec8f10>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite ura_c at 0x7fc448ec8430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DURIK1',\n",
       "  'name': 'Deoxyuridine kinase (ATP:Deoxyuridine)',\n",
       "  'notes': {'EC Number': '2.7.1.21',\n",
       "   'BioCyc': 'META:DURIDKI-RXN',\n",
       "   'BioPath Reaction': 'RXN00622',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1812',\n",
       "   'RHEA': '28206;28207;28208;28209',\n",
       "   'KEGG Reaction': 'R02099'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38003_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38003_1 at 0x7fc44986aa30>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite duri_c at 0x7fc448ec8b50>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite dump_c at 0x7fc448ec8a90>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DURIPP',\n",
       "  'name': 'Deoxyuridine phosphorylase',\n",
       "  'notes': {'RHEA': '22824;22825;22826;22827',\n",
       "   'BioCyc': 'META:URA-PHOSPH-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1813',\n",
       "   'KEGG Reaction': 'R02484',\n",
       "   'EC Number': '2.4.2.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36960_1 or EDU36961_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36960_1 at 0x7fc44986aa60>,\n",
       "   <Gene EDU36961_1 at 0x7fc44986aa90>},\n",
       "  '_metabolites': {<Metabolite duri_c at 0x7fc448ec8b50>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite 2dr1p_c at 0x7fc463adc4f0>: 1.0,\n",
       "   <Metabolite ura_c at 0x7fc448ec8430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DUTPDP',\n",
       "  'name': 'DUTP diphosphatase',\n",
       "  'notes': {'EC Number': '3.6.1.23',\n",
       "   'BioCyc': 'META:DUTP-PYROP-RXN',\n",
       "   'BioPath Reaction': 'RXN00353',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1816',\n",
       "   'KEGG Reaction': 'R02100',\n",
       "   'RHEA': '10248;10249;10250;10251',\n",
       "   'UniPathway Reaction': 'UCR02100;UER00666'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37571_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37571_1 at 0x7fc44986aac0>},\n",
       "  '_metabolites': {<Metabolite dutp_c at 0x7fc448ec8b20>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite dump_c at 0x7fc448ec8a90>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DXPRIi',\n",
       "  'name': '1-deoxy-D-xylulose reductoisomerase',\n",
       "  'notes': {'EC Number': '1.1.1.267',\n",
       "   'BioCyc': 'META:DXPREDISOM-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1818',\n",
       "   'KEGG Reaction': 'R05688',\n",
       "   'RHEA': '13717;13718;13719;13720',\n",
       "   'UniPathway Reaction': 'UCR05688;UER00092'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36408_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36408_1 at 0x7fc44986aaf0>},\n",
       "  '_metabolites': {<Metabolite dxyl5p_c at 0x7fc448eef550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 2me4p_c at 0x7fc448eef520>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'DXPS',\n",
       "  'name': '1-deoxy-D-xylulose 5-phosphate synthase',\n",
       "  'notes': {'EC Number': '2.2.1.7',\n",
       "   'BioCyc': 'META:DXS-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55306',\n",
       "   'KEGG Reaction': 'R05636',\n",
       "   'RHEA': '12605;12606;12607;12608',\n",
       "   'UniPathway Reaction': 'UCR05636;UER00091'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36941_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36941_1 at 0x7fc44986ab20>},\n",
       "  '_metabolites': {<Metabolite g3p_c at 0x7fc448ec8c40>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite dxyl5p_c at 0x7fc448eef550>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'E4PD',\n",
       "  'name': 'Erythrose 4-phosphate dehydrogenase',\n",
       "  'notes': {'EC Number': '1.2.1.72',\n",
       "   'BioCyc': 'META:ERYTH4PDEHYDROG-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1822',\n",
       "   'KEGG Reaction': 'R01825',\n",
       "   'RHEA': '12056;12057;12058;12059',\n",
       "   'UniPathway Reaction': 'UCR01825;UER00309'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite e4p_c at 0x7fc448e74d90>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 4per_c at 0x7fc448eef580>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR100x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C10:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68768',\n",
       "   'KEGG Reaction': 'R04961',\n",
       "   'EC Number': '2.3.1.-'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite tdec2eACP_c at 0x7fc463adcb80>: -1.0,\n",
       "   <Metabolite dcaACP_c at 0x7fc463a82520>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR120x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C12:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9661',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68770',\n",
       "   'KEGG Reaction': 'R04724',\n",
       "   'EC Number': '2.3.1.-'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite tddec2eACP_c at 0x7fc463adcbb0>: -1.0,\n",
       "   <Metabolite ddcaACP_c at 0x7fc463a82580>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR121x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C12:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68772', 'EC Number': '1.3.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite t3c5ddeceACP_c at 0x7fc463adcbe0>: -1.0,\n",
       "   <Metabolite cddec5eACP_c at 0x7fc463a825b0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR140x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C14:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9662',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68773',\n",
       "   'KEGG Reaction': 'R04966',\n",
       "   'EC Number': '2.3.1.-'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite tmrs2eACP_c at 0x7fc463adcc10>: -1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR141x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68775', 'EC Number': '1.3.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite t3c7mrseACP_c at 0x7fc463adcc40>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite tdeACP_c at 0x7fc463a82610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR160x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68777', 'EC Number': '1.3.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite tpalm2eACP_c at 0x7fc463adcca0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite palmACP_c at 0x7fc463a82640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR161x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82413', 'EC Number': '1.3.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite t3c9palmeACP_c at 0x7fc463adcd00>: -1.0,\n",
       "   <Metabolite hdeACP_c at 0x7fc463a82670>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR180x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82415', 'EC Number': '1.3.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite toctd2eACP_c at 0x7fc463adcd60>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite ocdcaACP_c at 0x7fc463aae5b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR181x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82417', 'EC Number': '1.3.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite t3c11vaceACP_c at 0x7fc463adcdc0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite octeACP_c at 0x7fc463aae580>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR40x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C4:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1843',\n",
       "   'KEGG Reaction': 'R04429',\n",
       "   'EC Number': '2.3.1.-'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite but2eACP_c at 0x7fc463adce20>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite butACP_c at 0x7fc463a826a0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR60x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C6:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9658',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68780',\n",
       "   'KEGG Reaction': 'R04955',\n",
       "   'EC Number': '2.3.1.-'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite thex2eACP_c at 0x7fc463adce50>: -1.0,\n",
       "   <Metabolite hexACP_c at 0x7fc463a826d0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EAR80x',\n",
       "  'name': 'Enoyl-[acyl-carrier-protein] reductase (NADH) (n-C8:0)',\n",
       "  'notes': {'BioCyc': 'META:RXN-9659',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68782',\n",
       "   'KEGG Reaction': 'R04958',\n",
       "   'EC Number': '2.3.1.-'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38143_1 at 0x7fc44986ab80>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite toct2eACP_c at 0x7fc463adce80>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite ocACP_c at 0x7fc463a824f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECHH_0',\n",
       "  'name': 'ECHH 0',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68792'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2_c at 0x7fc463a827c0>: -1.0,\n",
       "   <Metabolite fdox_c at 0x7fc448ec8190>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite fdred_c at 0x7fc448ec80a0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH1',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxybutanoyl-CoA)',\n",
       "  'notes': {'EC Number': '4.2.1.55',\n",
       "   'BioCyc': 'META:RXN-11667',\n",
       "   'BioPath Reaction': 'RXN00285;RXN00497',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1863',\n",
       "   'RHEA': '26558;26559;26560;26561',\n",
       "   'KEGG Reaction': 'R03026'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3hbcoa_c at 0x7fc448eef5e0>: -1.0,\n",
       "   <Metabolite b2coa_c at 0x7fc448e1c1f0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH2',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxyhexanoyl-CoA)',\n",
       "  'notes': {'EC Number': '4.2.1.74',\n",
       "   'BioCyc': 'META:RXN-12567',\n",
       "   'BioPath Reaction': 'RXN01234',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1866',\n",
       "   'RHEA': '30547;30548;30549;30550',\n",
       "   'KEGG Reaction': 'R04749'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3hhcoa_c at 0x7fc448eef640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite hx2coa_c at 0x7fc448e1c220>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH3',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxyoctanoyl-CoA)',\n",
       "  'notes': {'EC Number': '4.2.1.74',\n",
       "   'BioCyc': 'META:RXN-14276',\n",
       "   'BioPath Reaction': 'RXN01410',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1867',\n",
       "   'RHEA': '31199;31200;31201;31202',\n",
       "   'KEGG Reaction': 'R04746'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3hocoa_c at 0x7fc448eef670>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite oc2coa_c at 0x7fc448e1c280>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH4',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxydecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '4.2.1.74',\n",
       "   'BioCyc': 'META:RXN-13616',\n",
       "   'BioPath Reaction': 'RXN01141',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1868',\n",
       "   'RHEA': '31191;31192;31193;31194',\n",
       "   'KEGG Reaction': 'R04744'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3hdcoa_c at 0x7fc448e9f280>: -1.0,\n",
       "   <Metabolite dc2coa_c at 0x7fc448e1c2b0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH5',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxydodecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '4.2.1.74',\n",
       "   'BioPath Reaction': 'RXN00749',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1869',\n",
       "   'RHEA': '31075;31076;31077;31078',\n",
       "   'KEGG Reaction': 'R04170'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3hddcoa_c at 0x7fc448e9f1f0>: -1.0,\n",
       "   <Metabolite dd2coa_c at 0x7fc448e1c2e0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH6',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxytetradecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '4.2.1.74',\n",
       "   'BioCyc': 'META:RXN-14273',\n",
       "   'BioPath Reaction': 'RXN00436',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1870',\n",
       "   'RHEA': '31171;31172;31173;31174',\n",
       "   'KEGG Reaction': 'R04740'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3htdcoa_c at 0x7fc448e9f070>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite td2coa_c at 0x7fc45d333c10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH7',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxyhexadecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '4.2.1.74',\n",
       "   'BioCyc': 'META:RXN-14272',\n",
       "   'BioPath Reaction': 'RXN00073',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1871',\n",
       "   'RHEA': '31163;31164;31165;31166',\n",
       "   'KEGG Reaction': 'R04738'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3hhdcoa_c at 0x7fc448eef6a0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite hdd2coa_c at 0x7fc463a40f10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH8',\n",
       "  'name': '3-hydroxyacyl-CoA dehydratase (3-hydroxyoctadecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '35347;35348;35349;35350',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55317',\n",
       "   'KEGG Reaction': 'R07760',\n",
       "   'EC Number': '4.2.1.134'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39339_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39339_1 at 0x7fc44985db80>},\n",
       "  '_metabolites': {<Metabolite 3hodcoa_c at 0x7fc463a82070>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite od2coa_c at 0x7fc448e1c3a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ECOAH9ir',\n",
       "  'name': 'ECOAH9ir',\n",
       "  'notes': {'EC Number': '4.2.1.17',\n",
       "   'BioCyc': 'META:TIGLYLCOA-HYDROXY-RXN',\n",
       "   'BioPath Reaction': 'RXN00987',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1873',\n",
       "   'RHEA': '31119;31120;31121;31122',\n",
       "   'KEGG Reaction': 'R04204'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 2mb2coa_c at 0x7fc463adc760>: -1.0,\n",
       "   <Metabolite 3hmbcoa_c at 0x7fc448eef6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EDA',\n",
       "  'name': '2-dehydro-3-deoxy-phosphogluconate aldolase',\n",
       "  'notes': {'EC Number': '4.1.2.55',\n",
       "   'BioCyc': 'META:KDPGALDOL-RXN',\n",
       "   'BioPath Reaction': 'RXN01469',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55318',\n",
       "   'KEGG Reaction': 'R05605',\n",
       "   'RHEA': '17089;17090;17091;17092',\n",
       "   'UniPathway Reaction': 'UCR05605;UER00829'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37827_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37827_1 at 0x7fc44986abb0>},\n",
       "  '_metabolites': {<Metabolite 2ddg6p_c at 0x7fc448ec8c10>: -1.0,\n",
       "   <Metabolite g3p_c at 0x7fc448ec8c40>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EDTXS1',\n",
       "  'name': 'Endotoxin Synthesis (lauroyl transferase)',\n",
       "  'notes': {'BioCyc': 'META:LAUROYLACYLTRAN-RXN',\n",
       "   'EC Number': '2.3.1.241',\n",
       "   'KEGG Reaction': 'R05146',\n",
       "   'UniPathway Reaction': 'UCR05146;UER00485',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR37169'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ddcaACP_c at 0x7fc463a82580>: -1.0,\n",
       "   <Metabolite kdo2lipid4_c at 0x7fc448eef700>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite kdo2lipid4L_c at 0x7fc448eef730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EDTXS2',\n",
       "  'name': 'Endotoxin Synthesis (myristoyl transferase)',\n",
       "  'notes': {'BioCyc': 'META:MYRISTOYLACYLTRAN-RXN',\n",
       "   'EC Number': '2.3.1.243',\n",
       "   'KEGG Reaction': 'R05075',\n",
       "   'UniPathway Reaction': 'UCR05075;UER00486',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR26675'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite kdo2lipid4L_c at 0x7fc448eef730>: -1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite lipa_c at 0x7fc448eef760>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EDTXS3',\n",
       "  'name': 'Endotoxin Synthesis (palmitoleoyl ACP)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82423'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite hdeACP_c at 0x7fc463a82670>: -1.0,\n",
       "   <Metabolite kdo2lipid4_c at 0x7fc448eef700>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite kdo2lipid4p_c at 0x7fc448eef790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'EDTXS4',\n",
       "  'name': 'Endotoxin Synthesis (myristoyl transferase)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68796'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite kdo2lipid4p_c at 0x7fc448eef790>: -1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite lipa_cold_c at 0x7fc448eef7c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ENLIPAabctex',\n",
       "  'name': 'Phosphoethanolamine lipid A transport via ABC system (periplasm to extracellular)',\n",
       "  'notes': {'RHEA': '35259;35260;35261;35262',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR65718'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37217_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37217_1 at 0x7fc44985d700>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite enlipa_p at 0x7fc448eef820>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite enlipa_e at 0x7fc448eef7f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ENO',\n",
       "  'name': 'Enolase',\n",
       "  'notes': {'EC Number': '4.2.1.11',\n",
       "   'BioCyc': 'META:2PGADEHYDRAT-RXN',\n",
       "   'BioPath Reaction': 'RXN01306',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1898',\n",
       "   'KEGG Reaction': 'R00658',\n",
       "   'RHEA': '10164;10165;10166;10167',\n",
       "   'UniPathway Reaction': 'UCR00658;UER00187'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37912_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37912_1 at 0x7fc44986abe0>},\n",
       "  '_metabolites': {<Metabolite 2pg_c at 0x7fc463adc790>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite pep_c at 0x7fc463aaec40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ETHAt',\n",
       "  'name': 'Ethanolamine transport via diffusion  extracellular ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1924',\n",
       "   'BioCyc': 'META:TRANS-RXN0-476'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite etha_e at 0x7fc448eefa00>: -1.0,\n",
       "   <Metabolite etha_c at 0x7fc448eef9d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ETOHtex',\n",
       "  'name': 'Ethanol transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1930',\n",
       "   'BioCyc': 'META:TRANS-RXN0-546'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite etoh_e at 0x7fc448eefa30>: -1.0,\n",
       "   <Metabolite etoh_p at 0x7fc448eefa60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ETOHtrpp',\n",
       "  'name': 'Ethanol reversible transport via diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1930',\n",
       "   'BioCyc': 'META:TRANS-RXN0-546'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite etoh_p at 0x7fc448eefa60>: -1.0,\n",
       "   <Metabolite etoh_c at 0x7fc463aaefa0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'F6PP',\n",
       "  'name': 'D-fructose 6-phosphate phosphatase',\n",
       "  'notes': {'RHEA': '28558;28559;28560;28561',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR79325'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37643_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37643_1 at 0x7fc44986ac10>},\n",
       "  '_metabolites': {<Metabolite f6p_c at 0x7fc448e49430>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fru_c at 0x7fc448eefac0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAE100',\n",
       "  'name': 'Fatty-acid-CoA thioesterase (decanoate)',\n",
       "  'notes': {'RHEA': '40059;40060;40061;40062',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1984',\n",
       "   'EC Number': '3.1.2.2',\n",
       "   'BioCyc': 'META:RXN-9628'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite dcacoa_c at 0x7fc463aae940>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite dca_c at 0x7fc463aae5e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAE140',\n",
       "  'name': 'Fatty-acid-CoA thioesterase (tetradecanoate)',\n",
       "  'notes': {'RHEA': '40119;40120;40121;40122',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1986',\n",
       "   'EC Number': '3.1.2.2',\n",
       "   'BioCyc': 'META:RXN-9626'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite tdcoa_c at 0x7fc463aaea60>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ttdca_c at 0x7fc463a6dc70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAE141',\n",
       "  'name': 'Fatty-acid-CoA thioesterase (tetradecenoate)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68847'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite tdecoa_c at 0x7fc448ec8490>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ttdcea_c at 0x7fc463a6dd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAE160',\n",
       "  'name': 'Fatty-acid-CoA thioesterase (hexadecanoate)',\n",
       "  'notes': {'RHEA': '16645;16646;16647;16648',\n",
       "   'BioCyc': 'META:PALMITOYL-COA-HYDROLASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1988',\n",
       "   'KEGG Reaction': 'R01274',\n",
       "   'EC Number': '3.1.2.22'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite pmtcoa_c at 0x7fc463a821c0>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAE161',\n",
       "  'name': 'Fatty-acid-CoA thioesterase (hexadecenoate)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82688'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hdcoa_c at 0x7fc448e1cd90>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAE180',\n",
       "  'name': 'Fatty-acid-CoA thioesterase (octadecanoate)',\n",
       "  'notes': {'RHEA': '30139;30140;30141;30142',\n",
       "   'BioCyc': 'META:RXN-9624',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1990',\n",
       "   'KEGG Reaction': 'R08174',\n",
       "   'EC Number': '3.1.2.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite stcoa_c at 0x7fc448e1c370>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ocdca_c at 0x7fc463a6de20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAE80',\n",
       "  'name': 'Fatty-acid-CoA thioesterase (octanoate)',\n",
       "  'notes': {'RHEA': '30143;30144;30145;30146',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1993'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite occoa_c at 0x7fc463aae8e0>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite octa_c at 0x7fc463aae610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL100t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (decanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1995'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite dca_p at 0x7fc448ec8a60>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite dcacoa_c at 0x7fc463aae940>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL120t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (dodecanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1997'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite ddca_p at 0x7fc448ec8bb0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite ddcacoa_c at 0x7fc463aae9d0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL140',\n",
       "  'name': 'Fatty acid  CoA ligase  tetradecanoate ',\n",
       "  'notes': {'RHEA': '33619;33620;33621;33622',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR1998'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite ttdca_c at 0x7fc463a6dc70>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite tdcoa_c at 0x7fc463aaea60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL140t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (tetradecanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR1999'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ttdca_p at 0x7fc448eefaf0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite tdcoa_c at 0x7fc463aaea60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL141t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (tetradecenoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68851'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ttdcea_p at 0x7fc448eefb20>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite tdecoa_c at 0x7fc448ec8490>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL160',\n",
       "  'name': 'Fatty acid  CoA ligase  hexadecanoate ',\n",
       "  'notes': {'RHEA': '30751;30752;30753;30754',\n",
       "   'BioCyc': 'META:RXN-9623',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2003',\n",
       "   'KEGG Reaction': 'R01280',\n",
       "   'EC Number': '6.2.1.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite pmtcoa_c at 0x7fc463a821c0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL160t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (hexadecanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2004'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite hdca_p at 0x7fc448eefb50>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pmtcoa_c at 0x7fc463a821c0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL161',\n",
       "  'name': 'Fatty acid  CoA ligase  hexadecenoate ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68852'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite hdcoa_c at 0x7fc448e1cd90>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL161t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (hexadecenoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68853'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite hdcea_p at 0x7fc448eefb80>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hdcoa_c at 0x7fc448e1cd90>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL180t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (octadecanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2009'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ocdca_p at 0x7fc448eefbb0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite stcoa_c at 0x7fc448e1c370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL181',\n",
       "  'name': 'Fatty acid  CoA ligase  octadecenoate ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82695'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite ocdcea_c at 0x7fc463a6de80>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite odecoa_c at 0x7fc448e1cdc0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL181t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (octadecenoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82699'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ocdcea_p at 0x7fc448eefbe0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite odecoa_c at 0x7fc448e1cdc0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL40It2pp',\n",
       "  'name': 'Fatty-acid--CoA ligase (isobutanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82710'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ibt_p at 0x7fc448eefc10>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite ibcoa_c at 0x7fc448e1c400>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL50It2pp',\n",
       "  'name': 'Fatty-acid--CoA ligase (isovalerate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82712'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite 3mb_p at 0x7fc463adcfa0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite ivcoa_c at 0x7fc448e9f250>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL60t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (hexanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2040'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite hxa_p at 0x7fc448eefc40>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hxcoa_c at 0x7fc463aae880>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FACOAL80t2pp',\n",
       "  'name': 'Fatty-acid-CoA ligase (octanoate transport via vectoral Co-A coupling)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2042'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36256_1 or EDU39252_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36256_1 at 0x7fc449820220>,\n",
       "   <Gene EDU39252_1 at 0x7fc44986ac40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite octa_p at 0x7fc448eefc70>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite occoa_c at 0x7fc463aae8e0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FADMQOR',\n",
       "  'name': 'Menaquinone:FAD oxidoreductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82715'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36228_1 or EDU39259_1 or (EDU38860_1 and EDU39259_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36228_1 at 0x7fc44986acd0>,\n",
       "   <Gene EDU38860_1 at 0x7fc44986ac70>,\n",
       "   <Gene EDU39259_1 at 0x7fc44986aca0>},\n",
       "  '_metabolites': {<Metabolite fadh2_c at 0x7fc463a82bb0>: -1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: 1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FAO3',\n",
       "  'name': 'FAO3',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82740'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -9.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: -8.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -8.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -8.0,\n",
       "   <Metabolite ocdca_c at 0x7fc463a6de20>: -1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: 9.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite fadh2_c at 0x7fc463a82bb0>: 8.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 8.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 8.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FBA',\n",
       "  'name': 'Fructose-bisphosphate aldolase',\n",
       "  'notes': {'RHEA': '14729;14730;14731;14732',\n",
       "   'EC Number': '4.1.2.13',\n",
       "   'KEGG Reaction': 'R01068',\n",
       "   'UniPathway Reaction': 'UCR01068;UER00183',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR13125'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37949_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37949_1 at 0x7fc44986ad00>},\n",
       "  '_metabolites': {<Metabolite fdp_c at 0x7fc448eef0d0>: -1.0,\n",
       "   <Metabolite dhap_c at 0x7fc448e49d90>: 1.0,\n",
       "   <Metabolite g3p_c at 0x7fc448ec8c40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FBP',\n",
       "  'name': 'Fructose-bisphosphatase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82809',\n",
       "   'KEGG Reaction': 'R00762',\n",
       "   'EC Number': '3.1.3.11'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36223_1 or EDU38341_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36223_1 at 0x7fc44986ad30>,\n",
       "   <Gene EDU38341_1 at 0x7fc44986ad60>},\n",
       "  '_metabolites': {<Metabolite fdp_c at 0x7fc448eef0d0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite f6p_c at 0x7fc448e49430>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FCLT',\n",
       "  'name': 'Ferrochelatase',\n",
       "  'notes': {'EC Number': '4.99.1.1',\n",
       "   'BioCyc': 'META:PROTOHEMEFERROCHELAT-RXN',\n",
       "   'BioPath Reaction': 'RXN00238',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR70157',\n",
       "   'KEGG Reaction': 'R00310',\n",
       "   'RHEA': '22584;22585;22586;22587',\n",
       "   'UniPathway Reaction': 'UCR00310;UER00325'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite fe2_c at 0x7fc448e74490>: -1.0,\n",
       "   <Metabolite ppp9_c at 0x7fc448eefd30>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite pheme_c at 0x7fc448eefd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FDMO2',\n",
       "  'name': 'FMNH2-dependent monooxygenase (methanesulfonate)',\n",
       "  'notes': {'RHEA': '29307;29308;29309;29310',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR82816',\n",
       "   'KEGG Reaction': 'R10206',\n",
       "   'EC Number': '1.14.14.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite fmnh2_c at 0x7fc448e744c0>: -1.0,\n",
       "   <Metabolite mso3_c at 0x7fc448eefdc0>: -1.0,\n",
       "   <Metabolite o2_c at 0x7fc463a6d610>: -1.0,\n",
       "   <Metabolite fald_c at 0x7fc463adc4c0>: 1.0,\n",
       "   <Metabolite fmn_c at 0x7fc448e1c550>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite so3_c at 0x7fc463a82850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE2abc',\n",
       "  'name': 'Iron (II) transport via ABC system',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2211'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37165_1 or EDU37323_1 or EDU37516_1 or EDU38659_1 or (EDU38887_1 and EDU38888_1) or (EDU38953_1 and EDU38954_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37165_1 at 0x7fc44986ae80>,\n",
       "   <Gene EDU37323_1 at 0x7fc44986ae20>,\n",
       "   <Gene EDU37516_1 at 0x7fc44986adf0>,\n",
       "   <Gene EDU38659_1 at 0x7fc44986aeb0>,\n",
       "   <Gene EDU38887_1 at 0x7fc44986aee0>,\n",
       "   <Gene EDU38888_1 at 0x7fc44986adc0>,\n",
       "   <Gene EDU38953_1 at 0x7fc44986af10>,\n",
       "   <Gene EDU38954_1 at 0x7fc44986ae50>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fe2_e at 0x7fc448eefe20>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE2abcpp',\n",
       "  'name': 'Iron (II) transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2211'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38451_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38451_1 at 0x7fc44986af40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fe2_p at 0x7fc448eefe50>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE2Gabcpp',\n",
       "  'name': 'Iron (II) transport via ABC system (GTP) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82823'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35859_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35859_1 at 0x7fc44986ad90>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fe2_p at 0x7fc448eefe50>: -1.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: 1.0,\n",
       "   <Metabolite gdp_c at 0x7fc463adc850>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE2t3pp',\n",
       "  'name': 'Iron (Fe+2) transport out via proton antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2213',\n",
       "   'BioCyc': 'META:RXN0-2421;META:RXN0-6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39203_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39203_1 at 0x7fc44985d2b0>},\n",
       "  '_metabolites': {<Metabolite fe2_c at 0x7fc448e74490>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite fe2_p at 0x7fc448eefe50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE2t4pp',\n",
       "  'name': 'Iron(II) transport out via antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82825'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37535_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37535_1 at 0x7fc44985d2e0>},\n",
       "  '_metabolites': {<Metabolite fe2_c at 0x7fc448e74490>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite k_p at 0x7fc448e9f550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite k_c at 0x7fc448e74910>: 1.0,\n",
       "   <Metabolite fe2_p at 0x7fc448eefe50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE2tex',\n",
       "  'name': 'Iron (II) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2212',\n",
       "   'BioCyc': 'META:TRANS-RXN-8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite fe2_e at 0x7fc448eefe20>: -1.0,\n",
       "   <Metabolite fe2_p at 0x7fc448eefe50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE2tpp',\n",
       "  'name': 'Iron (+2) transport in via permease (no H+)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2212',\n",
       "   'BioCyc': 'META:TRANS-RXN-8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37639_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37639_1 at 0x7fc44985d310>},\n",
       "  '_metabolites': {<Metabolite fe2_p at 0x7fc448eefe50>: -1.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE3abc',\n",
       "  'name': 'FE3abc',\n",
       "  'notes': {'RHEA': '12332;12333;12334;12335',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2227',\n",
       "   'EC Number': '3.6.3.30',\n",
       "   'BioCyc': 'META:3.6.3.30-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35947_1 or EDU35948_1 or EDU35949_1 or EDU37165_1 or EDU37448_1 or EDU37449_1 or EDU37816_1 or EDU38820_1 or EDU39397_1 or (EDU35947_1 and EDU35948_1) or (EDU35947_1 and EDU35949_1) or (EDU35947_1 and EDU37448_1) or (EDU35947_1 and EDU37449_1) or (EDU35947_1 and EDU37816_1) or (EDU35947_1 and EDU38820_1) or (EDU35948_1 and EDU35949_1) or (EDU35948_1 and EDU37449_1) or (EDU35948_1 and EDU38820_1) or (EDU35949_1 and EDU37448_1) or (EDU35949_1 and EDU37816_1) or (EDU37448_1 and EDU37449_1) or (EDU37448_1 and EDU38820_1) or (EDU37449_1 and EDU37816_1) or (EDU37816_1 and EDU38820_1) or (EDU35947_1 and EDU35948_1 and EDU35949_1) or (EDU35947_1 and EDU35948_1 and EDU37449_1) or (EDU35947_1 and EDU35948_1 and EDU38820_1) or (EDU35947_1 and EDU35949_1 and EDU37448_1) or (EDU35947_1 and EDU35949_1 and EDU37816_1) or (EDU35947_1 and EDU37448_1 and EDU37449_1) or (EDU35947_1 and EDU37448_1 and EDU38820_1) or (EDU35947_1 and EDU37449_1 and EDU37816_1) or (EDU35947_1 and EDU37816_1 and EDU38820_1) or (EDU37449_1 and EDU37816_1 and EDU38821_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35947_1 at 0x7fc4498790a0>,\n",
       "   <Gene EDU35948_1 at 0x7fc44986afd0>,\n",
       "   <Gene EDU35949_1 at 0x7fc4498790d0>,\n",
       "   <Gene EDU37165_1 at 0x7fc44986ae80>,\n",
       "   <Gene EDU37448_1 at 0x7fc449879130>,\n",
       "   <Gene EDU37449_1 at 0x7fc44986afa0>,\n",
       "   <Gene EDU37816_1 at 0x7fc449879040>,\n",
       "   <Gene EDU38820_1 at 0x7fc44986a070>,\n",
       "   <Gene EDU38821_1 at 0x7fc44986af70>,\n",
       "   <Gene EDU39397_1 at 0x7fc44986a100>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fe3_e at 0x7fc448e74460>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite fe3_c at 0x7fc448e9ff70>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE3abcpp',\n",
       "  'name': 'Iron (III) transport via ABC system (periplasm to cytoplasm)',\n",
       "  'notes': {'RHEA': '12332;12333;12334;12335',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2227',\n",
       "   'EC Number': '3.6.3.30',\n",
       "   'BioCyc': 'META:3.6.3.30-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39370_1 or (EDU37662_1 and EDU39370_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37662_1 at 0x7fc449879160>,\n",
       "   <Gene EDU39370_1 at 0x7fc44983bfd0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fe3_p at 0x7fc448eefe80>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite fe3_c at 0x7fc448e9ff70>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE3Gabcpp',\n",
       "  'name': 'Iron (III) transport via ABC system (GTP) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82829'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35859_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35859_1 at 0x7fc44986ad90>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite fe3_p at 0x7fc448eefe80>: -1.0,\n",
       "   <Metabolite fe3_c at 0x7fc448e9ff70>: 1.0,\n",
       "   <Metabolite gdp_c at 0x7fc463adc850>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FE3tex',\n",
       "  'name': 'Iron (III) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2228'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite fe3_e at 0x7fc448e74460>: -1.0,\n",
       "   <Metabolite fe3_p at 0x7fc448eefe80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FEDCabc',\n",
       "  'name': 'FEDCabc',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2236'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37448_1 or EDU37652_1 or EDU38532_1 or (EDU37448_1 and EDU37817_1) or (EDU37652_1 and EDU37817_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37448_1 at 0x7fc449879130>,\n",
       "   <Gene EDU37652_1 at 0x7fc44985da90>,\n",
       "   <Gene EDU37817_1 at 0x7fc449879190>,\n",
       "   <Gene EDU38532_1 at 0x7fc4498791c0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite cit_e at 0x7fc448e9f970>: -2.0,\n",
       "   <Metabolite fe3_e at 0x7fc448e74460>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite cit_c at 0x7fc463aaef10>: 2.0,\n",
       "   <Metabolite fe3_c at 0x7fc448e9ff70>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FEENTER2tpp',\n",
       "  'name': 'Enterochelin transport2 via ABC system periplasm ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82838'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36887_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36887_1 at 0x7fc4498791f0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite enter_c at 0x7fc448eef850>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite enter_p at 0x7fc448eef8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FEENTERabcpp',\n",
       "  'name': 'Fe-enterobactin transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68984',\n",
       "   'EC Number': '3.6.3.34',\n",
       "   'BioCyc': 'META:ABC-10-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37514_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37514_1 at 0x7fc449879220>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite feenter_p at 0x7fc448eefeb0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite feenter_c at 0x7fc448eef880>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FEENTERR2',\n",
       "  'name': 'Fe-enterobactin reduction (Fe(III)-unloading)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82840'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite feenter_c at 0x7fc448eef880>: -2.0,\n",
       "   <Metabolite fmnh2_c at 0x7fc448e744c0>: -1.0,\n",
       "   <Metabolite enter_c at 0x7fc448eef850>: 2.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: 2.0,\n",
       "   <Metabolite fmn_c at 0x7fc448e1c550>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FEENTERtex',\n",
       "  'name': 'Enterochelin transport (secretion periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2242'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite enter_p at 0x7fc448eef8e0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite enter_e at 0x7fc448eefee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FEENTERtonex',\n",
       "  'name': 'Fe-enterobactin transport via ton system (extracellular)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR68986'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite feenter_e at 0x7fc448eeffa0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite feenter_p at 0x7fc448eefeb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FERIRDe',\n",
       "  'name': 'Quinon:iron(III) oxidoreductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82845'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite fe3_e at 0x7fc448e74460>: -2.0,\n",
       "   <Metabolite qh2_c at 0x7fc448ec86d0>: -1.0,\n",
       "   <Metabolite fe2_e at 0x7fc448eefe20>: 2.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: 2.0,\n",
       "   <Metabolite q_c at 0x7fc448ec86a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FEROpp',\n",
       "  'name': 'Ferroxidase',\n",
       "  'notes': {'EC Number': '1.16.3.1',\n",
       "   'BioCyc': 'META:RXN0-1483',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2250',\n",
       "   'RHEA': '11148;11149;11150;11151;11436;11437;11438;11439',\n",
       "   'KEGG Reaction': 'R00078'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38903_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38903_1 at 0x7fc44985dc10>},\n",
       "  '_metabolites': {<Metabolite fe2_p at 0x7fc448eefe50>: -4.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -4.0,\n",
       "   <Metabolite o2_p at 0x7fc463a82130>: -1.0,\n",
       "   <Metabolite fe3_p at 0x7fc448eefe80>: 4.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FESD1s',\n",
       "  'name': 'Iron-sulfur cluster damage (peroxide, spontaneous)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82859'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite 4fe4s_c at 0x7fc448eeff10>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite h2o2_c at 0x7fc463a6d580>: -1.0,\n",
       "   <Metabolite 3fe4s_c at 0x7fc448eeffd0>: 2.0,\n",
       "   <Metabolite fe3_c at 0x7fc448e9ff70>: 2.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FESR',\n",
       "  'name': 'Iron-sulfur cluster repair',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82861'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38509_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38509_1 at 0x7fc449879250>},\n",
       "  '_metabolites': {<Metabolite 3fe4s_c at 0x7fc448eeffd0>: -1.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: -1.0,\n",
       "   <Metabolite 4fe4s_c at 0x7fc448eeff10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FFSD',\n",
       "  'name': 'Beta-fructofuranosidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82862', 'EC Number': '3.2.1.26'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite suc6p_c at 0x7fc448e74c10>: -1.0,\n",
       "   <Metabolite fru_c at 0x7fc448eefac0>: 1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FFSD1r',\n",
       "  'name': 'Beta-fructofuranosidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82863'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37844_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37844_1 at 0x7fc449879280>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite tre6p_c at 0x7fc448eeff40>: -1.0,\n",
       "   <Metabolite fru_c at 0x7fc448eefac0>: 1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FLDO',\n",
       "  'name': 'NAD(P)H-flavin oxidoreductase',\n",
       "  'notes': {'RHEA': '21624;21625;21626;21627',\n",
       "   'BioCyc': 'META:FMNREDUCT-RXN;META:RXN-12444',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR68997',\n",
       "   'KEGG Reaction': 'R05706',\n",
       "   'EC Number': '1.5.1.39'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38697_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38697_1 at 0x7fc4498792b0>},\n",
       "  '_metabolites': {<Metabolite fmnh2_c at 0x7fc448e744c0>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite fmn_c at 0x7fc448e1c550>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FLNDPR2r',\n",
       "  'name': 'Flavodoxin : nadp reductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82865'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite flxr_c at 0x7fc448e9f6d0>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -2.0,\n",
       "   <Metabolite flxso_c at 0x7fc448e9f700>: 2.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FMNAT',\n",
       "  'name': 'FMN adenylyltransferase',\n",
       "  'notes': {'EC Number': '2.7.7.2',\n",
       "   'BioCyc': 'META:FADSYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81572',\n",
       "   'KEGG Reaction': 'R00161',\n",
       "   'RHEA': '17237;17238;17239;17240',\n",
       "   'UniPathway Reaction': 'UCR00161;UER00407'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36420_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36420_1 at 0x7fc4498792e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite fmn_c at 0x7fc448e1c550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FMNRx',\n",
       "  'name': 'FMN reductase',\n",
       "  'notes': {'RHEA': '21620;21621;21622;21623',\n",
       "   'BioCyc': 'META:RXN-9510',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69003',\n",
       "   'KEGG Reaction': 'R05705',\n",
       "   'EC Number': '1.5.1.42'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36612_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36612_1 at 0x7fc449879310>},\n",
       "  '_metabolites': {<Metabolite fmn_c at 0x7fc448e1c550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite fmnh2_c at 0x7fc448e744c0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FNOR',\n",
       "  'name': 'Ferredoxin---NADP+ reductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82868'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36072_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36072_1 at 0x7fc449879340>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite fdxrd_c at 0x7fc448e9f340>: -2.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0,\n",
       "   <Metabolite fdxo_2_2_c at 0x7fc448ec8c70>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FOLD3',\n",
       "  'name': 'FOLD3',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82870'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38669_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38669_1 at 0x7fc449879370>},\n",
       "  '_metabolites': {<Metabolite 4abz_c at 0x7fc463a828e0>: -1.0,\n",
       "   <Metabolite 2ahhmd_c at 0x7fc448eeff70>: -1.0,\n",
       "   <Metabolite dhpt_c at 0x7fc448ec8df0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FORCT',\n",
       "  'name': 'Formyl-CoA Transferase',\n",
       "  'notes': {'EC Number': '2.8.3.16',\n",
       "   'BioCyc': 'META:RXN0-1382',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR8463',\n",
       "   'KEGG Reaction': 'R07290',\n",
       "   'RHEA': '16545;16546;16547;16548',\n",
       "   'UniPathway Reaction': 'UCR07290;UER00598'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36588_1 or EDU39253_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36588_1 at 0x7fc4498793d0>,\n",
       "   <Gene EDU39253_1 at 0x7fc4498793a0>},\n",
       "  '_metabolites': {<Metabolite forcoa_c at 0x7fc448eefd60>: -1.0,\n",
       "   <Metabolite oxa_c at 0x7fc448e9fc70>: -1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite oxalcoa_c at 0x7fc45d3339a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FORtex',\n",
       "  'name': 'Formate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2277',\n",
       "   'BioCyc': 'META:TRANS-RXN-1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite for_e at 0x7fc448e9fbe0>: -1.0,\n",
       "   <Metabolite for_p at 0x7fc448eefd90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FRNDPR2r_1',\n",
       "  'name': 'Ferredoxin : nadp reductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82878'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36559_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36559_1 at 0x7fc449879400>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -2.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 2.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FRUK',\n",
       "  'name': 'Fructose-1-phosphate kinase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2301', 'EC Number': '2.7.1.56'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37073_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37073_1 at 0x7fc449879430>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite f1p_c at 0x7fc448eefca0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite fdp_c at 0x7fc448eef0d0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FRUpts',\n",
       "  'name': 'D-fructose transport via PEP:Pyr PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82881'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1 or EDU36819_1 or EDU36820_1 or (EDU36437_1 and EDU39079_1) or (EDU36437_1 and EDU36819_1 and EDU39079_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU36819_1 at 0x7fc449879460>,\n",
       "   <Gene EDU36820_1 at 0x7fc449879490>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite fru_e at 0x7fc448f20130>: -1.0,\n",
       "   <Metabolite f1p_c at 0x7fc448eefca0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FRUpts2',\n",
       "  'name': 'Fructose transport via PEP:Pyr PTS (f6p generating)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82882'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite fru_e at 0x7fc448f20130>: -1.0,\n",
       "   <Metabolite f6p_c at 0x7fc448e49430>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FRUpts2pp',\n",
       "  'name': 'Fructose transport via PEP:Pyr PTS (f6p generating) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82882'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite fru_p at 0x7fc448f20160>: -1.0,\n",
       "   <Metabolite f6p_c at 0x7fc448e49430>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FRUptspp',\n",
       "  'name': 'D-fructose transport via PEP:Pyr PTS (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82881'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite fru_p at 0x7fc448f20160>: -1.0,\n",
       "   <Metabolite f1p_c at 0x7fc448eefca0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FRUtex',\n",
       "  'name': 'D-fructose transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82533'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite fru_e at 0x7fc448f20130>: -1.0,\n",
       "   <Metabolite fru_p at 0x7fc448f20160>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FTCD',\n",
       "  'name': 'FTCD',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82885', 'EC Number': '4.3.1.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite 5forthf_c at 0x7fc448f20190>: -1.0,\n",
       "   <Metabolite methf_c at 0x7fc45d380ac0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FTHFCL',\n",
       "  'name': '5 formethyltetrahydrofolate cyclo ligase',\n",
       "  'notes': {'RHEA': '10488;10489;10490;10491',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2316',\n",
       "   'KEGG Reaction': 'R02301',\n",
       "   'EC Number': '6.3.3.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36233_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36233_1 at 0x7fc4498794c0>},\n",
       "  '_metabolites': {<Metabolite 5fthf_c at 0x7fc463aae100>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite methf_c at 0x7fc45d380ac0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FTHFLi',\n",
       "  'name': 'Formate-tetrahydrofolate ligase',\n",
       "  'notes': {'RHEA': '20221;20222;20223;20224',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2319',\n",
       "   'KEGG Reaction': 'R00943',\n",
       "   'EC Number': '6.3.4.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38224_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38224_1 at 0x7fc4498794f0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: -1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: -1.0,\n",
       "   <Metabolite 10fthf_c at 0x7fc463a377c0>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FUM',\n",
       "  'name': 'Fumarase',\n",
       "  'notes': {'EC Number': '4.2.1.2',\n",
       "   'BioCyc': 'META:FUMHYDR-RXN',\n",
       "   'BioPath Reaction': 'RXN00540',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2338',\n",
       "   'KEGG Reaction': 'R01082',\n",
       "   'RHEA': '12460;12461;12462;12463',\n",
       "   'UniPathway Reaction': 'UCR01082;UER01007'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39075_1 and EDU39076_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39075_1 at 0x7fc449879550>,\n",
       "   <Gene EDU39076_1 at 0x7fc449879520>},\n",
       "  '_metabolites': {<Metabolite fum_c at 0x7fc448e1cd00>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mal__L_c at 0x7fc448e49730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FUMt2_2pp',\n",
       "  'name': 'Fumarate transport via proton symport (2 H) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2343',\n",
       "   'BioCyc': 'META:TRANS-RXN-121B'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39112_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39112_1 at 0x7fc44983beb0>},\n",
       "  '_metabolites': {<Metabolite fum_p at 0x7fc448f20040>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -2.0,\n",
       "   <Metabolite fum_c at 0x7fc448e1cd00>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FUMt2r',\n",
       "  'name': 'Fumarate reversible transport via symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2345'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38390_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38390_1 at 0x7fc449879580>},\n",
       "  '_metabolites': {<Metabolite fum_e at 0x7fc448f20070>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite fum_c at 0x7fc448e1cd00>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'FUMtex',\n",
       "  'name': 'Fumarate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2346',\n",
       "   'BioCyc': 'META:TRANS-RXN0-553'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite fum_e at 0x7fc448f20070>: -1.0,\n",
       "   <Metabolite fum_p at 0x7fc448f20040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G1PACT',\n",
       "  'name': 'Glucosamine-1-phosphate N-acetyltransferase',\n",
       "  'notes': {'EC Number': '2.3.1.157',\n",
       "   'BioCyc': 'META:2.3.1.157-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2417',\n",
       "   'KEGG Reaction': 'R05332',\n",
       "   'RHEA': '13725;13726;13727;13728',\n",
       "   'UniPathway Reaction': 'UCR05332;UER00532'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38203_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38203_1 at 0x7fc4498795b0>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite gam1p_c at 0x7fc448f200a0>: -1.0,\n",
       "   <Metabolite acgam1p_c at 0x7fc463aaed60>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G1SAT',\n",
       "  'name': 'Glutamate-1-semialdehyde aminotransferase',\n",
       "  'notes': {'EC Number': '5.4.3.8',\n",
       "   'BioCyc': 'META:GSAAMINOTRANS-RXN',\n",
       "   'BioPath Reaction': 'RXN00998',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2422',\n",
       "   'KEGG Reaction': 'R02272',\n",
       "   'RHEA': '14265;14266;14267;14268',\n",
       "   'UniPathway Reaction': 'UCR02272;UER00317'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite glu1sa_c at 0x7fc448f20100>: -1.0,\n",
       "   <Metabolite 5aop_c at 0x7fc463aae130>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PAT120',\n",
       "  'name': 'Glycerol-3-phosphate acyltransferase (C12:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR55513', 'EC Number': '2.3.1.15'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36377_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36377_1 at 0x7fc4498795e0>},\n",
       "  '_metabolites': {<Metabolite ddcaACP_c at 0x7fc463a82580>: -1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite 1ddecg3p_c at 0x7fc448e49070>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PAT140',\n",
       "  'name': 'Glycerol-3-phosphate acyltransferase (C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69065', 'EC Number': '2.3.1.15'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36377_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36377_1 at 0x7fc4498795e0>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite myrsACP_c at 0x7fc463a825e0>: -1.0,\n",
       "   <Metabolite 1tdecg3p_c at 0x7fc448e49100>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PAT160',\n",
       "  'name': 'Glycerol-3-phosphate acyltransferase (C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69067', 'EC Number': '2.3.1.15'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36377_1 or EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36377_1 at 0x7fc4498795e0>,\n",
       "   <Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite palmACP_c at 0x7fc463a82640>: -1.0,\n",
       "   <Metabolite 1hdecg3p_c at 0x7fc448e49250>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PAT161',\n",
       "  'name': 'Glycerol-3-phosphate acyltransferase (C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82925', 'EC Number': '2.3.1.15'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36377_1 or EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36377_1 at 0x7fc4498795e0>,\n",
       "   <Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite hdeACP_c at 0x7fc463a82670>: -1.0,\n",
       "   <Metabolite 1hdec9eg3p_c at 0x7fc448e492b0>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PAT180',\n",
       "  'name': 'Glycerol-3-phosphate acyltransferase (C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82926', 'EC Number': '2.3.1.15'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36377_1 or EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36377_1 at 0x7fc4498795e0>,\n",
       "   <Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite ocdcaACP_c at 0x7fc463aae5b0>: -1.0,\n",
       "   <Metabolite 1odecg3p_c at 0x7fc448e49310>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PAT181',\n",
       "  'name': 'Glycerol-3-phosphate acyltransferase (C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82928', 'EC Number': '2.3.1.15'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36377_1 or EDU36623_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36377_1 at 0x7fc4498795e0>,\n",
       "   <Gene EDU36623_1 at 0x7fc44983b760>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite octeACP_c at 0x7fc463aae580>: -1.0,\n",
       "   <Metabolite 1odec11eg3p_c at 0x7fc448e49370>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PD1',\n",
       "  'name': 'G3PD1',\n",
       "  'notes': {'EC Number': '1.1.1.94',\n",
       "   'BioCyc': 'META:1.1.1.8-RXN',\n",
       "   'BioPath Reaction': 'RXN00456',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2438',\n",
       "   'RHEA': '11092;11093;11094;11095',\n",
       "   'KEGG Reaction': 'R00842'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38945_1 at 0x7fc449879610>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite dhap_c at 0x7fc448e49d90>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PD2',\n",
       "  'name': 'Glycerol-3-phosphate dehydrogenase (NADP)',\n",
       "  'notes': {'RHEA': '11096;11097;11098;11099',\n",
       "   'BioCyc': 'META:GLYC3PDEHYDROGBIOSYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2439',\n",
       "   'KEGG Reaction': 'R00844',\n",
       "   'EC Number': '1.1.1.94'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36312_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36312_1 at 0x7fc449879640>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite dhap_c at 0x7fc448e49d90>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PEtex',\n",
       "  'name': 'Glycero-3-phosphoethanolamine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2445'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite g3pe_e at 0x7fc448f20220>: -1.0,\n",
       "   <Metabolite g3pe_p at 0x7fc448f201f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PGtex',\n",
       "  'name': 'Glycerophoglycerol transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2447'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite g3pg_e at 0x7fc448f202b0>: -1.0,\n",
       "   <Metabolite g3pg_p at 0x7fc448f20280>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G3PT',\n",
       "  'name': 'Glycerol-3-phosphatase',\n",
       "  'notes': {'BioCyc': 'META:RXN-14965',\n",
       "   'BioPath Reaction': 'RXN01129',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69078',\n",
       "   'KEGG Reaction': 'R00841',\n",
       "   'EC Number': '3.1.3.21'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37643_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37643_1 at 0x7fc44986ac10>},\n",
       "  '_metabolites': {<Metabolite glyc3p_c at 0x7fc448e742e0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glyc_c at 0x7fc448e49b80>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G5SADs',\n",
       "  'name': 'L-glutamate 5-semialdehyde dehydratase (spontaneous)',\n",
       "  'notes': {'BioCyc': 'META:SPONTPRO-RXN',\n",
       "   'BioPath Reaction': 'RXN01473',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69079',\n",
       "   'KEGG Reaction': 'R03314',\n",
       "   'RHEA': '28234;28235;28236;28237',\n",
       "   'UniPathway Reaction': 'UCR03314'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite glu5sa_c at 0x7fc448f20310>: -1.0,\n",
       "   <Metabolite 1pyr5c_c at 0x7fc448f202e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G6PDA',\n",
       "  'name': 'Glucosamine-6-phosphate deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82944', 'EC Number': '3.5.99.6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36011_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36011_1 at 0x7fc449879670>},\n",
       "  '_metabolites': {<Metabolite gam6p_c at 0x7fc463aaed30>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite f6p_c at 0x7fc448e49430>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G6PI',\n",
       "  'name': 'Glucose 6 phosphate isomerase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69084', 'EC Number': '5.3.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39266_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39266_1 at 0x7fc4498796a0>},\n",
       "  '_metabolites': {<Metabolite g6p_c at 0x7fc463aae340>: -1.0,\n",
       "   <Metabolite g6p_B_c at 0x7fc448f20340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'G6PP',\n",
       "  'name': 'Glucose-6-phosphate phosphatase',\n",
       "  'notes': {'RHEA': '16689;16690;16691;16692',\n",
       "   'BioCyc': 'META:RXN66-526',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69085',\n",
       "   'KEGG Reaction': 'R00303',\n",
       "   'EC Number': '3.1.3.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37643_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37643_1 at 0x7fc44986ac10>},\n",
       "  '_metabolites': {<Metabolite g6p_c at 0x7fc463aae340>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glc__D_c at 0x7fc463a6d4f0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GALabc',\n",
       "  'name': 'D-galactose transport via ABC system',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR50923'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37922_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37922_1 at 0x7fc44986a940>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gal_e at 0x7fc448f203d0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gal_c at 0x7fc448ec8ca0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GALabcpp',\n",
       "  'name': 'D-galactose transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR50923'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37921_1 or EDU37922_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37921_1 at 0x7fc449879730>,\n",
       "   <Gene EDU37922_1 at 0x7fc44986a940>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gal_p at 0x7fc448f203a0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gal_c at 0x7fc448ec8ca0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GALKr',\n",
       "  'name': 'Galactokinase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR32659', 'EC Number': '2.7.1.6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite gal_c at 0x7fc448ec8ca0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gal1p_c at 0x7fc448f20400>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GALTpts',\n",
       "  'name': 'Galactitol transport via PEP:Pyr PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2532'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite galt_e at 0x7fc448f20460>: -1.0,\n",
       "   <Metabolite galt1p_c at 0x7fc448f20430>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GALUi',\n",
       "  'name': 'UTP-glucose-1-phosphate uridylyltransferase (irreversible)',\n",
       "  'notes': {'EC Number': '2.7.7.9',\n",
       "   'BioCyc': 'META:GLUC1PURIDYLTRANS-RXN',\n",
       "   'BioPath Reaction': 'RXN00643',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69105',\n",
       "   'RHEA': '19889;19890;19891;19892',\n",
       "   'KEGG Reaction': 'R00289'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39435_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39435_1 at 0x7fc449879700>},\n",
       "  '_metabolites': {<Metabolite g1p_c at 0x7fc448e1cbb0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite utp_c at 0x7fc448ec84c0>: -1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite udpg_c at 0x7fc463a6d6a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GALURt2r',\n",
       "  'name': 'D-galacturonate transport via proton symport, reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR55544'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39001_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39001_1 at 0x7fc4498796d0>},\n",
       "  '_metabolites': {<Metabolite galur_e at 0x7fc448eef910>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite galur_c at 0x7fc448f20490>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GALURtex',\n",
       "  'name': 'D-galacturonate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR55545'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite galur_e at 0x7fc448eef910>: -1.0,\n",
       "   <Metabolite galur_p at 0x7fc448f204c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GAMpts',\n",
       "  'name': 'D-glucosamine transport via PEP:Pyr PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2545'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1 or (EDU36099_1 and EDU36437_1 and EDU39079_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36099_1 at 0x7fc449879760>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite gam_e at 0x7fc448f204f0>: -1.0,\n",
       "   <Metabolite gam6p_c at 0x7fc463aaed30>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GAPD',\n",
       "  'name': 'Glyceraldehyde-3-phosphate dehydrogenase',\n",
       "  'notes': {'EC Number': '1.2.1.59',\n",
       "   'BioCyc': 'META:GAPOXNPHOSPHN-RXN',\n",
       "   'BioPath Reaction': 'RXN01471',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR50924',\n",
       "   'KEGG Reaction': 'R01061',\n",
       "   'RHEA': '10300;10301;10302;10303',\n",
       "   'UniPathway Reaction': 'UCR01061;UER00184'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37916_1 or EDU38969_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37916_1 at 0x7fc449879790>,\n",
       "   <Gene EDU38969_1 at 0x7fc4498797c0>},\n",
       "  '_metabolites': {<Metabolite g3p_c at 0x7fc448ec8c40>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: -1.0,\n",
       "   <Metabolite 13dpg_c at 0x7fc448e1c7f0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GARFT',\n",
       "  'name': 'Phosphoribosylglycinamide formyltransferase',\n",
       "  'notes': {'EC Number': '2.1.2.2',\n",
       "   'BioPath Reaction': 'RXN00351',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2551',\n",
       "   'KEGG Reaction': 'R04325',\n",
       "   'RHEA': '15053;15054;15055;15056',\n",
       "   'UniPathway Reaction': 'UCR04325;UER00126'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35968_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35968_1 at 0x7fc4498797f0>},\n",
       "  '_metabolites': {<Metabolite 10fthf_c at 0x7fc463a377c0>: -1.0,\n",
       "   <Metabolite gar_c at 0x7fc448f20550>: -1.0,\n",
       "   <Metabolite fgam_c at 0x7fc448f20520>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GBBTNabc',\n",
       "  'name': 'Gamma butyrobetaine transport in via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37130_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37130_1 at 0x7fc44985d5e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gbbtn_e at 0x7fc448ec8340>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gbbtn_c at 0x7fc448ec82b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GBBTNtex',\n",
       "  'name': 'Gamma-butyrobetaine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2561'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite gbbtn_e at 0x7fc448ec8340>: -1.0,\n",
       "   <Metabolite gbbtn_p at 0x7fc448ec8370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GCALDD',\n",
       "  'name': 'Glycolaldehyde dehydrogenase',\n",
       "  'notes': {'RHEA': '20001;20002;20003;20004',\n",
       "   'BioCyc': 'META:GLYCOLALD-DEHYDROG-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2565',\n",
       "   'KEGG Reaction': 'R01333',\n",
       "   'EC Number': '1.2.1.21'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38973_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38973_1 at 0x7fc449820550>},\n",
       "  '_metabolites': {<Metabolite gcald_c at 0x7fc448e494f0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite glyclt_c at 0x7fc448f20580>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GCCa',\n",
       "  'name': 'Glycine-cleavage complex',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR55549',\n",
       "   'KEGG Reaction': 'R03425',\n",
       "   'EC Number': '1.4.4.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU37132_1 and EDU38548_1 and EDU38549_1) or (EDU37132_1 and EDU38547_1 and EDU38548_1 and EDU38549_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37132_1 at 0x7fc449820a30>,\n",
       "   <Gene EDU38547_1 at 0x7fc449879880>,\n",
       "   <Gene EDU38548_1 at 0x7fc449879820>,\n",
       "   <Gene EDU38549_1 at 0x7fc449879850>},\n",
       "  '_metabolites': {<Metabolite gly_c at 0x7fc448e49400>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite lpro_c at 0x7fc448f205e0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite alpro_c at 0x7fc448f205b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GCCb',\n",
       "  'name': 'Glycine cleavage complex',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82977', 'EC Number': '2.1.2.10'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38546_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38546_1 at 0x7fc4498798b0>},\n",
       "  '_metabolites': {<Metabolite thf_c at 0x7fc448e49640>: -1.0,\n",
       "   <Metabolite alpro_c at 0x7fc448f205b0>: -1.0,\n",
       "   <Metabolite mlthf_c at 0x7fc448eef130>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite dhlpro_c at 0x7fc448f20610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GCCc',\n",
       "  'name': 'Glycine-cleavage complex',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2572',\n",
       "   'KEGG Reaction': 'R03815',\n",
       "   'EC Number': '1.8.1.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37132_1 or (EDU37132_1 and EDU38548_1 and EDU38549_1) or (EDU37132_1 and EDU38547_1 and EDU38548_1 and EDU38549_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37132_1 at 0x7fc449820a30>,\n",
       "   <Gene EDU38547_1 at 0x7fc449879880>,\n",
       "   <Gene EDU38548_1 at 0x7fc449879820>,\n",
       "   <Gene EDU38549_1 at 0x7fc449879850>},\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite dhlpro_c at 0x7fc448f20610>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite lpro_c at 0x7fc448f205e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GDPMNP',\n",
       "  'name': 'GDP-mannose phyrophosphatase',\n",
       "  'notes': {'RHEA': '27978;27979;27980;27981',\n",
       "   'BioCyc': 'META:RXN0-5108',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2586',\n",
       "   'KEGG Reaction': 'R00881',\n",
       "   'EC Number': '3.6.1.21'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite gdpmann_c at 0x7fc448e74bb0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gmp_c at 0x7fc448e1cb50>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite man1p_c at 0x7fc448e1cbe0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GDPTPDP',\n",
       "  'name': \"Guanosine 3'-diphosphate 5'-triphosphate 3'-diphosphatase\",\n",
       "  'notes': {'RHEA': '33055;33056;33057;33058',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR78000',\n",
       "   'EC Number': '3.1.7.2',\n",
       "   'BioCyc': 'META:RXN0-6427'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite gdptp_c at 0x7fc448f20640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gtp_c at 0x7fc463adc880>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GF6PTA',\n",
       "  'name': 'Glutamine-fructose-6-phosphate transaminase',\n",
       "  'notes': {'EC Number': '2.6.1.16',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR79233',\n",
       "   'KEGG Reaction': 'R00768',\n",
       "   'RHEA': '13237;13238;13239;13240',\n",
       "   'UniPathway Reaction': 'UCR00768;UER00528'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39107_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39107_1 at 0x7fc4498798e0>},\n",
       "  '_metabolites': {<Metabolite f6p_c at 0x7fc448e49430>: -1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite gam6p_c at 0x7fc463aaed30>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GHMT2r',\n",
       "  'name': 'Glycine hydroxymethyltransferase, reversible',\n",
       "  'notes': {'RHEA': '15481;15482;15483;15484',\n",
       "   'EC Number': '2.1.2.1',\n",
       "   'KEGG Reaction': 'R00945',\n",
       "   'UniPathway Reaction': 'UCR00945;UER00914;UER01023',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2619'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36245_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36245_1 at 0x7fc449879910>},\n",
       "  '_metabolites': {<Metabolite ser__L_c at 0x7fc463aae400>: -1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: -1.0,\n",
       "   <Metabolite gly_c at 0x7fc448e49400>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite mlthf_c at 0x7fc448eef130>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GK1',\n",
       "  'name': 'Guanylate kinase (GMP:ATP)',\n",
       "  'notes': {'EC Number': '2.7.4.8',\n",
       "   'BioCyc': 'META:GUANYL-KIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00728',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69138',\n",
       "   'RHEA': '20780;20781;20782;20783',\n",
       "   'KEGG Reaction': 'R00332'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36318_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36318_1 at 0x7fc44986a2e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite gmp_c at 0x7fc448e1cb50>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gdp_c at 0x7fc463adc850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GK2',\n",
       "  'name': 'Guanylate kinase  GMPdATP ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69139', 'EC Number': '2.7.4.8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36318_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36318_1 at 0x7fc44986a2e0>},\n",
       "  '_metabolites': {<Metabolite datp_c at 0x7fc448e1ca30>: -1.0,\n",
       "   <Metabolite gmp_c at 0x7fc448e1cb50>: -1.0,\n",
       "   <Metabolite dadp_c at 0x7fc448e1c9d0>: 1.0,\n",
       "   <Metabolite gdp_c at 0x7fc463adc850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCAASE3',\n",
       "  'name': 'Beta-glucuronidase (tryptophanyl)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82993'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite trpglcur_c at 0x7fc448f206a0>: -1.0,\n",
       "   <Metabolite glcur_c at 0x7fc448f20670>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite trp__L_c at 0x7fc448eef940>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCabcpp',\n",
       "  'name': 'D-glucose transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2686'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37378_1 or EDU37921_1 or EDU37922_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37378_1 at 0x7fc449879940>,\n",
       "   <Gene EDU37921_1 at 0x7fc449879730>,\n",
       "   <Gene EDU37922_1 at 0x7fc44986a940>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glc__D_p at 0x7fc448f200d0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite glc__D_c at 0x7fc463a6d4f0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCNt2r',\n",
       "  'name': 'D-gluconate transport via proton symport, reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2667',\n",
       "   'BioCyc': 'META:TRANS-RXN0-209'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36576_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36576_1 at 0x7fc4498204c0>},\n",
       "  '_metabolites': {<Metabolite glcn_e at 0x7fc448f206d0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite glcn_c at 0x7fc463adc400>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCNt2rpp',\n",
       "  'name': 'D-gluconate transport via proton symport, reversible (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2667',\n",
       "   'BioCyc': 'META:TRANS-RXN0-209'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36576_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36576_1 at 0x7fc4498204c0>},\n",
       "  '_metabolites': {<Metabolite glcn_p at 0x7fc448f20370>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite glcn_c at 0x7fc463adc400>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCpts',\n",
       "  'name': 'D-glucose transport via PEP:Pyr PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69163'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1 or (EDU36006_1 and EDU36099_1 and EDU37465_1) or (EDU36006_1 and EDU36099_1 and EDU37832_1) or (EDU36006_1 and EDU36437_1 and EDU37160_1) or (EDU36006_1 and EDU36437_1 and EDU37855_1) or (EDU36437_1 and EDU37465_1 and EDU39079_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36006_1 at 0x7fc449820790>,\n",
       "   <Gene EDU36099_1 at 0x7fc449879760>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU37160_1 at 0x7fc449820700>,\n",
       "   <Gene EDU37465_1 at 0x7fc4498799a0>,\n",
       "   <Gene EDU37832_1 at 0x7fc4498799d0>,\n",
       "   <Gene EDU37855_1 at 0x7fc449879970>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite glc__D_e at 0x7fc463a6d520>: -1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCptspp',\n",
       "  'name': 'D-glucose transport via PEP:Pyr PTS (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69163'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1 or (EDU36006_1 and EDU36437_1) or (EDU36437_1 and EDU37843_1) or (EDU36006_1 and EDU36437_1 and EDU37160_1) or (EDU36006_1 and EDU36437_1 and EDU37855_1) or (EDU36437_1 and EDU37160_1 and EDU37843_1) or (EDU36437_1 and EDU37843_1 and EDU37855_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36006_1 at 0x7fc449820790>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU37160_1 at 0x7fc449820700>,\n",
       "   <Gene EDU37843_1 at 0x7fc4498207f0>,\n",
       "   <Gene EDU37855_1 at 0x7fc449879970>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite glc__D_p at 0x7fc448f200d0>: -1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCtex',\n",
       "  'name': 'Glucose transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2651'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite glc__D_e at 0x7fc463a6d520>: -1.0,\n",
       "   <Metabolite glc__D_p at 0x7fc448f200d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCURT',\n",
       "  'name': 'Glucuronosyltransferase (tryptophanyl)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83017'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite trp__L_c at 0x7fc448eef940>: -1.0,\n",
       "   <Metabolite udpglcur_c at 0x7fc448e74d60>: -1.0,\n",
       "   <Metabolite udp_c at 0x7fc463a6d670>: 1.0,\n",
       "   <Metabolite trpglcur_c at 0x7fc448f206a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCURt2r',\n",
       "  'name': 'D-glucuronate transport via proton symport, reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2684'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39001_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39001_1 at 0x7fc4498796d0>},\n",
       "  '_metabolites': {<Metabolite glcur_e at 0x7fc448f20730>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite glcur_c at 0x7fc448f20670>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLCURtex',\n",
       "  'name': 'D-glucuronat transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2685'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite glcur_e at 0x7fc448f20730>: -1.0,\n",
       "   <Metabolite glcur_p at 0x7fc448f20760>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLNabc',\n",
       "  'name': 'GLNabc',\n",
       "  'notes': {'RHEA': '29895;29896;29897;29898',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2704',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-12-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36875_1 or EDU38242_1 or (EDU36792_1 and EDU37362_1) or (EDU37361_1 and EDU37426_1 and EDU37635_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36792_1 at 0x7fc44983ba00>,\n",
       "   <Gene EDU36875_1 at 0x7fc449879a90>,\n",
       "   <Gene EDU37361_1 at 0x7fc449879a60>,\n",
       "   <Gene EDU37362_1 at 0x7fc449879af0>,\n",
       "   <Gene EDU37426_1 at 0x7fc449879a30>,\n",
       "   <Gene EDU37635_1 at 0x7fc449879b20>,\n",
       "   <Gene EDU38242_1 at 0x7fc449879ac0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gln__L_e at 0x7fc448e49940>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLNabcpp',\n",
       "  'name': 'L-glutamine transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29895;29896;29897;29898',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2704',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-12-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37427_1 or EDU38242_1 or EDU39132_1 or (EDU37362_1 and EDU38242_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37362_1 at 0x7fc449879af0>,\n",
       "   <Gene EDU37427_1 at 0x7fc44983ba90>,\n",
       "   <Gene EDU38242_1 at 0x7fc449879ac0>,\n",
       "   <Gene EDU39132_1 at 0x7fc449879b50>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gln__L_p at 0x7fc448f20790>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLNS',\n",
       "  'name': 'Glutamine synthetase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82966', 'EC Number': '6.3.1.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38185_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38185_1 at 0x7fc449879a00>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLNtex',\n",
       "  'name': 'L-glutamine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2708',\n",
       "   'BioCyc': 'META:TRANS-RXN-233'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite gln__L_e at 0x7fc448e49940>: -1.0,\n",
       "   <Metabolite gln__L_p at 0x7fc448f20790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLTPD',\n",
       "  'name': 'Galactitol-1-phosphate dehydrogenase',\n",
       "  'notes': {'RHEA': '28106;28107;28108;28109',\n",
       "   'BioCyc': 'META:GALACTITOLPDEHYD-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2713',\n",
       "   'KEGG Reaction': 'R05571',\n",
       "   'EC Number': '1.1.1.M6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite galt1p_c at 0x7fc448f20430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite tag6p__D_c at 0x7fc463aaebb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUabc',\n",
       "  'name': 'L-glutamate transport via ABC system',\n",
       "  'notes': {'RHEA': '29035;29036;29037;29038',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR39726',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-13-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37242_1 or (EDU36790_1 and EDU36792_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36790_1 at 0x7fc449879d30>,\n",
       "   <Gene EDU36792_1 at 0x7fc44983ba00>,\n",
       "   <Gene EDU37242_1 at 0x7fc44983be50>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glu__L_e at 0x7fc448ec87f0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUabcpp',\n",
       "  'name': 'L-glutamate transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29035;29036;29037;29038',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR39726',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-13-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37242_1 or EDU38743_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37242_1 at 0x7fc44983be50>,\n",
       "   <Gene EDU38743_1 at 0x7fc44983be80>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glu__L_p at 0x7fc448f207f0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUABUTt7',\n",
       "  'name': '4-aminobutyrate/glutamate antiport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR31598',\n",
       "   'BioCyc': 'META:TRANS-RXN-261'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36361_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36361_1 at 0x7fc449879b80>},\n",
       "  '_metabolites': {<Metabolite 4abut_c at 0x7fc463a82880>: -1.0,\n",
       "   <Metabolite glu__L_e at 0x7fc448ec87f0>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite 4abut_e at 0x7fc463aae7c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUABUTt7pp',\n",
       "  'name': '4-aminobutyrate/glutamate antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR31598',\n",
       "   'BioCyc': 'META:TRANS-RXN-261'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36361_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36361_1 at 0x7fc449879b80>},\n",
       "  '_metabolites': {<Metabolite 4abut_c at 0x7fc463a82880>: -1.0,\n",
       "   <Metabolite glu__L_p at 0x7fc448f207f0>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite 4abut_p at 0x7fc463aae790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUDC',\n",
       "  'name': 'Glutamate Decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.15',\n",
       "   'BioCyc': 'META:GLUTDECARBOX-RXN',\n",
       "   'BioPath Reaction': 'RXN01408',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31600',\n",
       "   'RHEA': '17785;17786;17787;17788',\n",
       "   'KEGG Reaction': 'R00261'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35915_1 or EDU36907_1 or EDU39421_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35915_1 at 0x7fc44983bc10>,\n",
       "   <Gene EDU36907_1 at 0x7fc449879bb0>,\n",
       "   <Gene EDU39421_1 at 0x7fc44983bbe0>},\n",
       "  '_metabolites': {<Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite 4abut_c at 0x7fc463a82880>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUDxi',\n",
       "  'name': 'Glutamate dehydrogenase  NAD ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82979', 'EC Number': '1.4.1.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37022_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37022_1 at 0x7fc449879be0>},\n",
       "  '_metabolites': {<Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite akg_c at 0x7fc463adc370>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUDy',\n",
       "  'name': 'Glutamate dehydrogenase (NADP)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82980', 'EC Number': '1.4.1.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37022_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37022_1 at 0x7fc449879be0>},\n",
       "  '_metabolites': {<Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite akg_c at 0x7fc463adc370>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUFORT',\n",
       "  'name': 'Formimidoyltransferase cyclodeaminase',\n",
       "  'notes': {'EC Number': '4.3.1.4',\n",
       "   'BioPath Reaction': 'RXN00985',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR32839',\n",
       "   'KEGG Reaction': 'R02287',\n",
       "   'RHEA': '15097;15098;15099;15100',\n",
       "   'UniPathway Reaction': 'UCR02287;UER00555'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite 5forthf_c at 0x7fc448f20190>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: 1.0,\n",
       "   <Metabolite forglu_c at 0x7fc45d333d00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUN',\n",
       "  'name': 'Glutaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82967', 'EC Number': '6.3.5.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36037_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36037_1 at 0x7fc449879c10>},\n",
       "  '_metabolites': {<Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUPRT',\n",
       "  'name': 'Glutamine phosphoribosyldiphosphate amidotransferase',\n",
       "  'notes': {'EC Number': '2.4.2.14',\n",
       "   'BioCyc': 'META:PRPPAMIDOTRANS-RXN',\n",
       "   'BioPath Reaction': 'RXN01011',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69167',\n",
       "   'KEGG Reaction': 'R01072',\n",
       "   'RHEA': '14905;14906;14907;14908',\n",
       "   'UniPathway Reaction': 'UCR01072;UER00124'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35931_1 or EDU35966_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35931_1 at 0x7fc449879c40>,\n",
       "   <Gene EDU35966_1 at 0x7fc449879c70>},\n",
       "  '_metabolites': {<Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite prpp_c at 0x7fc463aae0d0>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite pram_c at 0x7fc448f20820>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUR',\n",
       "  'name': 'Glutamate racemase',\n",
       "  'notes': {'RHEA': '12813;12814;12815;12816',\n",
       "   'BioCyc': 'META:GLUTRACE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31606',\n",
       "   'KEGG Reaction': 'R00260',\n",
       "   'EC Number': '5.1.1.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38186_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38186_1 at 0x7fc449879ca0>},\n",
       "  '_metabolites': {<Metabolite glu__D_c at 0x7fc448e49a00>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUSy',\n",
       "  'name': 'Glutamate synthase (NADPH)',\n",
       "  'notes': {'EC Number': '1.4.1.13',\n",
       "   'BioCyc': 'META:GLUTAMATESYN-RXN',\n",
       "   'BioPath Reaction': 'RXN00859',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31609',\n",
       "   'KEGG Reaction': 'R00114',\n",
       "   'RHEA': '15501;15502;15503;15504',\n",
       "   'UniPathway Reaction': 'UCR00114;UER00689'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38634_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38634_1 at 0x7fc449879cd0>},\n",
       "  '_metabolites': {<Metabolite akg_c at 0x7fc463adc370>: -1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 2.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUt2r',\n",
       "  'name': 'L glutamate transport via proton symport  reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR31615'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36849_1 or EDU38390_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36849_1 at 0x7fc449879d60>,\n",
       "   <Gene EDU38390_1 at 0x7fc449879580>},\n",
       "  '_metabolites': {<Metabolite glu__L_e at 0x7fc448ec87f0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUt4',\n",
       "  'name': 'Na+/glutamate symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR31616'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36849_1 or EDU38513_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36849_1 at 0x7fc449879d60>,\n",
       "   <Gene EDU38513_1 at 0x7fc449879d90>},\n",
       "  '_metabolites': {<Metabolite glu__L_e at 0x7fc448ec87f0>: -1.0,\n",
       "   <Metabolite na1_e at 0x7fc463adc610>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUt4pp',\n",
       "  'name': 'Na+/glutamate symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR31616'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38513_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38513_1 at 0x7fc449879d90>},\n",
       "  '_metabolites': {<Metabolite glu__L_p at 0x7fc448f207f0>: -1.0,\n",
       "   <Metabolite na1_p at 0x7fc463adcfd0>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUtex',\n",
       "  'name': 'L-glutamate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR31619',\n",
       "   'BioCyc': 'META:TRANS-RXN-232;META:TRANS-RXN-234'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite glu__L_e at 0x7fc448ec87f0>: -1.0,\n",
       "   <Metabolite glu__L_p at 0x7fc448f207f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUTRR',\n",
       "  'name': 'Glutamyl-tRNA reductase',\n",
       "  'notes': {'EC Number': '1.2.1.70',\n",
       "   'KEGG Reaction': 'R04109',\n",
       "   'UniPathway Reaction': 'UCR04109;UER00316',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR48203'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite glutrna_c at 0x7fc448f20850>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite glu1sa_c at 0x7fc448f20100>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0,\n",
       "   <Metabolite trnaglu_c at 0x7fc448f20880>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLUTRS',\n",
       "  'name': 'Glutamyl-tRNA synthetase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR48752',\n",
       "   'KEGG Reaction': 'R05578',\n",
       "   'EC Number': '6.1.1.24'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38968_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38968_1 at 0x7fc449879d00>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite trnaglu_c at 0x7fc448f20880>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite glutrna_c at 0x7fc448f20850>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLXCL',\n",
       "  'name': 'Glyoxalate carboligase',\n",
       "  'notes': {'EC Number': '4.1.1.47',\n",
       "   'BioCyc': 'META:GLYOCARBOLIG-RXN',\n",
       "   'BioPath Reaction': 'RXN00883',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2736',\n",
       "   'KEGG Reaction': 'R00013',\n",
       "   'RHEA': '10136;10137;10138;10139',\n",
       "   'UniPathway Reaction': 'UCR00013;UER00831'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite glx_c at 0x7fc448e493d0>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite 2h3oppan_c at 0x7fc448f20700>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLY3Pt2',\n",
       "  'name': 'Glycerol 3 phosphate transport in via proton symport',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38948_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38948_1 at 0x7fc449879dc0>},\n",
       "  '_metabolites': {<Metabolite glyc3p_e at 0x7fc448f208b0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYALDDr',\n",
       "  'name': 'GLYALDDr',\n",
       "  'notes': {'RHEA': '30783;30784;30785;30786',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2740',\n",
       "   'KEGG Reaction': 'R01752',\n",
       "   'EC Number': '1.2.1.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38973_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38973_1 at 0x7fc449820550>},\n",
       "  '_metabolites': {<Metabolite glyald_c at 0x7fc448e49b50>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite glyc__R_c at 0x7fc448e497c0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYBabc',\n",
       "  'name': 'Glycine betaine transport via ABC system',\n",
       "  'notes': {'RHEA': '32783;32784;32785;32786',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2744'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36889_1 or EDU37130_1 or EDU38425_1 or (EDU36480_1 and EDU37129_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36480_1 at 0x7fc44985d5b0>,\n",
       "   <Gene EDU36889_1 at 0x7fc44985da60>,\n",
       "   <Gene EDU37129_1 at 0x7fc44985d610>,\n",
       "   <Gene EDU37130_1 at 0x7fc44985d5e0>,\n",
       "   <Gene EDU38425_1 at 0x7fc449879df0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glyb_e at 0x7fc448f20910>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite glyb_c at 0x7fc448e74be0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYBt2r',\n",
       "  'name': 'Glycine betaine transport via proton symport, reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2745',\n",
       "   'BioCyc': 'META:TRANS-RXN-29A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37680_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37680_1 at 0x7fc449879e20>},\n",
       "  '_metabolites': {<Metabolite glyb_e at 0x7fc448f20910>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite glyb_c at 0x7fc448e74be0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYC3Pabc',\n",
       "  'name': 'Sn-Glycerol 3-phosphate transport via ABC system',\n",
       "  'notes': {'RHEA': '21668;21669;21670;21671',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2750',\n",
       "   'EC Number': '3.6.3.20',\n",
       "   'BioCyc': 'META:3.6.3.20-RXN;META:ABC-34-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37262_1 or (EDU37261_1 and EDU37262_1 and EDU37263_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37261_1 at 0x7fc449879e50>,\n",
       "   <Gene EDU37262_1 at 0x7fc449879eb0>,\n",
       "   <Gene EDU37263_1 at 0x7fc449879e80>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glyc3p_e at 0x7fc448f208b0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCDx',\n",
       "  'name': 'Glycerol dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.6',\n",
       "   'BioCyc': 'META:GLYCDEH-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69172',\n",
       "   'KEGG Reaction': 'R01034',\n",
       "   'RHEA': '13769;13770;13771;13772',\n",
       "   'UniPathway Reaction': 'UCR01034;UER00668'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36810_1 or EDU38823_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36810_1 at 0x7fc449879ee0>,\n",
       "   <Gene EDU38823_1 at 0x7fc449879f10>},\n",
       "  '_metabolites': {<Metabolite glyc_c at 0x7fc448e49b80>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite dha_c at 0x7fc448e49d60>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCK',\n",
       "  'name': 'Glycerate kinase',\n",
       "  'notes': {'EC Number': '2.7.1.31',\n",
       "   'BioCyc': 'META:GLY3KIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00057',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2756',\n",
       "   'KEGG Reaction': 'R01514',\n",
       "   'RHEA': '23516;23517;23518;23519',\n",
       "   'UniPathway Reaction': 'UCR01514;UER00833;UER00917'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36537_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36537_1 at 0x7fc449879f40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite glyc__R_c at 0x7fc448e497c0>: -1.0,\n",
       "   <Metabolite 3pg_c at 0x7fc463a82790>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCL',\n",
       "  'name': 'Glycine Cleavage System',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83036', 'EC Number': '2.1.2.10'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37132_1 or (EDU37132_1 and EDU38546_1 and EDU38548_1 and EDU38549_1) or (EDU37132_1 and EDU38546_1 and EDU38547_1 and EDU38548_1 and EDU38549_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37132_1 at 0x7fc449820a30>,\n",
       "   <Gene EDU38546_1 at 0x7fc4498798b0>,\n",
       "   <Gene EDU38547_1 at 0x7fc449879880>,\n",
       "   <Gene EDU38548_1 at 0x7fc449879820>,\n",
       "   <Gene EDU38549_1 at 0x7fc449879850>},\n",
       "  '_metabolites': {<Metabolite gly_c at 0x7fc448e49400>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite mlthf_c at 0x7fc448eef130>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCt1',\n",
       "  'name': 'Glycerol transport via uniport (facilitated diffusion)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69175',\n",
       "   'BioCyc': 'META:TRANS-RXN-131'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36066_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36066_1 at 0x7fc449879fa0>},\n",
       "  '_metabolites': {<Metabolite glyc_e at 0x7fc448f20970>: -1.0,\n",
       "   <Metabolite glyc_c at 0x7fc448e49b80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCtex',\n",
       "  'name': 'Glycerol transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69175',\n",
       "   'BioCyc': 'META:TRANS-RXN-131'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite glyc_e at 0x7fc448f20970>: -1.0,\n",
       "   <Metabolite glyc_p at 0x7fc448e9ffd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCTO1',\n",
       "  'name': 'GLYCTO1',\n",
       "  'notes': {'EC Number': '1.1.3.15',\n",
       "   'BioCyc': 'META:RXN-969',\n",
       "   'BioPath Reaction': 'RXN00927',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2764',\n",
       "   'KEGG Reaction': 'R00475',\n",
       "   'RHEA': '25311;25312;25313;25314',\n",
       "   'UniPathway Reaction': 'UCR00475;UER00912'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38862_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38862_1 at 0x7fc449879f70>},\n",
       "  '_metabolites': {<Metabolite glyclt_c at 0x7fc448f20580>: -1.0,\n",
       "   <Metabolite o2_c at 0x7fc463a6d610>: -1.0,\n",
       "   <Metabolite glx_c at 0x7fc448e493d0>: 1.0,\n",
       "   <Metabolite h2o2_c at 0x7fc463a6d580>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCTO3',\n",
       "  'name': 'Glycolate oxidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69173'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38862_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38862_1 at 0x7fc449879f70>},\n",
       "  '_metabolites': {<Metabolite glyclt_c at 0x7fc448f20580>: -1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite glx_c at 0x7fc448e493d0>: 1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYCtpp',\n",
       "  'name': 'Glycerol transport via channel (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69175',\n",
       "   'BioCyc': 'META:TRANS-RXN-131'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite glyc_c at 0x7fc448e49b80>: -1.0,\n",
       "   <Metabolite glyc_p at 0x7fc448e9ffd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYK',\n",
       "  'name': 'Glycerol kinase',\n",
       "  'notes': {'EC Number': '2.7.1.30',\n",
       "   'BioCyc': 'META:GLYCEROL-KIN-RXN',\n",
       "   'BioPath Reaction': 'RXN00925',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69177',\n",
       "   'KEGG Reaction': 'R00847',\n",
       "   'RHEA': '21644;21645;21646;21647',\n",
       "   'UniPathway Reaction': 'UCR00847;UER00672'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36067_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36067_1 at 0x7fc449879fd0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite glyc_c at 0x7fc448e49b80>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYOX',\n",
       "  'name': 'Hydroxyacylglutathione hydrolase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83039'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35789_1 or EDU38576_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35789_1 at 0x7fc449879070>,\n",
       "   <Gene EDU38576_1 at 0x7fc449879100>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite lgt__S_c at 0x7fc448f209a0>: -1.0,\n",
       "   <Metabolite gthrd_c at 0x7fc448e74a60>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lac__D_c at 0x7fc448e1c0a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYt2r',\n",
       "  'name': 'Glycine reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2778',\n",
       "   'BioCyc': 'META:TRANS-RXN-62B'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36264_1 or EDU37508_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36264_1 at 0x7fc44983b340>,\n",
       "   <Gene EDU37508_1 at 0x7fc44983b310>},\n",
       "  '_metabolites': {<Metabolite gly_e at 0x7fc448e49970>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite gly_c at 0x7fc448e49400>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYt4pp',\n",
       "  'name': 'Glycine transport in via sodium symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2779'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35942_1 or EDU36264_1 or EDU37508_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35942_1 at 0x7fc44983b3a0>,\n",
       "   <Gene EDU36264_1 at 0x7fc44983b340>,\n",
       "   <Gene EDU37508_1 at 0x7fc44983b310>},\n",
       "  '_metabolites': {<Metabolite gly_p at 0x7fc448f208e0>: -1.0,\n",
       "   <Metabolite na1_p at 0x7fc463adcfd0>: -1.0,\n",
       "   <Metabolite gly_c at 0x7fc448e49400>: 1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GLYtex',\n",
       "  'name': 'Glycine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2782',\n",
       "   'BioCyc': 'META:TRANS-RXN0-537'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite gly_e at 0x7fc448e49970>: -1.0,\n",
       "   <Metabolite gly_p at 0x7fc448f208e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GMPS2',\n",
       "  'name': 'GMP synthase',\n",
       "  'notes': {'EC Number': '6.3.5.2',\n",
       "   'BioCyc': 'META:GMP-SYN-GLUT-RXN',\n",
       "   'BioPath Reaction': 'RXN00486',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR32727',\n",
       "   'KEGG Reaction': 'R01231',\n",
       "   'RHEA': '11680;11681;11682;11683',\n",
       "   'UniPathway Reaction': 'UCR01231;UER00296'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39249_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39249_1 at 0x7fc4498870d0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite xmp_c at 0x7fc448f209d0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite gmp_c at 0x7fc448e1cb50>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GPDDA2',\n",
       "  'name': 'Glycerophosphodiester phosphodiesterase (Glycerophosphoethanolamine)',\n",
       "  'notes': {'EC Number': '3.1.4.46',\n",
       "   'BioCyc': 'META:RXN-14160',\n",
       "   'BioPath Reaction': 'RXN00830',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2803',\n",
       "   'RHEA': '29319;29320;29321;29322',\n",
       "   'KEGG Reaction': 'R01470'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35917_1 or EDU37625_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35917_1 at 0x7fc4498870a0>,\n",
       "   <Gene EDU37625_1 at 0x7fc449887040>},\n",
       "  '_metabolites': {<Metabolite g3pe_c at 0x7fc448f201c0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite etha_c at 0x7fc448eef9d0>: 1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GPDDA4',\n",
       "  'name': 'Glycerophosphodiester phosphodiesterase (Glycerophosphoglycerol)',\n",
       "  'notes': {'RHEA': '29907;29908;29909;29910',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69183',\n",
       "   'EC Number': '3.1.4.46',\n",
       "   'BioCyc': 'META:RXN-14073'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35917_1 or EDU37625_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35917_1 at 0x7fc4498870a0>,\n",
       "   <Gene EDU37625_1 at 0x7fc449887040>},\n",
       "  '_metabolites': {<Metabolite g3pg_c at 0x7fc448f20250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glyc_c at 0x7fc448e49b80>: 1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GRTT',\n",
       "  'name': 'Geranyltranstransferase',\n",
       "  'notes': {'EC Number': '2.5.1.10',\n",
       "   'BioCyc': 'META:FPPSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN00933',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69185',\n",
       "   'KEGG Reaction': 'R02003',\n",
       "   'RHEA': '19361;19362;19363;19364',\n",
       "   'UniPathway Reaction': 'UCR02003;UER00369'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36939_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36939_1 at 0x7fc44986a7c0>},\n",
       "  '_metabolites': {<Metabolite grdp_c at 0x7fc448eef1f0>: -1.0,\n",
       "   <Metabolite ipdp_c at 0x7fc448ec8af0>: -1.0,\n",
       "   <Metabolite frdp_c at 0x7fc448ec8ac0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GRXR',\n",
       "  'name': 'Glutaredoxin reductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83065'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37556_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37556_1 at 0x7fc449887130>},\n",
       "  '_metabolites': {<Metabolite grxox_c at 0x7fc448f20a90>: -1.0,\n",
       "   <Metabolite gthrd_c at 0x7fc448e74a60>: -2.0,\n",
       "   <Metabolite grxrd_c at 0x7fc448f20ac0>: 1.0,\n",
       "   <Metabolite gthox_c at 0x7fc448e74a30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GSPMDA',\n",
       "  'name': 'Glutathionylspermidine amidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83066'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35937_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35937_1 at 0x7fc449887160>},\n",
       "  '_metabolites': {<Metabolite gtspmd_c at 0x7fc448f20af0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gthrd_c at 0x7fc448e74a60>: 1.0,\n",
       "   <Metabolite spmd_c at 0x7fc448ec8460>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GSPMDS',\n",
       "  'name': 'Glutathionylspermidine synthetase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83067'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35937_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35937_1 at 0x7fc449887160>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite gthrd_c at 0x7fc448e74a60>: -1.0,\n",
       "   <Metabolite spmd_c at 0x7fc448ec8460>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gtspmd_c at 0x7fc448f20af0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTHOr',\n",
       "  'name': 'Glutathione oxidoreductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82983', 'EC Number': '1.8.1.7'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite gthox_c at 0x7fc448e74a30>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite gthrd_c at 0x7fc448e74a60>: 2.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTHRDabc2pp',\n",
       "  'name': 'Glutathione export via ABC system (cytoplasm to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83074'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37455_1 or EDU38449_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37455_1 at 0x7fc44985de80>,\n",
       "   <Gene EDU38449_1 at 0x7fc449887190>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite gthrd_c at 0x7fc448e74a60>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite gthrd_p at 0x7fc448eef3d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTHRDabcpp',\n",
       "  'name': 'Reduced glutathione via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83074'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU37081_1 and EDU37580_1) or (EDU37081_1 and EDU37423_1 and EDU37580_1) or (EDU37081_1 and EDU37580_1 and EDU38488_1) or (EDU37081_1 and EDU37423_1 and EDU37580_1 and EDU37581_1) or (EDU37081_1 and EDU37423_1 and EDU37580_1 and EDU38488_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37081_1 at 0x7fc4498871c0>,\n",
       "   <Gene EDU37423_1 at 0x7fc4498201f0>,\n",
       "   <Gene EDU37580_1 at 0x7fc44983b040>,\n",
       "   <Gene EDU37581_1 at 0x7fc4498871f0>,\n",
       "   <Gene EDU38488_1 at 0x7fc449887220>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite gthrd_p at 0x7fc448eef3d0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite gthrd_c at 0x7fc448e74a60>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTPCI',\n",
       "  'name': 'GTP cyclohydrolase I',\n",
       "  'notes': {'EC Number': '3.5.4.16',\n",
       "   'BioCyc': 'META:GTP-CYCLOHYDRO-I-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2834',\n",
       "   'KEGG Reaction': 'R00424',\n",
       "   'RHEA': '17473;17474;17475;17476',\n",
       "   'UniPathway Reaction': 'UER00151'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37207_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37207_1 at 0x7fc449887250>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ahdt_c at 0x7fc463aae2e0>: 1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTPCII',\n",
       "  'name': 'GTP cyclohydrolase II',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69190'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35981_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35981_1 at 0x7fc44986a250>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -3.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0,\n",
       "   <Metabolite 25dhpp_c at 0x7fc448ec8f70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTPCII2',\n",
       "  'name': 'GTP cyclohydrolase II (25drapp)',\n",
       "  'notes': {'EC Number': '3.5.4.25',\n",
       "   'BioCyc': 'META:GTP-CYCLOHYDRO-II-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2835',\n",
       "   'KEGG Reaction': 'R00425',\n",
       "   'RHEA': '23704;23705;23706;23707',\n",
       "   'UniPathway Reaction': 'UCR00425;UER00400'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35981_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35981_1 at 0x7fc44986a250>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -3.0,\n",
       "   <Metabolite 25drapp_c at 0x7fc448ec8f40>: 1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTPDPK',\n",
       "  'name': 'GTP diphosphokinase',\n",
       "  'notes': {'EC Number': '2.7.6.5',\n",
       "   'BioCyc': 'META:GTPPYPHOSKIN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69192',\n",
       "   'KEGG Reaction': 'R00429',\n",
       "   'RHEA': '22088;22089;22090;22091',\n",
       "   'UniPathway Reaction': 'UCR00429;UER00884'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35787_1 or EDU37957_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35787_1 at 0x7fc4498872b0>,\n",
       "   <Gene EDU37957_1 at 0x7fc449887280>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite gdptp_c at 0x7fc448f20640>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GTPHs',\n",
       "  'name': 'GTP amine hydrolysis (spontaneous)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83077'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite xtp_c at 0x7fc448f20b50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GUAD',\n",
       "  'name': 'Guanine deaminase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83078', 'EC Number': '3.5.4.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37522_1 or EDU39082_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37522_1 at 0x7fc4498872e0>,\n",
       "   <Gene EDU39082_1 at 0x7fc44983b4f0>},\n",
       "  '_metabolites': {<Metabolite gua_c at 0x7fc448f20a00>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite xan_c at 0x7fc448f20b80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GUAPRT',\n",
       "  'name': 'Guanine phosphoribosyltransferase',\n",
       "  'notes': {'EC Number': '2.4.2.8',\n",
       "   'BioCyc': 'META:GUANPRIBOSYLTRAN-RXN',\n",
       "   'BioPath Reaction': 'RXN00706',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69194',\n",
       "   'KEGG Reaction': 'R01229',\n",
       "   'RHEA': '25424;25425;25426;25427',\n",
       "   'UniPathway Reaction': 'UCR01229;UER00887'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35702_1 or EDU36804_1 or EDU38222_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35702_1 at 0x7fc449887310>,\n",
       "   <Gene EDU36804_1 at 0x7fc449820d00>,\n",
       "   <Gene EDU38222_1 at 0x7fc449887340>},\n",
       "  '_metabolites': {<Metabolite gua_c at 0x7fc448f20a00>: -1.0,\n",
       "   <Metabolite prpp_c at 0x7fc463aae0d0>: -1.0,\n",
       "   <Metabolite gmp_c at 0x7fc448e1cb50>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GUAt2pp',\n",
       "  'name': 'Guanine transport in via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2847'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39336_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39336_1 at 0x7fc449887370>},\n",
       "  '_metabolites': {<Metabolite gua_p at 0x7fc448f20be0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite gua_c at 0x7fc448f20a00>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GUAt2r',\n",
       "  'name': 'Guanine reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2847'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38414_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38414_1 at 0x7fc4498873a0>},\n",
       "  '_metabolites': {<Metabolite gua_e at 0x7fc448f20bb0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite gua_c at 0x7fc448f20a00>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'GUAtex',\n",
       "  'name': 'Guanine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2846',\n",
       "   'BioCyc': 'META:TRANS-RXN0-578'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite gua_e at 0x7fc448f20bb0>: -1.0,\n",
       "   <Metabolite gua_p at 0x7fc448f20be0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'H2O2tex',\n",
       "  'name': 'Hydrogen peroxide transport via diffusion (external)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2864',\n",
       "   'BioCyc': 'META:TRANS-RXN0-572'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o2_e at 0x7fc448f20b20>: -1.0,\n",
       "   <Metabolite h2o2_p at 0x7fc463adcf70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'H2Ot',\n",
       "  'name': 'H2O transport via diffusion',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2865',\n",
       "   'BioCyc': 'META:TRANS-RXN0-547;META:TRANS-RXN-145'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'H2Otex',\n",
       "  'name': 'H2O transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2865',\n",
       "   'BioCyc': 'META:TRANS-RXN0-547;META:TRANS-RXN-145'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1 or spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>,\n",
       "   <Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'H2Otpp',\n",
       "  'name': 'H2O transport via diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2865',\n",
       "   'BioCyc': 'META:TRANS-RXN0-547;META:TRANS-RXN-145'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'H2St',\n",
       "  'name': 'H2St',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR31666'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2s_e at 0x7fc448f20c10>: -1.0,\n",
       "   <Metabolite h2s_c at 0x7fc448e49580>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'H2tex',\n",
       "  'name': 'Hydrogen transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69200'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite h2_e at 0x7fc448eefa90>: -1.0,\n",
       "   <Metabolite h2_p at 0x7fc448f20c40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'H2tpp',\n",
       "  'name': 'Hydrogen transport diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69200'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2_p at 0x7fc448f20c40>: -1.0,\n",
       "   <Metabolite h2_c at 0x7fc463a827c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD1',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (acetoacetyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-11662',\n",
       "   'BioPath Reaction': 'RXN01071',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2899',\n",
       "   'RHEA': '30799;30800;30801;30802',\n",
       "   'KEGG Reaction': 'R01975'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite aacoa_c at 0x7fc463aae550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3hbcoa_c at 0x7fc448eef5e0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD1_2',\n",
       "  'name': '3 hydroxyacyl CoA dehydrogenase  acetoacetyl CoA ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite aacoa_c at 0x7fc463aae550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite 3hbycoa_c at 0x7fc448eef610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD1i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (acetoacetyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-11662',\n",
       "   'BioPath Reaction': 'RXN01071',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2899',\n",
       "   'RHEA': '30799;30800;30801;30802',\n",
       "   'KEGG Reaction': 'R01975'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3hbcoa_c at 0x7fc448eef5e0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite aacoa_c at 0x7fc463aae550>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD2',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxohexanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-12570',\n",
       "   'BioPath Reaction': 'RXN01362',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2901',\n",
       "   'RHEA': '31143;31144;31145;31146',\n",
       "   'KEGG Reaction': 'R04748'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite 3ohcoa_c at 0x7fc463aae7f0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3hhcoa_c at 0x7fc448eef640>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD2i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxohexanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-12570',\n",
       "   'BioPath Reaction': 'RXN01362',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2901',\n",
       "   'RHEA': '31143;31144;31145;31146',\n",
       "   'KEGG Reaction': 'R04748'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3hhcoa_c at 0x7fc448eef640>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3ohcoa_c at 0x7fc463aae7f0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD3',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxooctanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-14275',\n",
       "   'BioPath Reaction': 'RXN00413',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2902',\n",
       "   'RHEA': '31195;31196;31197;31198',\n",
       "   'KEGG Reaction': 'R04745'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite 3oocoa_c at 0x7fc463aae850>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3hocoa_c at 0x7fc448eef670>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD3i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxooctanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-14275',\n",
       "   'BioPath Reaction': 'RXN00413',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2902',\n",
       "   'RHEA': '31195;31196;31197;31198',\n",
       "   'KEGG Reaction': 'R04745'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3hocoa_c at 0x7fc448eef670>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3oocoa_c at 0x7fc463aae850>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD4',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxodecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-12490',\n",
       "   'BioPath Reaction': 'RXN00359',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2903',\n",
       "   'RHEA': '31187;31188;31189;31190',\n",
       "   'KEGG Reaction': 'R04743'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite 3odcoa_c at 0x7fc463aae8b0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3hdcoa_c at 0x7fc448e9f280>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD4i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxodecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-12490',\n",
       "   'BioPath Reaction': 'RXN00359',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2903',\n",
       "   'RHEA': '31187;31188;31189;31190',\n",
       "   'KEGG Reaction': 'R04743'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3hdcoa_c at 0x7fc448e9f280>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3odcoa_c at 0x7fc463aae8b0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD5',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxododecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '31179;31180;31181;31182',\n",
       "   'BioPath Reaction': 'RXN00663',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2904',\n",
       "   'KEGG Reaction': 'R04741',\n",
       "   'EC Number': '1.1.1.35'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite 3oddcoa_c at 0x7fc463aae910>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3hddcoa_c at 0x7fc448e9f1f0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD5i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxododecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '31179;31180;31181;31182',\n",
       "   'BioPath Reaction': 'RXN00663',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2904',\n",
       "   'KEGG Reaction': 'R04741',\n",
       "   'EC Number': '1.1.1.35'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3hddcoa_c at 0x7fc448e9f1f0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3oddcoa_c at 0x7fc463aae910>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD6',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxotetradecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '31167;31168;31169;31170',\n",
       "   'BioCyc': 'META:RXN-12507',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2905',\n",
       "   'KEGG Reaction': 'R04739',\n",
       "   'EC Number': '1.1.1.35'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite 3otdcoa_c at 0x7fc463aae9a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3htdcoa_c at 0x7fc448e9f070>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD6i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxotetradecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '31167;31168;31169;31170',\n",
       "   'BioCyc': 'META:RXN-12507',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2905',\n",
       "   'KEGG Reaction': 'R04739',\n",
       "   'EC Number': '1.1.1.35'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3htdcoa_c at 0x7fc448e9f070>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3otdcoa_c at 0x7fc463aae9a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD7',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxohexadecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-14271',\n",
       "   'BioPath Reaction': 'RXN00078',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2906',\n",
       "   'RHEA': '31159;31160;31161;31162',\n",
       "   'KEGG Reaction': 'R04737'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite 3ohdcoa_c at 0x7fc463aaea30>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3hhdcoa_c at 0x7fc448eef6a0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD7i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxohexadecanoyl-CoA)',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:RXN-14271',\n",
       "   'BioPath Reaction': 'RXN00078',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2906',\n",
       "   'RHEA': '31159;31160;31161;31162',\n",
       "   'KEGG Reaction': 'R04737'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3hhdcoa_c at 0x7fc448eef6a0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3ohdcoa_c at 0x7fc463aaea30>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD8',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxooctadecanoyl-CoA), peroxisomal',\n",
       "  'notes': {'RHEA': '35275;35276;35277;35278',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55617'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39340_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39340_1 at 0x7fc4498873d0>},\n",
       "  '_metabolites': {<Metabolite 3ohodcoa_c at 0x7fc463a820d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 3hodcoa_c at 0x7fc463a82070>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD8i',\n",
       "  'name': '3-hydroxyacyl-CoA dehydrogenase (3-oxooctadecanoyl-CoA)',\n",
       "  'notes': {'RHEA': '35275;35276;35277;35278',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55617'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38190_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38190_1 at 0x7fc44985dbb0>},\n",
       "  '_metabolites': {<Metabolite 3hodcoa_c at 0x7fc463a82070>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3ohodcoa_c at 0x7fc463a820d0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HACD9',\n",
       "  'name': 'HACD9',\n",
       "  'notes': {'EC Number': '1.1.1.35',\n",
       "   'BioCyc': 'META:1.1.1.178-RXN',\n",
       "   'BioPath Reaction': 'RXN00605',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2909',\n",
       "   'RHEA': '13281;13282;13283;13284',\n",
       "   'KEGG Reaction': 'R04203'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite 3hmbcoa_c at 0x7fc448eef6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite 2maacoa_c at 0x7fc448f20c70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HADPCOADH3',\n",
       "  'name': '3-hydroxyadipyl-CoA dehydrogenase (NAD+)',\n",
       "  'notes': {'RHEA': '34851;34852;34853;34854',\n",
       "   'BioCyc': 'META:RXN0-2044',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR60328',\n",
       "   'KEGG Reaction': 'R06941',\n",
       "   'EC Number': '1.1.1.35'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 3hadpcoa_c at 0x7fc448ec8d60>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite oxadpcoa_c at 0x7fc463a821f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HAMR',\n",
       "  'name': 'Hydroxylamine reductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83092', 'EC Number': '1.7.99.1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36059_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36059_1 at 0x7fc449887400>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite ham_c at 0x7fc448f20ca0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HBZOPT',\n",
       "  'name': 'Hydroxybenzoate octaprenyltransferase',\n",
       "  'notes': {'RHEA': '27782;27783;27784;27785',\n",
       "   'BioCyc': 'META:4OHBENZOATE-OCTAPRENYLTRANSFER-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55621',\n",
       "   'KEGG Reaction': 'R05615',\n",
       "   'EC Number': '2.5.1.39'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 4hbz_c at 0x7fc463a82a60>: -1.0,\n",
       "   <Metabolite octdp_c at 0x7fc448ec8e50>: -1.0,\n",
       "   <Metabolite 3ophb_c at 0x7fc448f20cd0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HCO3E',\n",
       "  'name': 'HCO3 equilibration reaction',\n",
       "  'notes': {'EC Number': '4.2.1.1',\n",
       "   'BioCyc': 'META:CARBODEHYDRAT-RXN;META:RXN0-5224',\n",
       "   'BioPath Reaction': 'RXN00996',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2856',\n",
       "   'RHEA': '10748;10749;10750;10751',\n",
       "   'KEGG Reaction': 'R00132;R10092'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite co2_c at 0x7fc463adc8e0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hco3_c at 0x7fc463adc5b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HCO3tex',\n",
       "  'name': 'Bicarbonate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69197'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite hco3_e at 0x7fc463adc640>: -1.0,\n",
       "   <Metabolite hco3_p at 0x7fc448e74b50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HDCAt2pp',\n",
       "  'name': 'Hexadecanoate transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2932'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite hdca_p at 0x7fc448eefb50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HETZK',\n",
       "  'name': 'Hydroxyethylthiazole kinase',\n",
       "  'notes': {'EC Number': '2.7.1.50',\n",
       "   'BioCyc': 'META:THIAZOLSYN3-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2946',\n",
       "   'KEGG Reaction': 'R04448',\n",
       "   'RHEA': '24212;24213;24214;24215',\n",
       "   'UniPathway Reaction': 'UCR04448;UER00139'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36512_1 or EDU38270_1 or EDU39123_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36512_1 at 0x7fc449887490>,\n",
       "   <Gene EDU38270_1 at 0x7fc449887460>,\n",
       "   <Gene EDU39123_1 at 0x7fc449887430>},\n",
       "  '_metabolites': {<Metabolite 4mhetz_c at 0x7fc448f20dc0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite 4mpetz_c at 0x7fc448f20df0>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HEX1',\n",
       "  'name': 'Hexokinase (D-glucose:ATP)',\n",
       "  'notes': {'EC Number': '2.7.1.2',\n",
       "   'BioCyc': 'META:RXN66-521',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69234',\n",
       "   'KEGG Reaction': 'R00299',\n",
       "   'RHEA': '17825;17826;17827;17828',\n",
       "   'UniPathway Reaction': 'UCR00299;UER00180'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38926_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38926_1 at 0x7fc4498874c0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite glc__D_c at 0x7fc463a6d4f0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HEX4',\n",
       "  'name': 'Hexokinase (D-mannose:ATP)',\n",
       "  'notes': {'RHEA': '11028;11029;11030;11031',\n",
       "   'BioCyc': 'META:MANNKIN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2949',\n",
       "   'KEGG Reaction': 'R01326',\n",
       "   'EC Number': '2.7.1.7'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite man_c at 0x7fc448e49fd0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite man6p_c at 0x7fc448e1c0d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HEX7',\n",
       "  'name': 'Hexokinase (D-fructose:ATP)',\n",
       "  'notes': {'RHEA': '16125;16126;16127;16128',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR79250',\n",
       "   'KEGG Reaction': 'R00760',\n",
       "   'EC Number': '2.7.1.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite fru_c at 0x7fc448eefac0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite f6p_c at 0x7fc448e49430>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HG2t3pp',\n",
       "  'name': 'Mercury (Hg+2) transport out via proton antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2957'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39203_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39203_1 at 0x7fc44985d2b0>},\n",
       "  '_metabolites': {<Metabolite hg2_c at 0x7fc448f20e20>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hg2_p at 0x7fc448f20e50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HG2tex',\n",
       "  'name': 'Mercury (Hg+2) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2958'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite hg2_e at 0x7fc448f20e80>: -1.0,\n",
       "   <Metabolite hg2_p at 0x7fc448f20e50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HGt',\n",
       "  'name': 'Mercury transport',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37497_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37497_1 at 0x7fc4498874f0>},\n",
       "  '_metabolites': {<Metabolite hg2_e at 0x7fc448f20e80>: -1.0,\n",
       "   <Metabolite hg2_c at 0x7fc448f20e20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HISabcpp',\n",
       "  'name': 'L-histidine transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29891;29892;29893;29894',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2971',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-14-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37427_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37427_1 at 0x7fc44983ba90>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite his__L_p at 0x7fc448f20ee0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite his__L_c at 0x7fc463aae4f0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HISDr',\n",
       "  'name': 'Histidase r',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83116', 'EC Number': '4.3.1.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite his__L_c at 0x7fc463aae4f0>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite urcan_c at 0x7fc448f20fa0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HISTD',\n",
       "  'name': 'Histidinol dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.23',\n",
       "   'BioCyc': 'META:RXN-8001',\n",
       "   'BioPath Reaction': 'RXN00505',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2968',\n",
       "   'KEGG Reaction': 'R01158',\n",
       "   'RHEA': '20641;20642;20643;20644',\n",
       "   'UniPathway Reaction': 'UCR01158;UER00014'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37197_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37197_1 at 0x7fc449887520>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite histd_c at 0x7fc448f20d00>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 3.0,\n",
       "   <Metabolite his__L_c at 0x7fc463aae4f0>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HIStex',\n",
       "  'name': 'L-histidine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2975'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite his__L_e at 0x7fc448f20eb0>: -1.0,\n",
       "   <Metabolite his__L_p at 0x7fc448f20ee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HISTP',\n",
       "  'name': 'Histidinol-phosphatase',\n",
       "  'notes': {'EC Number': '3.1.3.15',\n",
       "   'BioCyc': 'META:HISTIDPHOS-RXN',\n",
       "   'BioPath Reaction': 'RXN00548',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2969',\n",
       "   'KEGG Reaction': 'R03013',\n",
       "   'RHEA': '14465;14466;14467;14468',\n",
       "   'UniPathway Reaction': 'UCR03013;UER00013'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36270_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36270_1 at 0x7fc449887550>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hisp_c at 0x7fc448e741c0>: -1.0,\n",
       "   <Metabolite histd_c at 0x7fc448f20d00>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMBS',\n",
       "  'name': 'Hydroxymethylbilane synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83118', 'EC Number': '4.3.1.8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38766_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38766_1 at 0x7fc449887580>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ppbng_c at 0x7fc448f20f10>: -4.0,\n",
       "   <Metabolite hmbil_c at 0x7fc448f20fd0>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 4.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMGCOARi',\n",
       "  'name': 'Hydroxymethylglutaryl CoA reductase (ir)',\n",
       "  'notes': {'EC Number': '1.1.1.34',\n",
       "   'BioCyc': 'META:1.1.1.34-RXN',\n",
       "   'BioPath Reaction': 'RXN00349',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2981',\n",
       "   'KEGG Reaction': 'R02082',\n",
       "   'RHEA': '15989;15990;15991;15992',\n",
       "   'UniPathway Reaction': 'UCR02082;UER00103'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -2.0,\n",
       "   <Metabolite hmgcoa_c at 0x7fc448f20f70>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 2.0,\n",
       "   <Metabolite mev__R_c at 0x7fc448f20f40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMGCOASi',\n",
       "  'name': 'Hydroxymethylglutaryl CoA synthase (ir)',\n",
       "  'notes': {'EC Number': '2.3.3.10',\n",
       "   'BioCyc': 'META:HYDROXYMETHYLGLUTARYL-COA-SYNTHASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00820',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2982',\n",
       "   'KEGG Reaction': 'R01978',\n",
       "   'RHEA': '10188;10189;10190;10191',\n",
       "   'UniPathway Reaction': 'UCR01978;UER00102'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite aacoa_c at 0x7fc463aae550>: -1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hmgcoa_c at 0x7fc448f20f70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMPK1',\n",
       "  'name': 'Hydroxymethylpyrimidine kinase (ATP)',\n",
       "  'notes': {'EC Number': '2.7.1.49',\n",
       "   'BioCyc': 'META:OHMETPYRKIN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2985',\n",
       "   'KEGG Reaction': 'R03471',\n",
       "   'RHEA': '23096;23097;23098;23099',\n",
       "   'UniPathway Reaction': 'UCR03471;UER00137'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37385_1 or EDU38271_1 or EDU38611_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37385_1 at 0x7fc4498875e0>,\n",
       "   <Gene EDU38271_1 at 0x7fc449887610>,\n",
       "   <Gene EDU38611_1 at 0x7fc4498875b0>},\n",
       "  '_metabolites': {<Metabolite 4ahmmp_c at 0x7fc448e494c0>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite 4ampm_c at 0x7fc463adc730>: 1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMPK2',\n",
       "  'name': 'HMPK2',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69241'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38611_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38611_1 at 0x7fc4498875b0>},\n",
       "  '_metabolites': {<Metabolite 4ahmmp_c at 0x7fc448e494c0>: -1.0,\n",
       "   <Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite 4ampm_c at 0x7fc463adc730>: 1.0,\n",
       "   <Metabolite cdp_c at 0x7fc448e9f7f0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMPK3',\n",
       "  'name': 'HMPK3',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69242'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38611_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38611_1 at 0x7fc4498875b0>},\n",
       "  '_metabolites': {<Metabolite 4ahmmp_c at 0x7fc448e494c0>: -1.0,\n",
       "   <Metabolite utp_c at 0x7fc448ec84c0>: -1.0,\n",
       "   <Metabolite 4ampm_c at 0x7fc463adc730>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite udp_c at 0x7fc463a6d670>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMPK4',\n",
       "  'name': 'HMPK4',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69243'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38611_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38611_1 at 0x7fc4498875b0>},\n",
       "  '_metabolites': {<Metabolite 4ahmmp_c at 0x7fc448e494c0>: -1.0,\n",
       "   <Metabolite gtp_c at 0x7fc463adc880>: -1.0,\n",
       "   <Metabolite 4ampm_c at 0x7fc463adc730>: 1.0,\n",
       "   <Metabolite gdp_c at 0x7fc463adc850>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMSH',\n",
       "  'name': 'HMSH',\n",
       "  'notes': {'EC Number': '3.7.1.9',\n",
       "   'BioCyc': 'META:3.7.1.9-RXN',\n",
       "   'BioPath Reaction': 'RXN00624',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69246',\n",
       "   'RHEA': '14549;14550;14551;14552',\n",
       "   'KEGG Reaction': 'R02604'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36773_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36773_1 at 0x7fc449887640>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 2hmcnsad_c at 0x7fc448e9f3d0>: -1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite op4en_c at 0x7fc463a82e50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HMSH2',\n",
       "  'name': 'HMSH2',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83120',\n",
       "   'KEGG Reaction': 'R05865',\n",
       "   'EC Number': '3.7.1.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36773_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36773_1 at 0x7fc449887640>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hmccms_c at 0x7fc463a82dc0>: -1.0,\n",
       "   <Metabolite for_c at 0x7fc463a82e20>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 2hchd_c at 0x7fc463a82e80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HOPNTAL',\n",
       "  'name': '4-hydroxy-2-oxopentanoate aldolase',\n",
       "  'notes': {'EC Number': '4.1.3.39',\n",
       "   'BioCyc': 'META:MHPELY-RXN',\n",
       "   'BioPath Reaction': 'RXN01058',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2997',\n",
       "   'UniPathway Reaction': 'UCR00750;UER00797;UER00907',\n",
       "   'KEGG Reaction': 'R00750'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 4h2opntn_c at 0x7fc448f4b160>: -1.0,\n",
       "   <Metabolite acald_c at 0x7fc463aae2b0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HOPNTAL2',\n",
       "  'name': 'HOPNTAL2',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR2998',\n",
       "   'KEGG Reaction': 'R05298',\n",
       "   'EC Number': '4.1.3.43'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 4hoxoh_c at 0x7fc448f4b190>: -1.0,\n",
       "   <Metabolite ppal_c at 0x7fc463a6d460>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HPAtex',\n",
       "  'name': '4-hydroxyphenylacetate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69251'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 4hphac_e at 0x7fc448f4b0a0>: -1.0,\n",
       "   <Metabolite 4hphac_p at 0x7fc463a82be0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HPPK',\n",
       "  'name': 'HPPK',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83125'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38671_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38671_1 at 0x7fc44986a5b0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite 2ahhmp_c at 0x7fc448ec8e80>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 2ahhmd_c at 0x7fc448eeff70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HPYRI',\n",
       "  'name': 'Hydroxypyruvate isomerase',\n",
       "  'notes': {'RHEA': '11952;11953;11954;11955',\n",
       "   'BioCyc': 'META:RXN0-305',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3014',\n",
       "   'KEGG Reaction': 'R01394',\n",
       "   'EC Number': '5.3.1.22'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36657_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36657_1 at 0x7fc449887670>},\n",
       "  '_metabolites': {<Metabolite hpyr_c at 0x7fc448f20940>: -1.0,\n",
       "   <Metabolite 2h3oppan_c at 0x7fc448f20700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HQNt6',\n",
       "  'name': 'Hydroquinone transport inout via proton symport',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite hqn_e at 0x7fc448f4b040>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hqn_c at 0x7fc463aae700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HSDy',\n",
       "  'name': 'Homoserine dehydrogenase (NADPH)',\n",
       "  'notes': {'EC Number': '1.1.1.3',\n",
       "   'BioCyc': 'META:HOMOSERDEHYDROG-RXN',\n",
       "   'BioPath Reaction': 'RXN01170',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3045',\n",
       "   'KEGG Reaction': 'R01775',\n",
       "   'RHEA': '15761;15762;15763;15764',\n",
       "   'UniPathway Reaction': 'UCR01775;UER00950'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36947_1 or EDU37117_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36947_1 at 0x7fc4498876d0>,\n",
       "   <Gene EDU37117_1 at 0x7fc4498876a0>},\n",
       "  '_metabolites': {<Metabolite hom__L_c at 0x7fc448f4b130>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite aspsa_c at 0x7fc448e74700>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HSERTA',\n",
       "  'name': 'Homoserine O trans acetylase',\n",
       "  'notes': {'EC Number': '2.3.1.31',\n",
       "   'BioCyc': 'META:HOMOSERINE-O-ACETYLTRANSFERASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3046',\n",
       "   'KEGG Reaction': 'R01776',\n",
       "   'RHEA': '13701;13702;13703;13704',\n",
       "   'UniPathway Reaction': 'UCR01776;UER00074'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36948_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36948_1 at 0x7fc449887700>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite hom__L_c at 0x7fc448f4b130>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite achms_c at 0x7fc463aaefd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HSK',\n",
       "  'name': 'Homoserine kinase',\n",
       "  'notes': {'EC Number': '2.7.1.39',\n",
       "   'BioCyc': 'META:HOMOSERKIN-RXN',\n",
       "   'BioPath Reaction': 'RXN01457',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3047',\n",
       "   'KEGG Reaction': 'R01771',\n",
       "   'RHEA': '13985;13986;13987;13988',\n",
       "   'UniPathway Reaction': 'UCR01771;UER00064'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37119_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37119_1 at 0x7fc449887730>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite hom__L_c at 0x7fc448f4b130>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite phom_c at 0x7fc448f4b070>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HSST',\n",
       "  'name': 'Homoserine O-succinyltransferase',\n",
       "  'notes': {'EC Number': '2.3.1.46',\n",
       "   'BioCyc': 'META:HOMSUCTRAN-RXN',\n",
       "   'BioPath Reaction': 'RXN01415',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3051',\n",
       "   'KEGG Reaction': 'R01777',\n",
       "   'RHEA': '22008;22009;22010;22011',\n",
       "   'UniPathway Reaction': 'UCR01777;UER00075'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36948_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36948_1 at 0x7fc449887700>},\n",
       "  '_metabolites': {<Metabolite hom__L_c at 0x7fc448f4b130>: -1.0,\n",
       "   <Metabolite succoa_c at 0x7fc463a82250>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite suchms_c at 0x7fc448f4b0d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HSTPT',\n",
       "  'name': 'Histidinol-phosphate transaminase',\n",
       "  'notes': {'EC Number': '2.6.1.9',\n",
       "   'BioCyc': 'META:HISTAMINOTRANS-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31722',\n",
       "   'KEGG Reaction': 'R03243',\n",
       "   'RHEA': '23744;23745;23746;23747',\n",
       "   'UniPathway Reaction': 'UCR03243;UER00012'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37193_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37193_1 at 0x7fc449887760>},\n",
       "  '_metabolites': {<Metabolite glu__L_c at 0x7fc463a6d190>: -1.0,\n",
       "   <Metabolite imacp_c at 0x7fc448e741f0>: -1.0,\n",
       "   <Metabolite akg_c at 0x7fc463adc370>: 1.0,\n",
       "   <Metabolite hisp_c at 0x7fc448e741c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'Htex',\n",
       "  'name': 'Proton transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3066', 'BioCyc': 'META:RXN-14452'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HXAND',\n",
       "  'name': 'Hypoxanthine dehydrogenase',\n",
       "  'notes': {'EC Number': '1.17.1.4',\n",
       "   'BioCyc': 'META:RXN-7682',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3054',\n",
       "   'KEGG Reaction': 'R01768',\n",
       "   'RHEA': '24670;24671;24672;24673',\n",
       "   'UniPathway Reaction': 'UCR01768;UER00661'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35951_1 or EDU35953_1 or EDU35963_1 or (EDU37523_1 and EDU37524_1 and EDU37525_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35951_1 at 0x7fc4498877c0>,\n",
       "   <Gene EDU35953_1 at 0x7fc449887850>,\n",
       "   <Gene EDU35963_1 at 0x7fc449887790>,\n",
       "   <Gene EDU37523_1 at 0x7fc449887880>,\n",
       "   <Gene EDU37524_1 at 0x7fc449887820>,\n",
       "   <Gene EDU37525_1 at 0x7fc4498877f0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite hxan_c at 0x7fc448e1c970>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite xan_c at 0x7fc448f20b80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HXANt2pp',\n",
       "  'name': 'Hypoxanthine transport in via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3055'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39336_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39336_1 at 0x7fc449887370>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite hxan_p at 0x7fc448f4b100>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hxan_c at 0x7fc448e1c970>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HXANt2r',\n",
       "  'name': 'Hypoxanthine reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3055'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38414_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38414_1 at 0x7fc4498873a0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite hxan_e at 0x7fc448f4b1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hxan_c at 0x7fc448e1c970>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HXAtex',\n",
       "  'name': 'Hexanoate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3057'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite hxa_e at 0x7fc448f4b1f0>: -1.0,\n",
       "   <Metabolite hxa_p at 0x7fc448eefc40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HXPRT',\n",
       "  'name': 'Hypoxanthine phosphoribosyltransferase (Hypoxanthine)',\n",
       "  'notes': {'EC Number': '2.4.2.8',\n",
       "   'BioCyc': 'META:HYPOXANPRIBOSYLTRAN-RXN',\n",
       "   'BioPath Reaction': 'RXN01159',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69265',\n",
       "   'KEGG Reaction': 'R01132',\n",
       "   'RHEA': '17973;17974;17975;17976',\n",
       "   'UniPathway Reaction': 'UCR01132;UER00648'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35702_1 or EDU38222_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35702_1 at 0x7fc449887310>,\n",
       "   <Gene EDU38222_1 at 0x7fc449887340>},\n",
       "  '_metabolites': {<Metabolite hxan_c at 0x7fc448e1c970>: -1.0,\n",
       "   <Metabolite prpp_c at 0x7fc463aae0d0>: -1.0,\n",
       "   <Metabolite imp_c at 0x7fc448e1cd60>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HYD2',\n",
       "  'name': 'Hydrogenase (menaquinone8: 2 protons)',\n",
       "  'notes': {'RHEA': '29051;29052;29053;29054',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69267'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36991_1 and EDU36992_1 and EDU36993_1 and EDU36994_1 and EDU36996_1 and EDU36997_1 and EDU36999_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36991_1 at 0x7fc4498878e0>,\n",
       "   <Gene EDU36992_1 at 0x7fc449887940>,\n",
       "   <Gene EDU36993_1 at 0x7fc4498879d0>,\n",
       "   <Gene EDU36994_1 at 0x7fc4498879a0>,\n",
       "   <Gene EDU36996_1 at 0x7fc4498878b0>,\n",
       "   <Gene EDU36997_1 at 0x7fc449887970>,\n",
       "   <Gene EDU36999_1 at 0x7fc449887910>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite h2_c at 0x7fc463a827c0>: -1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HYDFDN',\n",
       "  'name': 'Ferredoxin NADH linked hydrogenase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83134'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37003_1 or (EDU36977_1 and EDU36978_1 and EDU36979_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36977_1 at 0x7fc449887a60>,\n",
       "   <Gene EDU36978_1 at 0x7fc449887a30>,\n",
       "   <Gene EDU36979_1 at 0x7fc449887a90>,\n",
       "   <Gene EDU37003_1 at 0x7fc449887a00>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -3.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: -1.0,\n",
       "   <Metabolite h2_c at 0x7fc463a827c0>: 2.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HYDFDN2r',\n",
       "  'name': 'Ferredoxin NADPH linked hydrogenase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83135'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36977_1 and EDU36978_1 and EDU36979_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36977_1 at 0x7fc449887a60>,\n",
       "   <Gene EDU36978_1 at 0x7fc449887a30>,\n",
       "   <Gene EDU36979_1 at 0x7fc449887a90>},\n",
       "  '_metabolites': {<Metabolite h2_c at 0x7fc463a827c0>: -2.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 3.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HYPOE',\n",
       "  'name': 'Hypothetical enyme',\n",
       "  'notes': {'EC Number': '3.1.3.74',\n",
       "   'BioCyc': 'META:RXN-14046',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3064',\n",
       "   'KEGG Reaction': 'R02494',\n",
       "   'RHEA': '25135;25136;25137;25138',\n",
       "   'UniPathway Reaction': 'UCR02494;UER00302'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite pyam5p_c at 0x7fc448e49a30>: -1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite pydam_c at 0x7fc448f4b220>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'HYXNtex',\n",
       "  'name': 'Hypoxanthine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3056',\n",
       "   'BioCyc': 'META:TRANS-RXN0-562;META:TRANS-RXN0-579'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite hxan_e at 0x7fc448f4b1c0>: -1.0,\n",
       "   <Metabolite hxan_p at 0x7fc448f4b100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'I2FE2SR',\n",
       "  'name': 'ISC [2Fe-2S] regeneration',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83138'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1 or (EDU36273_1 and EDU36274_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>,\n",
       "   <Gene EDU36274_1 at 0x7fc449887ac0>},\n",
       "  '_metabolites': {<Metabolite 2fe1s_c at 0x7fc448e74f10>: -1.0,\n",
       "   <Metabolite iscssh_c at 0x7fc448f4b2b0>: -1.0,\n",
       "   <Metabolite iscu_c at 0x7fc448f4b2e0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 4.0,\n",
       "   <Metabolite iscs_c at 0x7fc448f4b280>: 1.0,\n",
       "   <Metabolite iscu_2fe2s_c at 0x7fc448f4b310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'I2FE2SS2',\n",
       "  'name': 'ISC [2Fe-2S] Synthesis II',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83140'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1 or (EDU36273_1 and EDU36274_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>,\n",
       "   <Gene EDU36274_1 at 0x7fc449887ac0>},\n",
       "  '_metabolites': {<Metabolite fadh2_c at 0x7fc463a82bb0>: -1.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: -2.0,\n",
       "   <Metabolite iscssh_c at 0x7fc448f4b2b0>: -2.0,\n",
       "   <Metabolite iscu_2fe2s_c at 0x7fc448f4b310>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 6.0,\n",
       "   <Metabolite iscs_c at 0x7fc448f4b280>: 2.0,\n",
       "   <Metabolite iscu_2fe2s2_c at 0x7fc448f4b340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'I2FE2ST',\n",
       "  'name': 'ISC [2Fe-2S] Transfer',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83141'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1 and EDU36274_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>,\n",
       "   <Gene EDU36274_1 at 0x7fc449887ac0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -4.0,\n",
       "   <Metabolite iscu_2fe2s_c at 0x7fc448f4b310>: -1.0,\n",
       "   <Metabolite 2fe2s_c at 0x7fc448e74f70>: 1.0,\n",
       "   <Metabolite iscu_c at 0x7fc448f4b2e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'I4FE4SR',\n",
       "  'name': 'ISC [4Fe-4S] Reduction',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83142'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1 and EDU36274_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>,\n",
       "   <Gene EDU36274_1 at 0x7fc449887ac0>},\n",
       "  '_metabolites': {<Metabolite fadh2_c at 0x7fc463a82bb0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite iscu_2fe2s2_c at 0x7fc448f4b340>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: 1.0,\n",
       "   <Metabolite iscu_4fe4s_c at 0x7fc448f4b370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'I4FE4ST',\n",
       "  'name': 'ISC [4Fe-4S] Transfer',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83143'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1 and EDU36274_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>,\n",
       "   <Gene EDU36274_1 at 0x7fc449887ac0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -4.0,\n",
       "   <Metabolite iscu_4fe4s_c at 0x7fc448f4b370>: -1.0,\n",
       "   <Metabolite 4fe4s_c at 0x7fc448eeff10>: 1.0,\n",
       "   <Metabolite iscu_c at 0x7fc448f4b2e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IBTtex',\n",
       "  'name': 'Isobutyrate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82565'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite ibt_e at 0x7fc448f4b3a0>: -1.0,\n",
       "   <Metabolite ibt_p at 0x7fc448eefc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ICDHx',\n",
       "  'name': 'Isocitrate dehydrogenase (NAD)',\n",
       "  'notes': {'EC Number': '1.1.1.41',\n",
       "   'KEGG Reaction': 'R00709',\n",
       "   'UniPathway Reaction': 'UCR00709;UER00996',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31728'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38241_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38241_1 at 0x7fc449887af0>},\n",
       "  '_metabolites': {<Metabolite icit_c at 0x7fc448e1c4c0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite akg_c at 0x7fc463adc370>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ICITt2',\n",
       "  'name': 'Isocitrate transport inout via proton symport',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite icit_e at 0x7fc448f4b3d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite icit_c at 0x7fc448e1c4c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ICYSDS',\n",
       "  'name': 'ISC Cysteine desulfuration',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83144'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>},\n",
       "  '_metabolites': {<Metabolite cys__L_c at 0x7fc463aae3d0>: -1.0,\n",
       "   <Metabolite iscs_c at 0x7fc448f4b280>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite iscssh_c at 0x7fc448f4b2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ID3ACALDt',\n",
       "  'name': 'Indoleacetaldehyde transport  extracellular ',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3075'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite id3acald_c at 0x7fc448e49c10>: -1.0,\n",
       "   <Metabolite id3acald_e at 0x7fc448f4b400>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IDONt2r',\n",
       "  'name': 'L-idonate transport via proton symport, reversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69276',\n",
       "   'EC Number': '3.6.3',\n",
       "   'BioCyc': 'META:TRANS-RXN-181A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36576_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36576_1 at 0x7fc4498204c0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite idon__L_e at 0x7fc448f4b430>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite idon__L_c at 0x7fc463adc430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IDONt2rpp',\n",
       "  'name': 'L-idonate transport via proton symport, reversible (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69276',\n",
       "   'EC Number': '3.6.3',\n",
       "   'BioCyc': 'META:TRANS-RXN-181A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36576_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36576_1 at 0x7fc4498204c0>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite idon__L_p at 0x7fc448f4b460>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite idon__L_c at 0x7fc463adc430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IG3PS',\n",
       "  'name': 'Imidazole-glycerol-3-phosphate synthase',\n",
       "  'notes': {'EC Number': '4.1.3.-',\n",
       "   'BioCyc': 'META:GLUTAMIDOTRANS-RXN',\n",
       "   'BioPath Reaction': 'RXN01139',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31741',\n",
       "   'KEGG Reaction': 'R04558',\n",
       "   'RHEA': '24793;24794;24795;24796',\n",
       "   'UniPathway Reaction': 'UCR04558;UER00010;UER00899'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37192_1 and EDU37195_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37192_1 at 0x7fc449887b20>,\n",
       "   <Gene EDU37195_1 at 0x7fc449887b50>},\n",
       "  '_metabolites': {<Metabolite gln__L_c at 0x7fc463aae3a0>: -1.0,\n",
       "   <Metabolite prlp_c at 0x7fc448f4b4c0>: -1.0,\n",
       "   <Metabolite aicar_c at 0x7fc448e1cc40>: 1.0,\n",
       "   <Metabolite eig3p_c at 0x7fc448f4b490>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IGPDH',\n",
       "  'name': 'Imidazoleglycerol-phosphate dehydratase',\n",
       "  'notes': {'EC Number': '4.2.1.19',\n",
       "   'BioCyc': 'META:IMIDPHOSDEHYD-RXN',\n",
       "   'BioPath Reaction': 'RXN01204',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3090',\n",
       "   'KEGG Reaction': 'R03457',\n",
       "   'RHEA': '11040;11041;11042;11043',\n",
       "   'UniPathway Reaction': 'UCR03457;UER00011'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37196_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37196_1 at 0x7fc449887b80>},\n",
       "  '_metabolites': {<Metabolite eig3p_c at 0x7fc448f4b490>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite imacp_c at 0x7fc448e741f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ILEabc',\n",
       "  'name': 'L-isoleucine transport via ABC system',\n",
       "  'notes': {'RHEA': '29819;29820;29821;29822',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3094',\n",
       "   'BioCyc': 'META:ABC-15-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU38276_1 and EDU38278_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1 and EDU38279_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36982_1 at 0x7fc44983b280>,\n",
       "   <Gene EDU38275_1 at 0x7fc44983b220>,\n",
       "   <Gene EDU38276_1 at 0x7fc44983b250>,\n",
       "   <Gene EDU38278_1 at 0x7fc44983b2e0>,\n",
       "   <Gene EDU38279_1 at 0x7fc44983b2b0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ile__L_e at 0x7fc448f4b4f0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ile__L_c at 0x7fc448e74b80>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ILEabcpp',\n",
       "  'name': 'L-isoleucine transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29819;29820;29821;29822',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3094',\n",
       "   'BioCyc': 'META:ABC-15-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38279_1 or (EDU36982_1 and EDU38275_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1) or (EDU36982_1 and EDU38275_1 and EDU38279_1) or (EDU38276_1 and EDU38277_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38277_1 and EDU38279_1) or (EDU38275_1 and EDU38276_1 and EDU38277_1 and EDU38279_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36982_1 at 0x7fc44983b280>,\n",
       "   <Gene EDU38275_1 at 0x7fc44983b220>,\n",
       "   <Gene EDU38276_1 at 0x7fc44983b250>,\n",
       "   <Gene EDU38277_1 at 0x7fc449887c10>,\n",
       "   <Gene EDU38279_1 at 0x7fc44983b2b0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ile__L_p at 0x7fc448f4b520>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ile__L_c at 0x7fc448e74b80>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ILEDHr',\n",
       "  'name': 'Isoleucine dehydrogenase',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ile__L_c at 0x7fc448e74b80>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite 2o3mpt_c at 0x7fc448ec8d90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ILEt2r',\n",
       "  'name': 'L isoleucine reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3095'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38654_1 or EDU38655_1 or EDU39251_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38654_1 at 0x7fc449887c70>,\n",
       "   <Gene EDU38655_1 at 0x7fc449887c40>,\n",
       "   <Gene EDU39251_1 at 0x7fc449887ca0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite ile__L_e at 0x7fc448f4b4f0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ile__L_c at 0x7fc448e74b80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ILEt2rpp',\n",
       "  'name': 'L-isoleucine reversible transport via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3095'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38655_1 or EDU39251_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38655_1 at 0x7fc449887c40>,\n",
       "   <Gene EDU39251_1 at 0x7fc449887ca0>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite ile__L_p at 0x7fc448f4b520>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ile__L_c at 0x7fc448e74b80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ILETA',\n",
       "  'name': 'Isoleucine transaminase',\n",
       "  'notes': {'EC Number': '2.6.1.42',\n",
       "   'BioCyc': 'META:BRANCHED-CHAINAMINOTRANSFERILEU-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31743',\n",
       "   'KEGG Reaction': 'R02199',\n",
       "   'RHEA': '24801;24802;24803;24804',\n",
       "   'UniPathway Reaction': 'UCR02199;UER00058'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36097_1 or EDU38746_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36097_1 at 0x7fc449887be0>,\n",
       "   <Gene EDU38746_1 at 0x7fc449887bb0>},\n",
       "  '_metabolites': {<Metabolite akg_c at 0x7fc463adc370>: -1.0,\n",
       "   <Metabolite ile__L_c at 0x7fc448e74b80>: -1.0,\n",
       "   <Metabolite 3mop_c at 0x7fc463adcee0>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ILEtex',\n",
       "  'name': 'L-isoleucine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3097'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite ile__L_e at 0x7fc448f4b4f0>: -1.0,\n",
       "   <Metabolite ile__L_p at 0x7fc448f4b520>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IMPC',\n",
       "  'name': 'IMP cyclohydrolase',\n",
       "  'notes': {'EC Number': '3.5.4.10',\n",
       "   'BioCyc': 'META:IMPCYCLOHYDROLASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00218',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3099',\n",
       "   'KEGG Reaction': 'R01127',\n",
       "   'RHEA': '18445;18446;18447;18448',\n",
       "   'UniPathway Reaction': 'UCR01127;UER00135'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35969_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35969_1 at 0x7fc449820fa0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite imp_c at 0x7fc448e1cd60>: -1.0,\n",
       "   <Metabolite fprica_c at 0x7fc448e49610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IMPD',\n",
       "  'name': 'IMP dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.205',\n",
       "   'BioCyc': 'META:IMP-DEHYDROG-RXN',\n",
       "   'BioPath Reaction': 'RXN00783',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3100',\n",
       "   'KEGG Reaction': 'R01130',\n",
       "   'RHEA': '11708;11709;11710;11711',\n",
       "   'UniPathway Reaction': 'UCR01130;UER00295'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37943_1 or EDU38951_1 or EDU39248_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37943_1 at 0x7fc449887d00>,\n",
       "   <Gene EDU38951_1 at 0x7fc449887d30>,\n",
       "   <Gene EDU39248_1 at 0x7fc449887cd0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite imp_c at 0x7fc448e1cd60>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite xmp_c at 0x7fc448f209d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'INOSR',\n",
       "  'name': 'Ketoinositol reductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83150'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35676_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35676_1 at 0x7fc449887d60>},\n",
       "  '_metabolites': {<Metabolite inost_c at 0x7fc448e49130>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0,\n",
       "   <Metabolite 2ins_c at 0x7fc463adc6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'INOSTt4pp',\n",
       "  'name': 'Na+/myo-inositol symporter (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3108'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37759_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37759_1 at 0x7fc449887d90>},\n",
       "  '_metabolites': {<Metabolite inost_p at 0x7fc448f20a60>: -1.0,\n",
       "   <Metabolite na1_p at 0x7fc463adcfd0>: -1.0,\n",
       "   <Metabolite inost_c at 0x7fc448e49130>: 1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'INS2D',\n",
       "  'name': 'Inositol 2-dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.18',\n",
       "   'BioCyc': 'META:MYO-INOSITOL-2-DEHYDROGENASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3109',\n",
       "   'KEGG Reaction': 'R01183',\n",
       "   'RHEA': '16949;16950;16951;16952',\n",
       "   'UniPathway Reaction': 'UCR01183;UER00143'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35676_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35676_1 at 0x7fc449887d60>},\n",
       "  '_metabolites': {<Metabolite inost_c at 0x7fc448e49130>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite 2ins_c at 0x7fc463adc6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'INSH',\n",
       "  'name': 'Inosine hydrolase',\n",
       "  'notes': {'RHEA': '16657;16658;16659;16660',\n",
       "   'BioCyc': 'META:INOSINE-NUCLEOSIDASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69281',\n",
       "   'KEGG Reaction': 'R01770',\n",
       "   'EC Number': '3.2.2.8'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36217_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36217_1 at 0x7fc449887dc0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ins_c at 0x7fc448e1c820>: -1.0,\n",
       "   <Metabolite hxan_c at 0x7fc448e1c970>: 1.0,\n",
       "   <Metabolite rib__D_c at 0x7fc448e1caf0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'INSTabc',\n",
       "  'name': 'Inositol transport in via ABC transport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83151'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37264_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37264_1 at 0x7fc449887df0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite inost_e at 0x7fc448f4b580>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite inost_c at 0x7fc448e49130>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'INSTt2',\n",
       "  'name': 'Inositol transport in via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3112'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite inost_e at 0x7fc448f4b580>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite inost_c at 0x7fc448e49130>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'INSTtex',\n",
       "  'name': 'Inositol transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3114'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite inost_e at 0x7fc448f4b580>: -1.0,\n",
       "   <Metabolite inost_p at 0x7fc448f20a60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IOR2b',\n",
       "  'name': 'Indolepyruvate ferredoxin oxidoreductase (phenylpyvurate)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83152'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36039_1 or (EDU39094_1 and EDU39095_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36039_1 at 0x7fc449887e20>,\n",
       "   <Gene EDU39094_1 at 0x7fc449887e50>,\n",
       "   <Gene EDU39095_1 at 0x7fc449887e80>},\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite phpyr_c at 0x7fc448f4b5e0>: -1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite phaccoa_c at 0x7fc463a827f0>: 1.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IOR3b',\n",
       "  'name': 'Indolepyruvate ferredoxin oxidoreductase (hydroxyphenylpyvurate)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83154'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36039_1 or (EDU39094_1 and EDU39095_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36039_1 at 0x7fc449887e20>,\n",
       "   <Gene EDU39094_1 at 0x7fc449887e50>,\n",
       "   <Gene EDU39095_1 at 0x7fc449887e80>},\n",
       "  '_metabolites': {<Metabolite 34hpp_c at 0x7fc463adc9d0>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hphaccoa_c at 0x7fc463a82f40>: 1.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IORb',\n",
       "  'name': 'Indolepyruvate ferredoxin oxidoreductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83155'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36039_1 or (EDU39094_1 and EDU39095_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36039_1 at 0x7fc449887e20>,\n",
       "   <Gene EDU39094_1 at 0x7fc449887e50>,\n",
       "   <Gene EDU39095_1 at 0x7fc449887e80>},\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite indpyr_c at 0x7fc448f4b550>: -1.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: 1.0,\n",
       "   <Metabolite indaccoa_c at 0x7fc448f4b5b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IPDPS',\n",
       "  'name': '1-hydroxy-2-methyl-2-(E)-butenyl 4-diphosphate reductase (ipdp)',\n",
       "  'notes': {'RHEA': '24492;24493;24494;24495',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR55660',\n",
       "   'KEGG Reaction': 'R08209',\n",
       "   'EC Number': '1.17.1.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37026_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37026_1 at 0x7fc44986a7f0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2mb4p_c at 0x7fc448eef250>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite ipdp_c at 0x7fc448ec8af0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ITCY',\n",
       "  'name': \"ITP:cytidine 5'-phosphotransferase\",\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR73711',\n",
       "   'KEGG Reaction': 'R00962',\n",
       "   'EC Number': '2.7.1.48'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36287_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36287_1 at 0x7fc44985df70>},\n",
       "  '_metabolites': {<Metabolite cytd_c at 0x7fc463a820a0>: -1.0,\n",
       "   <Metabolite itp_c at 0x7fc463aaec10>: -1.0,\n",
       "   <Metabolite cmp_c at 0x7fc448e1cdf0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite idp_c at 0x7fc463aaebe0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'IZPN',\n",
       "  'name': 'Imidazolonepropionase',\n",
       "  'notes': {'EC Number': '3.5.2.7',\n",
       "   'BioCyc': 'META:IMIDAZOLONEPROPIONASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01175',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69293',\n",
       "   'KEGG Reaction': 'R02288',\n",
       "   'RHEA': '23660;23661;23662;23663',\n",
       "   'UniPathway Reaction': 'UCR02288;UER00551'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38653_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38653_1 at 0x7fc449887eb0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 4izp_c at 0x7fc463a82cd0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite forglu_c at 0x7fc45d333d00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'K2L4Aabcpp',\n",
       "  'name': 'KDO(2)-lipid IV A transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '35247;35248;35249;35250',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3143'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37495_1 or EDU38920_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37495_1 at 0x7fc449887ee0>,\n",
       "   <Gene EDU38920_1 at 0x7fc449887f10>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite kdo2lipid4_c at 0x7fc448eef700>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite kdo2lipid4_p at 0x7fc448f4b640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'K2L4Aabctex',\n",
       "  'name': 'KDO(2)-lipid IV A transport via ABC system (periplasm to extracellular)',\n",
       "  'notes': {'RHEA': '35247;35248;35249;35250',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3143'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37217_1 or EDU39397_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37217_1 at 0x7fc44985d700>,\n",
       "   <Gene EDU39397_1 at 0x7fc44986a100>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite kdo2lipid4_p at 0x7fc448f4b640>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite kdo2lipid4_e at 0x7fc448f4b610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'Kabc',\n",
       "  'name': 'Potassium ABC transporter',\n",
       "  'notes': {'RHEA': '16777;16778;16779;16780',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR2979',\n",
       "   'EC Number': '3.6.3.12',\n",
       "   'BioCyc': 'META:3.6.3.12-RXN;META:TRANS-RXN-2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39181_1 and EDU39182_1 and EDU39183_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39181_1 at 0x7fc449887fa0>,\n",
       "   <Gene EDU39182_1 at 0x7fc449887fd0>,\n",
       "   <Gene EDU39183_1 at 0x7fc449887100>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite k_e at 0x7fc448e74940>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite k_c at 0x7fc448e74910>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KARA1',\n",
       "  'name': 'Ketol-acid reductoisomerase (2,3-dihydroxy-3-methylbutanoate)',\n",
       "  'notes': {'EC Number': '1.1.1.86',\n",
       "   'BioCyc': 'META:ACETOLACTREDUCTOISOM-RXN;META:RXN-16061',\n",
       "   'BioPath Reaction': 'RXN01210',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69294',\n",
       "   'KEGG Reaction': 'R04439',\n",
       "   'RHEA': '22068;22069;22070;22071',\n",
       "   'UniPathway Reaction': 'UER00060'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 23dhmb_c at 0x7fc463a6d7f0>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite alac__S_c at 0x7fc448e1c130>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KARA2',\n",
       "  'name': 'Ketol-acid reductoisomerase (2-Acetolactate)',\n",
       "  'notes': {'EC Number': '1.1.1.86',\n",
       "   'BioCyc': 'META:ACETOOHBUTREDUCTOISOM-RXN;META:RXN-16062',\n",
       "   'BioPath Reaction': 'RXN01320',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR72831',\n",
       "   'RHEA': '13493;13494;13495;13496',\n",
       "   'UniPathway Reaction': 'UER00056'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2ahbut_c at 0x7fc463aaee80>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 23dhmp_c at 0x7fc463a6d8b0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS13',\n",
       "  'name': 'B-ketoacyl synthetase (octadecanoate)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3151', 'EC Number': '2.3.1.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38327_1 or EDU38560_1 or (EDU36255_1 and EDU38327_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38327_1 at 0x7fc449887f40>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -23.0,\n",
       "   <Metabolite malcoa_c at 0x7fc463a82100>: -8.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -16.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 8.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 9.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 7.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 16.0,\n",
       "   <Metabolite ocdca_c at 0x7fc463a6de20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS14',\n",
       "  'name': 'Beta-ketoacyl-ACP synthase',\n",
       "  'notes': {'BioCyc': 'META:3-OXOACYL-ACP-SYNTH-BASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR83172',\n",
       "   'KEGG Reaction': 'R04355',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38146_1 or EDU38560_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38146_1 at 0x7fc449820340>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite acACP_c at 0x7fc448e1c430>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite actACP_c at 0x7fc463a82430>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS15',\n",
       "  'name': 'Beta-ketoacyl-ACP synthase (2)',\n",
       "  'notes': {'BioCyc': 'META:2.3.1.180-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3153',\n",
       "   'KEGG Reaction': 'R10707',\n",
       "   'EC Number': '2.3.1.180'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1 or EDU38141_1 or EDU38560_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38141_1 at 0x7fc449820970>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite actACP_c at 0x7fc463a82430>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS16',\n",
       "  'name': '3-hydroxy-myristoyl-ACP synthesis',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69295'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38560_1 or (EDU37093_1 and EDU38560_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37093_1 at 0x7fc449887f70>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite ddcaACP_c at 0x7fc463a82580>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 3hmrsACP_c at 0x7fc463adcac0>: 1.0,\n",
       "   <Metabolite ACP_c at 0x7fc463adca30>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS17',\n",
       "  'name': 'B-ketoacyl synthetase (n-C18:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83173'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38327_1 or EDU38560_1 or (EDU36255_1 and EDU38327_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38327_1 at 0x7fc449887f40>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -22.0,\n",
       "   <Metabolite malcoa_c at 0x7fc463a82100>: -8.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -15.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 8.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 9.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 7.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 15.0,\n",
       "   <Metabolite ocdcea_c at 0x7fc463a6de80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS2',\n",
       "  'name': 'B-ketoacyl synthetase (n-C14:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3157', 'EC Number': '2.3.1.41'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38327_1 or EDU38560_1 or (EDU36255_1 and EDU38327_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38327_1 at 0x7fc449887f40>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -17.0,\n",
       "   <Metabolite malcoa_c at 0x7fc463a82100>: -6.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -12.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 6.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 7.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 5.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 12.0,\n",
       "   <Metabolite ttdca_c at 0x7fc463a6dc70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS7',\n",
       "  'name': 'B-ketoacyl synthetase (n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3163'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38327_1 or EDU38560_1 or (EDU36255_1 and EDU38327_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>,\n",
       "   <Gene EDU38327_1 at 0x7fc449887f40>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -19.0,\n",
       "   <Metabolite malcoa_c at 0x7fc463a82100>: -7.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -13.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 7.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 8.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 6.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 13.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KAS8',\n",
       "  'name': 'B-ketoacyl synthetase (palmitate, n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3164', 'EC Number': '2.3.1.85'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38327_1 or EDU38560_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38327_1 at 0x7fc449887f40>,\n",
       "   <Gene EDU38560_1 at 0x7fc449820940>},\n",
       "  '_metabolites': {<Metabolite accoa_c at 0x7fc463a82700>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -20.0,\n",
       "   <Metabolite malcoa_c at 0x7fc463a82100>: -7.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -14.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 7.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 8.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 6.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 14.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KDOCT2',\n",
       "  'name': '3-deoxy-manno-octulosonate cytidylyltransferase',\n",
       "  'notes': {'BioCyc': 'META:CPM-KDOSYNTH-RXN',\n",
       "   'EC Number': '2.7.7.38',\n",
       "   'KEGG Reaction': 'R03351',\n",
       "   'UniPathway Reaction': 'UCR03351;UER00476',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69298'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite kdo_c at 0x7fc448f4b6a0>: -1.0,\n",
       "   <Metabolite ckdo_c at 0x7fc448f4b670>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KDOPP',\n",
       "  'name': '3-deoxy-manno-octulosonate-8-phosphatase',\n",
       "  'notes': {'BioCyc': 'META:KDO-8PPHOSPHAT-RXN',\n",
       "   'EC Number': '3.1.3.45',\n",
       "   'KEGG Reaction': 'R03350',\n",
       "   'UniPathway Reaction': 'UCR03350;UER00475',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3175'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite kdo8p_c at 0x7fc448f4b6d0>: -1.0,\n",
       "   <Metabolite kdo_c at 0x7fc448f4b6a0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'KDOPS',\n",
       "  'name': '3-deoxy -D-manno-octulosonic -acid 8-phosphate synthase',\n",
       "  'notes': {'BioCyc': 'META:KDO-8PSYNTH-RXN',\n",
       "   'EC Number': '2.5.1.55',\n",
       "   'KEGG Reaction': 'R03254',\n",
       "   'UniPathway Reaction': 'UCR03254;UER00474',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3176'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ara5p_c at 0x7fc463aae490>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite kdo8p_c at 0x7fc448f4b6d0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'Kt2pp',\n",
       "  'name': 'Potassium transport in via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3188',\n",
       "   'BioCyc': 'META:TRANS-RXN-3;META:TRANS-RXN-42'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37113_1 or EDU37475_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37113_1 at 0x7fc4498960d0>,\n",
       "   <Gene EDU37475_1 at 0x7fc449887070>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite k_p at 0x7fc448e9f550>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite k_c at 0x7fc448e74910>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'Kt2r',\n",
       "  'name': 'Potassium reversible transport  via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3188',\n",
       "   'BioCyc': 'META:TRANS-RXN-3;META:TRANS-RXN-42'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37474_1 or EDU39406_1 or (EDU39405_1 and EDU39406_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37474_1 at 0x7fc449896130>,\n",
       "   <Gene EDU39405_1 at 0x7fc449896040>,\n",
       "   <Gene EDU39406_1 at 0x7fc4498960a0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite k_e at 0x7fc448e74940>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite k_c at 0x7fc448e74910>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'Ktex',\n",
       "  'name': 'Potassium transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3189',\n",
       "   'BioCyc': 'META:TRANS-RXN-143;META:TRANS-RXN-185;META:TRANS-RXN-186;META:TRANS-RXN-187;META:TRANS-RXN-190'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite k_e at 0x7fc448e74940>: -1.0,\n",
       "   <Metabolite k_p at 0x7fc448e9f550>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LACD',\n",
       "  'name': 'Lactate dehydrogenase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83182'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38862_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38862_1 at 0x7fc449879f70>},\n",
       "  '_metabolites': {<Metabolite mql8_c at 0x7fc448e49f70>: -1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: -2.0,\n",
       "   <Metabolite dhlplarg_c at 0x7fc448f20d30>: -1.0,\n",
       "   <Metabolite lac__D_c at 0x7fc448e1c0a0>: 2.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: 1.0,\n",
       "   <Metabolite lplarg_c at 0x7fc448f20d60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LacR',\n",
       "  'name': 'LacR',\n",
       "  'notes': {'EC Number': '5.1.2.1',\n",
       "   'BioCyc': 'META:LACTATE-RACEMASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00541',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3361',\n",
       "   'RHEA': '10960;10961;10962;10963',\n",
       "   'KEGG Reaction': 'R01450'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36789_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36789_1 at 0x7fc449896430>},\n",
       "  '_metabolites': {<Metabolite lac__L_c at 0x7fc463adc8b0>: -1.0,\n",
       "   <Metabolite lac__D_c at 0x7fc448e1c0a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LACZpp',\n",
       "  'name': 'B-galactosidase',\n",
       "  'notes': {'RHEA': '10076;10077;10078;10079',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31780',\n",
       "   'KEGG Reaction': 'R01100;R06114',\n",
       "   'EC Number': '3.2.1.23'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite lcts_p at 0x7fc448f4b760>: -1.0,\n",
       "   <Metabolite gal_p at 0x7fc448f203a0>: 1.0,\n",
       "   <Metabolite glc__D_p at 0x7fc448f200d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LALALGLUtex',\n",
       "  'name': 'L-alanyl-L-glutamate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82471'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite LalaLglu_e at 0x7fc448f4b790>: -1.0,\n",
       "   <Metabolite LalaLglu_p at 0x7fc448f4b7c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LALALGLUtpp',\n",
       "  'name': 'L-alanyl-L-glutamate transport in via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83184'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36499_1 or EDU36616_1 or EDU37071_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36499_1 at 0x7fc449896160>,\n",
       "   <Gene EDU36616_1 at 0x7fc449896190>,\n",
       "   <Gene EDU37071_1 at 0x7fc4498961c0>},\n",
       "  '_metabolites': {<Metabolite LalaLglu_p at 0x7fc448f4b7c0>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite LalaLglu_c at 0x7fc448e499a0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LALGP',\n",
       "  'name': 'L-alanyl-gamma-L-glutamate peptidase',\n",
       "  'notes': {'RHEA': '29335;29336;29337;29338',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR39728',\n",
       "   'EC Number': '3.4.13.18',\n",
       "   'BioCyc': 'META:RXN0-6981'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35978_1 or EDU36540_1 or EDU37387_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35978_1 at 0x7fc44983b670>,\n",
       "   <Gene EDU36540_1 at 0x7fc44983b640>,\n",
       "   <Gene EDU37387_1 at 0x7fc44983b6a0>},\n",
       "  '_metabolites': {<Metabolite LalaLglu_c at 0x7fc448e499a0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite ala__L_c at 0x7fc463a82c10>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LCARS',\n",
       "  'name': 'Lacaldehyde reductase (S-propane-1,2-diol forming)',\n",
       "  'notes': {'RHEA': '15933;15934;15935;15936',\n",
       "   'BioCyc': 'META:LACTALDREDUCT-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR81650',\n",
       "   'KEGG Reaction': 'R02258',\n",
       "   'EC Number': '1.1.1.77'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38973_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38973_1 at 0x7fc449820550>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite lald__L_c at 0x7fc448e49bb0>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 12ppd__S_c at 0x7fc463a6d4c0>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LCTSabc',\n",
       "  'name': 'LCTSabc',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3207'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37262_1 or (EDU37262_1 and EDU37263_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37262_1 at 0x7fc449879eb0>,\n",
       "   <Gene EDU37263_1 at 0x7fc449879e80>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite lcts_e at 0x7fc448f4b700>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lcts_c at 0x7fc448f4b730>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LCTStpp',\n",
       "  'name': 'Lactose transport via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3208'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite lcts_p at 0x7fc448f4b760>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lcts_c at 0x7fc448f4b730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LCYSTabc',\n",
       "  'name': 'Cysteate transport via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36524_1 and EDU39295_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36524_1 at 0x7fc4498961f0>,\n",
       "   <Gene EDU39295_1 at 0x7fc44983bfa0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite Lcyst_e at 0x7fc448f4b7f0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite Lcyst_c at 0x7fc463adc340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LCYSTabcpp',\n",
       "  'name': 'L-cysteate transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR27800'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39370_1 or (EDU39288_1 and EDU39293_1 and EDU39295_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39288_1 at 0x7fc449896250>,\n",
       "   <Gene EDU39293_1 at 0x7fc449896220>,\n",
       "   <Gene EDU39295_1 at 0x7fc44983bfa0>,\n",
       "   <Gene EDU39370_1 at 0x7fc44983bfd0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite Lcyst_p at 0x7fc448f4b820>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite Lcyst_c at 0x7fc463adc340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LCYSTtex',\n",
       "  'name': 'L-cysteate transport via ABC system  (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR82472'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite Lcyst_e at 0x7fc448f4b7f0>: -1.0,\n",
       "   <Metabolite Lcyst_p at 0x7fc448f4b820>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LDGUNPD',\n",
       "  'name': 'Linear diguanylate phosphodiesterase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83191'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 5pg35pg_c at 0x7fc448e9f760>: -1.0,\n",
       "   <Metabolite gmp_c at 0x7fc448e1cb50>: 2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LDH_D',\n",
       "  'name': 'D-lactate dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.28',\n",
       "   'BioCyc': 'META:DLACTDEHYDROGNAD-RXN',\n",
       "   'BioPath Reaction': 'RXN00296',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3212',\n",
       "   'RHEA': '16369;16370;16371;16372',\n",
       "   'KEGG Reaction': 'R00704'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39261_1 or EDU39262_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39261_1 at 0x7fc4498962b0>,\n",
       "   <Gene EDU39262_1 at 0x7fc449896280>},\n",
       "  '_metabolites': {<Metabolite lac__D_c at 0x7fc448e1c0a0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LDH_L',\n",
       "  'name': 'L-lactate dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.27',\n",
       "   'BioCyc': 'META:L-LACTATE-DEHYDROGENASE-RXN',\n",
       "   'BioPath Reaction': 'RXN00795',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3215',\n",
       "   'KEGG Reaction': 'R00703',\n",
       "   'RHEA': '23444;23445;23446;23447',\n",
       "   'UniPathway Reaction': 'UCR00703;UER00611'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37260_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37260_1 at 0x7fc4498962e0>},\n",
       "  '_metabolites': {<Metabolite lac__L_c at 0x7fc463adc8b0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LEUabc',\n",
       "  'name': 'L-leucine transport via ABC system',\n",
       "  'notes': {'RHEA': '29807;29808;29809;29810',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3222',\n",
       "   'BioCyc': 'META:ABC-35-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU38276_1 and EDU38278_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1 and EDU38279_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36982_1 at 0x7fc44983b280>,\n",
       "   <Gene EDU38275_1 at 0x7fc44983b220>,\n",
       "   <Gene EDU38276_1 at 0x7fc44983b250>,\n",
       "   <Gene EDU38278_1 at 0x7fc44983b2e0>,\n",
       "   <Gene EDU38279_1 at 0x7fc44983b2b0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite leu__L_e at 0x7fc463adc9a0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite leu__L_c at 0x7fc463adc970>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LEUabcpp',\n",
       "  'name': 'L-leucine transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29807;29808;29809;29810',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3222',\n",
       "   'BioCyc': 'META:ABC-35-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38279_1 or (EDU36982_1 and EDU38275_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1) or (EDU36982_1 and EDU38275_1 and EDU38279_1) or (EDU38276_1 and EDU38277_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38276_1 and EDU38279_1) or (EDU36982_1 and EDU38275_1 and EDU38277_1 and EDU38279_1) or (EDU38275_1 and EDU38276_1 and EDU38277_1 and EDU38279_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36982_1 at 0x7fc44983b280>,\n",
       "   <Gene EDU38275_1 at 0x7fc44983b220>,\n",
       "   <Gene EDU38276_1 at 0x7fc44983b250>,\n",
       "   <Gene EDU38277_1 at 0x7fc449887c10>,\n",
       "   <Gene EDU38279_1 at 0x7fc44983b2b0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite leu__L_p at 0x7fc448f4b850>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite leu__L_c at 0x7fc463adc970>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LEUt2r',\n",
       "  'name': 'L leucine reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3223',\n",
       "   'BioCyc': 'META:TRANS-RXN0-270'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38654_1 or EDU38655_1 or EDU39251_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38654_1 at 0x7fc449887c70>,\n",
       "   <Gene EDU38655_1 at 0x7fc449887c40>,\n",
       "   <Gene EDU39251_1 at 0x7fc449887ca0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite leu__L_e at 0x7fc463adc9a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite leu__L_c at 0x7fc463adc970>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LEUt2rpp',\n",
       "  'name': 'L-leucine reversible transport via proton symport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3223',\n",
       "   'BioCyc': 'META:TRANS-RXN0-270'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38655_1 or EDU39251_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38655_1 at 0x7fc449887c40>,\n",
       "   <Gene EDU39251_1 at 0x7fc449887ca0>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite leu__L_p at 0x7fc448f4b850>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite leu__L_c at 0x7fc463adc970>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LEUt4rpp',\n",
       "  'name': 'L-leucine transport in via sodium symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3224',\n",
       "   'BioCyc': 'META:TRANS-RXN-126B'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38655_1 or EDU39251_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38655_1 at 0x7fc449887c40>,\n",
       "   <Gene EDU39251_1 at 0x7fc449887ca0>},\n",
       "  '_metabolites': {<Metabolite leu__L_p at 0x7fc448f4b850>: -1.0,\n",
       "   <Metabolite na1_p at 0x7fc463adcfd0>: -1.0,\n",
       "   <Metabolite leu__L_c at 0x7fc463adc970>: 1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LEUTA',\n",
       "  'name': 'Leucine transaminase',\n",
       "  'notes': {'EC Number': '2.6.1.67',\n",
       "   'BioCyc': 'META:BRANCHED-CHAINAMINOTRANSFERLEU-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR31792',\n",
       "   'KEGG Reaction': 'R01090',\n",
       "   'RHEA': '18321;18322;18323;18324',\n",
       "   'UniPathway Reaction': 'UCR01090;UER00073;UER00857'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36097_1 or EDU38746_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36097_1 at 0x7fc449887be0>,\n",
       "   <Gene EDU38746_1 at 0x7fc449887bb0>},\n",
       "  '_metabolites': {<Metabolite akg_c at 0x7fc463adc370>: -1.0,\n",
       "   <Metabolite leu__L_c at 0x7fc463adc970>: -1.0,\n",
       "   <Metabolite 4mop_c at 0x7fc463a82d00>: 1.0,\n",
       "   <Metabolite glu__L_c at 0x7fc463a6d190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LEUtex',\n",
       "  'name': 'L-leucine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3225', 'BioCyc': 'META:RXN66-447'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite leu__L_e at 0x7fc463adc9a0>: -1.0,\n",
       "   <Metabolite leu__L_p at 0x7fc448f4b850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LGTHL',\n",
       "  'name': 'Lactoylglutathione lyase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83040'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38457_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38457_1 at 0x7fc449896310>},\n",
       "  '_metabolites': {<Metabolite gthrd_c at 0x7fc448e74a60>: -1.0,\n",
       "   <Metabolite mthgxl_c at 0x7fc463aae640>: -1.0,\n",
       "   <Metabolite lgt__S_c at 0x7fc448f209a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LIPAabcpp',\n",
       "  'name': 'Lipid A transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '35251;35252;35253;35254',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3240'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37495_1 or EDU38920_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37495_1 at 0x7fc449887ee0>,\n",
       "   <Gene EDU38920_1 at 0x7fc449887f10>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite lipa_c at 0x7fc448eef760>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite lipa_p at 0x7fc448f4b8b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LIPAabctex',\n",
       "  'name': 'Lipid A transport via ABC system (periplasm to extracellular)',\n",
       "  'notes': {'RHEA': '35251;35252;35253;35254',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3240'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37217_1 or EDU39397_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37217_1 at 0x7fc44985d700>,\n",
       "   <Gene EDU39397_1 at 0x7fc44986a100>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite lipa_p at 0x7fc448f4b8b0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite lipa_e at 0x7fc448f4b880>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LIPACabcpp',\n",
       "  'name': 'Lipid (cold) A transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69316'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37495_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37495_1 at 0x7fc449887ee0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite lipa_cold_c at 0x7fc448eef7c0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite lipa_cold_p at 0x7fc448e9fc40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LIPOS',\n",
       "  'name': 'Lipoate synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83200'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37446_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37446_1 at 0x7fc449896340>},\n",
       "  '_metabolites': {<Metabolite 4fe4s_c at 0x7fc448eeff10>: -1.0,\n",
       "   <Metabolite amet_c at 0x7fc463adc940>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite octapb_c at 0x7fc448f4b910>: -1.0,\n",
       "   <Metabolite 2fe2s_c at 0x7fc448e74f70>: 1.0,\n",
       "   <Metabolite dad_5_c at 0x7fc463aae070>: 2.0,\n",
       "   <Metabolite fe2_c at 0x7fc448e74490>: 2.0,\n",
       "   <Metabolite lipopb_c at 0x7fc448f4b8e0>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 2.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'L_LACD2',\n",
       "  'name': 'L-Lactate dehydrogenase (ubiquinone)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR55668', 'EC Number': '1.1.2.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39422_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39422_1 at 0x7fc449896400>},\n",
       "  '_metabolites': {<Metabolite lac__L_c at 0x7fc463adc8b0>: -1.0,\n",
       "   <Metabolite q8_c at 0x7fc448e749a0>: -1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0,\n",
       "   <Metabolite q8h2_c at 0x7fc448e749d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'L_LACD3',\n",
       "  'name': 'L-Lactate dehydrogenase (menaquinone)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69304', 'EC Number': '1.1.2.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39422_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39422_1 at 0x7fc449896400>},\n",
       "  '_metabolites': {<Metabolite lac__L_c at 0x7fc463adc8b0>: -1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPADSS',\n",
       "  'name': 'Lipid A disaccaride synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83210'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite lipidX_c at 0x7fc448f4b970>: -1.0,\n",
       "   <Metabolite u23ga_c at 0x7fc448f4b9a0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lipidAds_c at 0x7fc448f4b940>: 1.0,\n",
       "   <Metabolite udp_c at 0x7fc463a6d670>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPD5',\n",
       "  'name': 'Dihydrolipoamide dehydrogenase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83219'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37132_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37132_1 at 0x7fc449820a30>},\n",
       "  '_metabolites': {<Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite dhlplarg_c at 0x7fc448f20d30>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite lplarg_c at 0x7fc448f20d60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1E120pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoethanolamine, n-C12:0) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69338'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpe120_p at 0x7fc448f4b9d0>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite ddca_p at 0x7fc448ec8bb0>: 1.0,\n",
       "   <Metabolite g3pe_p at 0x7fc448f201f0>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1E140pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoethanolamine, n-C14:0) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69339'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpe140_p at 0x7fc448f4ba00>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite g3pe_p at 0x7fc448f201f0>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0,\n",
       "   <Metabolite ttdca_p at 0x7fc448eefaf0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1E181pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoethanolamine, n-C18:1) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3282'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpe181_p at 0x7fc448f4ba30>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite g3pe_p at 0x7fc448f201f0>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0,\n",
       "   <Metabolite ocdcea_p at 0x7fc448eefbe0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1G120pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoglycerol, n-C12:0) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69344'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpg120_p at 0x7fc448f4ba60>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite ddca_p at 0x7fc448ec8bb0>: 1.0,\n",
       "   <Metabolite g3pg_p at 0x7fc448f20280>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1G140pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoglycerol, n-C14:0) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3284'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpg140_p at 0x7fc45d333d30>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite g3pg_p at 0x7fc448f20280>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0,\n",
       "   <Metabolite ttdca_p at 0x7fc448eefaf0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1G160pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoglycerol, n-C16:0) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3286'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpg160_p at 0x7fc45d375670>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite g3pg_p at 0x7fc448f20280>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0,\n",
       "   <Metabolite hdca_p at 0x7fc448eefb50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1G180pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoglycerol, n-C18:0) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3288'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpg180_p at 0x7fc448f4baf0>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite g3pg_p at 0x7fc448f20280>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0,\n",
       "   <Metabolite ocdca_p at 0x7fc448eefbb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL1G181pp',\n",
       "  'name': 'Lysophospholipase L1 (2-acylglycerophosphoglycerol, n-C18:1) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83223'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35945_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35945_1 at 0x7fc449896370>},\n",
       "  '_metabolites': {<Metabolite 1agpg181_p at 0x7fc45d333460>: -1.0,\n",
       "   <Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite g3pg_p at 0x7fc448f20280>: 1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: 1.0,\n",
       "   <Metabolite ocdcea_p at 0x7fc448eefbe0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL2A161',\n",
       "  'name': 'Lysophospholipase L2 (2-acylglycerophosphotidate, n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3294'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2hdec9eg3p_c at 0x7fc463a6d760>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite glyc3p_c at 0x7fc448e742e0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL2E160',\n",
       "  'name': 'Lysophospholipase L2 (2-acylglycerophosphoethanolamine, n-C16:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69370', 'EC Number': '3.1.1.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38330_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38330_1 at 0x7fc4498963a0>},\n",
       "  '_metabolites': {<Metabolite 2agpe160_c at 0x7fc463a6da30>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite g3pe_c at 0x7fc448f201c0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hdca_c at 0x7fc463a6dd60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL2E180',\n",
       "  'name': 'Lysophospholipase L2 (2-acylglycerophosphoethanolamine, n-C18:0)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69371', 'EC Number': '3.1.1.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38330_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38330_1 at 0x7fc4498963a0>},\n",
       "  '_metabolites': {<Metabolite 2agpe180_c at 0x7fc463a6dac0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite g3pe_c at 0x7fc448f201c0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ocdca_c at 0x7fc463a6de20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL2G141',\n",
       "  'name': 'Lysophospholipase L2 (2-acylglycerophosphoglycerol, n-C14:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83226'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg141_c at 0x7fc463a6dcd0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite g3pg_c at 0x7fc448f20250>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite ttdcea_c at 0x7fc463a6dd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LPLIPAL2G161',\n",
       "  'name': 'Lysophospholipase L2 (2-acylglycerophosphoglycerol, n-C16:1)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3322'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 2agpg161_c at 0x7fc463adc130>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite g3pg_c at 0x7fc448f20250>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite hdcea_c at 0x7fc463a6ddc0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LSERDHr',\n",
       "  'name': 'L-serine dehydrogenase',\n",
       "  'notes': {'RHEA': '21596;21597;21598;21599',\n",
       "   'BioCyc': 'META:RXN0-2201',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69383',\n",
       "   'KEGG Reaction': 'R06126',\n",
       "   'EC Number': '1.1.1.276'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36536_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36536_1 at 0x7fc44983bf10>},\n",
       "  '_metabolites': {<Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite ser__L_c at 0x7fc463aae400>: -1.0,\n",
       "   <Metabolite 2amsa_c at 0x7fc448eef430>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LYSabcpp',\n",
       "  'name': 'L-lysine transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '29887;29888;29889;29890',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3354',\n",
       "   'EC Number': '3.6.3.21',\n",
       "   'BioCyc': 'META:ABC-3-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37427_1 or (EDU37362_1 and EDU37652_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37362_1 at 0x7fc449879af0>,\n",
       "   <Gene EDU37427_1 at 0x7fc44983ba90>,\n",
       "   <Gene EDU37652_1 at 0x7fc44985da90>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite lys__L_p at 0x7fc448e9f3a0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lys__L_c at 0x7fc448e74310>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LYSDC',\n",
       "  'name': 'Lysine decarboxylase',\n",
       "  'notes': {'EC Number': '4.1.1.18',\n",
       "   'BioCyc': 'META:LYSDECARBOX-RXN',\n",
       "   'BioPath Reaction': 'RXN00827',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3346',\n",
       "   'RHEA': '22352;22353;22354;22355',\n",
       "   'KEGG Reaction': 'R00462'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38084_1 or EDU39429_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38084_1 at 0x7fc44983b8b0>,\n",
       "   <Gene EDU39429_1 at 0x7fc44983b880>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite lys__L_c at 0x7fc448e74310>: -1.0,\n",
       "   <Metabolite 15dap_c at 0x7fc448e742b0>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LYSt2r',\n",
       "  'name': 'L lysine reversible transport via proton symport',\n",
       "  'notes': {'RHEA': '28911;28912;28913;28914',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3355',\n",
       "   'BioCyc': 'META:TRANS-RXN-58'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37871_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37871_1 at 0x7fc4498963d0>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite lys__L_e at 0x7fc448e745b0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite lys__L_c at 0x7fc448e74310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'LYStex',\n",
       "  'name': 'L-lysine transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3357',\n",
       "   'BioCyc': 'META:TRANS-RXN0-569'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite lys__L_e at 0x7fc448e745b0>: -1.0,\n",
       "   <Metabolite lys__L_p at 0x7fc448e9f3a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'M1PD',\n",
       "  'name': 'Mannitol-1-phosphate 5-dehydrogenase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83262',\n",
       "   'KEGG Reaction': 'R00758',\n",
       "   'EC Number': '1.1.1.17'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite mnl1p_c at 0x7fc448f4bbe0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite f6p_c at 0x7fc448e49430>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MACCOAT',\n",
       "  'name': 'MACCOAT',\n",
       "  'notes': {'EC Number': '2.3.1.9',\n",
       "   'BioCyc': 'META:METHYLACETOACETYLCOATHIOL-RXN',\n",
       "   'BioPath Reaction': 'RXN01541',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR342',\n",
       "   'RHEA': '30719;30720;30721;30722',\n",
       "   'KEGG Reaction': 'R00927'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39341_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39341_1 at 0x7fc449820370>},\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite 2maacoa_c at 0x7fc448f20c70>: -1.0,\n",
       "   <Metabolite accoa_c at 0x7fc463a82700>: 1.0,\n",
       "   <Metabolite ppcoa_c at 0x7fc463aaeb20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MACPD',\n",
       "  'name': 'Malonyl-ACP decarboxylase',\n",
       "  'notes': {'BioCyc': 'META:MALONYL-ACPDECARBOX-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3381',\n",
       "   'KEGG Reaction': 'R09482',\n",
       "   'EC Number': '4.1.1.87'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36255_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36255_1 at 0x7fc4498202b0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: -1.0,\n",
       "   <Metabolite acACP_c at 0x7fc448e1c430>: 1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MAL6PG',\n",
       "  'name': 'Maltose 6  phosphate glucosidase',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37844_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37844_1 at 0x7fc449879280>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite malt6p_c at 0x7fc448f4bb80>: -1.0,\n",
       "   <Metabolite g6p_c at 0x7fc463aae340>: 1.0,\n",
       "   <Metabolite glc__D_c at 0x7fc463a6d4f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALT',\n",
       "  'name': 'Alpha glucosidase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3391'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37161_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37161_1 at 0x7fc449896460>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite malt_c at 0x7fc448e49fa0>: -1.0,\n",
       "   <Metabolite glc__D_c at 0x7fc463a6d4f0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALt2_2pp',\n",
       "  'name': 'Malate transport via proton symport (2 H) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3407',\n",
       "   'BioCyc': 'META:TRANS-RXN-121A'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39112_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39112_1 at 0x7fc44983beb0>},\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -2.0,\n",
       "   <Metabolite mal__L_p at 0x7fc448f4bb50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite mal__L_c at 0x7fc448e49730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALt2r',\n",
       "  'name': 'L malate reversible transport via proton symport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3409'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38390_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38390_1 at 0x7fc449879580>},\n",
       "  '_metabolites': {<Metabolite h_e at 0x7fc463adc580>: -1.0,\n",
       "   <Metabolite mal__L_e at 0x7fc448e49760>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mal__L_c at 0x7fc448e49730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALTabc',\n",
       "  'name': 'Maltose transport via ABC system',\n",
       "  'notes': {'RHEA': '22132;22133;22134;22135',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3403',\n",
       "   'EC Number': '3.6.3.19',\n",
       "   'BioCyc': 'META:3.6.3.19-RXN;META:ABC-16-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite malt_e at 0x7fc448e74070>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite malt_c at 0x7fc448e49fa0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALtpp',\n",
       "  'name': 'Malate transport via diffusion (periplasm to cytosol)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3409'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite mal__L_p at 0x7fc448f4bb50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mal__L_c at 0x7fc448e49730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALTpts',\n",
       "  'name': 'Maltose transport via PEP:Pyr PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3404'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU36437_1 and EDU39079_1) or (EDU36006_1 and EDU36437_1 and EDU37855_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36006_1 at 0x7fc449820790>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU37855_1 at 0x7fc449879970>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite malt_e at 0x7fc448e74070>: -1.0,\n",
       "   <Metabolite malt6p_c at 0x7fc448f4bb80>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALTptspp',\n",
       "  'name': 'Maltose transport via PEP:Pyr PTS (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3404'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '(EDU36006_1 and EDU36437_1) or (EDU36437_1 and EDU37843_1) or (EDU36006_1 and EDU36437_1 and EDU37855_1) or (EDU36437_1 and EDU37843_1 and EDU37855_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36006_1 at 0x7fc449820790>,\n",
       "   <Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU37843_1 at 0x7fc4498207f0>,\n",
       "   <Gene EDU37855_1 at 0x7fc449879970>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite malt_p at 0x7fc448f4bbb0>: -1.0,\n",
       "   <Metabolite malt6p_c at 0x7fc448f4bb80>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MALTtexi',\n",
       "  'name': 'MaltoseMaltotriose transport via diffusion (extracellular to periplasm) irreversible',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3405',\n",
       "   'BioCyc': 'META:RXN0-1741;META:RXN-14351;META:TRANS-RXN0-575;META:TRANS-RXN-239'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite malt_e at 0x7fc448e74070>: -1.0,\n",
       "   <Metabolite malt_p at 0x7fc448f4bbb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MAN1PT2',\n",
       "  'name': 'Mannose-1-phosphate guanylyltransferase (GDP)',\n",
       "  'notes': {'EC Number': '2.7.7.22',\n",
       "   'BioCyc': 'META:MANNPGUANYLTRANGDP-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69400',\n",
       "   'KEGG Reaction': 'R00883',\n",
       "   'RHEA': '12905;12906;12907;12908',\n",
       "   'UniPathway Reaction': 'UCR00883;UER00425'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38036_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38036_1 at 0x7fc449896490>},\n",
       "  '_metabolites': {<Metabolite gdp_c at 0x7fc463adc850>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite man1p_c at 0x7fc448e1cbe0>: -1.0,\n",
       "   <Metabolite gdpmann_c at 0x7fc448e74bb0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MAN6PI',\n",
       "  'name': 'Mannose-6-phosphate isomerase',\n",
       "  'notes': {'RHEA': '12356;12357;12358;12359',\n",
       "   'EC Number': '5.3.1.8',\n",
       "   'KEGG Reaction': 'R00772',\n",
       "   'UniPathway Reaction': 'UCR00772;UER00423',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR83276'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37708_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37708_1 at 0x7fc4498964c0>},\n",
       "  '_metabolites': {<Metabolite man6p_c at 0x7fc448e1c0d0>: -1.0,\n",
       "   <Metabolite f6p_c at 0x7fc448e49430>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MANA1',\n",
       "  'name': 'Mannase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83277'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite manb_c at 0x7fc448f4ba90>: -1.0,\n",
       "   <Metabolite man_c at 0x7fc448e49fd0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MANA2',\n",
       "  'name': 'Mannase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83278'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mantr_c at 0x7fc448f4bb20>: -1.0,\n",
       "   <Metabolite man_c at 0x7fc448e49fd0>: 1.0,\n",
       "   <Metabolite manb_c at 0x7fc448f4ba90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MANBabc',\n",
       "  'name': 'Mannobiose transport via ABC transporter',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83280'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37083_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37083_1 at 0x7fc44985d550>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite manb_e at 0x7fc448f4bac0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite manb_c at 0x7fc448f4ba90>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MANpts',\n",
       "  'name': 'D-mannose transport via PEP:Pyr PTS',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3429'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1 or EDU37941_1 or (EDU36437_1 and EDU39079_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU37941_1 at 0x7fc4498964f0>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite man_e at 0x7fc448f4bc10>: -1.0,\n",
       "   <Metabolite man6p_c at 0x7fc448e1c0d0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MANTRabc',\n",
       "  'name': 'Mannotriose transport via ABC transporter',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83281'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37083_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37083_1 at 0x7fc44985d550>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mantr_e at 0x7fc448f4bc40>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite mantr_c at 0x7fc448f4bb20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCOATA',\n",
       "  'name': 'Malonyl-CoA-ACP transacylase',\n",
       "  'notes': {'BioCyc': 'META:MALONYL-COA-ACP-TRANSACYL-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR35619',\n",
       "   'KEGG Reaction': 'R01626',\n",
       "   'EC Number': '2.3.1.86'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36378_1 and EDU38144_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36378_1 at 0x7fc4498209a0>,\n",
       "   <Gene EDU38144_1 at 0x7fc449896520>},\n",
       "  '_metabolites': {<Metabolite ACP_c at 0x7fc463adca30>: -1.0,\n",
       "   <Metabolite malcoa_c at 0x7fc463a82100>: -1.0,\n",
       "   <Metabolite coa_c at 0x7fc463a6d730>: 1.0,\n",
       "   <Metabolite malACP_c at 0x7fc463a824c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCSNAH',\n",
       "  'name': '5-Methylcytosine aminohydrolase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83288', 'EC Number': '3.5.4.1'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38989_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38989_1 at 0x7fc44985db50>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 5mcsn_c at 0x7fc448ec88b0>: -1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite thym_c at 0x7fc448eef4f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCTP1Aex',\n",
       "  'name': 'Murein crosslinking transpeptidase 1A:(A2pm->D-ala) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69415'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35700_1 or EDU35846_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35700_1 at 0x7fc449896580>,\n",
       "   <Gene EDU35846_1 at 0x7fc449896550>},\n",
       "  '_metabolites': {<Metabolite murein5p5p_e at 0x7fc448f4bca0>: -1.0,\n",
       "   <Metabolite ala__D_e at 0x7fc448e49910>: 1.0,\n",
       "   <Metabolite murein5px4p_e at 0x7fc448f4bc70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCTP1App',\n",
       "  'name': 'Murein crosslinking transpeptidase 1A:(A2pm->D-ala) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69415'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39141_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39141_1 at 0x7fc4498965b0>},\n",
       "  '_metabolites': {<Metabolite murein5p5p_p at 0x7fc448f4bcd0>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 1.0,\n",
       "   <Metabolite murein5px4p_p at 0x7fc448f4bd00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCTP1Bex',\n",
       "  'name': 'Murein crosslinking transpeptidase 1B:(A2pm->A2pm) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69416'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35700_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35700_1 at 0x7fc449896580>},\n",
       "  '_metabolites': {<Metabolite murein5p5p_e at 0x7fc448f4bca0>: -1.0,\n",
       "   <Metabolite alaala_e at 0x7fc448e49880>: 1.0,\n",
       "   <Metabolite murein5px3p_e at 0x7fc448f4bd30>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCTP1Bpp',\n",
       "  'name': 'Murein crosslinking transpeptidase 1B:(A2pm->A2pm) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69416'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39141_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39141_1 at 0x7fc4498965b0>},\n",
       "  '_metabolites': {<Metabolite murein5p5p_p at 0x7fc448f4bcd0>: -1.0,\n",
       "   <Metabolite alaala_p at 0x7fc448e49850>: 1.0,\n",
       "   <Metabolite murein5px3p_p at 0x7fc448f4bd60>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCTP2Aex',\n",
       "  'name': 'Murein crosslinking transpeptidase 1A:(A2pm->D-ala) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83289'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35700_1 or EDU35846_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35700_1 at 0x7fc449896580>,\n",
       "   <Gene EDU35846_1 at 0x7fc449896550>},\n",
       "  '_metabolites': {<Metabolite murein5p5p5p_e at 0x7fc448f4bd90>: -1.0,\n",
       "   <Metabolite ala__D_e at 0x7fc448e49910>: 2.0,\n",
       "   <Metabolite murein5px4px4p_e at 0x7fc448f4bdc0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MCTP2App',\n",
       "  'name': 'Murein crosslinking transpeptidase 1A:(A2pm->D-ala) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83289'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39141_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39141_1 at 0x7fc4498965b0>},\n",
       "  '_metabolites': {<Metabolite murein5p5p5p_p at 0x7fc448f4bdf0>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 2.0,\n",
       "   <Metabolite murein5px4px4p_p at 0x7fc448f4be20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP1ex',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5px4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69418', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36966_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36966_1 at 0x7fc4498965e0>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite murein5px4p_e at 0x7fc448f4bc70>: -1.0,\n",
       "   <Metabolite ala__D_e at 0x7fc448e49910>: 1.0,\n",
       "   <Metabolite murein4px4p_e at 0x7fc448f4be50>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP1pp',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5px4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69418', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36973_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36973_1 at 0x7fc449896640>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein5px4p_p at 0x7fc448f4bd00>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 1.0,\n",
       "   <Metabolite murein4px4p_p at 0x7fc448f4be80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP2ex',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5px4px4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69419', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36966_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36966_1 at 0x7fc4498965e0>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite murein5px4px4p_e at 0x7fc448f4bdc0>: -1.0,\n",
       "   <Metabolite ala__D_e at 0x7fc448e49910>: 1.0,\n",
       "   <Metabolite murein4px4px4p_e at 0x7fc448f4beb0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP2pp',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5px4px4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69419', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36973_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36973_1 at 0x7fc449896640>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein5px4px4p_p at 0x7fc448f4be20>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 1.0,\n",
       "   <Metabolite murein4px4px4p_p at 0x7fc448f4bee0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP3ex',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5p5p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83290', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36966_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36966_1 at 0x7fc4498965e0>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite murein5p5p_e at 0x7fc448f4bca0>: -1.0,\n",
       "   <Metabolite ala__D_e at 0x7fc448e49910>: 1.0,\n",
       "   <Metabolite murein5p4p_e at 0x7fc448f4bfa0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP3pp',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5p5p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83290', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36973_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36973_1 at 0x7fc449896640>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein5p5p_p at 0x7fc448f4bcd0>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 1.0,\n",
       "   <Metabolite murein5p4p_p at 0x7fc448f4bfd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP4ex',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5p4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83291', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36966_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36966_1 at 0x7fc4498965e0>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite murein5p4p_e at 0x7fc448f4bfa0>: -1.0,\n",
       "   <Metabolite ala__D_e at 0x7fc448e49910>: 1.0,\n",
       "   <Metabolite murein4p4p_e at 0x7fc448f4bf10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP4pp',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5p4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83291', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36973_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36973_1 at 0x7fc449896640>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein5p4p_p at 0x7fc448f4bfd0>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 1.0,\n",
       "   <Metabolite murein4p4p_p at 0x7fc448f4bf70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP5ex',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5p3p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83292', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36966_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36966_1 at 0x7fc4498965e0>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_e at 0x7fc463a6d550>: -1.0,\n",
       "   <Metabolite murein5p3p_e at 0x7fc448f74130>: -1.0,\n",
       "   <Metabolite ala__D_e at 0x7fc448e49910>: 1.0,\n",
       "   <Metabolite murein4p3p_e at 0x7fc448f4bf40>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDCP5pp',\n",
       "  'name': 'Murein D,D-carboxypeptidase (murein5p3p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83292', 'EC Number': '3.4.16.4'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36973_1 or EDU37369_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36973_1 at 0x7fc449896640>,\n",
       "   <Gene EDU37369_1 at 0x7fc449896610>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein5p3p_p at 0x7fc448f74190>: -1.0,\n",
       "   <Metabolite ala__D_p at 0x7fc463a82f70>: 1.0,\n",
       "   <Metabolite murein4p3p_p at 0x7fc448f74160>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDEP1pp',\n",
       "  'name': 'Murein D,D-endopeptidase (murein4px4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69423'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein4px4p_p at 0x7fc448f4be80>: -1.0,\n",
       "   <Metabolite murein4p4p_p at 0x7fc448f4bf70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDDEP4pp',\n",
       "  'name': 'Murein D,D-endopeptidase (murein4px4px4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69426'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein4px4px4p_p at 0x7fc448f4bee0>: -1.0,\n",
       "   <Metabolite murein4px4p4p_p at 0x7fc448f740a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDH',\n",
       "  'name': 'Malate dehydrogenase',\n",
       "  'notes': {'EC Number': '1.1.1.37',\n",
       "   'BioCyc': 'META:MALATE-DEH-RXN',\n",
       "   'BioPath Reaction': 'RXN00015',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3464',\n",
       "   'KEGG Reaction': 'R00342',\n",
       "   'RHEA': '21432;21433;21434;21435',\n",
       "   'UniPathway Reaction': 'UCR00342;UER00721'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37260_1 or EDU38642_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37260_1 at 0x7fc4498962e0>,\n",
       "   <Gene EDU38642_1 at 0x7fc449896670>},\n",
       "  '_metabolites': {<Metabolite mal__L_c at 0x7fc448e49730>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite oaa_c at 0x7fc463aaef70>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MDRPD',\n",
       "  'name': '5-Methylthio-5-deoxy-D-ribulose 1-phosphate dehydratase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR50983'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 5mdru1p_c at 0x7fc448f74040>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite dkmpp_c at 0x7fc463a6d850>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ME1',\n",
       "  'name': 'Malic enzyme (NAD)',\n",
       "  'notes': {'EC Number': '1.1.1.39',\n",
       "   'BioCyc': 'META:1.1.1.39-RXN',\n",
       "   'BioPath Reaction': 'RXN00044',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3470',\n",
       "   'RHEA': '12653;12654;12655;12656',\n",
       "   'KEGG Reaction': 'R00214'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37932_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37932_1 at 0x7fc4498966a0>},\n",
       "  '_metabolites': {<Metabolite mal__L_c at 0x7fc448e49730>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'ME2',\n",
       "  'name': 'Malic enzyme (NADP)',\n",
       "  'notes': {'EC Number': '1.1.1.40',\n",
       "   'BioCyc': 'META:MALIC-NADP-RXN',\n",
       "   'BioPath Reaction': 'RXN00625',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3471',\n",
       "   'RHEA': '18253;18254;18255;18256',\n",
       "   'KEGG Reaction': 'R00216'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37932_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37932_1 at 0x7fc4498966a0>},\n",
       "  '_metabolites': {<Metabolite mal__L_c at 0x7fc448e49730>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite co2_c at 0x7fc463adc8e0>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MECDPDH2',\n",
       "  'name': '2C-methyl-D-erythritol 2,4 cyclodiphosphate dehydratase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR55700'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36410_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36410_1 at 0x7fc4498966d0>},\n",
       "  '_metabolites': {<Metabolite 2mecdp_c at 0x7fc448f74070>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite h2mb4p_c at 0x7fc448eef250>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MECDPS',\n",
       "  'name': '2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase',\n",
       "  'notes': {'EC Number': '4.6.1.12',\n",
       "   'BioCyc': 'META:RXN0-302',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3474',\n",
       "   'KEGG Reaction': 'R05637',\n",
       "   'RHEA': '23864;23865;23866;23867',\n",
       "   'UniPathway Reaction': 'UCR05637;UER00095'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38071_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38071_1 at 0x7fc449896700>},\n",
       "  '_metabolites': {<Metabolite 2p4c2me_c at 0x7fc448e9f790>: -1.0,\n",
       "   <Metabolite 2mecdp_c at 0x7fc448f74070>: 1.0,\n",
       "   <Metabolite cmp_c at 0x7fc448e1cdf0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MEHLER',\n",
       "  'name': 'Mehler reaction',\n",
       "  'notes': {'RHEA': '37795;37796;37797;37798',\n",
       "   'BioCyc': 'META:RXN-14690',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR83299',\n",
       "   'KEGG Reaction': 'R10518',\n",
       "   'EC Number': '1.6.3.2'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36072_1 or EDU36220_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36072_1 at 0x7fc449879340>,\n",
       "   <Gene EDU36220_1 at 0x7fc449896730>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite o2_c at 0x7fc463a6d610>: -0.5,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MEOHtex',\n",
       "  'name': 'Methanol transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3482',\n",
       "   'BioCyc': 'META:TRANS-RXN0-459'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite meoh_e at 0x7fc448e74cd0>: -1.0,\n",
       "   <Metabolite meoh_p at 0x7fc448f74100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MEOHtrpp',\n",
       "  'name': 'Methanol reversible transport via diffusion (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3482',\n",
       "   'BioCyc': 'META:TRANS-RXN0-459'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite meoh_p at 0x7fc448f74100>: -1.0,\n",
       "   <Metabolite meoh_c at 0x7fc448e49b20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MEPCT',\n",
       "  'name': '2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase',\n",
       "  'notes': {'EC Number': '2.7.7.60',\n",
       "   'BioCyc': 'META:2.7.7.60-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69431',\n",
       "   'KEGG Reaction': 'R05633',\n",
       "   'RHEA': '13429;13430;13431;13432',\n",
       "   'UniPathway Reaction': 'UCR05633;UER00093'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39013_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39013_1 at 0x7fc449896760>},\n",
       "  '_metabolites': {<Metabolite 2me4p_c at 0x7fc448eef520>: -1.0,\n",
       "   <Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite 4c2me_c at 0x7fc448e9f7c0>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METabc',\n",
       "  'name': 'L-methionine transport via ABC system',\n",
       "  'notes': {'RHEA': '29779;29780;29781;29782',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3503',\n",
       "   'BioCyc': 'META:RXN0-4522'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37371_1 or (EDU37278_1 and EDU37279_1 and EDU38960_1) or (EDU38958_1 and EDU38959_1 and EDU38961_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37278_1 at 0x7fc4498969a0>,\n",
       "   <Gene EDU37279_1 at 0x7fc449896a00>,\n",
       "   <Gene EDU37371_1 at 0x7fc449896910>,\n",
       "   <Gene EDU38958_1 at 0x7fc449896970>,\n",
       "   <Gene EDU38959_1 at 0x7fc449896940>,\n",
       "   <Gene EDU38960_1 at 0x7fc4498969d0>,\n",
       "   <Gene EDU38961_1 at 0x7fc4498968e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite met__L_e at 0x7fc448f741c0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METAT',\n",
       "  'name': 'Methionine adenosyltransferase',\n",
       "  'notes': {'EC Number': '2.5.1.6',\n",
       "   'BioCyc': 'META:S-ADENMETSYN-RXN',\n",
       "   'BioPath Reaction': 'RXN00344',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3487',\n",
       "   'KEGG Reaction': 'R00177',\n",
       "   'RHEA': '21080;21081;21082;21083',\n",
       "   'UniPathway Reaction': 'UCR00177;UER00080'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37973_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37973_1 at 0x7fc449896790>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: -1.0,\n",
       "   <Metabolite amet_c at 0x7fc463adc940>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METGL',\n",
       "  'name': 'METGL',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83302', 'EC Number': '4.4.1.11'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38988_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38988_1 at 0x7fc44985dd30>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: -1.0,\n",
       "   <Metabolite 2obut_c at 0x7fc463adc550>: 1.0,\n",
       "   <Metabolite nh4_c at 0x7fc463a6d5e0>: 1.0,\n",
       "   <Metabolite ch4s_c at 0x7fc448e49550>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METOX1s',\n",
       "  'name': 'Methionine oxidation (spontaneous)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3493'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2o2_c at 0x7fc463a6d580>: -1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite metsox_S__L_c at 0x7fc448f741f0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METOX2s',\n",
       "  'name': 'Methionine oxidation 2 (spontaneous)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69433'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2o2_c at 0x7fc463a6d580>: -1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite metsox_R__L_c at 0x7fc448f74220>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METR',\n",
       "  'name': 'Methyltetrahydrofolate:corrinoid/iron-sulfur protein methyltransferase (MeTr)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83305'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite 5mthf_c at 0x7fc463aae250>: -1.0,\n",
       "   <Metabolite cfesp_c at 0x7fc448ec8040>: -1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: 1.0,\n",
       "   <Metabolite mecfsp_c at 0x7fc448ec8070>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METS',\n",
       "  'name': 'Methionine synthase',\n",
       "  'notes': {'EC Number': '2.1.1.13',\n",
       "   'BioPath Reaction': 'RXN00753',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR69434',\n",
       "   'UniPathway Reaction': 'UCR00946;UER00081',\n",
       "   'RHEA': '11172;11173;11174;11175',\n",
       "   'KEGG Reaction': 'R00946'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37142_1 or EDU37143_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37142_1 at 0x7fc4498967c0>,\n",
       "   <Gene EDU37143_1 at 0x7fc4498967f0>},\n",
       "  '_metabolites': {<Metabolite 5mthf_c at 0x7fc463aae250>: -1.0,\n",
       "   <Metabolite hcys__L_c at 0x7fc448e49460>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METSabc',\n",
       "  'name': 'L methionine S oxide transport via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37371_1 or (EDU38958_1 and EDU38959_1 and EDU38961_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37371_1 at 0x7fc449896910>,\n",
       "   <Gene EDU38958_1 at 0x7fc449896970>,\n",
       "   <Gene EDU38959_1 at 0x7fc449896940>,\n",
       "   <Gene EDU38961_1 at 0x7fc4498968e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite metox_e at 0x7fc448f74280>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite metox_c at 0x7fc448f74250>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METSOXR1',\n",
       "  'name': 'L-methionine-S-oxide reductase',\n",
       "  'notes': {'BioCyc': 'META:1.8.4.13-RXN;META:1.8.4.14-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR83306',\n",
       "   'KEGG Reaction': 'R02025;R07606;R07608',\n",
       "   'EC Number': '1.8.4.14'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36488_1 or EDU36895_1 or EDU37384_1 or EDU37548_1 or (EDU36895_1 and EDU37548_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36488_1 at 0x7fc449896880>,\n",
       "   <Gene EDU36895_1 at 0x7fc4498968b0>,\n",
       "   <Gene EDU37384_1 at 0x7fc449896850>,\n",
       "   <Gene EDU37548_1 at 0x7fc449896820>},\n",
       "  '_metabolites': {<Metabolite metsox_S__L_c at 0x7fc448f741f0>: -1.0,\n",
       "   <Metabolite trdrd_c at 0x7fc448e495e0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0,\n",
       "   <Metabolite trdox_c at 0x7fc448e495b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METSOXR2',\n",
       "  'name': 'L-methionine-R-sulfoxide reductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83307'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36488_1 or EDU37548_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36488_1 at 0x7fc449896880>,\n",
       "   <Gene EDU37548_1 at 0x7fc449896820>},\n",
       "  '_metabolites': {<Metabolite metsox_R__L_c at 0x7fc448f74220>: -1.0,\n",
       "   <Metabolite trdrd_c at 0x7fc448e495e0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0,\n",
       "   <Metabolite trdox_c at 0x7fc448e495b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METSRabc',\n",
       "  'name': 'L methionine R oxide transport via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37371_1 or (EDU38958_1 and EDU38959_1 and EDU38961_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37371_1 at 0x7fc449896910>,\n",
       "   <Gene EDU38958_1 at 0x7fc449896970>,\n",
       "   <Gene EDU38959_1 at 0x7fc449896940>,\n",
       "   <Gene EDU38961_1 at 0x7fc4498968e0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite metox__R_e at 0x7fc448f742e0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite metox__R_c at 0x7fc448f742b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METSR_R1',\n",
       "  'name': 'L methionine R oxide reductase  trdrd ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36488_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36488_1 at 0x7fc449896880>},\n",
       "  '_metabolites': {<Metabolite trdrd_c at 0x7fc448e495e0>: -1.0,\n",
       "   <Metabolite metox__R_c at 0x7fc448f742b0>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0,\n",
       "   <Metabolite trdox_c at 0x7fc448e495b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METSR_R2',\n",
       "  'name': 'L methionine R oxide  H2O2 ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36488_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36488_1 at 0x7fc449896880>},\n",
       "  '_metabolites': {<Metabolite h2o2_c at 0x7fc463a6d580>: -1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite metox__R_c at 0x7fc448f742b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'METSR_S1',\n",
       "  'name': 'L methionine S oxide reductase  trdrd ',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite trdrd_c at 0x7fc448e495e0>: -1.0,\n",
       "   <Metabolite metox_c at 0x7fc448f74250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite met__L_c at 0x7fc448e49520>: 1.0,\n",
       "   <Metabolite trdox_c at 0x7fc448e495b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MEVK1',\n",
       "  'name': 'Mevalonate kinase  atp ',\n",
       "  'notes': {'EC Number': '2.7.1.36',\n",
       "   'BioCyc': 'META:MEVALONATE-KINASE-RXN',\n",
       "   'BioPath Reaction': 'RXN01123',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3507',\n",
       "   'KEGG Reaction': 'R02245',\n",
       "   'RHEA': '17065;17066;17067;17068',\n",
       "   'UniPathway Reaction': 'UCR02245;UER00098'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38800_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38800_1 at 0x7fc449896a30>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite mev__R_c at 0x7fc448f20f40>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 5pmev_c at 0x7fc448f74310>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MG2abc',\n",
       "  'name': 'Magnesium transport via ABC system',\n",
       "  'notes': {'RHEA': '10260;10261;10262;10263',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3513',\n",
       "   'EC Number': '3.6.3.2',\n",
       "   'BioCyc': 'META:3.6.3.2-RXN;META:TRANS-RXN-250'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38088_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38088_1 at 0x7fc449896a60>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mg2_e at 0x7fc448e9f9d0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mg2_c at 0x7fc448e9f9a0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MG2tex',\n",
       "  'name': 'Magnesium (Mg+2) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3516',\n",
       "   'BioCyc': 'META:TRANS-RXN-141'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite mg2_e at 0x7fc448e9f9d0>: -1.0,\n",
       "   <Metabolite mg2_p at 0x7fc448f74340>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MG2uabcpp',\n",
       "  'name': 'Magnesium (Mg+2) ABC transporter (ubtake, periplasm)',\n",
       "  'notes': {'RHEA': '10260;10261;10262;10263',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3513',\n",
       "   'EC Number': '3.6.3.2',\n",
       "   'BioCyc': 'META:3.6.3.2-RXN;META:TRANS-RXN-250'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35976_1 or EDU37686_1 or EDU38046_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35976_1 at 0x7fc449896af0>,\n",
       "   <Gene EDU37686_1 at 0x7fc449896ac0>,\n",
       "   <Gene EDU38046_1 at 0x7fc449896a90>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mg2_p at 0x7fc448f74340>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mg2_c at 0x7fc448e9f9a0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MGSA',\n",
       "  'name': 'Methylglyoxal synthase',\n",
       "  'notes': {'RHEA': '17937;17938;17939;17940',\n",
       "   'BioCyc': 'META:METHGLYSYN-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3520',\n",
       "   'KEGG Reaction': 'R01016',\n",
       "   'EC Number': '4.2.3.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37942_1 or spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37942_1 at 0x7fc449896b20>,\n",
       "   <Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite dhap_c at 0x7fc448e49d90>: -1.0,\n",
       "   <Metabolite mthgxl_c at 0x7fc463aae640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MGt5',\n",
       "  'name': 'MGt5',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3516',\n",
       "   'BioCyc': 'META:TRANS-RXN-141'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36234_1 or EDU37686_1 or EDU37939_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36234_1 at 0x7fc449896b50>,\n",
       "   <Gene EDU37686_1 at 0x7fc449896ac0>,\n",
       "   <Gene EDU37939_1 at 0x7fc449896b80>},\n",
       "  '_metabolites': {<Metabolite mg2_c at 0x7fc448e9f9a0>: -1.0,\n",
       "   <Metabolite mg2_e at 0x7fc448e9f9d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 0.0},\n",
       " {'_id': 'MLDCP3App',\n",
       "  'name': 'Murein L,D-carboxypeptidase (murein5px3p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69456', 'EC Number': '3.4.17.13'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein5px3p_p at 0x7fc448f4bd60>: -1.0,\n",
       "   <Metabolite alaala_p at 0x7fc448e49850>: 1.0,\n",
       "   <Metabolite murein3px3p_p at 0x7fc448f743a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MLDEP1pp',\n",
       "  'name': 'Murein L,D-endopeptidase (murein3px3p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69457'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36423_1 or EDU38138_1 or EDU39164_1 or EDU39365_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36423_1 at 0x7fc449896be0>,\n",
       "   <Gene EDU38138_1 at 0x7fc449896c40>,\n",
       "   <Gene EDU39164_1 at 0x7fc449896bb0>,\n",
       "   <Gene EDU39365_1 at 0x7fc449896c10>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein3px3p_p at 0x7fc448f743a0>: -1.0,\n",
       "   <Metabolite murein3p3p_p at 0x7fc448f74370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MLDEP2pp',\n",
       "  'name': 'Murein L,D-endopeptidase (murein5px3p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3568'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36423_1 or EDU38138_1 or EDU39164_1 or EDU39365_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36423_1 at 0x7fc449896be0>,\n",
       "   <Gene EDU38138_1 at 0x7fc449896c40>,\n",
       "   <Gene EDU39164_1 at 0x7fc449896bb0>,\n",
       "   <Gene EDU39365_1 at 0x7fc449896c10>},\n",
       "  '_metabolites': {<Metabolite h2o_p at 0x7fc463a6d880>: -1.0,\n",
       "   <Metabolite murein5px3p_p at 0x7fc448f4bd60>: -1.0,\n",
       "   <Metabolite murein5p3p_p at 0x7fc448f74190>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MLTGY1pp',\n",
       "  'name': 'Murein lytic transglycosylase (murein4p4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69464'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35834_1 or EDU39270_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35834_1 at 0x7fc449896ca0>,\n",
       "   <Gene EDU39270_1 at 0x7fc449896c70>},\n",
       "  '_metabolites': {<Metabolite murein4p4p_p at 0x7fc448f4bf70>: -1.0,\n",
       "   <Metabolite anhgm4p_p at 0x7fc448e1cfd0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MLTGY2pp',\n",
       "  'name': 'Murein lytic transglycosylase (murein4p3p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69465'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35834_1 or EDU39270_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35834_1 at 0x7fc449896ca0>,\n",
       "   <Gene EDU39270_1 at 0x7fc449896c70>},\n",
       "  '_metabolites': {<Metabolite murein4p3p_p at 0x7fc448f74160>: -1.0,\n",
       "   <Metabolite anhgm3p_p at 0x7fc448e1ceb0>: 1.0,\n",
       "   <Metabolite anhgm4p_p at 0x7fc448e1cfd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MLTGY3pp',\n",
       "  'name': 'Murein lytic transglycosylase (murein3p3p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69466'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35834_1 or EDU39270_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35834_1 at 0x7fc449896ca0>,\n",
       "   <Gene EDU39270_1 at 0x7fc449896c70>},\n",
       "  '_metabolites': {<Metabolite murein3p3p_p at 0x7fc448f74370>: -1.0,\n",
       "   <Metabolite anhgm3p_p at 0x7fc448e1ceb0>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MLTGY4pp',\n",
       "  'name': 'Murein lytic transglycosylase (murein4px4p4p) (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69467'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35834_1 or EDU39270_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35834_1 at 0x7fc449896ca0>,\n",
       "   <Gene EDU39270_1 at 0x7fc449896c70>},\n",
       "  '_metabolites': {<Metabolite murein4px4p4p_p at 0x7fc448f740a0>: -1.0,\n",
       "   <Metabolite anhgm4p_p at 0x7fc448e1cfd0>: 1.0,\n",
       "   <Metabolite murein4px4p_p at 0x7fc448f4be80>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MMSAD2',\n",
       "  'name': 'Methylmalonate-semialdehyde dehydrogenase (propanol)',\n",
       "  'notes': {'RHEA': '36027;36028;36029;36030',\n",
       "   'BioCyc': 'META:RXN-12736',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3606',\n",
       "   'KEGG Reaction': 'R09097',\n",
       "   'EC Number': '1.2.1.87'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite coa_c at 0x7fc463a6d730>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite ppal_c at 0x7fc463a6d460>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 1.0,\n",
       "   <Metabolite ppcoa_c at 0x7fc463aaeb20>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MN2t3pp',\n",
       "  'name': 'Manganese (Mn+2) transport out via proton antiport (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69476',\n",
       "   'BioCyc': 'META:TRANS-RXN-241'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39203_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU39203_1 at 0x7fc44985d2b0>},\n",
       "  '_metabolites': {<Metabolite mn2_c at 0x7fc448e9fa00>: -1.0,\n",
       "   <Metabolite h_p at 0x7fc45d333b50>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mn2_p at 0x7fc448f743d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MN2tpp',\n",
       "  'name': 'Manganese transport in via permease (no H+)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69477',\n",
       "   'BioCyc': 'META:RXN0-16;META:TRANS-RXN0-487'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37639_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37639_1 at 0x7fc44985d310>},\n",
       "  '_metabolites': {<Metabolite mn2_p at 0x7fc448f743d0>: -1.0,\n",
       "   <Metabolite mn2_c at 0x7fc448e9fa00>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MN6PP',\n",
       "  'name': 'Mannose 6-phosphate phosphatase',\n",
       "  'notes': {'RHEA': '28554;28555;28556;28557',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3612'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37643_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37643_1 at 0x7fc44986ac10>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite man6p_c at 0x7fc448e1c0d0>: -1.0,\n",
       "   <Metabolite man_c at 0x7fc448e49fd0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MNabc',\n",
       "  'name': 'Manganese transport via ABC system',\n",
       "  'notes': {'RHEA': '17365;17366;17367;17368',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR19986',\n",
       "   'EC Number': '3.6.3.35',\n",
       "   'BioCyc': 'META:3.6.3.35-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37059_1 or EDU39167_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37059_1 at 0x7fc449896d00>,\n",
       "   <Gene EDU39167_1 at 0x7fc449896cd0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mn2_e at 0x7fc448e9fa30>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mn2_c at 0x7fc448e9fa00>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MNabc_1',\n",
       "  'name': 'Manganese Mn2 transport via ABC system periplasm ',\n",
       "  'notes': {'RHEA': '17365;17366;17367;17368',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR19986',\n",
       "   'EC Number': '3.6.3.35',\n",
       "   'BioCyc': 'META:3.6.3.35-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38941_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38941_1 at 0x7fc44985dc40>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mn2_p at 0x7fc448f743d0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mn2_c at 0x7fc448e9fa00>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MNLpts',\n",
       "  'name': 'Mannitol transport via PEP:Pyr PTS',\n",
       "  'notes': {'RHEA': '33363;33364;33365;33366',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR38088'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36437_1 or (EDU36437_1 and EDU39079_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36437_1 at 0x7fc449820760>,\n",
       "   <Gene EDU39079_1 at 0x7fc449820730>},\n",
       "  '_metabolites': {<Metabolite pep_c at 0x7fc463aaec40>: -1.0,\n",
       "   <Metabolite mnl_e at 0x7fc448f74400>: -1.0,\n",
       "   <Metabolite mnl1p_c at 0x7fc448f4bbe0>: 1.0,\n",
       "   <Metabolite pyr_c at 0x7fc463adc2b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MNtex',\n",
       "  'name': 'Manganese (Mn+2) transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69477',\n",
       "   'BioCyc': 'META:RXN0-16;META:TRANS-RXN0-487'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite mn2_e at 0x7fc448e9fa30>: -1.0,\n",
       "   <Metabolite mn2_p at 0x7fc448f743d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOADSUx',\n",
       "  'name': 'MoaD sulfuration (nadh, assumed)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83388'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36273_1 or (EDU36273_1 and EDU36884_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36273_1 at 0x7fc44983b610>,\n",
       "   <Gene EDU36884_1 at 0x7fc449896d30>},\n",
       "  '_metabolites': {<Metabolite iscssh_c at 0x7fc448f4b2b0>: -1.0,\n",
       "   <Metabolite moadamp_c at 0x7fc448f74430>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite iscs_c at 0x7fc448f4b280>: 1.0,\n",
       "   <Metabolite moadcosh_c at 0x7fc448f74460>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOAT',\n",
       "  'name': '3-deoxy-D-manno-octulosonic acid transferase',\n",
       "  'notes': {'EC Number': '2.4.99.12',\n",
       "   'BioCyc': 'META:KDOTRANS-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3619',\n",
       "   'KEGG Reaction': 'R04658',\n",
       "   'RHEA': '28066;28067;28068;28069',\n",
       "   'UniPathway Reaction': 'UCR04658;UER00483'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ckdo_c at 0x7fc448f4b670>: -1.0,\n",
       "   <Metabolite lipidA_c at 0x7fc448f744c0>: -1.0,\n",
       "   <Metabolite cmp_c at 0x7fc448e1cdf0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite kdolipid4_c at 0x7fc448f74490>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOAT2',\n",
       "  'name': '3-deoxy-D-manno-octulosonic acid transferase',\n",
       "  'notes': {'EC Number': '2.4.99.13',\n",
       "   'BioCyc': 'META:KDOTRANS2-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3620',\n",
       "   'KEGG Reaction': 'R05074;R06285',\n",
       "   'RHEA': '28062;28063;28064;28065',\n",
       "   'UniPathway Reaction': 'UCR05074;UER00484'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite ckdo_c at 0x7fc448f4b670>: -1.0,\n",
       "   <Metabolite kdolipid4_c at 0x7fc448f74490>: -1.0,\n",
       "   <Metabolite cmp_c at 0x7fc448e1cdf0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite kdo2lipid4_c at 0x7fc448eef700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOBDabc',\n",
       "  'name': 'Molybdate transport via ABC system',\n",
       "  'notes': {'RHEA': '22020;22021;22022;22023',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3623',\n",
       "   'EC Number': '3.6.3.29',\n",
       "   'BioCyc': 'META:3.6.3.29-RXN;META:ABC-19-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36878_1 or EDU36982_1 or EDU37291_1 or EDU39138_1 or (EDU36982_1 and EDU37293_1) or (EDU36982_1 and EDU37699_1) or (EDU37291_1 and EDU37293_1) or (EDU37291_1 and EDU37699_1) or (EDU37292_1 and EDU37293_1) or (EDU36878_1 and EDU37292_1 and EDU37293_1) or (EDU36982_1 and EDU37293_1 and EDU37699_1) or (EDU37291_1 and EDU37293_1 and EDU37699_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36878_1 at 0x7fc449896d60>,\n",
       "   <Gene EDU36982_1 at 0x7fc44983b280>,\n",
       "   <Gene EDU37291_1 at 0x7fc449896dc0>,\n",
       "   <Gene EDU37292_1 at 0x7fc449896df0>,\n",
       "   <Gene EDU37293_1 at 0x7fc449896d90>,\n",
       "   <Gene EDU37699_1 at 0x7fc449896e50>,\n",
       "   <Gene EDU39138_1 at 0x7fc449896e20>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mobd_e at 0x7fc448f74520>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mobd_c at 0x7fc448f744f0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOBDabcpp',\n",
       "  'name': 'Molybdate transport via ABC system (periplasm)',\n",
       "  'notes': {'RHEA': '22020;22021;22022;22023',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3623',\n",
       "   'EC Number': '3.6.3.29',\n",
       "   'BioCyc': 'META:3.6.3.29-RXN;META:ABC-19-RXN'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37242_1 or EDU37652_1 or (EDU37242_1 and EDU37661_1) or (EDU37242_1 and EDU37699_1) or (EDU37291_1 and EDU37293_1) or (EDU37378_1 and EDU37699_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37242_1 at 0x7fc44983be50>,\n",
       "   <Gene EDU37291_1 at 0x7fc449896dc0>,\n",
       "   <Gene EDU37293_1 at 0x7fc449896d90>,\n",
       "   <Gene EDU37378_1 at 0x7fc449879940>,\n",
       "   <Gene EDU37652_1 at 0x7fc44985da90>,\n",
       "   <Gene EDU37661_1 at 0x7fc449896e80>,\n",
       "   <Gene EDU37699_1 at 0x7fc449896e50>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mobd_p at 0x7fc448f74550>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mobd_c at 0x7fc448f744f0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOBDtex',\n",
       "  'name': 'Molybdate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3624'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite mobd_e at 0x7fc448f74520>: -1.0,\n",
       "   <Metabolite mobd_p at 0x7fc448f74550>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOCDS',\n",
       "  'name': 'Molybdopterin cytidine dinucleotide synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83392'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35953_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35953_1 at 0x7fc449887850>},\n",
       "  '_metabolites': {<Metabolite ctp_c at 0x7fc448e1c1c0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite moco_c at 0x7fc448e74df0>: -1.0,\n",
       "   <Metabolite mococdp_c at 0x7fc448f74580>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOCOS',\n",
       "  'name': 'Molybdenum cofactor synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83393'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36877_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36877_1 at 0x7fc449896eb0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite mobd_c at 0x7fc448f744f0>: -1.0,\n",
       "   <Metabolite mptamp_c at 0x7fc448e74e20>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite cu2_c at 0x7fc448e74dc0>: 1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite moco_c at 0x7fc448e74df0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MOHMT',\n",
       "  'name': '3-methyl-2-oxobutanoate hydroxymethyltransferase',\n",
       "  'notes': {'RHEA': '11824;11825;11826;11827',\n",
       "   'EC Number': '2.1.2.11',\n",
       "   'KEGG Reaction': 'R01226',\n",
       "   'UniPathway Reaction': 'UCR01226;UER00003',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3626'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38247_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38247_1 at 0x7fc449896ee0>},\n",
       "  '_metabolites': {<Metabolite 3mob_c at 0x7fc463adc490>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mlthf_c at 0x7fc448eef130>: -1.0,\n",
       "   <Metabolite 2dhp_c at 0x7fc463adc460>: 1.0,\n",
       "   <Metabolite thf_c at 0x7fc448e49640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MPTAT',\n",
       "  'name': 'Molybdopterin adenylyltransferase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83416'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36876_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36876_1 at 0x7fc449896f10>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite mpt_c at 0x7fc448f745b0>: -1.0,\n",
       "   <Metabolite mptamp_c at 0x7fc448e74e20>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MPTG',\n",
       "  'name': 'Murein polymerizing transglycosylase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69483'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35686_1 or EDU37319_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35686_1 at 0x7fc449896f40>,\n",
       "   <Gene EDU37319_1 at 0x7fc449896f70>},\n",
       "  '_metabolites': {<Metabolite uaagmda_c at 0x7fc448f207c0>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite udcpdp_c at 0x7fc448eef5b0>: 2.0,\n",
       "   <Metabolite murein5p5p_p at 0x7fc448f4bcd0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MPTG2',\n",
       "  'name': 'Murein polymerizing transglycosylase 2 (three linked units)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83417'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35686_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35686_1 at 0x7fc449896f40>},\n",
       "  '_metabolites': {<Metabolite uaagmda_c at 0x7fc448f207c0>: -1.0,\n",
       "   <Metabolite murein5p5p_p at 0x7fc448f4bcd0>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite udcpdp_c at 0x7fc448eef5b0>: 1.0,\n",
       "   <Metabolite murein5p5p5p_p at 0x7fc448f4bdf0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MPTG2ex',\n",
       "  'name': 'Murein polymerizing transglycosylase 2 (three linked units)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83417'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35700_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35700_1 at 0x7fc449896580>},\n",
       "  '_metabolites': {<Metabolite murein5p5p_e at 0x7fc448f4bca0>: -1.0,\n",
       "   <Metabolite uaagmda_e at 0x7fc448f745e0>: -1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: 1.0,\n",
       "   <Metabolite murein5p5p5p_e at 0x7fc448f4bd90>: 1.0,\n",
       "   <Metabolite udcpdp_e at 0x7fc448f74610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MPTGex',\n",
       "  'name': 'Murein polymerizing transglycosylase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69483'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU35700_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU35700_1 at 0x7fc449896580>},\n",
       "  '_metabolites': {<Metabolite uaagmda_e at 0x7fc448f745e0>: -2.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: 2.0,\n",
       "   <Metabolite murein5p5p_e at 0x7fc448f4bca0>: 1.0,\n",
       "   <Metabolite udcpdp_e at 0x7fc448f74610>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MPTS',\n",
       "  'name': 'Molybdopterin synthase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83418'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36881_1 and EDU36884_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36881_1 at 0x7fc449896fa0>,\n",
       "   <Gene EDU36884_1 at 0x7fc449896d30>},\n",
       "  '_metabolites': {<Metabolite cpmp_c at 0x7fc448ec8100>: -1.0,\n",
       "   <Metabolite cu2_c at 0x7fc448e74dc0>: -1.0,\n",
       "   <Metabolite moadcosh_c at 0x7fc448f74460>: -2.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 5.0,\n",
       "   <Metabolite moadcoo_c at 0x7fc448f74640>: 2.0,\n",
       "   <Metabolite mpt_c at 0x7fc448f745b0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MPTSS',\n",
       "  'name': 'Molybdopterin synthase sulfurylase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83419'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36885_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36885_1 at 0x7fc449896fd0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite moadcoo_c at 0x7fc448f74640>: -1.0,\n",
       "   <Metabolite moadamp_c at 0x7fc448f74430>: 1.0,\n",
       "   <Metabolite ppi_c at 0x7fc463a6dc10>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MSO3abc',\n",
       "  'name': 'Methanesulfonate transport via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36524_1 and EDU39295_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36524_1 at 0x7fc4498961f0>,\n",
       "   <Gene EDU39295_1 at 0x7fc44983bfa0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mso3_e at 0x7fc448f74670>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mso3_c at 0x7fc448eefdc0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MSO3abcpp',\n",
       "  'name': 'Methanesulfonate transport via ABC system (periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3635'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36524_1 or EDU36526_1 or EDU39369_1 or EDU39370_1 or (EDU36524_1 and EDU36526_1) or (EDU36526_1 and EDU39167_1) or (EDU36526_1 and EDU39369_1) or (EDU36524_1 and EDU36526_1 and EDU39167_1) or (EDU39288_1 and EDU39293_1 and EDU39295_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36524_1 at 0x7fc4498961f0>,\n",
       "   <Gene EDU36526_1 at 0x7fc449896070>,\n",
       "   <Gene EDU39167_1 at 0x7fc449896cd0>,\n",
       "   <Gene EDU39288_1 at 0x7fc449896250>,\n",
       "   <Gene EDU39293_1 at 0x7fc449896220>,\n",
       "   <Gene EDU39295_1 at 0x7fc44983bfa0>,\n",
       "   <Gene EDU39369_1 at 0x7fc449896100>,\n",
       "   <Gene EDU39370_1 at 0x7fc44983bfd0>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mso3_p at 0x7fc448f746a0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite mso3_c at 0x7fc448eefdc0>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MSO3tex',\n",
       "  'name': 'Methanesulfonate transport via diffusion (extracellular to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3636'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite mso3_e at 0x7fc448f74670>: -1.0,\n",
       "   <Metabolite mso3_p at 0x7fc448f746a0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTAN',\n",
       "  'name': 'Methylthioadenosine nucleosidase',\n",
       "  'notes': {'EC Number': '3.2.2.9',\n",
       "   'BioCyc': 'META:METHYLTHIOADENOSINE-NUCLEOSIDASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3638',\n",
       "   'KEGG Reaction': 'R01401',\n",
       "   'RHEA': '13617;13618;13619;13620',\n",
       "   'UniPathway Reaction': 'UCR01401;UER00871'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37304_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37304_1 at 0x7fc449820f10>},\n",
       "  '_metabolites': {<Metabolite 5mta_c at 0x7fc463aae220>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite 5mtr_c at 0x7fc463aae280>: 1.0,\n",
       "   <Metabolite ade_c at 0x7fc463aae040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTHFC',\n",
       "  'name': 'Methenyltetrahydrofolate cyclohydrolase',\n",
       "  'notes': {'RHEA': '23700;23701;23702;23703',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3641',\n",
       "   'KEGG Reaction': 'R01655',\n",
       "   'EC Number': '3.5.4.9'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37644_1 or EDU37645_1 or EDU38224_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37644_1 at 0x7fc4498a40d0>,\n",
       "   <Gene EDU37645_1 at 0x7fc4498a4040>,\n",
       "   <Gene EDU38224_1 at 0x7fc4498794f0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite methf_c at 0x7fc45d380ac0>: -1.0,\n",
       "   <Metabolite 10fthf_c at 0x7fc463a377c0>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTHFD',\n",
       "  'name': 'Methylenetetrahydrofolate dehydrogenase (NADP)',\n",
       "  'notes': {'RHEA': '22812;22813;22814;22815',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3642',\n",
       "   'KEGG Reaction': 'R01220',\n",
       "   'EC Number': '1.5.1.5'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37644_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37644_1 at 0x7fc4498a40d0>},\n",
       "  '_metabolites': {<Metabolite mlthf_c at 0x7fc448eef130>: -1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: -1.0,\n",
       "   <Metabolite methf_c at 0x7fc45d380ac0>: 1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTHFR2',\n",
       "  'name': '5,10-methylenetetrahydrofolate reductase (NADH)',\n",
       "  'notes': {'RHEA': '19821;19822;19823;19824',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3644',\n",
       "   'KEGG Reaction': 'R07168',\n",
       "   'EC Number': '1.5.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37144_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37144_1 at 0x7fc4498a40a0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite mlthf_c at 0x7fc448eef130>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 5mthf_c at 0x7fc463aae250>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTHFR2_1',\n",
       "  'name': '5,10-methylenetetrahydrofolate reductase (NADH)',\n",
       "  'notes': {'RHEA': '19821;19822;19823;19824',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3644',\n",
       "   'KEGG Reaction': 'R07168',\n",
       "   'EC Number': '1.5.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37144_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37144_1 at 0x7fc4498a40a0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite mlthf_c at 0x7fc448eef130>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite 5mthf_c at 0x7fc463aae250>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTHFR3',\n",
       "  'name': '5 10 methylenetetrahydrofolatereductase  NADPH ',\n",
       "  'notes': {'RHEA': '19817;19818;19819;19820',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3645',\n",
       "   'KEGG Reaction': 'R01224',\n",
       "   'EC Number': '1.5.1.20'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37144_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37144_1 at 0x7fc4498a40a0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -2.0,\n",
       "   <Metabolite mlthf_c at 0x7fc448eef130>: -1.0,\n",
       "   <Metabolite nadph_c at 0x7fc45d375370>: -1.0,\n",
       "   <Metabolite 5mthf_c at 0x7fc463aae250>: 1.0,\n",
       "   <Metabolite nadp_c at 0x7fc463a6d790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTHTHFSs',\n",
       "  'name': '(2R,4S)-2-methyl-2,3,3,4-tetrahydroxytetrahydrofuran synthesis (spontaneous)',\n",
       "  'notes': {'RHEA': '35159;35160;35161;35162',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR62125',\n",
       "   'BioCyc': 'META:RXN-10017'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'spontaneous',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene spontaneous at 0x7fc449820610>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite mdhdhf_c at 0x7fc448eef190>: -1.0,\n",
       "   <Metabolite mththf_c at 0x7fc448f74700>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTRI',\n",
       "  'name': '5-methylthioribose-1-phosphate isomerase',\n",
       "  'notes': {'EC Number': '5.3.1.23',\n",
       "   'BioCyc': 'META:5.3.1.23-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3650',\n",
       "   'KEGG Reaction': 'R04420',\n",
       "   'RHEA': '19989;19990;19991;19992',\n",
       "   'UniPathway Reaction': 'UCR04420;UER00874'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37540_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37540_1 at 0x7fc4498a4130>},\n",
       "  '_metabolites': {<Metabolite 5mdr1p_c at 0x7fc448f746d0>: -1.0,\n",
       "   <Metabolite 5mdru1p_c at 0x7fc448f74040>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'MTRK',\n",
       "  'name': '5-methylthioribose kinase',\n",
       "  'notes': {'EC Number': '2.7.1.100',\n",
       "   'BioCyc': 'META:5-METHYLTHIORIBOSE-KINASE-RXN',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3651',\n",
       "   'KEGG Reaction': 'R04143',\n",
       "   'RHEA': '22312;22313;22314;22315',\n",
       "   'UniPathway Reaction': 'UCR04143;UER00872'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU37541_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU37541_1 at 0x7fc4498a4160>},\n",
       "  '_metabolites': {<Metabolite 5mtr_c at 0x7fc463aae280>: -1.0,\n",
       "   <Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite 5mdr1p_c at 0x7fc448f746d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'N2OR',\n",
       "  'name': 'Nitrous oxide reductase (nadh)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR29250', 'EC Number': '1.7.99.6'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite n2o_c at 0x7fc45d375550>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite n2_c at 0x7fc448f74730>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'N2tex',\n",
       "  'name': 'Nitrogen exchange, diffusion (extra-organism to periplasm)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3686'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36547_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36547_1 at 0x7fc4498201c0>},\n",
       "  '_metabolites': {<Metabolite n2_e at 0x7fc448f74760>: -1.0,\n",
       "   <Metabolite n2_p at 0x7fc448f74790>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NAabcO',\n",
       "  'name': 'Sodium transport out via ABC system',\n",
       "  'notes': {},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36532_1 and EDU36533_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36532_1 at 0x7fc4498a4340>,\n",
       "   <Gene EDU36533_1 at 0x7fc4498a4310>},\n",
       "  '_metabolites': {<Metabolite atp_c at 0x7fc463a6d250>: -1.0,\n",
       "   <Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite na1_c at 0x7fc463adc5e0>: -1.0,\n",
       "   <Metabolite adp_c at 0x7fc463a6d160>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 1.0,\n",
       "   <Metabolite pi_c at 0x7fc463a6d100>: 1.0,\n",
       "   <Metabolite na1_e at 0x7fc463adc610>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NACt',\n",
       "  'name': 'Nicotinic acid transport',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR3743'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': '',\n",
       "  'subsystem': '',\n",
       "  '_genes': set(),\n",
       "  '_metabolites': {<Metabolite nac_e at 0x7fc448f747f0>: -1.0,\n",
       "   <Metabolite nac_c at 0x7fc448f747c0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NADDP',\n",
       "  'name': 'NAD diphosphatase',\n",
       "  'notes': {'EC Number': '3.6.1.9',\n",
       "   'BioCyc': 'META:NADPYROPHOSPHAT-RXN',\n",
       "   'BioPath Reaction': 'RXN00220',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR3745',\n",
       "   'RHEA': '11800;11801;11802;11803',\n",
       "   'KEGG Reaction': 'R00103'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU38996_1 or EDU39274_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38996_1 at 0x7fc4498a4190>,\n",
       "   <Gene EDU39274_1 at 0x7fc4498a41c0>},\n",
       "  '_metabolites': {<Metabolite h2o_c at 0x7fc463a6d430>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -1.0,\n",
       "   <Metabolite amp_c at 0x7fc463a6db80>: 1.0,\n",
       "   <Metabolite h_c at 0x7fc463a6d640>: 2.0,\n",
       "   <Metabolite nmn_c at 0x7fc448e74100>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NADFADOR',\n",
       "  'name': 'NAD:FAD oxidoreductase',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83427'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU39259_1 or (EDU38860_1 and EDU39259_1)',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU38860_1 at 0x7fc44986ac70>,\n",
       "   <Gene EDU39259_1 at 0x7fc44986aca0>},\n",
       "  '_metabolites': {<Metabolite fadh2_c at 0x7fc463a82bb0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: -2.0,\n",
       "   <Metabolite fdxr_42_c at 0x7fc463a82af0>: -1.0,\n",
       "   <Metabolite fad_c at 0x7fc463a82b80>: 1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: 2.0,\n",
       "   <Metabolite fdxo_42_c at 0x7fc463a82ac0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': -1000.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NADH10',\n",
       "  'name': 'NADH dehydrogenase (menaquinone-8 & 0 protons)',\n",
       "  'notes': {'RHEA': '29243;29244;29245;29246',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR83428'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36458_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36458_1 at 0x7fc4498a41f0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NADH5',\n",
       "  'name': 'NADH dehydrogenase (ubiquinone-8 )',\n",
       "  'notes': {'RHEA': '29107;29108;29109;29110',\n",
       "   'MetaNetX (MNX) Equation': 'MNXR83430'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36458_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36458_1 at 0x7fc4498a41f0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite q8_c at 0x7fc448e749a0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite q8h2_c at 0x7fc448e749d0>: 1.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NADH6',\n",
       "  'name': 'NADH dehydrogenase (ubiquinone-8 & 3.5 protons)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR83431', 'EC Number': '1.6.5.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36458_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36458_1 at 0x7fc4498a41f0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -4.5,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite q8_c at 0x7fc448e749a0>: -1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite q8h2_c at 0x7fc448e749d0>: 1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: 3.5},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " {'_id': 'NADH7',\n",
       "  'name': 'NADH dehydrogenase (menaquinone-8 & 2 protons)',\n",
       "  'notes': {'MetaNetX (MNX) Equation': 'MNXR69516', 'EC Number': '1.6.5.3'},\n",
       "  '_annotation': {},\n",
       "  '_gene_reaction_rule': 'EDU36458_1',\n",
       "  'subsystem': '',\n",
       "  '_genes': {<Gene EDU36458_1 at 0x7fc4498a41f0>},\n",
       "  '_metabolites': {<Metabolite h_c at 0x7fc463a6d640>: -3.0,\n",
       "   <Metabolite mqn8_c at 0x7fc448e74130>: -1.0,\n",
       "   <Metabolite nadh_c at 0x7fc463a6d700>: -1.0,\n",
       "   <Metabolite mql8_c at 0x7fc448e49f70>: 1.0,\n",
       "   <Metabolite nad_c at 0x7fc463a6d6d0>: 1.0,\n",
       "   <Metabolite h_e at 0x7fc463adc580>: 2.0},\n",
       "  '_model': <Model Clostridium_sporogenes_ATCC_15579 at 0x7fc463a378e0>,\n",
       "  '_lower_bound': 0.0,\n",
       "  '_upper_bound': 1000.0},\n",
       " ...]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x.__dict__ for x in model.reactions]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'azimuth_id': '',\n",
       " 'id': 'ALAt2r',\n",
       " 'source': [],\n",
       " 'version': '',\n",
       " 'status': '',\n",
       " 'reactants': ['ala__L', 'h'],\n",
       " 'products': ['ala__L', 'h'],\n",
       " 'enzymes': [],\n",
       " 'genes': ['EDU37508_1', 'EDU36264_1', 'EDU37559_1'],\n",
       " 'pathways': [],\n",
       " 'ontologies': [],\n",
       " 'species': '',\n",
       " 'compartments': [],\n",
       " 'cell_types': [],\n",
       " 'tissues': [],\n",
       " 'db_ids': [('MetaNetX (MNX) Equation', 'MNXR636'),\n",
       "  ('BioCyc', 'META:RXN0-5202')]}"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# port reactions, to include genes and enzymes\n",
    "def port_reaction(R):\n",
    "    new = Reaction()\n",
    "    new.id = R.id\n",
    "    new.reactants = [remove_compartment_by_split(m.id,'_') for m in R.reactants] # decompartmentalization\n",
    "    new.products = [remove_compartment_by_split(m.id,'_') for m in R.products]   # decompartmentalization\n",
    "    new.genes = [g.id for g in R.genes]\n",
    "    new.enzymes = R.notes.get('EC Number',[])\n",
    "    new.db_ids = dict2listOfTuple(R.notes,';') # not sure ; is useful\n",
    "    return new\n",
    "    \n",
    "test99 = port_reaction(model.reactions[200])\n",
    "test99.__dict__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1715\n"
     ]
    }
   ],
   "source": [
    "## Reactions to port\n",
    "myRxns = []\n",
    "for R in model.reactions:\n",
    "    myRxns.append( port_reaction(R) )\n",
    "    \n",
    "print(len(myRxns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove duplicated reactions after decompartmentalization\n",
    "myRxns = remove_duplicate_rxn(myRxns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1457"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(myRxns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'azimuth_id': '',\n",
       " 'id': '2AGPEAT120',\n",
       " 'source': [],\n",
       " 'version': '',\n",
       " 'status': '',\n",
       " 'reactants': ['2agpe120', 'atp', 'ddca'],\n",
       " 'products': ['amp', 'pe120', 'ppi'],\n",
       " 'enzymes': [],\n",
       " 'genes': ['EDU36256_1'],\n",
       " 'pathways': [],\n",
       " 'ontologies': [],\n",
       " 'species': '',\n",
       " 'compartments': [],\n",
       " 'cell_types': [],\n",
       " 'tissues': [],\n",
       " 'db_ids': [('MetaNetX (MNX) Equation', 'MNXR81336')]}"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myRxns[0].__dict__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pathway information will be parsed later when looking back to the original data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Collected data; now output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import datetime\n",
    "today =  str(datetime.today()).split(\" \")[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2022-04-27'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "today"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "note = \"\"\"CarveME model.\"\"\"\n",
    "\n",
    "## metabolicModel to export\n",
    "MM = MetabolicModel()\n",
    "MM.id = f'az_{model_name}_{today}' #\n",
    "MM.meta_data = {\n",
    "            'species': model_name.split('-')[0],\n",
    "            'version': '',\n",
    "            'sources': [f'https://github.com/SysBioChalmers/{model_name}, retrieved {today}'], #\n",
    "            'status': '',\n",
    "            'last_update': today,  #\n",
    "            'note': note,\n",
    "        }\n",
    "# MM.list_of_pathways = [P.serialize() for P in myPathways]\n",
    "MM.list_of_reactions = [R.serialize() for R in  myRxns]\n",
    "MM.list_of_compounds = [C.serialize() for C in myCpds]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "import os\n",
    "\n",
    "# Write pickle file\n",
    "export_pickle(os.path.join(output_fdr,f'{MM.id}.pickle'), MM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write json file\n",
    "export_json(os.path.join(output_fdr,f'{MM.id}.json'), MM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write dataframe \n",
    "import pandas as pd\n",
    "export_table(os.path.join(output_fdr,f'{MM.id}_list_of_compounds.csv'),MM, 'list_of_compounds')\n",
    "export_table(os.path.join(output_fdr,f'{MM.id}_list_of_reactions.csv'),MM, 'list_of_reactions')\n",
    "# export_table(os.path.join(output_fdr,f'{MM.id}_list_of_pathways.csv'),MM, 'list_of_pathways')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary\n",
    "\n",
    "This ports reactions, pathways and compounds. Gene and enzyme information is now included. \n",
    "\n",
    "The exported pickle can be re-imported and uploaded to Database easily.\n",
    "\n",
    "This notebook, the pickle file and the JSON file go to GitHub repo (https://github.com/shuzhao-li/Azimuth)."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "metadata": {
   "interpreter": {
    "hash": "aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
